// rozod/lib/cache.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cache = exports.ChromeStore = exports.LocalStorageStore = exports.MemoryStore = void 0;
class MemoryStore {
    store = {};
    async get(key) {
        return this.store[key];
    }
    async set(key, value) {
        this.store[key] = value;
    }
    async all() {
        return this.store;
    }
    async delete(key) {
        delete this.store[key];
    }
}
exports.MemoryStore = MemoryStore;
class LocalStorageStore {
    async get(key) {
        const value = localStorage.getItem(key);
        return value ? JSON.parse(value) : null;
    }
    async set(key, value) {
        localStorage.setItem(key, JSON.stringify(value));
    }
    async all() {
        if (!globalThis.localStorage)
            return {};
        const keys = Object.keys(localStorage);
        const entries = {};
        keys.forEach((key) => {
            if (!key.startsWith('rozod_cache:'))
                return;
            const value = localStorage.getItem(key);
            if (value) {
                entries[key] = JSON.parse(value);
            }
        });
        return entries;
    }
    async delete(key) {
        localStorage.removeItem(key);
    }
}
exports.LocalStorageStore = LocalStorageStore;
class ChromeStore {
    async get(key) {
        return new Promise((resolve) => {
            // @ts-ignore
            chrome.storage.local.get(key, (result) => {
                resolve(result[key]);
            });
        });
    }
    async set(key, value) {
        return new Promise((resolve) => {
            let item = {};
            item[key] = value;
            // @ts-ignore
            chrome.storage.local.set(item, resolve);
        });
    }
    async all() {
        if (!globalThis.chrome)
            return {};
        return new Promise((resolve) => {
            // @ts-ignore
            chrome?.storage?.local?.get?.(null, (result) => {
                if (!result) {
                    resolve({});
                    return;
                }
                const filtered = {};
                for (const key of Object.keys(result)) {
                    if (key.startsWith('rozod_cache:')) {
                        filtered[key] = result[key];
                    }
                }
                resolve(filtered);
            });
        });
    }
    async delete(key) {
        return new Promise((resolve) => {
            // @ts-ignore
            chrome.storage.local.remove(key, resolve);
        });
    }
}
exports.ChromeStore = ChromeStore;
class Cache {
    store;
    didCheckExpired = false;
    constructor(store) {
        this.store = store;
        this.checkExpired().catch();
    }
    async get(key) {
        const entry = await this.store.get(key);
        if (!entry) {
            return null;
        }
        if (entry.expiresAt && entry.expiresAt < Date.now()) {
            await this.delete(key);
            return null;
        }
        return entry.value;
    }
    async set(key, value, ttl) {
        await this.store.set(key, {
            value,
            expiresAt: ttl ? Date.now() + ttl : undefined,
        });
    }
    async delete(key) {
        await this.store.delete(key);
    }
    async checkExpired() {
        if (this.didCheckExpired)
            return;
        this.didCheckExpired = true;
        const entries = await this.store.all();
        if (!entries)
            return;
        Object.keys(entries).forEach(async (key) => {
            const entry = entries[key];
            if (entry.expiresAt && entry.expiresAt < Date.now()) {
                await this.delete(key);
            }
        });
    }
}
exports.Cache = Cache;
