// rozod/lib/opencloud/v1/assets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getOperationsOperationId = exports.getAssetsAssetIdVersionsVersionNumber = exports.postAssetsAssetIdVersionsRollback = exports.getAssetsAssetIdVersions = exports.postAssetsAssetIdRestore = exports.postAssetsAssetIdArchive = exports.patchAssetsAssetId = exports.getAssetsAssetId = exports.postAssets = void 0;
const zod_1 = require("zod");
const __1 = require("../..");
const AssetPrivacy = zod_1.z.enum(['default', 'restricted', 'openUse']);
const Creator = zod_1.z.object({
    userId: zod_1.z.number().int(),
    groupId: zod_1.z.number().int(),
});
const CreationContext = zod_1.z.object({
    assetPrivacy: AssetPrivacy,
    creator: Creator,
    expectedPrice: zod_1.z.number().int(),
});
const ModerationResult = zod_1.z.object({ moderationState: zod_1.z.string() });
const Preview = zod_1.z.object({ asset: zod_1.z.string(), altText: zod_1.z.string() });
const State = zod_1.z.enum(['Unspecified', 'Active', 'Archived']);
const SocialLink = zod_1.z.object({ title: zod_1.z.string(), uri: zod_1.z.string() });
const Asset = zod_1.z.object({
    assetType: zod_1.z.string(),
    assetId: zod_1.z.number().int(),
    creationContext: CreationContext,
    description: zod_1.z.string(),
    displayName: zod_1.z.string(),
    path: zod_1.z.string(),
    revisionId: zod_1.z.string(),
    revisionCreateTime: zod_1.z.string().datetime({ offset: true }),
    moderationResult: ModerationResult,
    icon: zod_1.z.string(),
    previews: zod_1.z.array(Preview),
    state: State,
    socialLink: SocialLink,
});
const Assets_CreateAsset_Body = zod_1.z.object({
    request: Asset,
    fileContent: zod_1.z.instanceof(File),
});
const Status = zod_1.z.object({ code: zod_1.z.number().int(), message: zod_1.z.string() });
const Operation = zod_1.z.object({
    path: zod_1.z.string(),
    done: zod_1.z.boolean(),
    error: Status,
    response: Asset,
});
const AssetVersion = zod_1.z.object({
    creationContext: CreationContext,
    path: zod_1.z.string(),
    moderationResult: ModerationResult,
    published: zod_1.z.boolean(),
});
/**
 * **Create Asset** · `BETA`
 *
 * Creates an asset with provided content and metadata.

You can't add [SocialLink](#SocialLink) objects when you create an asset. Instead, use [Update Asset](#PATCH-v1-assets-_assetId_).

Provide the [Asset](#Asset), binary asset file path, and [content type](/cloud/guides/usage-assets.md#supported-asset-types-and-limits) in the form data.
 *
 * **Scopes:** `asset:read`, `asset:write`
 *
 * @param body
 */
exports.postAssets = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/assets',
    baseUrl: 'https://apis.roblox.com/assets',
    scopes: ['asset:read', 'asset:write'],
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Assets_CreateAsset_Body,
    response: Operation,
    errors: [
        {
            status: 400,
            description: `Invalid argument. Failed to parse the request or the file.`,
        },
        {
            status: 401,
            description: `The API key is not valid for this operation / You don&#x27;t have the authorization.`,
        },
        {
            status: 500,
            description: `Server internal error / Unknown error.`,
        },
    ],
});
/**
 * **Get Asset** · `BETA`
 *
 * Retrieve specific asset metadata.
 *
 * **Scopes:** `asset:read`
 *
 * @param assetId The unique identifier of the asset.
 * @param readMask Asset metadata fields to retrieve, including the description, display name, icon, social links, and previews. Examples: `description%2CdisplayName`, `previews%2CtwitchSocialLink`.
 */
exports.getAssetsAssetId = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/assets/:assetId',
    baseUrl: 'https://apis.roblox.com/assets',
    scopes: ['asset:read'],
    requestFormat: 'json',
    serializationMethod: {
        assetId: {},
        readMask: {},
    },
    parameters: {
        assetId: zod_1.z.string(),
        readMask: zod_1.z.string().optional(),
    },
    response: Asset,
    errors: [
        {
            status: 400,
            description: `Malformed request, likely due to an invalid read mask.`,
        },
        {
            status: 401,
            description: `The API key is not valid for this operation / You don&#x27;t have the authorization.`,
        },
        {
            status: 403,
            description: `Doesn&#x27;t have the required permission.`,
        },
        {
            status: 404,
            description: `Asset doesn&#x27;t exist.`,
        },
        {
            status: 500,
            description: `Server internal error / Unknown error.`,
        },
    ],
});
/**
 * **Update Asset** · `BETA`
 *
 * Updates an asset with provided content and metadata, including the description, display name, icon, social links, and previews. Currently can only update the content body for **Models**. Icons and Previews must be **Image** assets. Icons must have square dimensions.

Provide the [Asset](#Asset), binary asset file path, and [content type](/cloud/guides/usage-assets.md#supported-asset-types-and-limits) in the form data.
 *
 * **Scopes:** `asset:read`, `asset:write`
 *
 * @param body
 * @param assetId The unique identifier of the asset.
 * @param updateMask Asset metadata fields to update, including the description, display name, icon, and previews. Examples: `description%2CdisplayName`, `previews%2CtwitchSocialLink`.
 */
exports.patchAssetsAssetId = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/assets/:assetId',
    baseUrl: 'https://apis.roblox.com/assets',
    scopes: ['asset:read', 'asset:write'],
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        assetId: {},
        updateMask: {},
    },
    parameters: {
        assetId: zod_1.z.string(),
        updateMask: zod_1.z.string().optional(),
    },
    body: Assets_CreateAsset_Body.partial(),
    response: Operation,
    errors: [
        {
            status: 400,
            description: `Invalid argument. Failed to parse the request or the file.`,
        },
        {
            status: 401,
            description: `The API key is not valid for this operation / You don&#x27;t have the authorization.`,
        },
        {
            status: 500,
            description: `Server internal error / Unknown error.`,
        },
    ],
});
/**
 * **Archive Asset** · `BETA`
 *
 * Archives the asset. Archived assets disappear from the website and are no longer usable or visible in Roblox experiences, but you can [restore](#POST-v1-assets-{assetId}:restore) them.
 *
 * **Scopes:** `asset:read`, `asset:write`
 *
 * @param assetId The unique identifier of the asset.
 */
exports.postAssetsAssetIdArchive = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/assets/:assetId:archive',
    baseUrl: 'https://apis.roblox.com/assets',
    scopes: ['asset:read', 'asset:write'],
    requestFormat: 'json',
    serializationMethod: {
        assetId: {},
    },
    parameters: {
        assetId: zod_1.z.string(),
    },
    response: Asset,
    errors: [
        {
            status: 400,
            description: `Bad request - invalid request.`,
        },
        {
            status: 403,
            description: `Forbidden - API key without Write scope or user doesn&#x27;t have access.`,
        },
        {
            status: 404,
            description: `Asset not found.`,
        },
    ],
});
/**
 * **Restore Asset** · `BETA`
 *
 * Restores an archived asset.
 *
 * **Scopes:** `asset:read`, `asset:write`
 *
 * @param assetId The unique identifier of the asset.
 */
exports.postAssetsAssetIdRestore = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/assets/:assetId:restore',
    baseUrl: 'https://apis.roblox.com/assets',
    scopes: ['asset:read', 'asset:write'],
    requestFormat: 'json',
    serializationMethod: {
        assetId: {},
    },
    parameters: {
        assetId: zod_1.z.string(),
    },
    response: Asset,
    errors: [
        {
            status: 400,
            description: `Bad request - invalid request.`,
        },
        {
            status: 403,
            description: `Forbidden - API key without Write scope or user doesn&#x27;t have access.`,
        },
        {
            status: 404,
            description: `Asset not found.`,
        },
    ],
});
/**
 * **List Asset Versions** · `BETA`
 *
 * List all versions of a specific asset, with optional pagination.
 *
 * **Scopes:** `asset:read`
 *
 * @param assetId The unique identifier of the asset.
 * @param maxPageSize Specifies the number of asset versions to include in the response. Valid values range from 1 to 50 (inclusive). Defaults to 8 when not provided.
 * @param pageToken A token for pagination. The value is obtained from a previous request and allows for retrieving the next page of asset versions.
 */
exports.getAssetsAssetIdVersions = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/assets/:assetId/versions',
    baseUrl: 'https://apis.roblox.com/assets',
    scopes: ['asset:read'],
    requestFormat: 'json',
    serializationMethod: {
        assetId: {},
        maxPageSize: {},
        pageToken: {},
    },
    parameters: {
        assetId: zod_1.z.string(),
        maxPageSize: zod_1.z.number().int().optional(),
        pageToken: zod_1.z.string().optional(),
    },
    response: zod_1.z.array(AssetVersion),
    errors: [
        {
            status: 400,
            description: `Bad request - invalid parameters.`,
        },
        {
            status: 403,
            description: `Forbidden - API key without Read scope or user doesn&#x27;t have access.`,
        },
        {
            status: 404,
            description: `Asset not found.`,
        },
    ],
});
/**
 * **Rollback Asset Version** · `BETA`
 *
 * Rollback an asset to a specific previous version.

 Provide the asset version path in the form data.
 *
 * **Scopes:** `asset:read`, `asset:write`
 *
 * @param body
 * @param assetId The unique identifier of the asset.
 */
exports.postAssetsAssetIdVersionsRollback = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/assets/:assetId/versions:rollback',
    baseUrl: 'https://apis.roblox.com/assets',
    scopes: ['asset:read', 'asset:write'],
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        assetId: {},
    },
    parameters: {
        assetId: zod_1.z.string(),
    },
    body: zod_1.z.object({ assetVersion: zod_1.z.string() }),
    response: AssetVersion,
    errors: [
        {
            status: 400,
            description: `Bad request - invalid request body.`,
        },
        {
            status: 403,
            description: `Forbidden - API key without Write scope or user doesn&#x27;t have access.`,
        },
        {
            status: 404,
            description: `Asset or Asset Version not found.`,
        },
    ],
});
/**
 * **Get Asset Version** · `BETA`
 *
 * Retrieve a specific asset version by the asset ID and the version number.
 *
 * **Scopes:** `asset:read`
 *
 * @param assetId The unique identifier of the asset.
 * @param versionNumber The version number.
 */
exports.getAssetsAssetIdVersionsVersionNumber = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/assets/:assetId/versions/:versionNumber',
    baseUrl: 'https://apis.roblox.com/assets',
    scopes: ['asset:read'],
    requestFormat: 'json',
    serializationMethod: {
        assetId: {},
        versionNumber: {},
    },
    parameters: {
        assetId: zod_1.z.string(),
        versionNumber: zod_1.z.string(),
    },
    response: AssetVersion,
    errors: [
        {
            status: 403,
            description: `Forbidden - API key without Read scope or user doesn&#x27;t have access.`,
        },
        {
            status: 404,
            description: `Asset or Asset Version not found.`,
        },
    ],
});
/**
 * **Get Operation** · `BETA`
 *
 * Get the result of an asset creation or update using the returned Operation ID. Requires **Read** for the API key permission and **asset:read** for OAuth 2.0 apps.
 *
 * **Scopes:** `asset:read`
 *
 * @param operationId The unique identifier of the operation.
 */
exports.getOperationsOperationId = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/operations/:operationId',
    baseUrl: 'https://apis.roblox.com/assets',
    scopes: ['asset:read'],
    requestFormat: 'json',
    serializationMethod: {
        operationId: {},
    },
    parameters: {
        operationId: zod_1.z.string(),
    },
    response: Operation,
    errors: [
        {
            status: 400,
            description: `Invalid argument. Failed to parse the request or the file.`,
        },
        {
            status: 401,
            description: `The API key is not valid for this operation / You don&#x27;t have the authorization.`,
        },
        {
            status: 500,
            description: `Server internal error / Unknown error.`,
        },
    ],
});
