// rozod/lib/opencloud/v1/asset-permissions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.patchAssetPermissionsApiV1AssetsPermissions = void 0;
const zod_1 = require("zod");
const __1 = require("../..");
const SubjectType = zod_1.z.enum(['Invalid', 'User', 'Group', 'GroupRoleset', 'All', 'Universe']);
const AssetAction = zod_1.z.enum(['Invalid', 'Edit', 'Use', 'Download', 'CopyFromRcc', 'UpdateFromRcc']);
const AssetGrantRequest = zod_1.z.object({
    assetId: zod_1.z.number().int(),
    grantToDependencies: zod_1.z.boolean(),
    parentVersionNumber: zod_1.z.number().int(),
});
const BatchGrantPermissionsRequest = zod_1.z.object({
    subjectType: SubjectType,
    subjectId: zod_1.z.string().nullable(),
    action: AssetAction,
    requests: zod_1.z.array(AssetGrantRequest).nullable(),
    assetIds: zod_1.z.array(zod_1.z.number()).nullable(),
    enableDeepAccessCheck: zod_1.z.boolean(),
});
const ErrorCode = zod_1.z.enum([
    'UnknownError',
    'InvalidRequest',
    'AssetNotFound',
    'CannotManageAsset',
    'PublicAssetCannotBeGrantedTo',
    'CannotManageSubject',
    'SubjectNotFound',
    'AssetTypeNotEnabled',
    'PermissionLimitReached',
    'DependenciesLimitReached',
]);
const GrantPermissionError = zod_1.z.object({
    assetId: zod_1.z.number().int(),
    code: ErrorCode,
});
const BatchGrantPermissionsResponse = zod_1.z.object({
    successAssetIds: zod_1.z.array(zod_1.z.number()).nullable(),
    errors: zod_1.z.array(GrantPermissionError).nullable(),
});
const Error = zod_1.z.object({ code: ErrorCode, message: zod_1.z.string().nullable() });
const ErrorResponse = zod_1.z.object({ error: Error });
/**
 * `BETA`
 *
 * Grant a subject permission to multiple assets.
            
Authorization is required to grant permissions to the subject and asset IDs in the request.
 *
 * **Scopes:** `asset-permissions:write`
 * **Engine:** Not available in-engine
 *
 * @param body
 */
exports.patchAssetPermissionsApiV1AssetsPermissions = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/asset-permissions-api/v1/assets/permissions',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['asset-permissions:write'],
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: BatchGrantPermissionsRequest.partial(),
    response: BatchGrantPermissionsResponse,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 500,
            description: `Internal Server Error`,
        },
    ],
});
