// rozod/lib/opencloud/v1/developer-products.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDeveloperProductsV2UniversesUniverseIdDeveloperProductsCreator = exports.getDeveloperProductsV2UniversesUniverseIdDeveloperProductsProductIdCreator = exports.patchDeveloperProductsV2UniversesUniverseIdDeveloperProductsProductId = exports.postDeveloperProductsV2UniversesUniverseIdDeveloperProducts = void 0;
const zod_1 = require("zod");
const __1 = require("../..");
const DeveloperProducts_CreateDeveloperProductV2_Body = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string().nullish(),
    isForSale: zod_1.z.boolean().nullish(),
    price: zod_1.z.number().int().nullish(),
    imageFile: zod_1.z.instanceof(File).nullish(),
    isRegionalPricingEnabled: zod_1.z.boolean().nullish(),
});
const PricingFeature = zod_1.z.enum(['Invalid', 'PriceOptimization', 'UserFixedPrice', 'RegionalPricing']);
const PriceInformationStruct = zod_1.z.object({
    defaultPriceInRobux: zod_1.z.number().int().nullable(),
    enabledFeatures: zod_1.z.array(PricingFeature),
});
const DeveloperProductConfigV2 = zod_1.z.object({
    productId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    iconImageAssetId: zod_1.z.number().int().nullable(),
    universeId: zod_1.z.number().int(),
    isForSale: zod_1.z.boolean(),
    storePageEnabled: zod_1.z.boolean(),
    priceInformation: PriceInformationStruct.nullable(),
    isImmutable: zod_1.z.boolean(),
    createdTimestamp: zod_1.z.string().datetime({ offset: true }),
    updatedTimestamp: zod_1.z.string().datetime({ offset: true }),
});
const ErrorCode = zod_1.z.enum([
    'InvalidImageFile',
    'UnauthorizedAccess',
    'NotAuthenticated',
    'NotFound',
    'Internal',
    'BadRequest',
    'InvalidProductId',
    'UnauthorizedProductAccess',
    'InvalidDeveloperProductId',
    'DuplicateProductName',
    'InvalidUniverseId',
    'UnauthorizedUniverseAccess',
    'InvalidShopId',
    'UnknownError',
    'InvalidPriceInRobux',
    'InvalidPostBody',
    'InvalidPageNumber',
    'InvalidPageSize',
    'UnsupportedDeveloperProductUpdate',
    'PendingProductsLimitExceeded',
    'InvalidCursor',
    'InvalidRegionalPricing',
    'ProductRetrievalLimitExceeded',
    'InvalidProductIds',
    'InvalidName',
    'InvalidDescription',
    'InvalidIsForSale',
    'InvalidStorePageEnabled',
    'Unavailable',
    'InvalidPrice',
    'Blocked',
    'Conflict',
]);
const ErrorResponse = zod_1.z.object({
    errorCode: ErrorCode,
    errorMessage: zod_1.z.string().nullable(),
    field: zod_1.z.string().nullable(),
    hint: zod_1.z.string().nullable(),
});
const DeveloperProducts_UpdateDeveloperProductV2_Body = zod_1.z.object({
    name: zod_1.z.string().nullable(),
    description: zod_1.z.string().nullable(),
    isForSale: zod_1.z.boolean().nullable(),
    price: zod_1.z.number().int().nullable(),
    imageFile: zod_1.z.instanceof(File).nullable(),
    isRegionalPricingEnabled: zod_1.z.boolean().nullable(),
    storePageEnabled: zod_1.z.boolean().nullable(),
});
const ListDeveloperProductConfigsV2Response = zod_1.z.object({
    developerProducts: zod_1.z.array(DeveloperProductConfigV2),
    nextPageToken: zod_1.z.string().nullable(),
});
/**
 * `BETA`
 *
 * Creates a new developer product with the provided configuration details.
 *
 * **Scopes:** `developer-product:write`
 * **Engine:** Not available in-engine
 *
 * @param body
 * @param universeId The universe ID.
 */
exports.postDeveloperProductsV2UniversesUniverseIdDeveloperProducts = (0, __1.endpoint)({
    method: 'POST',
    path: '/developer-products/v2/universes/:universeId/developer-products',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['developer-product:write'],
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        universeId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    body: DeveloperProducts_CreateDeveloperProductV2_Body,
    response: DeveloperProductConfigV2,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 401,
            description: `Unauthorized`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
    ],
});
/**
 * `BETA`
 *
 * Updates a developer product with the provided configuration details.
Note that only fields provided in the request will be updated.
 *
 * **Scopes:** `developer-product:write`
 * **Engine:** Not available in-engine
 *
 * @param body
 * @param universeId The universe ID.
 * @param productId The product ID of the developer product.
 */
exports.patchDeveloperProductsV2UniversesUniverseIdDeveloperProductsProductId = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/developer-products/v2/universes/:universeId/developer-products/:productId',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['developer-product:write'],
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        universeId: {},
        productId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        productId: zod_1.z.number().int(),
    },
    body: DeveloperProducts_UpdateDeveloperProductV2_Body.partial(),
    response: zod_1.z.void(),
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 401,
            description: `Unauthorized`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
        {
            status: 409,
            description: `Conflict`,
        },
    ],
});
/**
 * `BETA`
 *
 * Get developer product with configuration details
 *
 * **Scopes:** `developer-product:read`
 * **Engine:** Not available in-engine
 *
 * @param universeId The universe ID.
 * @param productId The product ID of the developer product.
 */
exports.getDeveloperProductsV2UniversesUniverseIdDeveloperProductsProductIdCreator = (0, __1.endpoint)({
    method: 'GET',
    path: '/developer-products/v2/universes/:universeId/developer-products/:productId/creator',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['developer-product:read'],
    requestFormat: 'json',
    serializationMethod: {
        universeId: {},
        productId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        productId: zod_1.z.number().int(),
    },
    response: DeveloperProductConfigV2,
    errors: [
        {
            status: 401,
            description: `Unauthorized`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
    ],
});
/**
 * `BETA`
 *
 * List developer products by universe with configuration details
 *
 * **Scopes:** `developer-product:read`
 * **Engine:** Not available in-engine
 *
 * @param universeId The universe ID.
 * @param pageSize The number of results to return. Defaults to 50.
 * @param pageToken The cursor token for pagination.
 */
exports.getDeveloperProductsV2UniversesUniverseIdDeveloperProductsCreator = (0, __1.endpoint)({
    method: 'GET',
    path: '/developer-products/v2/universes/:universeId/developer-products/creator',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['developer-product:read'],
    requestFormat: 'json',
    serializationMethod: {
        universeId: {},
        pageSize: {},
        pageToken: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        pageSize: zod_1.z.number().int().optional(),
        pageToken: zod_1.z.string().optional(),
    },
    response: ListDeveloperProductConfigsV2Response,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 401,
            description: `Unauthorized`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
    ],
});
