// rozod/lib/opencloud/v1/game-passes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getGamePassesV1UniversesUniverseIdGamePassesCreator = exports.getGamePassesV1UniversesUniverseIdGamePassesGamePassIdCreator = exports.patchGamePassesV1UniversesUniverseIdGamePassesGamePassId = exports.postGamePassesV1UniversesUniverseIdGamePasses = void 0;
const zod_1 = require("zod");
const __1 = require("../..");
const GamePasses_CreateGamePass_Body = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string().nullish(),
    imageFile: zod_1.z.instanceof(File).nullish(),
    isForSale: zod_1.z.boolean().nullish(),
    price: zod_1.z.number().int().nullish(),
    isRegionalPricingEnabled: zod_1.z.boolean().nullish(),
});
const PricingFeature = zod_1.z.enum(['Invalid', 'PriceOptimization', 'UserFixedPrice', 'RegionalPricing']);
const PriceInformationStruct = zod_1.z.object({
    defaultPriceInRobux: zod_1.z.number().int().nullable(),
    enabledFeatures: zod_1.z.array(PricingFeature),
});
const GamePassConfigV2 = zod_1.z.object({
    gamePassId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    isForSale: zod_1.z.boolean(),
    iconAssetId: zod_1.z.number().int(),
    createdTimestamp: zod_1.z.string().datetime({ offset: true }),
    updatedTimestamp: zod_1.z.string().datetime({ offset: true }),
    priceInformation: PriceInformationStruct.nullable(),
});
const ErrorCode = zod_1.z.enum([
    'InternalError',
    'BadRequest',
    'NotFound',
    'UnauthorizedAccess',
    'NotAuthenticated',
    'PassNotFound',
    'TargetUnauthorizedAccess',
    'UniverseNotFound',
    'MissingArgument',
    'AssetCreationFailure',
    'PassRevokeFailure',
    'PassAlreadyRevoked',
    'SalesNotFound',
    'CommissionRateNotFound',
    'AssetNotFound',
    'FileTooLarge',
    'ActiveInPO',
    'NotForSale',
    'NotSameUniverse',
    'PricingConfigError',
    'InvalidCount',
    'Blocked',
    'InvalidPageSize',
]);
const ErrorResponse = zod_1.z.object({
    errorCode: ErrorCode,
    errorMessage: zod_1.z.string().nullable(),
    field: zod_1.z.string().nullable(),
    hint: zod_1.z.string().nullable(),
});
const GamePasses_UpdateGamePass_Body = zod_1.z.object({
    name: zod_1.z.string().nullable(),
    description: zod_1.z.string().nullable(),
    file: zod_1.z.instanceof(File).nullable(),
    isForSale: zod_1.z.boolean().nullable(),
    price: zod_1.z.number().int().nullable(),
    isRegionalPricingEnabled: zod_1.z.boolean().nullable(),
});
const ListGamePassConfigsByUniverseResponse = zod_1.z.object({
    gamePasses: zod_1.z.array(GamePassConfigV2),
    nextPageToken: zod_1.z.string().nullable(),
});
/**
 * `BETA`
 *
 * Creates a new game pass with the provided configuration details.
 *
 * **Scopes:** `game-pass:write`
 * **Engine:** Not available in-engine
 *
 * @param body
 * @param universeId The universe ID.
 */
exports.postGamePassesV1UniversesUniverseIdGamePasses = (0, __1.endpoint)({
    method: 'POST',
    path: '/game-passes/v1/universes/:universeId/game-passes',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['game-pass:write'],
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        universeId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    body: GamePasses_CreateGamePass_Body,
    response: GamePassConfigV2,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 401,
            description: `Unauthorized`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
    ],
});
/**
 * `BETA`
 *
 * Updates a game pass with the provided configuration details.
Note that only fields provided in the request will be updated.
 *
 * **Scopes:** `game-pass:write`
 * **Engine:** Not available in-engine
 *
 * @param body
 * @param universeId The universe ID.
 * @param gamePassId The game pass ID.
 */
exports.patchGamePassesV1UniversesUniverseIdGamePassesGamePassId = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/game-passes/v1/universes/:universeId/game-passes/:gamePassId',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['game-pass:write'],
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        universeId: {},
        gamePassId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        gamePassId: zod_1.z.number().int(),
    },
    body: GamePasses_UpdateGamePass_Body.partial(),
    response: zod_1.z.void(),
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 401,
            description: `Unauthorized`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
        {
            status: 409,
            description: `Conflict`,
        },
    ],
});
/**
 * `BETA`
 *
 * Get game pass with configuration details
 *
 * **Scopes:** `game-pass:read`
 * **Engine:** Not available in-engine
 *
 * @param universeId The universe ID.
 * @param gamePassId The game pass ID.
 */
exports.getGamePassesV1UniversesUniverseIdGamePassesGamePassIdCreator = (0, __1.endpoint)({
    method: 'GET',
    path: '/game-passes/v1/universes/:universeId/game-passes/:gamePassId/creator',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['game-pass:read'],
    requestFormat: 'json',
    serializationMethod: {
        universeId: {},
        gamePassId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        gamePassId: zod_1.z.number().int(),
    },
    response: GamePassConfigV2,
    errors: [
        {
            status: 401,
            description: `Unauthorized`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
    ],
});
/**
 * `BETA`
 *
 * List game passes by universe with configuration details
 *
 * **Scopes:** `game-pass:read`
 * **Engine:** Not available in-engine
 *
 * @param universeId The universe ID.
 * @param pageSize The number of results to return. Defaults to 50 if not provided.
 * @param pageToken The cursor token for pagination.
 */
exports.getGamePassesV1UniversesUniverseIdGamePassesCreator = (0, __1.endpoint)({
    method: 'GET',
    path: '/game-passes/v1/universes/:universeId/game-passes/creator',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['game-pass:read'],
    requestFormat: 'json',
    serializationMethod: {
        universeId: {},
        pageSize: {},
        pageToken: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        pageSize: zod_1.z.number().int().optional(),
        pageToken: zod_1.z.string().optional(),
    },
    response: ListGamePassConfigsByUniverseResponse,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 401,
            description: `Unauthorized`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
    ],
});
