// rozod/lib/opencloud/v1/messaging.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postUniversesUniverseIdTopicsTopic = void 0;
const zod_1 = require("zod");
const __1 = require("../..");
const PublishRequest = zod_1.z.object({ message: zod_1.z.string().nullable() });
/**
 * **Publish a Message** · `BETA`
 *
 * Publish a message to a pre-defined topic of an experience, with the size of the message up to 1,024 characters (1 KB). Requires the **Publish** permission for API keys and the **universe-messaging-service:publish** scope for OAuth 2.0 apps. See [Cross-server messaging](/cloud-services/cross-server-messaging.md#subscribe-users-to-receive-messages) for defining and subscribing users to a topic.
 *
 * **Scopes:** `universe-messaging-service:publish`
 *
 * @param body
 * @param universeId The identifier of the experience in which you want to send your messages to. You can [copy your experience's Universe ID](/cloud/guides/usage-messaging.md#publishing-messages-to-live-servers) on **Creator Dashboard**.
 * @param topic The topic that you want to publish your message to, with up to 80 characters.
 * @see POST https://apis.roblox.com/cloud/v2/universes/{universe_id}:publishMessage
 * @deprecated Prefer the v2 alternative(s) listed above.
 */
exports.postUniversesUniverseIdTopicsTopic = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/universes/:universeId/topics/:topic',
    baseUrl: 'https://apis.roblox.com/messaging-service',
    scopes: ['universe-messaging-service:publish'],
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        universeId: {},
        topic: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        topic: zod_1.z.string().nullable(),
    },
    body: zod_1.z.object({ message: zod_1.z.string().nullable() }),
    response: zod_1.z.void(),
    errors: [
        {
            status: 400,
            description: `Invalid request.`,
        },
        {
            status: 401,
            description: `The API key is not valid for this operation / You don&#x27;t have the authorization.`,
        },
        {
            status: 403,
            description: `Publishing is not allowed on this experience.`,
        },
        {
            status: 500,
            description: `Server internal error / Unknown error.`,
        },
    ],
});
