// rozod/lib/opencloud/v1/open-eval.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getOpenEvalApiV1EvalRecordsJobId = exports.postOpenEvalApiV1Eval = void 0;
const zod_1 = require("zod");
const __1 = require("../..");
const CustomLlmInfo = zod_1.z.object({
    name: zod_1.z.string().min(1),
    model_version: zod_1.z.string().nullish(),
    url: zod_1.z.string().min(1),
    api_key: zod_1.z.string().nullish(),
});
const EvalRequest = zod_1.z.object({
    name: zod_1.z.string().min(1),
    description: zod_1.z.string().nullish(),
    use_reference_mode: zod_1.z.boolean().optional(),
    input_script: zod_1.z.string().min(10),
    custom_llm_info: CustomLlmInfo.optional(),
});
const EvalResponse = zod_1.z.object({
    job_id: zod_1.z.string().nullable(),
    status_code: zod_1.z.number().int(),
    error: zod_1.z.string().nullable(),
});
const User = zod_1.z.object({ id: zod_1.z.string().nullable() });
const InterruptionRecord = zod_1.z.object({
    check: zod_1.z.number().int(),
    type: zod_1.z.string().nullable(),
});
const EvalJsonRecord = zod_1.z.object({
    passes: zod_1.z.number().int(),
    fails: zod_1.z.number().int(),
    checks: zod_1.z.number().int(),
    warning: zod_1.z.string().nullable(),
    error: zod_1.z.string().nullable(),
    interruptions: zod_1.z.array(InterruptionRecord).nullable(),
});
const EvalResultRecord = zod_1.z.object({
    mode: zod_1.z.string().nullable(),
    result: EvalJsonRecord,
});
const EvalRecord = zod_1.z.object({
    id: zod_1.z.number().int().optional(),
    name: zod_1.z.string().nullable(),
    user: User.optional(),
    useReferenceMode: zod_1.z.string().nullish(),
    description: zod_1.z.string().nullish(),
    jobId: zod_1.z.string().uuid(),
    inputScript: zod_1.z.string().nullable(),
    jobStatus: zod_1.z.string().nullish(),
    results: zod_1.z.array(EvalResultRecord).nullish(),
    fullLogs: zod_1.z.string().nullish(),
    evalSucceeded: zod_1.z.string().nullish(),
    createUtc: zod_1.z.string().nullish(),
});
/**
 * `BETA`
 *
 * Performs the evaluation of the Lua script.
 *
 * **Scopes:** `studio-evaluations:create`
 * **Engine:** Not available in-engine
 *
 * @param body
 */
exports.postOpenEvalApiV1Eval = (0, __1.endpoint)({
    method: 'POST',
    path: '/open-eval-api/v1/eval',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['studio-evaluations:create'],
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: EvalRequest,
    response: EvalResponse,
    errors: [],
});
/**
 * `BETA`
 *
 * Gets the evaluation record by job ID.
 *
 * **Scopes:** `studio-evaluations:create`
 * **Engine:** Not available in-engine
 *
 * @param jobId The job ID of the evaluation record.
 */
exports.getOpenEvalApiV1EvalRecordsJobId = (0, __1.endpoint)({
    method: 'GET',
    path: '/open-eval-api/v1/eval-records/:jobId',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['studio-evaluations:create'],
    requestFormat: 'json',
    serializationMethod: {
        jobId: {},
    },
    parameters: {
        jobId: zod_1.z.string(),
    },
    response: EvalRecord,
    errors: [],
});
