// rozod/lib/opencloud/v1/secrets-store.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCloudV2UniversesUniverseIdSecretsPublicKey = exports.deleteCloudV2UniversesUniverseIdSecretsSecretId = exports.patchCloudV2UniversesUniverseIdSecretsSecretId = exports.getCloudV2UniversesUniverseIdSecrets = exports.postCloudV2UniversesUniverseIdSecrets = void 0;
const zod_1 = require("zod");
const __1 = require("../..");
const Secret = zod_1.z.object({
    id: zod_1.z.string().nullable(),
    secret: zod_1.z.string().nullable(),
    key_id: zod_1.z.string().nullable(),
    domain: zod_1.z.string().nullable(),
    create_time: zod_1.z.string().nullable(),
    update_time: zod_1.z.string().nullable(),
});
const ProblemDetails = zod_1.z.object({
    type: zod_1.z.string().nullable(),
    title: zod_1.z.string().nullable(),
    status: zod_1.z.string().nullable(),
    detail: zod_1.z.string().nullable(),
    instance: zod_1.z.string().nullable(),
});
const SecretPaginatedList = zod_1.z.object({
    secrets: zod_1.z.array(Secret).nullable(),
    nextPageCursor: zod_1.z.string().nullable(),
    previousPageCursor: zod_1.z.string().nullable(),
});
/**
 * `BETA`
 *
 * Creates a new secret. A maximum of 500 secrets per universe is allowed.
            
Only the owner of the universe can create secrets. For group-owned universes, only the group owner or authorized
members can create secrets.
            
To encrypt the secret:
1. Get the public key using the Get Public Key endpoint
2. Encrypt your secret using LibSodium sealed box
3. Base64 encode the encrypted content

Include the key_id from the public key response in the request.

For an example, see the [Secrets store guide](https://create.roblox.com/docs/cloud/guides/secrets-store).
 *
 * **Scopes:** `universe.secret:write`
 * **Engine:** Not available in-engine
 *
 * @param body The secret to create with encrypted content
 * @param universeId The universe ID
 */
exports.postCloudV2UniversesUniverseIdSecrets = (0, __1.endpoint)({
    method: 'POST',
    path: '/cloud/v2/universes/:universeId/secrets',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['universe.secret:write'],
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        universeId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    body: Secret,
    response: Secret,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 409,
            description: `Conflict`,
        },
    ],
});
/**
 * `BETA`
 *
 * Lists all secrets defined for a universe.
Secret content is not returned for security reasons - only metadata such as ID, domain, creation and update timestamps are included.

Only the owner of the universe can list secrets. For group-owned universes, only the group owner or authorized
members can list secrets.
 *
 * **Scopes:** `universe.secret:read`
 * **Engine:** Not available in-engine
 *
 * @param universeId The universe ID
 * @param limit Number of secrets to return per page (1-500, default 10)
 * @param cursor Pagination cursor from previous response
 */
exports.getCloudV2UniversesUniverseIdSecrets = (0, __1.endpoint)({
    method: 'GET',
    path: '/cloud/v2/universes/:universeId/secrets',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['universe.secret:read'],
    requestFormat: 'json',
    serializationMethod: {
        universeId: {},
        limit: {},
        cursor: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        limit: zod_1.z.number().int().optional().default(10),
        cursor: zod_1.z.string().optional(),
    },
    response: SecretPaginatedList,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
    ],
});
/**
 * `BETA`
 *
 * Updates an existing secret.

Only the owner of the universe can update secrets. For group-owned universes, only the group owner or authorized
members can update secrets.

Only the secret content, key_id, and domain can be updated - the secret ID cannot be changed.

To encrypt the updated secret:
1. Get the current public key using the GetPublicKey endpoint
2. Encrypt your new secret content using LibSodium sealed box
3. Base64 encode the encrypted content

Include the key_id from the public key response in the request.

For an example, see the [Secrets store guide](https://create.roblox.com/docs/cloud/guides/secrets-store).
 *
 * **Scopes:** `universe.secret:write`
 * **Engine:** Not available in-engine
 *
 * @param body The updated secret data with encrypted content
 * @param universeId The universe ID
 * @param secretId The ID of the secret to update
 */
exports.patchCloudV2UniversesUniverseIdSecretsSecretId = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/cloud/v2/universes/:universeId/secrets/:secretId',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['universe.secret:write'],
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        universeId: {},
        secretId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        secretId: zod_1.z.string(),
    },
    body: Secret.partial(),
    response: Secret,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
    ],
});
/**
 * `BETA`
 *
 * Permanently deletes a secret from a universe.

Only the owner of the universe can delete secrets. For group-owned universes, only the group owner or authorized
members can delete secrets.

This operation is irreversible. Make sure you no longer need the secret before deleting it.
 *
 * **Scopes:** `universe.secret:write`
 * **Engine:** Not available in-engine
 *
 * @param universeId The universe ID
 * @param secretId The ID of the secret to delete
 */
exports.deleteCloudV2UniversesUniverseIdSecretsSecretId = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/cloud/v2/universes/:universeId/secrets/:secretId',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['universe.secret:write'],
    requestFormat: 'json',
    serializationMethod: {
        universeId: {},
        secretId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        secretId: zod_1.z.string(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
    ],
});
/**
 * `BETA`
 *
 * Retrieves the public key for a universe. You need this key to encrypt secret content
before sending it to Roblox.

Only the owner of the universe can retrieve the public key. For group-owned universes, only the group owner or
authorized members can retrieve the public key.

The secret id field is static and always returns "public-key".

The returned public key in the secret field is universe-specific and derived from a master key using the universe ID.
Use this key with LibSodium sealed box encryption to encrypt your secret content before
creating or updating secrets.

Include the key_id from the public key response in the request to create or update a secret.
 *
 * **Scopes:** `universe.secret:read`
 * **Engine:** Not available in-engine
 *
 * @param universeId The universe ID
 */
exports.getCloudV2UniversesUniverseIdSecretsPublicKey = (0, __1.endpoint)({
    method: 'GET',
    path: '/cloud/v2/universes/:universeId/secrets/public-key',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['universe.secret:read'],
    requestFormat: 'json',
    serializationMethod: {
        universeId: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: Secret,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
    ],
});
