// rozod/lib/opencloud/v1/toolbox.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getToolboxServiceV2AssetsId = exports.getToolboxServiceV2AssetsSearch = void 0;
const zod_1 = require("zod");
const __1 = require("../..");
const SearchCategoryType = zod_1.z.enum(['Audio', 'Model', 'Decal', 'Plugin', 'MeshPart', 'Video', 'FontFamily']);
const searchCategoryType = SearchCategoryType;
const ModelSubType = zod_1.z.enum(['Ad', 'MaterialPack', 'Package']);
const SortDirection = zod_1.z.enum(['None', 'Ascending', 'Descending']);
const sortDirection = SortDirection.optional().default('None');
const SortCategory = zod_1.z.enum(['Relevance', 'Trending', 'Top', 'AudioDuration', 'CreateTime', 'UpdatedTime', 'Ratings']);
const sortCategory = SortCategory.optional().default('Relevance');
const SearchAudioTypeModel = zod_1.z.enum(['Music', 'SoundEffect']);
const ModelInstanceType = zod_1.z.enum(['Script', 'MeshPart', 'Decal', 'Animation', 'Audio', 'Tool']);
const SearchView = zod_1.z.enum(['IDs', 'Core', 'Full']);
const searchView = SearchView.optional().default('Core');
const MusicChartType = zod_1.z.enum(['None', 'Current', 'Week', 'Month', 'Year']);
const musicChartType = MusicChartType.optional().default('None');
const QueryFacets = zod_1.z.object({
    appliedFacets: zod_1.z.array(zod_1.z.string()).nullable(),
    availableFacets: zod_1.z.array(zod_1.z.string()).nullable(),
});
const VotingModel = zod_1.z.object({
    showVotes: zod_1.z.boolean(),
    upVotes: zod_1.z.number().int(),
    downVotes: zod_1.z.number().int(),
    canVote: zod_1.z.boolean(),
    userVote: zod_1.z.boolean().nullable(),
    hasVoted: zod_1.z.boolean(),
    voteCount: zod_1.z.number().int(),
    upVotePercent: zod_1.z.number().int(),
});
const CreatorModelV2 = zod_1.z.object({
    creator: zod_1.z.string().nullable(),
    userId: zod_1.z.number().int().nullable(),
    groupId: zod_1.z.number().int().nullable(),
    name: zod_1.z.string().nullable(),
    verified: zod_1.z.boolean().nullable(),
});
const Decimal = zod_1.z.object({ significand: zod_1.z.number().int(), exponent: zod_1.z.number().int() });
const Money = zod_1.z.object({ currencyCode: zod_1.z.string().nullable(), quantity: Decimal.nullable() });
const CreatorStoreProduct = zod_1.z.object({ purchasePrice: Money.nullable(), purchasable: zod_1.z.boolean() });
const LinkType = zod_1.z.union([
    zod_1.z.literal(0),
    zod_1.z.literal(1),
    zod_1.z.literal(2),
    zod_1.z.literal(3),
    zod_1.z.literal(4),
    zod_1.z.literal(5),
    zod_1.z.literal(6),
    zod_1.z.literal(7),
    zod_1.z.literal(8),
    zod_1.z.literal(9),
    zod_1.z.literal(10),
]);
const SocialLinkModel = zod_1.z.object({ linkType: LinkType, url: zod_1.z.string().nullable(), title: zod_1.z.string().nullable() });
const PreviewAssetsModel = zod_1.z.object({
    imagePreviewAssets: zod_1.z.array(zod_1.z.number()).nullable(),
    videoPreviewAssets: zod_1.z.array(zod_1.z.number()).nullable(),
});
const Asset = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string().nullable(),
    description: zod_1.z.string().nullable(),
    assetTypeId: zod_1.z.number().int().nullable(),
    socialLinks: zod_1.z.array(SocialLinkModel).nullable(),
    previewAssets: PreviewAssetsModel.nullable(),
    createTime: zod_1.z.string().datetime({ offset: true }).nullable(),
    updateTime: zod_1.z.string().datetime({ offset: true }).nullable(),
    categoryPath: zod_1.z.string().nullable(),
});
const Audio = Asset;
const Decal = Asset;
const MeshPart = Asset;
const Model = Asset;
const Music = Audio;
const Plugin = Asset;
const SoundEffect = Audio;
const CreatorStoreAsset = zod_1.z.object({
    voting: VotingModel.nullable(),
    creator: CreatorModelV2.nullable(),
    creatorStoreProduct: CreatorStoreProduct.nullable(),
    asset: zod_1.z.union([Asset, Audio, Decal, MeshPart, Model, Music, Plugin, SoundEffect]).nullable(),
});
const SearchCreatorStoreAssetsResponse = zod_1.z.object({
    nextPageToken: zod_1.z.string().nullable(),
    queryFacets: QueryFacets.nullable(),
    creatorStoreAssets: zod_1.z.array(CreatorStoreAsset).nullable(),
    totalResults: zod_1.z.number().int(),
    filteredKeyword: zod_1.z.string().nullable(),
});
const ProblemDetails = zod_1.z.unknown().nullable();
const HttpValidationProblemDetails = ProblemDetails;
/**
 * `BETA`
 *
 * Search Creator Store for assets.
 *
 * **Scopes:** `creator-store-product:read`
 * **Engine:** Usable with HttpService
 *
 * @param searchCategoryType The asset type to search within.
 * @param query The search terms used to filter the results.
 * @param modelSubTypes When searching for models, the subtypes associated with the search results.
 * @param excludedModelSubTypes When searching for models, the subtypes not associated with the search results.
 * @param creator Deprecated: Please refer to the 'userId' and 'groupId' properties instead. The creator type and ID. E.g. "user/123" or "group/456"
 * @param userId The User ID of the creator. Only one of 'userId' and 'groupId' can be present in a query.
 * @param groupId The Group ID of the creator. Only one of 'userId' and 'groupId' can be present in a query.
 * @param pageToken The identifier for the desired search results page. Only one of 'pageNumber' and 'pageToken' can be present in a query.
 * @param pageNumber The page number to retrieve, starting from 0. Only one of 'pageNumber' and 'pageToken' can be present in a query.
 * @param maxPageSize The number of assets to be returned. Cannot be larger than 100.
 * @param sortDirection The sort direction of the search results.
 * @param sortCategory The category to sort the search results by.
 * @param audioMinDurationSeconds When searching for audio, the minimum duration of the audio assets. If included, must be greater than or equal to 0.
 * @param audioMaxDurationSeconds When searching for audio, the maximum duration of the audio assets. If included, must be greater than or equal to 0.
 * @param audioArtist When searching for audio, the artist name of the audio assets.
 * @param audioAlbum When searching for audio, the album name of the audio assets.
 * @param includeTopCharts
 * @param audioTypes When searching for audio, the type of the audio assets.
 * @param includedInstanceTypes When searching for models, this filters that the following [Instance](https://create.roblox.com/docs/reference/engine/classes/Instance) types are included in the model.
 * @param includeOnlyVerifiedCreators Whether the results should only include assets created by verified creators.
 * @param minPriceCents The minimum price of the asset in cents. If included, must be greater than or equal to 0.
 * @param maxPriceCents The maximum price of the asset in cents. If included, must be greater than or equal to 0.
 * @param facets Additional keywords to query by.
 * @param categoryPath
 * @param searchView Indicates which fields will be populated in the response.
 * @param musicChartType Indicates which music charts to filter from.
 */
exports.getToolboxServiceV2AssetsSearch = (0, __1.endpoint)({
    method: 'GET',
    path: '/toolbox-service/v2/assets:search',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['creator-store-product:read'],
    requestFormat: 'json',
    serializationMethod: {
        searchCategoryType: {},
        query: {},
        modelSubTypes: {},
        excludedModelSubTypes: {},
        creator: {},
        userId: {},
        groupId: {},
        pageToken: {},
        pageNumber: {},
        maxPageSize: {},
        sortDirection: {},
        sortCategory: {},
        audioMinDurationSeconds: {},
        audioMaxDurationSeconds: {},
        audioArtist: {},
        audioAlbum: {},
        includeTopCharts: {},
        audioTypes: {},
        includedInstanceTypes: {},
        includeOnlyVerifiedCreators: {},
        minPriceCents: {},
        maxPriceCents: {},
        facets: {},
        categoryPath: {},
        searchView: {},
        musicChartType: {},
    },
    parameters: {
        searchCategoryType: searchCategoryType,
        query: zod_1.z.string().optional(),
        modelSubTypes: zod_1.z.array(ModelSubType).nullish(),
        excludedModelSubTypes: zod_1.z.array(ModelSubType).nullish(),
        creator: zod_1.z
            .string()
            .regex(/^(user|group)\/\d+$/)
            .optional(),
        userId: zod_1.z.number().int().gte(1).nullish(),
        groupId: zod_1.z.number().int().gte(1).nullish(),
        pageToken: zod_1.z.string().optional(),
        pageNumber: zod_1.z.number().int().gte(0).lte(2147483647).nullish(),
        maxPageSize: zod_1.z.number().int().gte(1).lte(100).optional().default(25),
        sortDirection: sortDirection,
        sortCategory: sortCategory,
        audioMinDurationSeconds: zod_1.z.number().int().gte(0).lte(2147483647).optional(),
        audioMaxDurationSeconds: zod_1.z.number().int().gte(1).lte(2147483647).nullish(),
        audioArtist: zod_1.z.string().optional(),
        audioAlbum: zod_1.z.string().optional(),
        includeTopCharts: zod_1.z.boolean().nullish(),
        audioTypes: zod_1.z.array(SearchAudioTypeModel).nullish(),
        includedInstanceTypes: zod_1.z.array(ModelInstanceType).nullish(),
        includeOnlyVerifiedCreators: zod_1.z.boolean().optional().default(true),
        minPriceCents: zod_1.z.number().int().gte(0).lte(2147483647).optional(),
        maxPriceCents: zod_1.z.number().int().gte(0).lte(2147483647).nullish(),
        facets: zod_1.z.array(zod_1.z.string()).nullish(),
        categoryPath: zod_1.z.string().optional(),
        searchView: searchView,
        musicChartType: musicChartType,
    },
    response: SearchCreatorStoreAssetsResponse.nullable(),
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 429,
            description: `Too Many Requests`,
        },
        {
            status: 500,
            description: `Server Error`,
        },
    ],
});
/**
 * `BETA`
 *
 * Get details for a single Creator Store asset.
 *
 * **Scopes:** `creator-store-product:read`
 * **Engine:** Usable with HttpService
 *
 * @param id The asset ID to retrieve details for.
 */
exports.getToolboxServiceV2AssetsId = (0, __1.endpoint)({
    method: 'GET',
    path: '/toolbox-service/v2/assets/:id',
    baseUrl: 'https://apis.roblox.com',
    scopes: ['creator-store-product:read'],
    requestFormat: 'json',
    serializationMethod: {
        id: {},
    },
    parameters: {
        id: zod_1.z.number().int(),
    },
    response: CreatorStoreAsset.nullable(),
    errors: [
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 404,
            description: `Not Found`,
        },
        {
            status: 429,
            description: `Too Many Requests`,
        },
    ],
});
