// rozod/lib/opencloud/v1/universes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postUniverseIdPlacesPlaceIdVersions = void 0;
const zod_1 = require("zod");
const __1 = require("../..");
/**
 * **Publish a Place** · `BETA`
 *
 * Publish a new place or update an existing place with a new version. Provide a RBXL or RBXLX file in the data-binary.
 *
 * **Scopes:** `universe-places:write`
 *
 * @param universeId The identifier of the experience in which you want to publish your place to. You can [copy your experience's Universe ID](/cloud/guides/usage-place-publishing.md#publishing-a-place) on **Creator Dashboard**.
 * @param placeId The identifier of your place. See [Publishing places with API keys](/cloud/guides/usage-place-publishing.md) on obtaining a Place ID.
 * @param versionType Can only be either:
- `Saved`: the place file should be saved, but not published.
 - `Published`: the place file should be saved and published.
 */
exports.postUniverseIdPlacesPlaceIdVersions = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/:universeId/places/:placeId/versions',
    baseUrl: 'https://apis.roblox.com/universes',
    scopes: ['universe-places:write'],
    requestFormat: 'json',
    serializationMethod: {
        universeId: {},
        placeId: {},
        versionType: {},
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        placeId: zod_1.z.number().int(),
        versionType: zod_1.z.string().nullish(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 400,
            description: `Invalid request / Invalid file content.`,
        },
        {
            status: 401,
            description: `API key not valid for operation, user does not have authorization.`,
        },
        {
            status: 403,
            description: `Publish not allowed on place.`,
        },
        {
            status: 404,
            description: `Place or universe does not exist.`,
        },
        {
            status: 409,
            description: `Place not part of the universe.`,
        },
        {
            status: 500,
            description: `Server internal error / Unknown error.`,
        },
    ],
});
