// rozod/lib/endpoints/accountinformationv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridRobloxBadges = exports.getUsersUseridPromotionChannels = exports.deleteStarCodeAffiliates = exports.postStarCodeAffiliates = exports.getStarCodeAffiliates = exports.postPromotionChannels = exports.getPromotionChannels = exports.postPhoneVerify = exports.postPhoneResend = exports.postPhoneDelete = exports.postPhone = exports.getPhone = exports.getMetadata = exports.postGender = exports.getGender = exports.postEmailVerify = exports.postDescription = exports.getDescription = exports.getBirthdate = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_AccountInformation_Api_Models_BirthdateResponse = zod_1.z.object({
    birthMonth: zod_1.z.number().int(),
    birthDay: zod_1.z.number().int(),
    birthYear: zod_1.z.number().int(),
});
const Roblox_AccountInformation_Api_Models_DescriptionResponse = zod_1.z.object({
    description: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_Models_DescriptionRequest = zod_1.z.object({
    description: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_Models_GenderResponse = zod_1.z.object({
    gender: zod_1.z.number().int(),
});
const Roblox_AccountInformation_Api_Models_GenderRequest = zod_1.z.object({
    gender: zod_1.z.string(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_AccountInformation_Api_Models_MetadataResponse = zod_1.z.object({
    isAllowedNotificationsEndpointDisabled: zod_1.z.boolean(),
    isAccountSettingsPolicyEnabled: zod_1.z.boolean(),
    isPhoneNumberEnabled: zod_1.z.boolean(),
    MaxUserDescriptionLength: zod_1.z.number().int(),
    isUserDescriptionEnabled: zod_1.z.boolean(),
    isUserBlockEndpointsUpdated: zod_1.z.boolean(),
    shouldUsePersonaForIdVerification: zod_1.z.boolean(),
    shouldDisplaySessionManagement: zod_1.z.boolean(),
    isPasswordRequiredForAgingDown: zod_1.z.boolean(),
});
const Roblox_AccountInformation_Api_Models_PhoneResponse = zod_1.z.object({
    countryCode: zod_1.z.string(),
    prefix: zod_1.z.string(),
    phone: zod_1.z.string(),
    isVerified: zod_1.z.boolean(),
    verificationCodeLength: zod_1.z.number().int(),
    canBypassPasswordForPhoneUpdate: zod_1.z.boolean(),
});
const Roblox_AccountInformation_Api_Models_PhoneRequest = zod_1.z.object({
    countryCode: zod_1.z.string(),
    prefix: zod_1.z.string(),
    phone: zod_1.z.string(),
    password: zod_1.z.string(),
    verificationChannel: zod_1.z.string(),
});
const Roblox_Platform_UserPhoneNumberVerification_Models_PendingVerificationResponse = zod_1.z.object({
    verificationChannel: zod_1.z.string(),
    data: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_Models_PromotionChannelsResponse = zod_1.z.object({
    promotionChannelsVisibilityPrivacy: zod_1.z.string(),
    facebook: zod_1.z.string(),
    twitter: zod_1.z.string(),
    youtube: zod_1.z.string(),
    twitch: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_Models_PromotionChannelsRequest = zod_1.z.object({
    facebook: zod_1.z.string(),
    twitter: zod_1.z.string(),
    youtube: zod_1.z.string(),
    twitch: zod_1.z.string(),
    promotionChannelsVisibilityPrivacy: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_Models_StarCodeAffiliateResponse = zod_1.z.object({
    userId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    code: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_Models_StarCodeAffiliateRequest = zod_1.z.object({
    code: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_Models_PromotionChannelsByUserIdResponse = zod_1.z.object({
    facebook: zod_1.z.string(),
    twitter: zod_1.z.string(),
    youtube: zod_1.z.string(),
    twitch: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_RobloxBadgeResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    imageUrl: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_Models_VerifyEmailRequest = zod_1.z.object({
    ticket: zod_1.z.string(),
});
const Roblox_AccountInformation_Api_Models_VerifyEmailResponse = zod_1.z.object({
    verifiedUserHatAssetId: zod_1.z.number().int(),
});
const Roblox_AccountInformation_Api_Models_EmptyRequest = zod_1.z.object({});
const Roblox_AccountInformation_Api_Models_VerifyPhoneRequest = zod_1.z.object({
    code: zod_1.z.string(),
});
/**
 * @api GET https://accountinformation.roblox.com/v1/birthdate
 * @summary Get the user's birthdate
 */
exports.getBirthdate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/birthdate',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    response: Roblox_AccountInformation_Api_Models_BirthdateResponse,
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://accountinformation.roblox.com/v1/description
 * @summary Get the user's description
 */
exports.getDescription = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/description',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ description: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://accountinformation.roblox.com/v1/description
 * @summary Update the user's description
 * @param body The Roblox.AccountInformation.Api.Models.DescriptionRequest
 */
exports.postDescription = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/description',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ description: zod_1.z.string() }),
    response: zod_1.z.object({ description: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
        {
            status: 503,
            description: `3: This feature is currently disabled. Please try again later.`,
        },
    ],
});
/**
 * @api POST https://accountinformation.roblox.com/v1/email/verify
 * @summary Verify the user's email address from token
 * @param body Roblox.AccountInformation.Api.Models.VerifyEmailRequest
 */
exports.postEmailVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/email/verify',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ ticket: zod_1.z.string() }),
    response: zod_1.z.object({ verifiedUserHatAssetId: zod_1.z.number().int() }),
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://accountinformation.roblox.com/v1/gender
 * @summary Get the user's gender
 */
exports.getGender = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/gender',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ gender: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://accountinformation.roblox.com/v1/gender
 * @summary Update the user's gender
 * @param body The Roblox.AccountInformation.Api.Models.GenderRequest
 */
exports.postGender = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/gender',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ gender: zod_1.z.string() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: User not found.
6: The gender provided is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api GET https://accountinformation.roblox.com/v1/metadata
 * @summary Get the metadata
 */
exports.getMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/metadata',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    response: Roblox_AccountInformation_Api_Models_MetadataResponse,
    errors: [],
});
/**
 * @api GET https://accountinformation.roblox.com/v1/phone
 * @summary Get Verified Phone Number
 */
exports.getPhone = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/phone',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    response: Roblox_AccountInformation_Api_Models_PhoneResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api POST https://accountinformation.roblox.com/v1/phone
 * @summary Set Phone Number
 * @param body Roblox.AccountInformation.Api.Models.PhoneRequest
 */
exports.postPhone = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/phone',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AccountInformation_Api_Models_PhoneRequest,
    response: Roblox_Platform_UserPhoneNumberVerification_Models_PendingVerificationResponse,
    errors: [
        {
            status: 400,
            description: `2: Invalid Phone Number
3: Phone Number Already Associated
8: Invalid Phone Number Type`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
5: Incorrect Password
10: `,
        },
        {
            status: 429,
            description: `6: Flooded`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
        {
            status: 503,
            description: `1: This feature is currently disabled. Please try again later.`,
        },
    ],
});
/**
 * @api POST https://accountinformation.roblox.com/v1/phone/delete
 * @summary Delete Phone
 * @param body Roblox.AccountInformation.Api.Models.PhoneRequest
 */
exports.postPhoneDelete = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/phone/delete',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AccountInformation_Api_Models_PhoneRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
5: Incorrect Password`,
        },
        {
            status: 429,
            description: `6: Flooded`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
        {
            status: 503,
            description: `1: This feature is currently disabled. Please try again later.`,
        },
    ],
});
/**
 * @api POST https://accountinformation.roblox.com/v1/phone/resend
 * @summary Resend Phone code
 * @param body Roblox.AccountInformation.Api.Models.PhoneRequest
 */
exports.postPhoneResend = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/phone/resend',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({}).optional(),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `6: Flooded`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
        {
            status: 503,
            description: `1: This feature is currently disabled. Please try again later.`,
        },
    ],
});
/**
 * @api POST https://accountinformation.roblox.com/v1/phone/verify
 * @summary Verify Phone
 * @param body Roblox.AccountInformation.Api.Models.VerifyPhoneRequest
 */
exports.postPhoneVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/phone/verify',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ code: zod_1.z.string() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: Invalid Phone Number
3: Phone Number Already Associated
7: Invalid Code`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `6: Flooded`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
        {
            status: 503,
            description: `1: This feature is currently disabled. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://accountinformation.roblox.com/v1/promotion-channels
 * @summary Get the user's promotion channels
 * @param alwaysReturnUrls Whether all promotion channel links should be returned as full URLs.
 * @param filterLink Whether all promotion channel links should be filtered.
 * @param onlyShortenTwitter Whether all promotion channels links except for Twitter should be returned as full URLs. If false, all promotion channels will be shortened.
 */
exports.getPromotionChannels = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/promotion-channels',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        alwaysReturnUrls: {
            style: 'form',
            explode: true,
        },
        filterLink: {
            style: 'form',
            explode: true,
        },
        onlyShortenTwitter: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        alwaysReturnUrls: zod_1.z.boolean().optional(),
        filterLink: zod_1.z.boolean().optional(),
        onlyShortenTwitter: zod_1.z.boolean().optional().default(true),
    },
    response: Roblox_AccountInformation_Api_Models_PromotionChannelsResponse,
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://accountinformation.roblox.com/v1/promotion-channels
 * @summary Update the user's promotion channels
 * @param body The Roblox.AccountInformation.Api.Models.PromotionChannelsRequest
 */
exports.postPromotionChannels = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/promotion-channels',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AccountInformation_Api_Models_PromotionChannelsRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: The request was empty.
11: The Facebook profile url is invalid.
12: The Twitter handle is invalid.
13: The YouTube url is invalid.
14: The Twitch profile url is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
4: Only users who are over twelve years of age may edit social network channels.`,
        },
    ],
});
/**
 * @api GET https://accountinformation.roblox.com/v1/star-code-affiliates
 * @summary Gets a star code affiliate supporter for the authenticated user
 */
exports.getStarCodeAffiliates = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/star-code-affiliates',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    response: Roblox_AccountInformation_Api_Models_StarCodeAffiliateResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api POST https://accountinformation.roblox.com/v1/star-code-affiliates
 * @summary Adds a star code affiliate supporter for the authenticated user
 * @param body Roblox.AccountInformation.Api.Models.StarCodeAffiliateRequest
 */
exports.postStarCodeAffiliates = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/star-code-affiliates',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ code: zod_1.z.string() }),
    response: Roblox_AccountInformation_Api_Models_StarCodeAffiliateResponse,
    errors: [
        {
            status: 400,
            description: `1: The code was invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api DELETE https://accountinformation.roblox.com/v1/star-code-affiliates
 * @summary Removes the star code affiliate supporter for the authenticated user
 */
exports.deleteStarCodeAffiliates = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/star-code-affiliates',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api GET https://accountinformation.roblox.com/v1/users/:userId/promotion-channels
 * @summary Get promotion channels for a given user ID
 * @param userId The ID of the user to fetch the promotion channels for.
 * @param alwaysReturnUrls Whether all promotion channel links should be returned as full URLs.
 * @param filterLink Whether all promotion channel links should be filtered.
 */
exports.getUsersUseridPromotionChannels = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/promotion-channels',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        alwaysReturnUrls: {
            style: 'form',
            explode: true,
        },
        filterLink: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        alwaysReturnUrls: zod_1.z.boolean().optional(),
        filterLink: zod_1.z.boolean().optional(),
    },
    response: Roblox_AccountInformation_Api_Models_PromotionChannelsByUserIdResponse,
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
    ],
});
/**
 * @api GET https://accountinformation.roblox.com/v1/users/:userId/roblox-badges
 * @summary Returns a list of Roblox badges belonging to a user.
 * @param userId
 */
exports.getUsersUseridRobloxBadges = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/roblox-badges',
    baseUrl: 'https://accountinformation.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.array(Roblox_AccountInformation_Api_RobloxBadgeResponse),
    errors: [],
});
