// rozod/lib/endpoints/accountsettingsv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postTradeValue = exports.getTradeValue = exports.postTradePrivacy = exports.getTradePrivacy = exports.getThemesTypes = exports.patchThemesConsumertypeConsumerid = exports.getThemesConsumertypeConsumerid = exports.getEmails = exports.postEmailVerify = exports.patchEmail = exports.postEmail = exports.getEmail = exports.getAccountSettingsMetadata = exports.postAccountSettingsAccountCountry = exports.getAccountSettingsAccountCountry = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_AccountSettings_Api_Models_Response_UserAccountCountry = zod_1.z.object({
    countryName: zod_1.z.string(),
    subdivisionIso: zod_1.z.string(),
    localizedSubdivision: zod_1.z.string(),
    localizedName: zod_1.z.string(),
    countryId: zod_1.z.number().int(),
});
const Roblox_AccountSettings_Api_Models_Response_AccountCountrySettingsResponse = zod_1.z.object({
    value: Roblox_AccountSettings_Api_Models_Response_UserAccountCountry,
});
const Roblox_AccountSettings_Api_UpdateAccountCountryRequest = zod_1.z.object({
    targetCountryId: zod_1.z.number().int(),
});
const Roblox_AccountSettings_Api_Models_Response_UpdateAccountCountryResponse = zod_1.z.object({});
const Roblox_AccountSettings_Api_Models_AccountsSettingsMetadataModel = zod_1.z.object({
    IsAccountsRestrictionsSpamBugFixEnabled: zod_1.z.boolean(),
    MaximumParentalControlsMonthlySpendLimitInUSD: zod_1.z.number().int(),
    IsParentalMonthlyLimitInUIEnabled: zod_1.z.boolean(),
    IsParentalNotificationSettingsInUIEnabled: zod_1.z.boolean(),
    IsContentControlsEnabled: zod_1.z.boolean(),
});
const Roblox_AccountSettings_Api_EmailResponse = zod_1.z.object({
    emailAddress: zod_1.z.string(),
    verified: zod_1.z.boolean(),
    canBypassPasswordForEmailUpdate: zod_1.z.boolean(),
});
const Roblox_AccountSettings_Api_UpdateEmailRequest = zod_1.z.object({
    password: zod_1.z.string(),
    emailAddress: zod_1.z.string(),
    skipVerificationEmail: zod_1.z.boolean(),
    isAdsAccount: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_AccountSettings_Api_CurrentEmailsResponse = zod_1.z.object({
    verifiedEmail: zod_1.z.string(),
    pendingEmail: zod_1.z.string(),
});
const Roblox_AccountSettings_Api_ThemeConfigurationResponse = zod_1.z.object({
    themeType: zod_1.z.string(),
});
const Roblox_AccountSettings_Api_ThemeConfigurationRequest = zod_1.z.object({
    themeType: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_System_String_ = zod_1.z.object({
    data: zod_1.z.array(zod_1.z.string()),
});
const Roblox_AccountSettings_Api_TradePrivacyResponse = zod_1.z.object({
    tradePrivacy: zod_1.z.string(),
});
const Roblox_AccountSettings_Api_UpdateTradePrivacyRequest = zod_1.z.object({
    tradePrivacy: zod_1.z.enum(['Undefined', 'Disabled', 'NoOne', 'Friends', 'TopFriends', 'Following', 'Followers', 'All']),
});
const Roblox_AccountSettings_Api_TradePrivacyUpdateResponse = zod_1.z.object({
    tradePrivacy: zod_1.z.enum(['Undefined', 'Disabled', 'NoOne', 'Friends', 'TopFriends', 'Following', 'Followers', 'All']),
    inventoryPrivacy: zod_1.z.enum([
        'NoOne',
        'Friends',
        'FriendsAndFollowing',
        'FriendsFollowingAndFollowers',
        'AllAuthenticatedUsers',
        'AllUsers',
    ]),
    privacySettingResponse: zod_1.z.enum(['Success', 'Conflict']),
});
const Roblox_AccountSettings_Api_TradeValueResponse = zod_1.z.object({
    tradeValue: zod_1.z.string(),
});
const Roblox_AccountSettings_Api_TradeValueRequest = zod_1.z.object({
    tradeValue: zod_1.z.enum(['Undefined', 'None', 'Low', 'Medium', 'High']),
});
const Roblox_AccountSettings_Api_SendVerifyEmailRequest = zod_1.z.object({
    freeItem: zod_1.z.boolean(),
    isAdsAccount: zod_1.z.boolean(),
});
/**
 * @api GET https://accountsettings.roblox.com/v1/account/settings/account-country
 * @summary Get a user's current account country setting.
 */
exports.getAccountSettingsAccountCountry = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/account/settings/account-country',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    response: Roblox_AccountSettings_Api_Models_Response_AccountCountrySettingsResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://accountsettings.roblox.com/v1/account/settings/account-country
 * @summary Updates the user's account country.
 * @param body
 */
exports.postAccountSettingsAccountCountry = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/account/settings/account-country',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ targetCountryId: zod_1.z.number().int() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: InvalidRequest`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: OperationNotPermitted`,
        },
        {
            status: 404,
            description: `2: OperationNotPermitted`,
        },
        {
            status: 500,
            description: `0: Unknown`,
        },
    ],
});
/**
 * @api GET https://accountsettings.roblox.com/v1/account/settings/metadata
 * @summary Returns metadata used by the account settings page
 */
exports.getAccountSettingsMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/account/settings/metadata',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    response: Roblox_AccountSettings_Api_Models_AccountsSettingsMetadataModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://accountsettings.roblox.com/v1/email
 * @summary Gets the authenticated user's email address and verified status
 */
exports.getEmail = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/email',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    response: Roblox_AccountSettings_Api_EmailResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://accountsettings.roblox.com/v1/email
 * @summary Updates the authenticated user's email address
 * @param body The request body.
 */
exports.postEmail = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/email',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AccountSettings_Api_UpdateEmailRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `8: Password is incorrect.
9: Invalid email address.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: This feature is currently disabled. Please try again later.
3: There are too many accounts associated with this email address.
11: You must be on the Corporate network to log in.`,
        },
        {
            status: 409,
            description: `4: This is already the current email.`,
        },
        {
            status: 429,
            description: `6: Too many attempts to update email. Please try again later.
7: Too many attempts to send verification email. Please try again later.`,
        },
        {
            status: 503,
            description: `2: This feature is currently disabled. Please try again later.`,
        },
    ],
});
/**
 * @api PATCH https://accountsettings.roblox.com/v1/email
 * @summary Updates the authenticated user's email address
 * @param body The request body.
 */
exports.patchEmail = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/email',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AccountSettings_Api_UpdateEmailRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `8: Password is incorrect.
9: Invalid email address.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: This feature is currently disabled. Please try again later.
3: There are too many accounts associated with this email address.
11: You must be on the Corporate network to log in.`,
        },
        {
            status: 409,
            description: `4: This is already the current email.`,
        },
        {
            status: 429,
            description: `6: Too many attempts to update email. Please try again later.
7: Too many attempts to send verification email. Please try again later.`,
        },
        {
            status: 503,
            description: `2: This feature is currently disabled. Please try again later.`,
        },
    ],
});
/**
 * @api POST https://accountsettings.roblox.com/v1/email/verify
 * @summary Send verify email to the authenticated user's email address
 * @param body The request body.
 */
exports.postEmailVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/email/verify',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AccountSettings_Api_SendVerifyEmailRequest.optional(),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `10: No email address is associated with the account.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: This feature is currently disabled. Please try again later.
11: You must be on the Corporate network to log in.`,
        },
        {
            status: 409,
            description: `5: The email is already verified.`,
        },
        {
            status: 429,
            description: `7: Too many attempts to send verification email. Please try again later.`,
        },
        {
            status: 503,
            description: `2: This feature is currently disabled. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://accountsettings.roblox.com/v1/emails
 * @summary Gets the authenticated user's verified email and pending (unverified) email if one exists
 */
exports.getEmails = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/emails',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    response: Roblox_AccountSettings_Api_CurrentEmailsResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://accountsettings.roblox.com/v1/themes/:consumerType/:consumerId
 * @summary returns the theme type for a specific consumer.
 * @param consumerType The consumer type
 * @param consumerId The consumer's theme configuration to get. If the consumerType is User always return the AuthenticatedUser's theme type.
 */
exports.getThemesConsumertypeConsumerid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/themes/:consumerType/:consumerId',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        consumerType: {
            style: 'simple',
        },
        consumerId: {
            style: 'simple',
        },
    },
    parameters: {
        consumerType: zod_1.z.literal(1),
        consumerId: zod_1.z.string(),
    },
    response: zod_1.z.object({ themeType: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `3: Invalid consumer type.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api PATCH https://accountsettings.roblox.com/v1/themes/:consumerType/:consumerId
 * @summary Modify the theme type for consumer.
 * @param body An Roblox.AccountSettings.Api.ThemeConfigurationRequest.
 * @param consumerType The consumer type
 * @param consumerId The consumer's theme configuration to modify. If the consumerType is User always modify the AuthenticatedUser's theme type.
 */
exports.patchThemesConsumertypeConsumerid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/themes/:consumerType/:consumerId',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        consumerType: {
            style: 'simple',
        },
        consumerId: {
            style: 'simple',
        },
    },
    parameters: {
        consumerType: zod_1.z.literal(1),
        consumerId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ themeType: zod_1.z.string() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: Invalid theme type.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://accountsettings.roblox.com/v1/themes/types
 * @summary returns all the enabled theme types.
 */
exports.getThemesTypes = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/themes/types',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_System_String_,
    errors: [],
});
/**
 * @api GET https://accountsettings.roblox.com/v1/trade-privacy
 * @summary Get a user's trade privacy setting
 */
exports.getTradePrivacy = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/trade-privacy',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ tradePrivacy: zod_1.z.string() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://accountsettings.roblox.com/v1/trade-privacy
 * @summary Updates a user's trade privacy setting
 * @param body
 */
exports.postTradePrivacy = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/trade-privacy',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AccountSettings_Api_UpdateTradePrivacyRequest,
    response: Roblox_AccountSettings_Api_TradePrivacyUpdateResponse,
    errors: [
        {
            status: 400,
            description: `Roblox.AccountSettings.Api.ResponseEnums.TradeSettingsErrors.InvalidTradePrivacy`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Roblox.AccountSettings.Api.ResponseEnums.TradeSettingsErrors.AccountLocked
            OR
            Roblox.AccountSettings.Api.ResponseEnums.TradeSettingsErrors.UserCannotTrade
0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://accountsettings.roblox.com/v1/trade-value
 * @summary Get a user's trade quality filter setting
 */
exports.getTradeValue = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/trade-value',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ tradeValue: zod_1.z.string() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://accountsettings.roblox.com/v1/trade-value
 * @summary Updates a user's trade quality filter setting
 * @param body
 */
exports.postTradeValue = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/trade-value',
    baseUrl: 'https://accountsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AccountSettings_Api_TradeValueRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `Roblox.AccountSettings.Api.ResponseEnums.TradeSettingsErrors.InvalidTradeValue`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Roblox.AccountSettings.Api.ResponseEnums.TradeSettingsErrors.AccountLocked
            OR
            Roblox.AccountSettings.Api.ResponseEnums.TradeSettingsErrors.UserCannotTrade
0: Token Validation Failed`,
        },
    ],
});
