// rozod/lib/endpoints/adconfigurationv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSponsoredGamesUniverses = exports.postSponsoredGamesStop = exports.postSponsoredGamesCreate = exports.getSponsoredGames = exports.postSponsoredCampaignsStop = exports.getSponsoredCampaignsMultiGetCanUserSponsor = exports.postSponsoredCampaignsEligibleCampaignTargets = exports.getSponsoredCampaignsEligibleAssetTypeIds = exports.postSponsoredCampaignsCreate = exports.getSponsoredCampaigns = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_AdConfiguration_Api_SponsoredCampaignModel = zod_1.z.object({
    adId: zod_1.z.number().int(),
    adSetId: zod_1.z.number().int(),
    adName: zod_1.z.string(),
    adStatus: zod_1.z.enum(['Undefined', 'Draft', 'PendingApproval', 'Approved', 'Moderated']),
    creativeType: zod_1.z.enum(['Undefined', 'Image', 'Video']),
    creativeTargetId: zod_1.z.number().int(),
    bidAmountInRobux: zod_1.z.number().int(),
    budgetInRobux: zod_1.z.number().int(),
    adSetStatus: zod_1.z.enum(['Undefined', 'Draft', 'Scheduled', 'Running', 'Paused', 'Stopped', 'Completed']),
    startDate: zod_1.z.string().datetime({ offset: true }),
    endDate: zod_1.z.string().datetime({ offset: true }),
    targetGender: zod_1.z.enum(['Undefined', 'Male', 'Female']),
    targetAgeBracket: zod_1.z.enum(['Undefined', 'AgeUnder13', 'Age13OrOver', 'Age13To16', 'Age17OrOver']),
    targetDeviceType: zod_1.z.enum(['Undefined', 'Computer', 'Phone', 'Tablet', 'Console', 'VR']),
    campaignTargetType: zod_1.z.enum(['Undefined', 'Universe', 'Asset', 'ImmersiveAd']),
    campaignTargetId: zod_1.z.number().int(),
    totalSpendInRobux: zod_1.z.number().int(),
    totalImpressions: zod_1.z.number().int(),
    totalClicks: zod_1.z.number().int(),
    totalConversions: zod_1.z.number().int(),
    impressionConversions: zod_1.z.number().int(),
    clickConversions: zod_1.z.number().int(),
});
const Roblox_AdConfiguration_Api_GetSponsoredCampaignsResponse = zod_1.z.object({
    sponsoredCampaigns: zod_1.z.array(Roblox_AdConfiguration_Api_SponsoredCampaignModel),
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
});
const Roblox_AdConfiguration_Api_SponsoredGameV2Model = zod_1.z.object({
    adId: zod_1.z.number().int(),
    adSetId: zod_1.z.number().int(),
    adName: zod_1.z.string(),
    adStatus: zod_1.z.enum(['Undefined', 'Draft', 'PendingApproval', 'Approved', 'Moderated']),
    creativeType: zod_1.z.enum(['Undefined', 'Image', 'Video']),
    creativeTargetId: zod_1.z.number().int(),
    creativeUrl: zod_1.z.string(),
    bidAmountInRobux: zod_1.z.number().int(),
    budgetInRobux: zod_1.z.number().int(),
    adSetStatus: zod_1.z.enum(['Undefined', 'Draft', 'Scheduled', 'Running', 'Paused', 'Stopped', 'Completed']),
    startDate: zod_1.z.string().datetime({ offset: true }),
    endDate: zod_1.z.string().datetime({ offset: true }),
    targetGender: zod_1.z.enum(['Undefined', 'Male', 'Female']),
    targetAgeBracket: zod_1.z.enum(['Undefined', 'AgeUnder13', 'Age13OrOver', 'Age13To16', 'Age17OrOver']),
    targetDeviceType: zod_1.z.enum(['Undefined', 'Computer', 'Phone', 'Tablet', 'Console', 'VR']),
    campaignTargetType: zod_1.z.enum(['Undefined', 'Universe', 'Asset', 'ImmersiveAd']),
    campaignTargetId: zod_1.z.number().int(),
    totalSpendInRobux: zod_1.z.number().int(),
    totalImpressions: zod_1.z.number().int(),
    totalClicks: zod_1.z.number().int(),
    totalConversions: zod_1.z.number().int(),
    impressionConversions: zod_1.z.number().int(),
    clickConversions: zod_1.z.number().int(),
});
const Roblox_AdConfiguration_Api_GetSponsoredGamesResponse = zod_1.z.object({
    sponsoredGames: zod_1.z.array(Roblox_AdConfiguration_Api_SponsoredGameV2Model),
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
});
const Roblox_AdConfiguration_Api_Models_UniverseModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
});
const Roblox_AdConfiguration_Api_Models_GetRecentAdsRankedUniversesResponse = zod_1.z.object({
    universes: zod_1.z.array(Roblox_AdConfiguration_Api_Models_UniverseModel),
});
const Roblox_AdConfiguration_Api_CreativeModel = zod_1.z.object({
    creativeId: zod_1.z.number().int(),
    creativeType: zod_1.z.enum(['Undefined', 'Image', 'Video']),
});
const Roblox_AdConfiguration_Api_Models_CreateSponsoredCampaignRequest = zod_1.z.object({
    campaignTargetId: zod_1.z.number().int(),
    campaignTargetType: zod_1.z.enum(['Undefined', 'Universe', 'Asset', 'ImmersiveAd']),
    targetGender: zod_1.z.enum(['Undefined', 'Male', 'Female']),
    targetAgeBracket: zod_1.z.enum(['Undefined', 'AgeUnder13', 'Age13OrOver', 'Age13To16', 'Age17OrOver']),
    startDate: zod_1.z.string().datetime({ offset: true }),
    endDate: zod_1.z.string().datetime({ offset: true }),
    targetDeviceType: zod_1.z.enum(['Undefined', 'Computer', 'Phone', 'Tablet', 'Console', 'VR']),
    campaignName: zod_1.z.string(),
    dailyBidAmountInRobux: zod_1.z.number().int(),
    placementLocation: zod_1.z.enum([
        'Undefined',
        'GameSort',
        'AvatarShop',
        'ItemDetails',
        'HomePage',
        'Billboard300x250',
        'Billboard600x300',
        'Billboard300x600',
    ]),
    creativeModel: Roblox_AdConfiguration_Api_CreativeModel,
});
const Roblox_AdConfiguration_Api_Models_GetEligibleCampaignTargetsRequest = zod_1.z.object({
    campaignTargetTypes: zod_1.z.array(zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)])),
    groupId: zod_1.z.number().int(),
});
const Roblox_AdConfiguration_Api_Models_CampaignTargetModel = zod_1.z.object({
    campaignTargetType: zod_1.z.enum(['Undefined', 'Universe', 'Asset', 'ImmersiveAd']),
    campaignTargetId: zod_1.z.number().int(),
    name: zod_1.z.string(),
});
const Roblox_AdConfiguration_Api_Models_GetCampaignTargetsResponse = zod_1.z.object({
    campaignTargetModels: zod_1.z.array(Roblox_AdConfiguration_Api_Models_CampaignTargetModel),
});
const Roblox_AdConfiguration_Api_Models_StopSponsoredCampaignRequest = zod_1.z.object({ adSetId: zod_1.z.number().int() });
const Roblox_AdConfiguration_Api_CreateSponsoredGameV2Request = zod_1.z.object({
    universeId: zod_1.z.number().int(),
    targetGender: zod_1.z.enum(['Undefined', 'Male', 'Female']),
    targetAgeBracket: zod_1.z.enum(['Undefined', 'AgeUnder13', 'Age13OrOver', 'Age13To16', 'Age17OrOver']),
    budgetInRobux: zod_1.z.number().int(),
    startDate: zod_1.z.string().datetime({ offset: true }),
    endDate: zod_1.z.string().datetime({ offset: true }),
    targetDeviceType: zod_1.z.enum(['Undefined', 'Computer', 'Phone', 'Tablet', 'Console', 'VR']),
    adName: zod_1.z.string(),
    bidAmountInRobux: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_AdConfiguration_Api_Models_StopSponsoredGameV2Request = zod_1.z.object({
    adSetId: zod_1.z.number().int(),
});
/**
 * @api GET https://adconfiguration.roblox.com/v2/sponsored-campaigns
 * @summary Gets a page of Roblox.AdConfiguration.Api.SponsoredCampaignModel with specified input parameters.
 * @param campaignTargetType The campaign target type enum value
 * @param campaignTargetId The id of the campaign target
 * @param includeReportingStats Indicates whether to include reporting stats in the response
 * @param isArchived Indicates whether to retrieve archived ads
 * @param pageCursor The cursor of the page to retrieve. If empty, fetches the first page
 */
exports.getSponsoredCampaigns = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/sponsored-campaigns',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        campaignTargetType: {
            style: 'form',
            explode: true,
        },
        campaignTargetId: {
            style: 'form',
            explode: true,
        },
        includeReportingStats: {
            style: 'form',
            explode: true,
        },
        isArchived: {
            style: 'form',
            explode: true,
        },
        pageCursor: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        campaignTargetType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
        campaignTargetId: zod_1.z.number().int(),
        includeReportingStats: zod_1.z.boolean().optional(),
        isArchived: zod_1.z.boolean().optional(),
        pageCursor: zod_1.z.string().optional(),
    },
    response: Roblox_AdConfiguration_Api_GetSponsoredCampaignsResponse,
    errors: [
        {
            status: 400,
            description: `22: Invalid campaign target ID.
23: Invalid campaign target type.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 503,
            description: `1: This feature is disabled.`,
        },
    ],
});
/**
 * @api POST https://adconfiguration.roblox.com/v2/sponsored-campaigns/create
 * @summary Creates a complete ad. Including ad campaign, ad set, escrow, and the ad.
Currently intended for creation of sponsorships only.
 * @param body Roblox.AdConfiguration.Api.Models.CreateSponsoredCampaignRequest
 */
exports.postSponsoredCampaignsCreate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/sponsored-campaigns/create',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AdConfiguration_Api_Models_CreateSponsoredCampaignRequest,
    response: zod_1.z.number().int(),
    errors: [
        {
            status: 400,
            description: `2: Daily budget is lower than minimum allowed.
3: Total budget must be greater than 0.
4: Ad name cannot be empty.
5: Start date must not be a future date.
6: End date must be a future date.
7: Start date must be earlier than end date.
8: Total budget does not match daily spend and number of days being scheduled
9: Cannot load the universe for the specified universe id.
11: Invalid target age bracket.
12: Invalid target gender.
13: Invalid target device type.
14: Invalid ad set id.
15: Ad name cannot exceed 255 characters.
16: Insufficient Robux balance.
17: Name has already been taken.
18: Daily budget is higher than maximum allowed.
19: Invalid group id.
20: Number of days scheduled exceeded maximum days allowed.
21: Your experience is currently not eligible for advertising.
22: Invalid campaign target ID.
23: Invalid campaign target type.
24: Invalid creative ID.
25: Invalid creative type.
28: Total budget must be campaign duration * daily bid
29: The target is not eligible for new campaigns
30: Invalid user ID`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
10: Insufficient permissions.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `0: An unknown error occurred.
31: Internal server error`,
        },
        {
            status: 503,
            description: `1: This feature is disabled.`,
        },
    ],
});
/**
 * @api GET https://adconfiguration.roblox.com/v2/sponsored-campaigns/eligible-asset-type-ids
 * @summary Get all asset type IDs that are eligible to be sponsored.
 */
exports.getSponsoredCampaignsEligibleAssetTypeIds = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/sponsored-campaigns/eligible-asset-type-ids',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.array(zod_1.z.number()),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `10: Insufficient permissions.`,
        },
    ],
});
/**
 * @api POST https://adconfiguration.roblox.com/v2/sponsored-campaigns/eligible-campaign-targets
 * @summary Returns a collection of Roblox.AdConfiguration.Api.Models.CampaignTargetModel that the user is authorized to sponsor, ordered by most recently advertised
 * @param body Roblox.AdConfiguration.Api.Models.GetEligibleCampaignTargetsRequest
 */
exports.postSponsoredCampaignsEligibleCampaignTargets = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/sponsored-campaigns/eligible-campaign-targets',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AdConfiguration_Api_Models_GetEligibleCampaignTargetsRequest,
    response: Roblox_AdConfiguration_Api_Models_GetCampaignTargetsResponse,
    errors: [
        {
            status: 400,
            description: `19: Invalid group id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
10: Insufficient permissions.`,
        },
    ],
});
/**
 * @api GET https://adconfiguration.roblox.com/v2/sponsored-campaigns/multi-get-can-user-sponsor
 * @summary Checks whether the targets are eligible for sponsorship, and
if the user is authorized to sponsor the targets.
 * @param campaignTargetType Ads.Management.Service.CampaignTargetType.
 * @param campaignTargetIds The IDs of the campaign targets.
 */
exports.getSponsoredCampaignsMultiGetCanUserSponsor = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/sponsored-campaigns/multi-get-can-user-sponsor',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        campaignTargetType: {
            style: 'form',
            explode: true,
        },
        campaignTargetIds: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        campaignTargetType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
        campaignTargetIds: zod_1.z.array(zod_1.z.number()),
    },
    response: zod_1.z.boolean(),
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 500,
            description: `Server Error`,
        },
    ],
});
/**
 * @api POST https://adconfiguration.roblox.com/v2/sponsored-campaigns/stop
 * @summary Stops a sponsored campaign / ad (ad set) from running. Initiated by a user.
 * @param body Roblox.AdConfiguration.Api.Models.StopSponsoredCampaignRequest
 */
exports.postSponsoredCampaignsStop = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/sponsored-campaigns/stop',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ adSetId: zod_1.z.number().int() }),
    response: zod_1.z.number().int(),
    errors: [
        {
            status: 400,
            description: `10: Insufficient permissions.
14: Invalid ad set id.
31: Internal server error`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `1: This feature is disabled.`,
        },
    ],
});
/**
 * @api GET https://adconfiguration.roblox.com/v2/sponsored-games
 * @summary Gets a page of Roblox.AdConfiguration.Api.SponsoredGameV2Model with specified input parameters.
 * @param universeId The universe id of the ad campaign.
 * @param includeReportingStats Indicates whether to include reporting stats in the response.
 * @param isArchived Indicates whether to retrieve archived ads.
 * @param pageCursor The cursor of the page to retrieve.
 */
exports.getSponsoredGames = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/sponsored-games',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'form',
            explode: true,
        },
        includeReportingStats: {
            style: 'form',
            explode: true,
        },
        isArchived: {
            style: 'form',
            explode: true,
        },
        pageCursor: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        includeReportingStats: zod_1.z.boolean().optional(),
        isArchived: zod_1.z.boolean().optional(),
        pageCursor: zod_1.z.string().optional(),
    },
    response: Roblox_AdConfiguration_Api_GetSponsoredGamesResponse,
    errors: [
        {
            status: 400,
            description: `9: Cannot load the universe for the specified universe id.
9: Cannot load the universe for the specified universe id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 503,
            description: `1: This feature is disabled.`,
        },
    ],
});
/**
 * @api POST https://adconfiguration.roblox.com/v2/sponsored-games/create
 * @summary Creates a new sponsored game ad with specified input parameters.
 * @param body Roblox.AdConfiguration.Api.CreateSponsoredGameV2Request
 */
exports.postSponsoredGamesCreate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/sponsored-games/create',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_AdConfiguration_Api_CreateSponsoredGameV2Request,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: Daily budget is lower than minimum allowed.
3: Total budget must be greater than 0.
4: Ad name cannot be empty.
5: Start date must not be a future date.
6: End date must be a future date.
7: Start date must be earlier than end date.
8: Total budget does not match daily spend and number of days being scheduled
9: Cannot load the universe for the specified universe id.
11: Invalid target age bracket.
12: Invalid target gender.
13: Invalid target device type.
14: Invalid ad set id.
15: Ad name cannot exceed 255 characters.
16: Insufficient Robux balance.
17: Name has already been taken.
18: Daily budget is higher than maximum allowed.
19: Invalid group id.
20: Number of days scheduled exceeded maximum days allowed.
21: Your experience is currently not eligible for advertising.
22: Invalid campaign target ID.
23: Invalid campaign target type.
24: Invalid creative ID.
25: Invalid creative type.
28: Total budget must be campaign duration * daily bid
29: The target is not eligible for new campaigns
30: Invalid user ID`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
10: Insufficient permissions.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `0: An unknown error occurred.
31: Internal server error`,
        },
        {
            status: 503,
            description: `1: This feature is disabled.`,
        },
    ],
});
/**
 * @api POST https://adconfiguration.roblox.com/v2/sponsored-games/stop
 * @summary To stop a sponsored-game ad (ad set) from running, initiated by a user.
 * @param body Roblox.AdConfiguration.Api.Models.StopSponsoredGameV2Request
 */
exports.postSponsoredGamesStop = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/sponsored-games/stop',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ adSetId: zod_1.z.number().int() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `14: Invalid ad set id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
10: Insufficient permissions.`,
        },
        {
            status: 503,
            description: `1: This feature is disabled.`,
        },
    ],
});
/**
 * @api GET https://adconfiguration.roblox.com/v2/sponsored-games/universes
 * @summary Gets a list of universes for the authenticated user, or the given group, ordered by most recently created sponsored game ads.
 * @param groupId The group id, if applicable.
 */
exports.getSponsoredGamesUniverses = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/sponsored-games/universes',
    baseUrl: 'https://adconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int().optional(),
    },
    response: Roblox_AdConfiguration_Api_Models_GetRecentAdsRankedUniversesResponse,
    errors: [
        {
            status: 400,
            description: `19: Invalid group id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `10: Insufficient permissions.`,
        },
    ],
});
