// rozod/lib/endpoints/assetdeliveryv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getOpencloudAssetidAssetidVersionVersionnumber = exports.getOpencloudAssetidAssetid = exports.getMarassethashMarassethashMarchecksumMarchecksum = exports.postAssetsBatch = exports.getAssetidAssetidVersionVersionnumber = exports.getAssetidAssetid = exports.getAsset = exports.getAliasAlias = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_Assets_IAssetItemError = zod_1.z.object({
    Code: zod_1.z.number().int(),
    Message: zod_1.z.string(),
    CustomErrorCode: zod_1.z.enum([
        'UnknownError',
        'NoPermissionToAsset',
        'AssetPermissionCheckFailed',
        'NotAuthorizedForAgeRecommendation',
        'AgeRecommendationCheckFailed',
        'InvalidPlaceRequestFromGameServer',
        'BlockedAssetTypeRequestedFromInsertService',
        'BlockedAssetTypeRequestedFromGameServer',
        'AssetTypeMismatch',
        'MissingAssetTypeInRequestHeader',
        'AssetNotTrustedForPlace',
        'NoAuthentication',
        'AssetContentRepresentationBlockedDueToModeration',
        'AssetNotFound',
        'AssetVersionNotFound',
        'AssetContentRepresentationNotFound',
        'BlockedByAgeGeoRestriction',
        'BlockedAssetTypeRequestedFromNonGameServer',
        'AssetPendingReview',
        'NotApprovedForRequestor',
        'NotApprovedByContentCompliance',
        'AssetContentRepresentationGenerating',
    ]),
});
const Roblox_Web_Assets_AssetContentRepresentationSpecifier = zod_1.z.object({
    format: zod_1.z.string(),
    majorVersion: zod_1.z.string(),
    fidelity: zod_1.z.string(),
    skipGenerationIfNotExist: zod_1.z.boolean(),
});
const Roblox_AssetDelivery_Api_AssetMetadata = zod_1.z.object({
    metadataType: zod_1.z.literal('UncompressedSize'),
    value: zod_1.z.string(),
});
const Roblox_Web_Assets_AssetResponseItemV1 = zod_1.z.object({
    location: zod_1.z.string(),
    errors: zod_1.z.array(Roblox_Web_Assets_IAssetItemError),
    requestId: zod_1.z.string(),
    isArchived: zod_1.z.boolean(),
    assetTypeId: zod_1.z.number().int(),
    contentRepresentationSpecifier: Roblox_Web_Assets_AssetContentRepresentationSpecifier,
    assetMetadatas: zod_1.z.array(Roblox_AssetDelivery_Api_AssetMetadata),
    isRecordable: zod_1.z.boolean(),
});
const Roblox_Web_Assets_BatchAssetRequestItem = zod_1.z.object({
    assetName: zod_1.z.string(),
    assetType: zod_1.z.string(),
    clientInsert: zod_1.z.boolean(),
    placeId: zod_1.z.number().int(),
    requestId: zod_1.z.string(),
    scriptInsert: zod_1.z.boolean(),
    serverPlaceId: zod_1.z.number().int(),
    universeId: zod_1.z.number().int(),
    accept: zod_1.z.string(),
    encoding: zod_1.z.string(),
    hash: zod_1.z.string(),
    userAssetId: zod_1.z.number().int(),
    assetId: zod_1.z.number().int(),
    version: zod_1.z.number().int(),
    assetVersionId: zod_1.z.number().int(),
    modulePlaceId: zod_1.z.number().int(),
    assetFormat: zod_1.z.string(),
    'roblox-assetFormat': zod_1.z.string(),
    assetResolutionMode: zod_1.z.string(),
    accessContext: zod_1.z.string(),
    contentRepresentationPriorityList: zod_1.z.string(),
    doNotFallbackToBaselineRepresentation: zod_1.z.boolean(),
});
/**
 * @api GET https://assetdelivery.roblox.com/v1/alias/:alias
 * @summary Retrieves an asset by its alias (universeID/name)
 * @param alias The alias of the asset to retrieve.
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param skipSigningScripts
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param expectedAssetType
 * @param accessContext
 */
exports.getAliasAlias = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/alias/:alias',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        alias: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        alias: zod_1.z.string().regex(/^[0-9]+\/.+/),
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV1,
    errors: [],
});
/**
 * @api GET https://assetdelivery.roblox.com/v1/asset
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param id
 * @param userAssetId
 * @param assetVersionId
 * @param version
 * @param universeId
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param assetName
 * @param hash
 * @param marAssetHash
 * @param marCheckSum
 * @param expectedAssetType
 * @param skipSigningScripts
 * @param permissionContext
 * @param doNotFallbackToBaselineRepresentation
 * @param contentRepresentationPriorityList
 * @param assetResolutionMode
 * @param accessContext
 */
exports.getAsset = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/asset',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        id: {
            style: 'form',
            explode: true,
        },
        userAssetId: {
            style: 'form',
            explode: true,
        },
        assetVersionId: {
            style: 'form',
            explode: true,
        },
        version: {
            style: 'form',
            explode: true,
        },
        universeId: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        assetName: {
            style: 'form',
            explode: true,
        },
        hash: {
            style: 'form',
            explode: true,
        },
        marAssetHash: {
            style: 'form',
            explode: true,
        },
        marCheckSum: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        permissionContext: {
            style: 'form',
            explode: true,
        },
        doNotFallbackToBaselineRepresentation: {
            style: 'form',
            explode: true,
        },
        contentRepresentationPriorityList: {
            style: 'form',
            explode: true,
        },
        assetResolutionMode: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        id: zod_1.z.number().int().optional(),
        userAssetId: zod_1.z.number().int().optional(),
        assetVersionId: zod_1.z.number().int().optional(),
        version: zod_1.z.number().int().optional(),
        universeId: zod_1.z.number().int().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.string().optional(),
        assetName: zod_1.z.string().optional(),
        hash: zod_1.z.string().optional(),
        marAssetHash: zod_1.z.string().optional(),
        marCheckSum: zod_1.z.string().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        permissionContext: zod_1.z.string().optional(),
        doNotFallbackToBaselineRepresentation: zod_1.z.boolean().optional(),
        contentRepresentationPriorityList: zod_1.z.string().optional(),
        assetResolutionMode: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://assetdelivery.roblox.com/v1/assetId/:assetId
 * @summary Retrieves an asset by its ID
 * @param assetId The ID of the asset to retrieve.
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param skipSigningScripts
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param expectedAssetType
 * @param doNotFallbackToBaselineRepresentation
 * @param contentRepresentationPriorityList
 * @param accessContext
 */
exports.getAssetidAssetid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/assetId/:assetId',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        doNotFallbackToBaselineRepresentation: {
            style: 'form',
            explode: true,
        },
        contentRepresentationPriorityList: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        doNotFallbackToBaselineRepresentation: zod_1.z.boolean().optional(),
        contentRepresentationPriorityList: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV1,
    errors: [],
});
/**
 * @api GET https://assetdelivery.roblox.com/v1/assetId/:assetId/version/:versionNumber
 * @summary Retrieves an asset by its ID and version number.
 * @param assetId The ID of the asset to retrieve.
 * @param versionNumber The version of the asset to retrieve.
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param skipSigningScripts
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param expectedAssetType
 * @param doNotFallbackToBaselineRepresentation
 * @param contentRepresentationPriorityList
 * @param accessContext
 */
exports.getAssetidAssetidVersionVersionnumber = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/assetId/:assetId/version/:versionNumber',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
        versionNumber: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        doNotFallbackToBaselineRepresentation: {
            style: 'form',
            explode: true,
        },
        contentRepresentationPriorityList: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
        versionNumber: zod_1.z.number().int(),
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        doNotFallbackToBaselineRepresentation: zod_1.z.boolean().optional(),
        contentRepresentationPriorityList: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV1,
    errors: [],
});
/**
 * @api POST https://assetdelivery.roblox.com/v1/assets/batch
 * @param body
 * @param Roblox-Place-Id
 * @param Accept
 * @param Roblox-Browser-Asset-Request
 */
exports.postAssetsBatch = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/assets/batch',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        'Roblox-Place-Id': {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        'Roblox-Browser-Asset-Request': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int(),
        Accept: zod_1.z.string(),
        'Roblox-Browser-Asset-Request': zod_1.z.string(),
    },
    body: zod_1.z.array(Roblox_Web_Assets_BatchAssetRequestItem),
    response: zod_1.z.array(Roblox_Web_Assets_AssetResponseItemV1),
    errors: [],
});
/**
 * @api GET https://assetdelivery.roblox.com/v1/marAssetHash/:marAssetHash/marCheckSum/:marCheckSum
 * @summary Retrieves an asset by its mar (moderation agnostic) hash and mar (moderation agnostic) checksum.
 * @param marAssetHash The mar (moderation agnostic) hash of the asset to retrieve.
 * @param marCheckSum The mar (moderation agnostic) checksum of the asset to retrieve.
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param skipSigningScripts
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param expectedAssetType
 */
exports.getMarassethashMarassethashMarchecksumMarchecksum = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/marAssetHash/:marAssetHash/marCheckSum/:marCheckSum',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        marAssetHash: {
            style: 'simple',
        },
        marCheckSum: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        marAssetHash: zod_1.z.string(),
        marCheckSum: zod_1.z.string(),
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV1,
    errors: [
        {
            status: 400,
            description: `2: invalid server request
3: Encoding cannot be empty`,
        },
        {
            status: 404,
            description: `5: Asset hash cannot be empty`,
        },
    ],
});
/**
 * @api GET https://assetdelivery.roblox.com/v1/openCloud/assetId/:assetId
 * @summary Retrieves an asset by its ID with OpenCloud auth.
 * @param assetId The ID of the asset to retrieve.
 * @param Accept-Encoding The Accept-Encoding header value specifying compression formats (e.g., "gzip, deflate"). Defaults to "gzip, deflate" if not provided.
 * @param Roblox-Place-Id The Roblox-Place-Id header value identifying the place making the request.
 * @param AssetType The AssetType header value specifying the expected asset type.
 * @param Accept The Accept header value specifying the preferred response content type.
 * @param AssetFormat The AssetFormat header value specifying the desired asset format. Overridden by robloxAssetFormat if both are provided.
 * @param Roblox-AssetFormat The Roblox-AssetFormat header value specifying the preferred Roblox-specific asset format. Takes precedence over assetFormat.
 * @param skipSigningScripts Whether to skip script signing for the returned asset. Used for script assets that don't require signing.
 * @param clientInsert Set to 1 to indicate this is a client insert request.
 * @param scriptinsert Set to 1 to indicate this is a script insert request.
 * @param modulePlaceId The place ID of the module making the request.
 * @param serverplaceid The server place ID making the request.
 * @param expectedAssetType The expected asset type as a fallback when assetType header is not provided.
 * @param doNotFallbackToBaselineRepresentation Whether to prevent fallback to baseline representation when specific content representations are not available.
 * @param contentRepresentationPriorityList Base64URL-encoded JSON string specifying the priority list of desired content representations (format, version, fidelity).
 * @param accessContext
 * @description Returns an object containing a `location` property which is a temporary CDN URL for the asset content. All asset types are supported.
You should request that URL with the `Accept-Encoding: gzip` header and decompress the result if the response is gzipped. If you are using cURL, the `--compressed` flag will automate these steps for you.
This endpoint is expected to be called with API key authentication through `apis.roblox.com/asset-delivery-api/v1/assetId/{assetId}`.
While you are able to make requests to this endpoint with Cookie authentication via `assetdelivery.roblox.com/v1/openCloud/assetId/{assetId}`, we highly discourage use this way.
Expect unannounced removal of this second route in the future.
 */
exports.getOpencloudAssetidAssetid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/openCloud/assetId/:assetId',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        doNotFallbackToBaselineRepresentation: {
            style: 'form',
            explode: true,
        },
        contentRepresentationPriorityList: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
        'Accept-Encoding': zod_1.z.string().optional(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
        AssetType: zod_1.z.string().optional(),
        Accept: zod_1.z.string().optional(),
        AssetFormat: zod_1.z.string().optional(),
        'Roblox-AssetFormat': zod_1.z.string().optional(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        doNotFallbackToBaselineRepresentation: zod_1.z.boolean().optional(),
        contentRepresentationPriorityList: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV1,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://assetdelivery.roblox.com/v1/openCloud/assetId/:assetId/version/:versionNumber
 * @summary Retrieves an asset by its ID and version number with OpenCloud auth.
 * @param assetId The ID of the asset to retrieve.
 * @param versionNumber The version number of the asset to retrieve.
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param skipSigningScripts
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param expectedAssetType
 * @param doNotFallbackToBaselineRepresentation
 * @param contentRepresentationPriorityList
 * @param accessContext
 * @description Refer to the assetId endpoint for details on usage.
This endpoint is expected to be called with API key authentication through `apis.roblox.com/asset-delivery-api/v1/assetId/{assetId}/version/{versionNumber}`.
While you are able to make requests to this endpoint with Cookie authentication via `assetdelivery.roblox.com/v1/openCloud/assetId/{assetId}/version/{versionNumber}`, we highly discourage use this way.
Expect unannounced removal of this second route in the future.
 */
exports.getOpencloudAssetidAssetidVersionVersionnumber = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/openCloud/assetId/:assetId/version/:versionNumber',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
        versionNumber: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        doNotFallbackToBaselineRepresentation: {
            style: 'form',
            explode: true,
        },
        contentRepresentationPriorityList: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
        versionNumber: zod_1.z.number().int(),
        'Accept-Encoding': zod_1.z.string().optional(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
        AssetType: zod_1.z.string().optional(),
        Accept: zod_1.z.string().optional(),
        AssetFormat: zod_1.z.string().optional(),
        'Roblox-AssetFormat': zod_1.z.string().optional(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        doNotFallbackToBaselineRepresentation: zod_1.z.boolean().optional(),
        contentRepresentationPriorityList: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV1,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
