// rozod/lib/endpoints/assetdeliveryv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMarassethashMarassethashMarchecksumMarchecksum = exports.postAssetsBatch = exports.getAssetidAssetidVersionVersionnumber = exports.getAssetidAssetid = exports.getAsset = exports.getAliasAlias = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_AssetDelivery_Api_AssetMetadata = zod_1.z.object({
    metadataType: zod_1.z.literal('UncompressedSize'),
    value: zod_1.z.string(),
});
const Roblox_Web_Assets_AssetFormatLocation = zod_1.z.object({
    assetFormat: zod_1.z.string(),
    location: zod_1.z.string(),
    assetMetadatas: zod_1.z.array(Roblox_AssetDelivery_Api_AssetMetadata),
});
const Roblox_Web_Assets_IAssetItemError = zod_1.z.object({
    Code: zod_1.z.number().int(),
    Message: zod_1.z.string(),
    CustomErrorCode: zod_1.z.enum([
        'UnknownError',
        'NoPermissionToAsset',
        'AssetPermissionCheckFailed',
        'NotAuthorizedForAgeRecommendation',
        'AgeRecommendationCheckFailed',
        'InvalidPlaceRequestFromGameServer',
        'BlockedAssetTypeRequestedFromInsertService',
        'BlockedAssetTypeRequestedFromGameServer',
        'AssetTypeMismatch',
        'MissingAssetTypeInRequestHeader',
        'AssetNotTrustedForPlace',
        'NoAuthentication',
        'AssetContentRepresentationBlockedDueToModeration',
        'AssetNotFound',
        'AssetVersionNotFound',
        'AssetContentRepresentationNotFound',
        'BlockedByAgeGeoRestriction',
        'BlockedAssetTypeRequestedFromNonGameServer',
        'AssetPendingReview',
        'NotApprovedForRequestor',
        'NotApprovedByContentCompliance',
        'AssetContentRepresentationGenerating',
    ]),
});
const Roblox_Web_Assets_AssetContentRepresentationSpecifier = zod_1.z.object({
    format: zod_1.z.string(),
    majorVersion: zod_1.z.string(),
    fidelity: zod_1.z.string(),
    skipGenerationIfNotExist: zod_1.z.boolean(),
});
const Roblox_Web_Assets_AssetResponseItemV2 = zod_1.z.object({
    locations: zod_1.z.array(Roblox_Web_Assets_AssetFormatLocation),
    errors: zod_1.z.array(Roblox_Web_Assets_IAssetItemError),
    requestId: zod_1.z.string(),
    isArchived: zod_1.z.boolean(),
    assetTypeId: zod_1.z.number().int(),
    contentRepresentationSpecifier: Roblox_Web_Assets_AssetContentRepresentationSpecifier,
    isRecordable: zod_1.z.boolean(),
});
const Roblox_Web_Assets_BatchAssetRequestItem = zod_1.z.object({
    assetName: zod_1.z.string(),
    assetType: zod_1.z.string(),
    clientInsert: zod_1.z.boolean(),
    placeId: zod_1.z.number().int(),
    requestId: zod_1.z.string(),
    scriptInsert: zod_1.z.boolean(),
    serverPlaceId: zod_1.z.number().int(),
    universeId: zod_1.z.number().int(),
    accept: zod_1.z.string(),
    encoding: zod_1.z.string(),
    hash: zod_1.z.string(),
    userAssetId: zod_1.z.number().int(),
    assetId: zod_1.z.number().int(),
    version: zod_1.z.number().int(),
    assetVersionId: zod_1.z.number().int(),
    modulePlaceId: zod_1.z.number().int(),
    assetFormat: zod_1.z.string(),
    'roblox-assetFormat': zod_1.z.string(),
    assetResolutionMode: zod_1.z.string(),
    accessContext: zod_1.z.string(),
    contentRepresentationPriorityList: zod_1.z.string(),
    doNotFallbackToBaselineRepresentation: zod_1.z.boolean(),
});
/**
 * @api GET https://assetdelivery.roblox.com/v2/alias/:alias
 * @param alias
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param skipSigningScripts
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param expectedAssetType
 * @param accessContext
 */
exports.getAliasAlias = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/alias/:alias',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        alias: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        alias: zod_1.z.string().regex(/^[0-9]+\/.+/),
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV2,
    errors: [],
});
/**
 * @api GET https://assetdelivery.roblox.com/v2/asset
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param id
 * @param userAssetId
 * @param assetVersionId
 * @param version
 * @param universeId
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param assetName
 * @param hash
 * @param marAssetHash
 * @param marCheckSum
 * @param expectedAssetType
 * @param skipSigningScripts
 * @param permissionContext
 * @param doNotFallbackToBaselineRepresentation
 * @param contentRepresentationPriorityList
 * @param accessContext
 */
exports.getAsset = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/asset',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        id: {
            style: 'form',
            explode: true,
        },
        userAssetId: {
            style: 'form',
            explode: true,
        },
        assetVersionId: {
            style: 'form',
            explode: true,
        },
        version: {
            style: 'form',
            explode: true,
        },
        universeId: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        assetName: {
            style: 'form',
            explode: true,
        },
        hash: {
            style: 'form',
            explode: true,
        },
        marAssetHash: {
            style: 'form',
            explode: true,
        },
        marCheckSum: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        permissionContext: {
            style: 'form',
            explode: true,
        },
        doNotFallbackToBaselineRepresentation: {
            style: 'form',
            explode: true,
        },
        contentRepresentationPriorityList: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        id: zod_1.z.number().int().optional(),
        userAssetId: zod_1.z.number().int().optional(),
        assetVersionId: zod_1.z.number().int().optional(),
        version: zod_1.z.number().int().optional(),
        universeId: zod_1.z.number().int().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.string().optional(),
        assetName: zod_1.z.string().optional(),
        hash: zod_1.z.string().optional(),
        marAssetHash: zod_1.z.string().optional(),
        marCheckSum: zod_1.z.string().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        permissionContext: zod_1.z.string().optional(),
        doNotFallbackToBaselineRepresentation: zod_1.z.boolean().optional(),
        contentRepresentationPriorityList: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV2,
    errors: [],
});
/**
 * @api GET https://assetdelivery.roblox.com/v2/assetId/:assetId
 * @param assetId
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param skipSigningScripts
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param expectedAssetType
 * @param doNotFallbackToBaselineRepresentation
 * @param contentRepresentationPriorityList
 * @param accessContext
 */
exports.getAssetidAssetid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/assetId/:assetId',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        doNotFallbackToBaselineRepresentation: {
            style: 'form',
            explode: true,
        },
        contentRepresentationPriorityList: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        doNotFallbackToBaselineRepresentation: zod_1.z.boolean().optional(),
        contentRepresentationPriorityList: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV2,
    errors: [],
});
/**
 * @api GET https://assetdelivery.roblox.com/v2/assetId/:assetId/version/:versionNumber
 * @param assetId
 * @param versionNumber
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param skipSigningScripts
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param expectedAssetType
 * @param doNotFallbackToBaselineRepresentation
 * @param contentRepresentationPriorityList
 * @param accessContext
 */
exports.getAssetidAssetidVersionVersionnumber = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/assetId/:assetId/version/:versionNumber',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
        versionNumber: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
        doNotFallbackToBaselineRepresentation: {
            style: 'form',
            explode: true,
        },
        contentRepresentationPriorityList: {
            style: 'form',
            explode: true,
        },
        accessContext: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
        versionNumber: zod_1.z.number().int(),
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
        doNotFallbackToBaselineRepresentation: zod_1.z.boolean().optional(),
        contentRepresentationPriorityList: zod_1.z.string().optional(),
        accessContext: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV2,
    errors: [],
});
/**
 * @api POST https://assetdelivery.roblox.com/v2/assets/batch
 * @param body
 * @param Roblox-Place-Id
 * @param Accept
 * @param Roblox-Browser-Asset-Request
 */
exports.postAssetsBatch = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/assets/batch',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        'Roblox-Place-Id': {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        'Roblox-Browser-Asset-Request': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int(),
        Accept: zod_1.z.string(),
        'Roblox-Browser-Asset-Request': zod_1.z.string(),
    },
    body: zod_1.z.array(Roblox_Web_Assets_BatchAssetRequestItem),
    response: zod_1.z.array(Roblox_Web_Assets_AssetResponseItemV2),
    errors: [],
});
/**
 * @api GET https://assetdelivery.roblox.com/v2/marAssetHash/:marAssetHash/marCheckSum/:marCheckSum
 * @param marAssetHash
 * @param marCheckSum
 * @param Accept-Encoding
 * @param Roblox-Place-Id
 * @param AssetType
 * @param Accept
 * @param AssetFormat
 * @param Roblox-AssetFormat
 * @param skipSigningScripts
 * @param clientInsert
 * @param scriptinsert
 * @param modulePlaceId
 * @param serverplaceid
 * @param expectedAssetType
 */
exports.getMarassethashMarassethashMarchecksumMarchecksum = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/marAssetHash/:marAssetHash/marCheckSum/:marCheckSum',
    baseUrl: 'https://assetdelivery.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        marAssetHash: {
            style: 'simple',
        },
        marCheckSum: {
            style: 'simple',
        },
        'Accept-Encoding': {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        AssetType: {
            style: 'simple',
        },
        Accept: {
            style: 'simple',
        },
        AssetFormat: {
            style: 'simple',
        },
        'Roblox-AssetFormat': {
            style: 'simple',
        },
        skipSigningScripts: {
            style: 'form',
            explode: true,
        },
        clientInsert: {
            style: 'form',
            explode: true,
        },
        scriptinsert: {
            style: 'form',
            explode: true,
        },
        modulePlaceId: {
            style: 'form',
            explode: true,
        },
        serverplaceid: {
            style: 'form',
            explode: true,
        },
        expectedAssetType: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        marAssetHash: zod_1.z.string(),
        marCheckSum: zod_1.z.string(),
        'Accept-Encoding': zod_1.z.string(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        AssetType: zod_1.z.string(),
        Accept: zod_1.z.string(),
        AssetFormat: zod_1.z.string(),
        'Roblox-AssetFormat': zod_1.z.string(),
        skipSigningScripts: zod_1.z.boolean().optional(),
        clientInsert: zod_1.z.number().int().optional(),
        scriptinsert: zod_1.z.number().int().optional(),
        modulePlaceId: zod_1.z.number().int().optional(),
        serverplaceid: zod_1.z.number().int().optional(),
        expectedAssetType: zod_1.z.string().optional(),
    },
    response: Roblox_Web_Assets_AssetResponseItemV2,
    errors: [],
});
