// rozod/lib/endpoints/authv3.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postUsersUseridTwoStepVerificationLogin = exports.postLogout = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Authentication_Api_Models_Request_LogoutV3Request = zod_1.z.object({
    logoutReason: zod_1.z.string(),
    url: zod_1.z.string(),
    userId: zod_1.z.string(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel = zod_1.z.object({
    clientPublicKey: zod_1.z.string(),
    clientEpochTimestamp: zod_1.z.number().int(),
    saiSignature: zod_1.z.string(),
    serverNonce: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_AccountLinkParameters = zod_1.z.object({
    LinkingPlatform: zod_1.z.enum([
        'Invalid',
        'Xbox',
        'Qq',
        'WeChat',
        'Facebook',
        'RobloxDeveloper',
        'RobloxGroupCreator',
        'Playstation',
        'ExternalProvider',
        'Example',
    ]),
});
const Roblox_Authentication_Api_TwoStepVerificationLoginRequest = zod_1.z.object({
    challengeId: zod_1.z.string(),
    verificationToken: zod_1.z.string(),
    rememberDevice: zod_1.z.boolean(),
    secureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
    accountBlob: zod_1.z.string(),
    accountLinkParameters: Roblox_Authentication_Api_Models_AccountLinkParameters,
});
const Roblox_Authentication_Api_Models_TwoStepVerificationV3LoginResponse = zod_1.z.object({
    identityVerificationLoginTicket: zod_1.z.string(),
    accountBlob: zod_1.z.string(),
});
/**
 * @api POST https://auth.roblox.com/v3/logout
 * @summary Destroys the current authentication session while reporting metrics like url and reason for logout.
 * @param body The logout request with postBody which includes reason, url, and userIdString.
 */
exports.postLogout = (0, __1.endpoint)({
    method: 'POST',
    path: '/v3/logout',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_LogoutV3Request,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v3/users/:userId/two-step-verification/login
 * @summary Authenticates as a user given a two step verification verification token.
 * @param body The Roblox.Authentication.Api.TwoStepVerificationLoginRequest.
 * @param userId The user ID to authenticate as.
 */
exports.postUsersUseridTwoStepVerificationLogin = (0, __1.endpoint)({
    method: 'POST',
    path: '/v3/users/:userId/two-step-verification/login',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_Authentication_Api_TwoStepVerificationLoginRequest,
    response: Roblox_Authentication_Api_Models_TwoStepVerificationV3LoginResponse,
    errors: [
        {
            status: 400,
            description: `1: User is invalid.
5: Invalid two step verification ticket.
10: Invalid verification token.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
11: Maxium logged in accounts limit reached.`,
        },
    ],
});
