// rozod/lib/endpoints/authv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postUsernamesValidate = exports.getUsernamesValidate = exports.postUsernamesRecover = exports.getUsernames = exports.postUsername = exports.postUserPasswordsChange = exports.postSignupLinked = exports.postSignup = exports.postSessionRefresh = exports.postRevertInvalidateTickets = exports.postRevertAccount = exports.getRevertAccount = exports.getRecoveryMetadata = exports.postPasswordsValidate = exports.getPasswordsValidate = exports.postPasswordsResetVerify = exports.postPasswordsResetSend = exports.postPasswordsReset = exports.getPasswordsReset = exports.getPasswordsCurrentStatus = exports.getMetadata = exports.postLogoutfromallsessionsandreauthenticate = exports.postLogout = exports.postLoginLinked = exports.postLogin = exports.postIdentityVerificationLogin = exports.getAuthMetadata = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Authentication_Api_Models_AuthMetaDataResponse = zod_1.z.object({
    cookieLawNoticeTimeout: zod_1.z.number().int(),
});
const Roblox_Authentication_Api_Models_MetadataResponse = zod_1.z.object({
    isUpdateUsernameEnabled: zod_1.z.boolean(),
    ftuxAvatarAssetMap: zod_1.z.string(),
    IsEmailUpsellAtLogoutEnabled: zod_1.z.boolean(),
    ShouldFetchEmailUpsellIXPValuesAtLogout: zod_1.z.boolean(),
    IsAccountRecoveryPromptEnabled: zod_1.z.boolean(),
    IsContactMethodRequiredAtSignup: zod_1.z.boolean(),
    IsUserAgreementsSignupIntegrationEnabled: zod_1.z.boolean(),
    IsPasswordRequiredForUsernameChange: zod_1.z.boolean(),
    IsPasskeyFeatureEnabled: zod_1.z.boolean(),
    IsAltBrowserTracker: zod_1.z.boolean(),
    IsLoginRedirectPageEnabled: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_PasswordStatusResponse = zod_1.z.object({
    valid: zod_1.z.boolean(),
});
const Roblox_Web_Responses_Users_LegacyUserResponse = zod_1.z.object({
    userId: zod_1.z.number().int(),
    username: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_PasswordResetMetadataResponse = zod_1.z.object({
    users: zod_1.z.array(Roblox_Web_Responses_Users_LegacyUserResponse),
});
const Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel = zod_1.z.object({
    clientPublicKey: zod_1.z.string(),
    clientEpochTimestamp: zod_1.z.number().int(),
    saiSignature: zod_1.z.string(),
    serverNonce: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_PasswordResetModel = zod_1.z.object({
    targetType: zod_1.z.enum(['Email', 'PhoneNumber', 'RecoverySessionID']),
    ticket: zod_1.z.string(),
    userId: zod_1.z.number().int(),
    password: zod_1.z.string(),
    passwordRepeated: zod_1.z.string(),
    twoStepVerificationChallengeId: zod_1.z.string(),
    twoStepVerificationToken: zod_1.z.string(),
    accountBlob: zod_1.z.string(),
    secureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
    newEmail: zod_1.z.string(),
    passkeySessionId: zod_1.z.string(),
    passkeyRegistrationResponse: zod_1.z.string(),
});
const Roblox_Web_Responses_Users_SkinnyUserResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_TwoStepVerificationSentResponse = zod_1.z.object({
    mediaType: zod_1.z.enum([
        'Email',
        'SMS',
        'Authenticator',
        'RecoveryCode',
        'SecurityKey',
        'CrossDevice',
        'Password',
        'Passkey',
    ]),
    ticket: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_LoginResponse = zod_1.z.object({
    user: Roblox_Web_Responses_Users_SkinnyUserResponse,
    twoStepVerificationData: Roblox_Authentication_Api_Models_TwoStepVerificationSentResponse,
    identityVerificationLoginTicket: zod_1.z.string(),
    isBanned: zod_1.z.boolean(),
    accountBlob: zod_1.z.string(),
    shouldUpdateEmail: zod_1.z.boolean(),
    recoveryEmail: zod_1.z.string(),
    passkeyRegistrationSucceeded: zod_1.z.boolean(),
    shouldAutoLoginFromRecovery: zod_1.z.boolean(),
    shouldPrompt2svRemoval: zod_1.z.boolean(),
    shouldPromptPasskeyAddition: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_PasswordValidationResponse = zod_1.z.object({
    code: zod_1.z.enum([
        'ValidPassword',
        'WeakPasswordError',
        'PasswordLengthError',
        'PasswordSameAsUsernameError',
        'ForbiddenPasswordError',
        'DumbStringsError',
    ]),
    message: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_PasswordValidationModel = zod_1.z.object({
    username: zod_1.z.string(),
    password: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_RecoveryMetadataResponse = zod_1.z.object({
    isOnPhone: zod_1.z.boolean(),
    codeLength: zod_1.z.number().int(),
    isPhoneFeatureEnabledForUsername: zod_1.z.boolean(),
    isPhoneFeatureEnabledForPassword: zod_1.z.boolean(),
    isBedev2CaptchaEnabledForPasswordReset: zod_1.z.boolean(),
    isUsernameRecoveryDeprecated: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_RevertAccountInfoResponse = zod_1.z.object({
    isTwoStepVerificationEnabled: zod_1.z.boolean(),
    isEmailVerified: zod_1.z.boolean(),
    isEmailChanged: zod_1.z.boolean(),
    isPhoneVerified: zod_1.z.boolean(),
    userId: zod_1.z.number().int(),
    username: zod_1.z.string(),
    ticket: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_RevertAccountSubmitRequest = zod_1.z.object({
    UserId: zod_1.z.number().int(),
    NewPassword: zod_1.z.string(),
    NewPasswordRepeated: zod_1.z.string(),
    Ticket: zod_1.z.string(),
    TwoStepVerificationChallengeId: zod_1.z.string(),
    TwoStepVerificationToken: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_UsernamesResponse = zod_1.z.object({
    usernames: zod_1.z.array(zod_1.z.string()),
});
const Roblox_Authentication_Api_Models_UsernameValidationResponse = zod_1.z.object({
    code: zod_1.z.enum([
        'ValidUsername',
        'AlreadyInUseError',
        'ModerationError',
        'InvalidLengthError',
        'StartsOrEndsWithUnderscoreError',
        'TooManyUnderscoresError',
        'ContainsSpacesError',
        'InvalidCharactersError',
        'ContainsPiiError',
        'ContainsReservedUsernameError',
    ]),
    message: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_UsernameValidationRequest = zod_1.z.object({
    username: zod_1.z.string(),
    birthday: zod_1.z.string().datetime({ offset: true }),
    context: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]),
});
const Roblox_Authentication_Api_Models_Request_IdentityVerificationLoginRequest = zod_1.z.object({
    loginTicket: zod_1.z.string(),
    resultToken: zod_1.z.string(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Authentication_Api_Models_AccountLinkParameters = zod_1.z.object({
    LinkingPlatform: zod_1.z.enum([
        'Invalid',
        'Xbox',
        'Qq',
        'WeChat',
        'Facebook',
        'RobloxDeveloper',
        'RobloxGroupCreator',
        'Playstation',
        'ExternalProvider',
        'Example',
    ]),
});
const Roblox_Authentication_Api_Models_LoginRequest = zod_1.z.object({
    ctype: zod_1.z.enum([
        'Email',
        'Username',
        'PhoneNumber',
        'EmailOtpSessionToken',
        'AuthToken',
        'Passkey',
        'AsUser',
        'TwoStepVerification',
        'XboxLive',
        'PlatformLive',
    ]),
    cvalue: zod_1.z.string(),
    password: zod_1.z.string(),
    userId: zod_1.z.number().int(),
    securityQuestionSessionId: zod_1.z.string(),
    securityQuestionRedemptionToken: zod_1.z.string(),
    secureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
    accountBlob: zod_1.z.string(),
    accountLinkParameters: Roblox_Authentication_Api_Models_AccountLinkParameters,
    captchaId: zod_1.z.string(),
    captchaToken: zod_1.z.string(),
    captchaProvider: zod_1.z.string(),
    challengeId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_LogoutFromAllSessionsAndReauthenticateRequest = zod_1.z.object({
    SecureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
});
const Roblox_Authentication_Api_Models_SendResetPasswordRequest = zod_1.z.object({
    targetType: zod_1.z.enum(['Email', 'PhoneNumber', 'RecoverySessionID']),
    target: zod_1.z.string(),
    captchaId: zod_1.z.string(),
    captchaToken: zod_1.z.string(),
    captchaProvider: zod_1.z.string(),
    challengeId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_SendResetPasswordResponse = zod_1.z.object({
    nonce: zod_1.z.string(),
    transmissionType: zod_1.z.enum(['Sms', 'Email']),
});
const Roblox_Authentication_Api_Models_PasswordResetVerificationRequest = zod_1.z.object({
    targetType: zod_1.z.enum(['Email', 'PhoneNumber', 'RecoverySessionID']),
    nonce: zod_1.z.string(),
    code: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_ForgotPasswordUserResponse = zod_1.z.object({
    user: Roblox_Web_Responses_Users_LegacyUserResponse,
    ticket: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_PasswordResetVerificationResponse = zod_1.z.object({
    userTickets: zod_1.z.array(Roblox_Authentication_Api_Models_ForgotPasswordUserResponse),
});
const Roblox_Authentication_Api_Models_ReferralDataModel = zod_1.z.object({
    acquisitionTime: zod_1.z.string().datetime({ offset: true }),
    acquisitionReferrer: zod_1.z.string(),
    medium: zod_1.z.string(),
    source: zod_1.z.string(),
    campaign: zod_1.z.string(),
    adGroup: zod_1.z.string(),
    keyword: zod_1.z.string(),
    matchType: zod_1.z.string(),
    sendInfo: zod_1.z.boolean(),
    requestSessionId: zod_1.z.string(),
    offerId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_OtpSessionModel = zod_1.z.object({
    otpSessionToken: zod_1.z.string(),
    otpContactType: zod_1.z.enum(['Unset', 'Email', 'Phone']),
});
const Roblox_Authentication_Api_Models_Request_AuditContentValue = zod_1.z.object({
    translationKey: zod_1.z.string(),
    translationNamespace: zod_1.z.string(),
    translatedSourceString: zod_1.z.string(),
    parameters: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_AuditSystemContent = zod_1.z.object({
    capturedAuditContent: Roblox_Authentication_Api_Models_Request_AuditContentValue,
    additionalAuditContent: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_SignupRequest = zod_1.z.object({
    username: zod_1.z.string(),
    password: zod_1.z.string(),
    gender: zod_1.z.enum(['Unknown', 'Male', 'Female']),
    birthday: zod_1.z.string().datetime({ offset: true }),
    displayName: zod_1.z.string(),
    isTosAgreementBoxChecked: zod_1.z.boolean(),
    email: zod_1.z.string(),
    locale: zod_1.z.string(),
    assetIds: zod_1.z.array(zod_1.z.number()),
    bodyColorId: zod_1.z.number().int(),
    bodyTypeScale: zod_1.z.number(),
    headScale: zod_1.z.number(),
    heightScale: zod_1.z.number(),
    widthScale: zod_1.z.number(),
    proportionScale: zod_1.z.number(),
    referralData: Roblox_Authentication_Api_Models_ReferralDataModel,
    agreementIds: zod_1.z.array(zod_1.z.string()),
    identityVerificationResultToken: zod_1.z.string(),
    secureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
    otpSession: Roblox_Authentication_Api_Models_Request_OtpSessionModel,
    dataToken: zod_1.z.string(),
    accountBlob: zod_1.z.string(),
    passkeySessionId: zod_1.z.string(),
    passkeyRegistrationResponse: zod_1.z.string(),
    accountLinkParameters: Roblox_Authentication_Api_Models_AccountLinkParameters,
    auditSystemContent: Roblox_Authentication_Api_Models_Request_AuditSystemContent,
    captchaId: zod_1.z.string(),
    captchaToken: zod_1.z.string(),
    captchaProvider: zod_1.z.string(),
    challengeId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_SignupResponse = zod_1.z.object({
    userId: zod_1.z.number().int(),
    starterPlaceId: zod_1.z.number().int(),
    returnUrl: zod_1.z.string(),
    accountBlob: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_PasswordChangeModel = zod_1.z.object({
    currentPassword: zod_1.z.string(),
    newPassword: zod_1.z.string(),
    secureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
});
const Roblox_Authentication_Api_Models_UsernameChangeRequest = zod_1.z.object({
    username: zod_1.z.string(),
    password: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_RecoverUsernameRequest = zod_1.z.object({
    targetType: zod_1.z.enum(['Email', 'PhoneNumber', 'RecoverySessionID']),
    target: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_RecoverUsernameResponse = zod_1.z.object({
    transmissionType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1)]),
});
/**
 * @api GET https://auth.roblox.com/v2/auth/metadata
 * @summary Gets Auth meta data
 */
exports.getAuthMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/auth/metadata',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ cookieLawNoticeTimeout: zod_1.z.number().int() }),
    errors: [],
});
/**
 * @api POST https://auth.roblox.com/v2/identity-verification/login
 * @summary Endpoint for login with identity verification
 * @param body
 */
exports.postIdentityVerificationLogin = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/identity-verification/login',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_IdentityVerificationLoginRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed
1: Invalid login ticket.
2: Invalid result token.
3: Invalid user.
4: Authentication failure.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/login
 * @summary Authenticates a user.
 * @param body Roblox.Authentication.Api.Models.LoginRequest.
 */
exports.postLogin = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/login',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_LoginRequest,
    response: Roblox_Authentication_Api_Models_LoginResponse,
    errors: [
        {
            status: 400,
            description: `0: An unexpected error occurred.
3: Username and Password are required. Please try again.
8: Login with received credential type is not supported.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: Incorrect username or password. Please try again.
2: You must pass the robot test before logging in.
4: Account has been locked. Please request a password reset.
5: Unable to login. Please use Social Network sign on.
6: Account issue. Please contact Support.
9: Unable to login with provided credentials. Default login is required.
10: Received credentials are unverified.
12: Existing login session found. Please log out first.
14: The account is unable to log in. Please log in to the LuoBu app.
15: Too many attempts. Please wait a bit.
27: The account is unable to login. Please log in with the VNG app.`,
        },
        {
            status: 429,
            description: `7: Too many attempts. Please wait a bit.`,
        },
        {
            status: 503,
            description: `11: Service unavailable. Please try again.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/login/linked
 * @summary Endpoint for logging in a user, specifically for linked
authentication on PCGDK
 * @param body Roblox.Authentication.Api.Models.LoginRequest
 */
exports.postLoginLinked = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/login/linked',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_LoginRequest,
    response: Roblox_Authentication_Api_Models_LoginResponse,
    errors: [
        {
            status: 400,
            description: `0: An unexpected error occurred.
3: Username and Password are required. Please try again.
8: Login with received credential type is not supported.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: Incorrect username or password. Please try again.
2: You must pass the robot test before logging in.
4: Account has been locked. Please request a password reset.
5: Unable to login. Please use Social Network sign on.
6: Account issue. Please contact Support.
9: Unable to login with provided credentials. Default login is required.
10: Received credentials are unverified.
12: Existing login session found. Please log out first.
14: The account is unable to log in. Please log in to the LuoBu app.
15: Too many attempts. Please wait a bit.
27: The account is unable to login. Please log in with the VNG app.`,
        },
        {
            status: 429,
            description: `7: Too many attempts. Please wait a bit.`,
        },
        {
            status: 503,
            description: `11: Service unavailable. Please try again.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/logout
 * @summary Destroys the current authentication session.
 */
exports.postLogout = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/logout',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/logoutfromallsessionsandreauthenticate
 * @summary Logs out user from all other sessions.
 * @param body
 */
exports.postLogoutfromallsessionsandreauthenticate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/logoutfromallsessionsandreauthenticate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_LogoutFromAllSessionsAndReauthenticateRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v2/metadata
 * @summary Get the metadata
 */
exports.getMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/metadata',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: Roblox_Authentication_Api_Models_MetadataResponse,
    errors: [],
});
/**
 * @api GET https://auth.roblox.com/v2/passwords/current-status
 * @summary Returns password status for current user, asynchronously.
 */
exports.getPasswordsCurrentStatus = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/passwords/current-status',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ valid: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v2/passwords/reset
 * @summary Gets metadata needed for the password reset view.
 * @param TargetType
 * @param Ticket
 */
exports.getPasswordsReset = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/passwords/reset',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        TargetType: {
            style: 'form',
            explode: true,
        },
        Ticket: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        TargetType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]),
        Ticket: zod_1.z.string(),
    },
    response: Roblox_Authentication_Api_Models_PasswordResetMetadataResponse,
    errors: [
        {
            status: 400,
            description: `3: Request was empty.
9: The target type is invalid.
11: The password reset ticket is invalid.
14: The nonce is invalid.`,
        },
        {
            status: 403,
            description: `11: The password reset ticket is invalid.
16: The ticket is expired.`,
        },
        {
            status: 500,
            description: `0: Unknown error occured.`,
        },
        {
            status: 503,
            description: `1: Feature temporarily disabled. Please try again later.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/passwords/reset
 * @summary Resets a password for a user that belongs to the password reset ticket.
 * @param body The request model including the target type, ticket, user id, and new password, Roblox.Authentication.Api.Models.PasswordResetModel
 * @description This will log the user out of all sessions and re-authenticate.
 */
exports.postPasswordsReset = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/passwords/reset',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_PasswordResetModel,
    response: Roblox_Authentication_Api_Models_LoginResponse,
    errors: [
        {
            status: 400,
            description: `3: Request was empty.
11: The password reset ticket is invalid.
12: The user is invalid.
20: The password is invalid.
21: Passwords do not match.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
16: The ticket is expired.
17: The nonce is expired.`,
        },
        {
            status: 500,
            description: `0: Unknown error occured.`,
        },
        {
            status: 503,
            description: `1: Feature temporarily disabled. Please try again later.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/passwords/reset/send
 * @summary Sends a password reset email or challenge to the specified target.
 * @param body The request model containing the target type and a target.
 * @description Phone target must be a csv with 3 values: "internationalPrefixNumber,nationalNumber,countryCode"
 */
exports.postPasswordsResetSend = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/passwords/reset/send',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_SendResetPasswordRequest,
    response: Roblox_Authentication_Api_Models_SendResetPasswordResponse,
    errors: [
        {
            status: 400,
            description: `3: Request was empty.
9: The target type is invalid.
10: The target is invalid.
12: The user is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
18: Captcha is required.
23: Authenticate with Luobu instead.`,
        },
        {
            status: 429,
            description: `2: Too many attempts. Please try again later.`,
        },
        {
            status: 500,
            description: `0: Unknown error occured.
19: Message could not be sent.`,
        },
        {
            status: 503,
            description: `1: Feature temporarily disabled. Please try again later.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/passwords/reset/verify
 * @summary Verifies a password reset challenge solution.
 * @param body The request model containing the nonce and the solution. Roblox.Authentication.Api.Models.PasswordResetVerificationRequest
 */
exports.postPasswordsResetVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/passwords/reset/verify',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_PasswordResetVerificationRequest,
    response: Roblox_Authentication_Api_Models_PasswordResetVerificationResponse,
    errors: [
        {
            status: 400,
            description: `3: Request was empty.
9: The target type is invalid.
14: The nonce is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
13: The code is invalid.`,
        },
        {
            status: 500,
            description: `0: Unknown error occured.`,
        },
        {
            status: 503,
            description: `1: Feature temporarily disabled. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v2/passwords/validate
 * @summary Endpoint for checking if a password is valid.
 * @param Username
 * @param Password
 */
exports.getPasswordsValidate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/passwords/validate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        Username: {
            style: 'form',
            explode: true,
        },
        Password: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        Username: zod_1.z.string(),
        Password: zod_1.z.string(),
    },
    response: Roblox_Authentication_Api_Models_PasswordValidationResponse,
    errors: [
        {
            status: 400,
            description: `1: Valid Username and Password are required. Please try again.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/passwords/validate
 * @summary Endpoint for checking if a password is valid.
 * @param body The Roblox.Authentication.Api.Models.PasswordValidationModel.
 */
exports.postPasswordsValidate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/passwords/validate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_PasswordValidationModel,
    response: Roblox_Authentication_Api_Models_PasswordValidationResponse,
    errors: [
        {
            status: 400,
            description: `1: Valid Username and Password are required. Please try again.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v2/recovery/metadata
 * @summary Get metadata for forgot endpoints
 */
exports.getRecoveryMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/recovery/metadata',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: Roblox_Authentication_Api_Models_RecoveryMetadataResponse,
    errors: [
        {
            status: 503,
            description: `7: The Roblox WeChat API is currently unavailable.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v2/revert/account
 * @summary Get Revert Account ticket info
 * @param ticket Ticket Guid to revert account.
 */
exports.getRevertAccount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/revert/account',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        ticket: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        ticket: zod_1.z.string(),
    },
    response: Roblox_Authentication_Api_Models_RevertAccountInfoResponse,
    errors: [
        {
            status: 400,
            description: `2: The account revert ticket is not valid`,
        },
        {
            status: 403,
            description: `13: Revert links are disabled for users in the Enhanced Protection Program.`,
        },
        {
            status: 503,
            description: `1: This feature is disabled`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/revert/account
 * @summary Submit Revert Account Request
 * @param body The Roblox.Authentication.Api.Models.RevertAccountSubmitRequest containing the necessary information to revert account.
 */
exports.postRevertAccount = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/revert/account',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_RevertAccountSubmitRequest,
    response: Roblox_Authentication_Api_Models_LoginResponse,
    errors: [
        {
            status: 400,
            description: `2: The account revert ticket is not valid
3: Password is not valid
4: Passwords do not match
5: Password cannot be used
8: The account security ticket is expired.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `0: Unknown
1: This feature is disabled`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/revert/invalidate-tickets
 * @summary Invalidates all account security tickets for the authenticated user.
This endpoint should be called before enrolling in EPP to ensure old revert links cannot be used.
 */
exports.postRevertInvalidateTickets = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/revert/invalidate-tickets',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `1: This feature is disabled`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/session/refresh
 * @summary Logs out user from the current session and create a new one.
 */
exports.postSessionRefresh = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/session/refresh',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/signup
 * @summary Endpoint for signing up a new user
 * @param body Roblox.Authentication.Api.Models.SignupRequest
 */
exports.postSignup = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/signup',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_SignupRequest,
    response: Roblox_Authentication_Api_Models_SignupResponse,
    errors: [
        {
            status: 400,
            description: `Bad request
16: User agreement ids are null.
21: Empty account switch blob required`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: Captcha Failed.
4: Invalid Birthday.
5: Invalid Username.
6: Username already taken.
7: Invalid Password.
8: Password and Username are same.
9: Password is too simple.
10: Email is invalid.
11: Asset is invalid.
12: Too many attempts. Please wait a bit.
17: One time Passcode session was not valid
22: Maximum logged in accounts limit reached.`,
        },
        {
            status: 429,
            description: `3: Too many attempts. Please wait a bit.`,
        },
        {
            status: 500,
            description: `Internal server error
15: Insert acceptances failed.
27: Pre-auth passkey registration failed`,
        },
        {
            status: 503,
            description: `Service unavailable`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/signup/linked
 * @summary Endpoint for signing up a new user, specifically for linked
authentication on PCGDK
 * @param body Roblox.Authentication.Api.Models.SignupRequest
 */
exports.postSignupLinked = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/signup/linked',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_SignupRequest,
    response: Roblox_Authentication_Api_Models_SignupResponse,
    errors: [
        {
            status: 400,
            description: `Bad request
16: User agreement ids are null.
21: Empty account switch blob required`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: Captcha Failed.
4: Invalid Birthday.
5: Invalid Username.
6: Username already taken.
7: Invalid Password.
8: Password and Username are same.
9: Password is too simple.
10: Email is invalid.
11: Asset is invalid.
12: Too many attempts. Please wait a bit.
17: One time Passcode session was not valid
22: Maximum logged in accounts limit reached.`,
        },
        {
            status: 429,
            description: `3: Too many attempts. Please wait a bit.`,
        },
        {
            status: 500,
            description: `Internal server error
15: Insert acceptances failed.
27: Pre-auth passkey registration failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/user/passwords/change
 * @summary Changes the password for the authenticated user.
 * @param body The request model including the current password, and the new password.
 * @description The current password is needed for verification that the password can be changed.
 */
exports.postUserPasswordsChange = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/user/passwords/change',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_PasswordChangeModel,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `Roblox.Web.Authentication.Passwords.PasswordResponseCodes.InvalidCurrentPassword
            OR
            Roblox.Web.Authentication.Passwords.PasswordResponseCodes.InvalidPassword`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Roblox.Web.Authentication.Passwords.PasswordResponseCodes.PinLocked
0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `Roblox.Web.Authentication.Passwords.PasswordResponseCodes.Flooded`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/username
 * @summary Change the user's username
 * @param body The Roblox.Authentication.Api.Models.UsernameChangeRequest
 */
exports.postUsername = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/username',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_UsernameChangeRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `5: You don&#x27;t have enough Robux to change your username.
10: This username is already in use
11: Username not appropriate for Roblox
12: Usernames can be 3 to 20 characters long
13: Usernames can’t start or end with _ and can have at most one _
14: Only a-z, A-Z, 0-9, and _ are allowed
15: Username is null
16: Username might contain private information
17: This username is not available
18: Username is same as current`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: PIN is locked.
2: A verified email is missing
3: Your password is incorrect.
100: Unknown birthday`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.
5: You don&#x27;t have enough Robux to change your username.`,
        },
        {
            status: 503,
            description: `4: The feature is currently not available. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v2/usernames
 * @summary Gets a list of existing usernames on Roblox based on the query parameters
 * @param username The username
 * @description This endpoint can be expanded in the future to include other query parameters such as "startsWith"
 */
exports.getUsernames = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/usernames',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        username: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        username: zod_1.z.string().optional(),
    },
    response: Roblox_Authentication_Api_Models_UsernamesResponse,
    errors: [],
});
/**
 * @api POST https://auth.roblox.com/v2/usernames/recover
 * @summary Sends an email of all accounts belonging to an email
 * @param body
 */
exports.postUsernamesRecover = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/usernames/recover',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_RecoverUsernameRequest,
    response: Roblox_Authentication_Api_Models_RecoverUsernameResponse,
    errors: [
        {
            status: 400,
            description: `20: Invalid Email
21: Invalid Phone
23: No Account Found`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
11: Too many attempts. Please wait a bit.`,
        },
        {
            status: 500,
            description: `0: An unexpected error occurred.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v2/usernames/validate
 * @summary Checks if a username is valid.
 * @param Username
 * @param Birthday
 * @param Context
 */
exports.getUsernamesValidate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/usernames/validate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        Username: {
            style: 'form',
            explode: true,
        },
        Birthday: {
            style: 'form',
            explode: true,
        },
        Context: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        Username: zod_1.z.string(),
        Birthday: zod_1.z.string().datetime({ offset: true }),
        Context: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]),
    },
    response: Roblox_Authentication_Api_Models_UsernameValidationResponse,
    errors: [
        {
            status: 400,
            description: `1: A valid username is required.
2: A valid birthday or authenticated user is required.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v2/usernames/validate
 * @summary Checks if a username is valid.
 * @param body The Roblox.Authentication.Api.Models.UsernameValidationRequest.
 */
exports.postUsernamesValidate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/usernames/validate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_UsernameValidationRequest,
    response: Roblox_Authentication_Api_Models_UsernameValidationResponse,
    errors: [
        {
            status: 400,
            description: `1: A valid username is required.
2: A valid birthday or authenticated user is required.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
