// rozod/lib/endpoints/authv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getValidatorsRecommendedusernamefromdisplayname = exports.getValidatorsEmail = exports.postUsernamesValidate = exports.getUsernamesValidate = exports.postUsernamesRecover = exports.getUsernames = exports.postUsername = exports.postUserPasswordsChange = exports.getSocialConnectedProviders = exports.postSocialProviderDisconnect = exports.postSignupLinked = exports.postSignup = exports.postSessionRefresh = exports.postRevertInvalidateTickets = exports.postRevertAccount = exports.getRevertAccount = exports.getRecoveryMetadata = exports.postPasswordsValidate = exports.getPasswordsValidate = exports.postPasskeyStartregistration = exports.postPasskeyStartauthentication = exports.postPasskeyStartPreauthRegistration = exports.postPasskeyStartAuthenticationByUser = exports.postPasskeyListcredentials = exports.postPasskeyFinishregistration = exports.postPasskeyFinishPreauthRegistration = exports.postPasskeyFinishArPreauthRegistration = exports.postPasskeyDeletecredentialbatch = exports.getMetadata = exports.postLogoutfromallsessionsandreauthenticate = exports.postLogout = exports.postLoginLinked = exports.postLogin = exports.postIdentityInitializeLogin = exports.postIdentityVerificationLogin = exports.postExternalUnlink = exports.postExternalSignup = exports.postExternalLoginandlink = exports.postExternalLogin = exports.postExternalAccess = exports.getExternalIdentityprovideridSsoOauthInit = exports.getExternalIdentityprovideridSsoOauthCallback = exports.getClientAssertion = exports.getAuthMetadata = exports.postAccountPinUnlock = exports.postAccountPinLock = exports.patchAccountPin = exports.deleteAccountPin = exports.postAccountPin = exports.getAccountPin = void 0;
exports.postXboxTranslate = exports.getXboxGetLoginConsecutiveDays = exports.postXboxDisconnect = exports.getXboxConnection = exports.postValidatorsUsername = exports.getValidatorsUsername = exports.postValidatorsRecommendedusernamefromdisplayname = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Authentication_Api_Models_AccountPinStatusResponse = zod_1.z.object({
    isEnabled: zod_1.z.boolean(),
    unlockedUntil: zod_1.z.number(),
});
const Roblox_Authentication_Api_Models_AccountPinRequest = zod_1.z.object({
    pin: zod_1.z.string(),
    reauthenticationToken: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiSuccessResponse = zod_1.z.object({
    success: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_AuthMetaDataResponse = zod_1.z.object({
    cookieLawNoticeTimeout: zod_1.z.number().int(),
});
const Roblox_Authentication_Api_Models_Response_GetClientAssertionResponse = zod_1.z.object({ clientAssertion: zod_1.z.string() });
const Roblox_Authentication_Api_Models_MetadataResponse = zod_1.z.object({
    isUpdateUsernameEnabled: zod_1.z.boolean(),
    ftuxAvatarAssetMap: zod_1.z.string(),
    IsEmailUpsellAtLogoutEnabled: zod_1.z.boolean(),
    ShouldFetchEmailUpsellIXPValuesAtLogout: zod_1.z.boolean(),
    IsAccountRecoveryPromptEnabled: zod_1.z.boolean(),
    IsContactMethodRequiredAtSignup: zod_1.z.boolean(),
    IsUserAgreementsSignupIntegrationEnabled: zod_1.z.boolean(),
    IsPasswordRequiredForUsernameChange: zod_1.z.boolean(),
    IsPasskeyFeatureEnabled: zod_1.z.boolean(),
    IsAltBrowserTracker: zod_1.z.boolean(),
    IsLoginRedirectPageEnabled: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_PasswordValidationResponse = zod_1.z.object({
    code: zod_1.z.enum([
        'ValidPassword',
        'WeakPasswordError',
        'PasswordLengthError',
        'PasswordSameAsUsernameError',
        'ForbiddenPasswordError',
        'DumbStringsError',
    ]),
    message: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_PasswordValidationModel = zod_1.z.object({
    username: zod_1.z.string(),
    password: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_RecoveryMetadataResponse = zod_1.z.object({
    isOnPhone: zod_1.z.boolean(),
    codeLength: zod_1.z.number().int(),
    isPhoneFeatureEnabledForUsername: zod_1.z.boolean(),
    isPhoneFeatureEnabledForPassword: zod_1.z.boolean(),
    isBedev2CaptchaEnabledForPasswordReset: zod_1.z.boolean(),
    isUsernameRecoveryDeprecated: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_RevertAccountInfoResponse = zod_1.z.object({
    isTwoStepVerificationEnabled: zod_1.z.boolean(),
    isEmailVerified: zod_1.z.boolean(),
    isEmailChanged: zod_1.z.boolean(),
    isPhoneVerified: zod_1.z.boolean(),
    userId: zod_1.z.number().int(),
    username: zod_1.z.string(),
    ticket: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_RevertAccountSubmitRequest = zod_1.z.object({
    UserId: zod_1.z.number().int(),
    NewPassword: zod_1.z.string(),
    NewPasswordRepeated: zod_1.z.string(),
    Ticket: zod_1.z.string(),
    TwoStepVerificationChallengeId: zod_1.z.string(),
    TwoStepVerificationToken: zod_1.z.string(),
});
const Roblox_Web_Responses_Users_SkinnyUserResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_TwoStepVerificationSentResponse = zod_1.z.object({
    mediaType: zod_1.z.enum([
        'Email',
        'SMS',
        'Authenticator',
        'RecoveryCode',
        'SecurityKey',
        'CrossDevice',
        'Password',
        'Passkey',
    ]),
    ticket: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_LoginResponse = zod_1.z.object({
    user: Roblox_Web_Responses_Users_SkinnyUserResponse,
    twoStepVerificationData: Roblox_Authentication_Api_Models_TwoStepVerificationSentResponse,
    identityVerificationLoginTicket: zod_1.z.string(),
    isBanned: zod_1.z.boolean(),
    accountBlob: zod_1.z.string(),
    shouldUpdateEmail: zod_1.z.boolean(),
    recoveryEmail: zod_1.z.string(),
    passkeyRegistrationSucceeded: zod_1.z.boolean(),
    shouldAutoLoginFromRecovery: zod_1.z.boolean(),
    shouldPrompt2svRemoval: zod_1.z.boolean(),
    shouldPromptPasskeyAddition: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_ProviderInfoModel = zod_1.z.object({
    provider: zod_1.z.string(),
    identifier: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_SocialProvidersResponse = zod_1.z.object({
    providers: zod_1.z.array(Roblox_Authentication_Api_Models_ProviderInfoModel),
});
const Roblox_Authentication_Api_Models_UsernamesResponse = zod_1.z.object({
    usernames: zod_1.z.array(zod_1.z.string()),
});
const Roblox_Authentication_Api_Models_UsernameValidationResponse = zod_1.z.object({
    code: zod_1.z.enum([
        'ValidUsername',
        'AlreadyInUseError',
        'ModerationError',
        'InvalidLengthError',
        'StartsOrEndsWithUnderscoreError',
        'TooManyUnderscoresError',
        'ContainsSpacesError',
        'InvalidCharactersError',
        'ContainsPiiError',
        'ContainsReservedUsernameError',
    ]),
    message: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_UsernameValidationRequest = zod_1.z.object({
    username: zod_1.z.string(),
    birthday: zod_1.z.string().datetime({ offset: true }),
    context: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]),
});
const Roblox_Authentication_Api_Models_EmailValidationResponse = zod_1.z.object({
    isEmailValid: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_RecommendedUsernameResponse = zod_1.z.object({
    didGenerateNewUsername: zod_1.z.boolean(),
    suggestedUsernames: zod_1.z.array(zod_1.z.string()),
});
const Roblox_Authentication_Api_Models_RecommendedUsernameFromDisplayNameRequest = zod_1.z.object({
    displayName: zod_1.z.string(),
    birthday: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Authentication_Api_Models_RecommendedUsernameRequest = zod_1.z.object({
    username: zod_1.z.string(),
    birthday: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Authentication_Api_Models_XboxConnectionModel = zod_1.z.object({
    hasConnectedXboxAccount: zod_1.z.boolean(),
    gamertag: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_XboxLoginConsecutiveDaysResponse = zod_1.z.object({ count: zod_1.z.number().int() });
const Roblox_Authentication_Api_Models_AccountPinResponse = zod_1.z.object({
    unlockedUntil: zod_1.z.number(),
});
const Roblox_Authentication_Api_Models_Request_ExternalAccessRequest = zod_1.z.object({
    authenticationProof: zod_1.z.string(),
    identityProviderPlatformType: zod_1.z.enum(['Undefined', 'Xbox', 'Playstation', 'Web']),
    additionalInfoPayload: zod_1.z.object({}),
});
const Roblox_Authentication_Api_Models_Response_ExternalIdentityGateway_ExternalIdentityAccessResponse = zod_1.z.object({
    placeId: zod_1.z.number().int(),
    isolationContext: zod_1.z.string(),
    launchData: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_ExternalLoginRequest = zod_1.z.object({
    identityProvider: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(999),
    ]),
    additionalData: zod_1.z.object({}),
    authenticationProof: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Response_ExternalIdentityGateway_ExternalLoginResponse = zod_1.z.object({
    success: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_Request_ExternalLoginAndLinkRequest = zod_1.z.object({
    ctype: zod_1.z.enum([
        'Email',
        'Username',
        'PhoneNumber',
        'EmailOtpSessionToken',
        'AuthToken',
        'Passkey',
        'AsUser',
        'TwoStepVerification',
        'XboxLive',
        'PlatformLive',
    ]),
    cvalue: zod_1.z.string(),
    password: zod_1.z.string(),
    authenticationProof: zod_1.z.string(),
    IdentityProviderPlatformType: zod_1.z.enum(['Undefined', 'Xbox', 'Playstation', 'Web']),
    additionalInfoPayload: zod_1.z.object({}),
});
const Roblox_Authentication_Api_Models_Request_ExternalSignupRequest = zod_1.z.object({
    username: zod_1.z.string(),
    password: zod_1.z.string(),
    birthday: zod_1.z.string().datetime({ offset: true }),
    locale: zod_1.z.string(),
    authenticationProof: zod_1.z.string(),
    IdentityProviderPlatformType: zod_1.z.enum(['Undefined', 'Xbox', 'Playstation', 'Web']),
    additionalInfoPayload: zod_1.z.object({}),
});
const Roblox_Authentication_Api_Models_Request_ExternalUnlinkRequest = zod_1.z.object({
    IdentityProviderPlatformType: zod_1.z.enum(['Undefined', 'Xbox', 'Playstation', 'Web']),
    additionalInfoPayload: zod_1.z.object({}),
});
const Roblox_Authentication_Api_Models_Request_IdentityVerificationLoginRequest = zod_1.z.object({
    loginTicket: zod_1.z.string(),
    resultToken: zod_1.z.string(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Authentication_Api_Models_Request_InitializeLoginRequest = zod_1.z.object({
    ctype: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
    ]),
    cvalue: zod_1.z.string().min(1),
    captchaId: zod_1.z.string().optional(),
    captchaToken: zod_1.z.string().optional(),
    captchaProvider: zod_1.z.string().optional(),
    challengeId: zod_1.z.string().optional(),
});
const Roblox_Authentication_Api_Models_LoginMethodModel = zod_1.z.object({
    method: zod_1.z.enum(['EmailOtp', 'Passkey', 'Password']),
    priority: zod_1.z.number().int(),
});
const Roblox_Authentication_Api_Models_Response_InitializeLoginResponse = zod_1.z.object({
    loginMethods: zod_1.z.array(Roblox_Authentication_Api_Models_LoginMethodModel),
});
const Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel = zod_1.z.object({
    clientPublicKey: zod_1.z.string(),
    clientEpochTimestamp: zod_1.z.number().int(),
    saiSignature: zod_1.z.string(),
    serverNonce: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_AccountLinkParameters = zod_1.z.object({
    LinkingPlatform: zod_1.z.enum([
        'Invalid',
        'Xbox',
        'Qq',
        'WeChat',
        'Facebook',
        'RobloxDeveloper',
        'RobloxGroupCreator',
        'Playstation',
        'ExternalProvider',
        'Example',
    ]),
});
const Roblox_Authentication_Api_Models_LoginRequest = zod_1.z.object({
    ctype: zod_1.z.enum([
        'Email',
        'Username',
        'PhoneNumber',
        'EmailOtpSessionToken',
        'AuthToken',
        'Passkey',
        'AsUser',
        'TwoStepVerification',
        'XboxLive',
        'PlatformLive',
    ]),
    cvalue: zod_1.z.string(),
    password: zod_1.z.string(),
    userId: zod_1.z.number().int(),
    securityQuestionSessionId: zod_1.z.string(),
    securityQuestionRedemptionToken: zod_1.z.string(),
    secureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
    accountBlob: zod_1.z.string(),
    accountLinkParameters: Roblox_Authentication_Api_Models_AccountLinkParameters,
    captchaId: zod_1.z.string(),
    captchaToken: zod_1.z.string(),
    captchaProvider: zod_1.z.string(),
    challengeId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_LogoutFromAllSessionsAndReauthenticateRequest = zod_1.z.object({
    SecureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
});
const Roblox_Authentication_Api_Models_Request_DeletePasskeysRequest = zod_1.z.object({
    credentialNicknames: zod_1.z.array(zod_1.z.string()),
});
const Roblox_Authentication_Api_Models_Request_FinishARPreAuthPasskeyRegistrationRequest = zod_1.z.object({
    recoverySession: zod_1.z.string(),
    passkeySessionId: zod_1.z.string(),
    passkeyRegistrationResponse: zod_1.z.string(),
    userId: zod_1.z.number().int(),
    isPostRecovery: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_Request_FinishPasskeyPreauthRegistrationRequest = zod_1.z.object({
    sessionId: zod_1.z.string(),
    registrationResponse: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_FinishPasskeyRegistrationRequest = zod_1.z.object({
    sessionId: zod_1.z.string(),
    credentialNickname: zod_1.z.string(),
    attestationResponse: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_ListPasskeysRequest = zod_1.z.object({
    all: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_Response_PasskeyCredential = zod_1.z.object({
    nickname: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Response_ListPasskeyCredentialResponse = zod_1.z.object({
    credentials: zod_1.z.array(Roblox_Authentication_Api_Models_Response_PasskeyCredential),
});
const Roblox_Authentication_Api_Models_Request_StartAuthenticationByUserRequest = zod_1.z.object({
    ctype: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
    ]),
    cvalue: zod_1.z.string().min(1),
});
const Roblox_Authentication_Api_Models_Response_StartAuthenticationByUserResponse = zod_1.z.object({
    authenticationOptions: zod_1.z.string(),
    sessionId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_StartPasskeyPreauthRegistrationRequest = zod_1.z.object({
    username: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Response_StartPasskeyPreauthRegistrationResponse = zod_1.z.object({
    creationOptions: zod_1.z.string(),
    sessionId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Response_StartAuthenticationResponse = zod_1.z.object({
    authenticationOptions: zod_1.z.string(),
    sessionId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_StartPasskeyRegistrationRequest = zod_1.z.object({
    isSilentUpgrade: zod_1.z.boolean(),
});
const Roblox_Authentication_Api_Models_Response_StartPasskeyRegistrationResponse = zod_1.z.object({
    creationOptions: zod_1.z.string(),
    sessionId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_ReferralDataModel = zod_1.z.object({
    acquisitionTime: zod_1.z.string().datetime({ offset: true }),
    acquisitionReferrer: zod_1.z.string(),
    medium: zod_1.z.string(),
    source: zod_1.z.string(),
    campaign: zod_1.z.string(),
    adGroup: zod_1.z.string(),
    keyword: zod_1.z.string(),
    matchType: zod_1.z.string(),
    sendInfo: zod_1.z.boolean(),
    requestSessionId: zod_1.z.string(),
    offerId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_OtpSessionModel = zod_1.z.object({
    otpSessionToken: zod_1.z.string(),
    otpContactType: zod_1.z.enum(['Unset', 'Email', 'Phone']),
});
const Roblox_Authentication_Api_Models_Request_AuditContentValue = zod_1.z.object({
    translationKey: zod_1.z.string(),
    translationNamespace: zod_1.z.string(),
    translatedSourceString: zod_1.z.string(),
    parameters: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_Request_AuditSystemContent = zod_1.z.object({
    capturedAuditContent: Roblox_Authentication_Api_Models_Request_AuditContentValue,
    additionalAuditContent: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_SignupRequest = zod_1.z.object({
    username: zod_1.z.string(),
    password: zod_1.z.string(),
    gender: zod_1.z.enum(['Unknown', 'Male', 'Female']),
    birthday: zod_1.z.string().datetime({ offset: true }),
    displayName: zod_1.z.string(),
    isTosAgreementBoxChecked: zod_1.z.boolean(),
    email: zod_1.z.string(),
    locale: zod_1.z.string(),
    assetIds: zod_1.z.array(zod_1.z.number()),
    bodyColorId: zod_1.z.number().int(),
    bodyTypeScale: zod_1.z.number(),
    headScale: zod_1.z.number(),
    heightScale: zod_1.z.number(),
    widthScale: zod_1.z.number(),
    proportionScale: zod_1.z.number(),
    referralData: Roblox_Authentication_Api_Models_ReferralDataModel,
    agreementIds: zod_1.z.array(zod_1.z.string()),
    identityVerificationResultToken: zod_1.z.string(),
    secureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
    otpSession: Roblox_Authentication_Api_Models_Request_OtpSessionModel,
    dataToken: zod_1.z.string(),
    accountBlob: zod_1.z.string(),
    passkeySessionId: zod_1.z.string(),
    passkeyRegistrationResponse: zod_1.z.string(),
    accountLinkParameters: Roblox_Authentication_Api_Models_AccountLinkParameters,
    auditSystemContent: Roblox_Authentication_Api_Models_Request_AuditSystemContent,
    captchaId: zod_1.z.string(),
    captchaToken: zod_1.z.string(),
    captchaProvider: zod_1.z.string(),
    challengeId: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_SignupResponse = zod_1.z.object({
    userId: zod_1.z.number().int(),
    starterPlaceId: zod_1.z.number().int(),
    returnUrl: zod_1.z.string(),
    accountBlob: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_SocialAuthenticationDisconnectRequest = zod_1.z.object({ Password: zod_1.z.string() });
const Roblox_Authentication_Api_Models_PasswordChangeModel = zod_1.z.object({
    currentPassword: zod_1.z.string(),
    newPassword: zod_1.z.string(),
    secureAuthenticationIntent: Roblox_Authentication_Api_Models_Request_SecureAuthenticationIntentModel,
});
const Roblox_Authentication_Api_Models_UsernameChangeRequest = zod_1.z.object({
    username: zod_1.z.string(),
    password: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_RecoverUsernameRequest = zod_1.z.object({
    targetType: zod_1.z.enum(['Email', 'PhoneNumber', 'RecoverySessionID']),
    target: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_RecoverUsernameResponse = zod_1.z.object({
    transmissionType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1)]),
});
const Roblox_Authentication_Api_Models_XboxTranslateRequest = zod_1.z.object({
    ids: zod_1.z.array(zod_1.z.string()),
});
const Roblox_Authentication_Api_Models_XboxUserModel = zod_1.z.object({
    Id: zod_1.z.string(),
    UserId: zod_1.z.number().int(),
    Username: zod_1.z.string(),
});
const Roblox_Authentication_Api_Models_XboxCollectionsOfUserResponse = zod_1.z.object({
    Users: zod_1.z.array(Roblox_Authentication_Api_Models_XboxUserModel),
});
/**
 * @api GET https://auth.roblox.com/v1/account/pin
 * @summary Gets the account pin status.
 */
exports.getAccountPin = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/account/pin',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: Roblox_Authentication_Api_Models_AccountPinStatusResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/account/pin
 * @summary Request to create the account pin.
 * @param body The Roblox.Authentication.Api.Models.AccountPinRequest.
 */
exports.postAccountPin = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/account/pin',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_AccountPinRequest,
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api DELETE https://auth.roblox.com/v1/account/pin
 * @summary Request for deletes the account pin from the account.
 * @param body
 */
exports.deleteAccountPin = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/account/pin',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_AccountPinRequest.optional(),
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api PATCH https://auth.roblox.com/v1/account/pin
 * @summary Request made to update the account pin on the account.
 * @param body The request body.
 */
exports.patchAccountPin = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/account/pin',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_AccountPinRequest,
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/account/pin/lock
 * @summary Request to locks the account which has an account pin enabled.
 */
exports.postAccountPinLock = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/account/pin/lock',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/account/pin/unlock
 * @summary Requests to unlock the account pin.
 * @param body The Roblox.Authentication.Api.Models.AccountPinRequest containing the entered pin.
 */
exports.postAccountPinUnlock = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/account/pin/unlock',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_AccountPinRequest,
    response: zod_1.z.object({ unlockedUntil: zod_1.z.number() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/auth/metadata
 * @summary Gets Auth meta data
 */
exports.getAuthMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/auth/metadata',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ cookieLawNoticeTimeout: zod_1.z.number().int() }),
    errors: [],
});
/**
 * @api GET https://auth.roblox.com/v1/client-assertion
 * @summary Creates a client assertion to be used when generating an auth ticket.
 */
exports.getClientAssertion = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/client-assertion',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ clientAssertion: zod_1.z.string() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/external/:identityProviderId/sso/oauth/callback
 * @summary Callback function that external identity provider calls post user authentication.
 * @param identityProviderId
 * @param code
 * @param state
 */
exports.getExternalIdentityprovideridSsoOauthCallback = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/external/:identityProviderId/sso/oauth/callback',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        identityProviderId: {
            style: 'simple',
        },
        code: {
            style: 'form',
            explode: true,
        },
        state: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        identityProviderId: zod_1.z.number().int(),
        code: zod_1.z.string(),
        state: zod_1.z.string(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 302,
            description: `Redirect`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/external/:identityProviderId/sso/oauth/init
 * @summary Signs a user up for Roblox and links the account to the authenticated external provider ID.
 * @param identityProviderId
 */
exports.getExternalIdentityprovideridSsoOauthInit = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/external/:identityProviderId/sso/oauth/init',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        identityProviderId: {
            style: 'simple',
        },
    },
    parameters: {
        identityProviderId: zod_1.z.number().int(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 302,
            description: `Redirect`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/external/access
 * @summary Signs a user up for Roblox and links the account to the authenticated external provider ID.
 * @param body
 */
exports.postExternalAccess = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/external/access',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_ExternalAccessRequest,
    response: Roblox_Authentication_Api_Models_Response_ExternalIdentityGateway_ExternalIdentityAccessResponse,
    errors: [],
});
/**
 * @api POST https://auth.roblox.com/v1/external/login
 * @summary Logs in a user to Roblox based on the user's authenticated external provider session
 * @param body
 */
exports.postExternalLogin = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/external/login',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_ExternalLoginRequest,
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `30: Platform is not supported for SSO login.`,
        },
        {
            status: 401,
            description: `31: No linked account found for SSO login.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `0: An unexpected error occurred.`,
        },
        {
            status: 501,
            description: `0: An unexpected error occurred.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/external/loginAndLink
 * @summary Deprecated endpoint
Logins in a user to Roblox, then links the Roblox account to the external provider ID
 * @param body
 */
exports.postExternalLoginandlink = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/external/loginAndLink',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_ExternalLoginAndLinkRequest,
    response: Roblox_Authentication_Api_Models_LoginResponse,
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/external/signup
 * @summary Signs a user up for Roblox and links the account to the authenticated external provider ID
 * @param body
 */
exports.postExternalSignup = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/external/signup',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_ExternalSignupRequest,
    response: zod_1.z.void(),
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/external/unlink
 * @summary Unlink the logged in Roblox account from the current external provider ID
 * @param body
 */
exports.postExternalUnlink = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/external/unlink',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_ExternalUnlinkRequest,
    response: zod_1.z.void(),
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/identity-verification/login
 * @summary Endpoint for login with identity verification
 * @param body
 */
exports.postIdentityVerificationLogin = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/identity-verification/login',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_IdentityVerificationLoginRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed
1: Invalid login ticket.
2: Invalid result token.
3: Invalid user.
4: Authentication failure.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/identity/initialize-login
 * @summary Initiates identifier-first login flow by returning a list of login methods for user(s).
 * @param body
 */
exports.postIdentityInitializeLogin = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/identity/initialize-login',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_InitializeLoginRequest,
    response: Roblox_Authentication_Api_Models_Response_InitializeLoginResponse,
    errors: [
        {
            status: 400,
            description: `3: User identifier and type are required.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: Invalid user identifier.`,
        },
        {
            status: 500,
            description: `0: An unknown error occurred.`,
        },
        {
            status: 503,
            description: `1: This feature is disabled.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/login
 * @summary Authenticates a user.
 * @param body Roblox.Authentication.Api.Models.LoginRequest.
 */
exports.postLogin = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/login',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_LoginRequest,
    response: Roblox_Authentication_Api_Models_LoginResponse,
    errors: [
        {
            status: 400,
            description: `0: An unexpected error occurred.
3: Username and Password are required. Please try again.
8: Login with received credential type is not supported.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: Incorrect username or password. Please try again.
2: You must pass the robot test before logging in.
4: Account has been locked. Please request a password reset.
5: Unable to login. Please use Social Network sign on.
6: Account issue. Please contact Support.
9: Unable to login with provided credentials. Default login is required.
10: Received credentials are unverified.
12: Existing login session found. Please log out first.
14: The account is unable to log in. Please log in to the LuoBu app.
15: Too many attempts. Please wait a bit.
27: The account is unable to login. Please log in with the VNG app.`,
        },
        {
            status: 429,
            description: `7: Too many attempts. Please wait a bit.`,
        },
        {
            status: 503,
            description: `11: Service unavailable. Please try again.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/login/linked
 * @summary Endpoint for logging in a user, specifically for linked
authentication on PCGDK
 * @param body Roblox.Authentication.Api.Models.LoginRequest
 */
exports.postLoginLinked = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/login/linked',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_LoginRequest,
    response: Roblox_Authentication_Api_Models_LoginResponse,
    errors: [
        {
            status: 400,
            description: `0: An unexpected error occurred.
3: Username and Password are required. Please try again.
8: Login with received credential type is not supported.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: Incorrect username or password. Please try again.
2: You must pass the robot test before logging in.
4: Account has been locked. Please request a password reset.
5: Unable to login. Please use Social Network sign on.
6: Account issue. Please contact Support.
9: Unable to login with provided credentials. Default login is required.
10: Received credentials are unverified.
12: Existing login session found. Please log out first.
14: The account is unable to log in. Please log in to the LuoBu app.
15: Too many attempts. Please wait a bit.
27: The account is unable to login. Please log in with the VNG app.`,
        },
        {
            status: 429,
            description: `7: Too many attempts. Please wait a bit.`,
        },
        {
            status: 503,
            description: `11: Service unavailable. Please try again.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/logout
 * @summary Destroys the current authentication session.
 */
exports.postLogout = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/logout',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/logoutfromallsessionsandreauthenticate
 * @summary Logs out user from all other sessions.
 * @param body
 */
exports.postLogoutfromallsessionsandreauthenticate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/logoutfromallsessionsandreauthenticate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_LogoutFromAllSessionsAndReauthenticateRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/metadata
 * @summary Get the metadata
 */
exports.getMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/metadata',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: Roblox_Authentication_Api_Models_MetadataResponse,
    errors: [],
});
/**
 * @api POST https://auth.roblox.com/v1/passkey/DeleteCredentialBatch
 * @summary Disables a batch of credentials for the specified user.
 * @param body The request body!:DisableTwoStepVerificationRequest.
 */
exports.postPasskeyDeletecredentialbatch = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passkey/DeleteCredentialBatch',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_DeletePasskeysRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
0: An unknown error occurred with the request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `2: Feature disabled.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/passkey/finish-ar-preauth-registration
 * @summary Finishes account recovery pre-auth passkey registration by validating the recovery session,
deactivating the user's password, and completing passkey registration.
 * @param body The request body containing recovery session and passkey details.
 */
exports.postPasskeyFinishArPreauthRegistration = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passkey/finish-ar-preauth-registration',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_FinishARPreAuthPasskeyRegistrationRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `0: An unknown error occurred with the request.`,
        },
        {
            status: 401,
            description: `0: An unknown error occurred with the request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: Reached limit of pass keys registered.`,
        },
        {
            status: 500,
            description: `0: An unknown error occurred with the request.`,
        },
        {
            status: 503,
            description: `2: Feature disabled.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/passkey/finish-preauth-registration
 * @param body
 */
exports.postPasskeyFinishPreauthRegistration = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passkey/finish-preauth-registration',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_FinishPasskeyPreauthRegistrationRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `0: An unknown error occurred with the request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
0: An unknown error occurred with the request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: Reached limit of pass keys registered.`,
        },
        {
            status: 500,
            description: `0: An unknown error occurred with the request.`,
        },
        {
            status: 503,
            description: `2: Feature disabled.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/passkey/FinishRegistration
 * @summary Complete Passkey registration by providing credential creation options.
 * @param body The request body.Roblox.Authentication.Api.Models.Request.FinishPasskeyRegistrationRequest.
 */
exports.postPasskeyFinishregistration = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passkey/FinishRegistration',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_FinishPasskeyRegistrationRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `0: An unknown error occurred with the request.
3: Invalid security key nickname.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
0: An unknown error occurred with the request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
3: Invalid security key nickname.`,
        },
        {
            status: 503,
            description: `2: Feature disabled.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/passkey/ListCredentials
 * @summary List a user's registered passkeys.
 * @param body
 */
exports.postPasskeyListcredentials = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passkey/ListCredentials',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ all: zod_1.z.boolean() }),
    response: Roblox_Authentication_Api_Models_Response_ListPasskeyCredentialResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
0: An unknown error occurred with the request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `2: Feature disabled.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/passkey/start-authentication-by-user
 * @summary Initializes passkey authentication for the user(s) corresponding to the identifier provided.
 * @param body
 */
exports.postPasskeyStartAuthenticationByUser = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passkey/start-authentication-by-user',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_Request_StartAuthenticationByUserRequest,
    response: Roblox_Authentication_Api_Models_Response_StartAuthenticationByUserResponse,
    errors: [
        {
            status: 400,
            description: `5: User identifier and type are required.
6: Multi-user passkey authentication is not supported for this credential type.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
4: No passkeys registered for any users found.`,
        },
        {
            status: 503,
            description: `2: Feature disabled.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/passkey/start-preauth-registration
 * @summary Initiates  Passkey preauthenticated registration by providing credential creation options.
 * @param body
 */
exports.postPasskeyStartPreauthRegistration = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passkey/start-preauth-registration',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ username: zod_1.z.string() }),
    response: Roblox_Authentication_Api_Models_Response_StartPasskeyPreauthRegistrationResponse,
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `2: Feature disabled.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/passkey/StartAuthentication
 * @summary Provides a challenge for the Passkey to authenticate.
 */
exports.postPasskeyStartauthentication = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passkey/StartAuthentication',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: Roblox_Authentication_Api_Models_Response_StartAuthenticationResponse,
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `2: Feature disabled.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/passkey/StartRegistration
 * @summary Initiates Passkey registration by providing credential creation options.
 * @param body
 * @param flow
 */
exports.postPasskeyStartregistration = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passkey/StartRegistration',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        flow: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        flow: zod_1.z.string().optional(),
    },
    body: zod_1.z.object({ isSilentUpgrade: zod_1.z.boolean() }),
    response: Roblox_Authentication_Api_Models_Response_StartPasskeyRegistrationResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
0: An unknown error occurred with the request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: Reached limit of pass keys registered.`,
        },
        {
            status: 503,
            description: `2: Feature disabled.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/passwords/validate
 * @summary Endpoint for checking if a password is valid.
 * @param Username
 * @param Password
 */
exports.getPasswordsValidate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/passwords/validate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        Username: {
            style: 'form',
            explode: true,
        },
        Password: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        Username: zod_1.z.string(),
        Password: zod_1.z.string(),
    },
    response: Roblox_Authentication_Api_Models_PasswordValidationResponse,
    errors: [
        {
            status: 400,
            description: `1: Valid Username and Password are required. Please try again.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/passwords/validate
 * @summary Endpoint for checking if a password is valid.
 * @param body The Roblox.Authentication.Api.Models.PasswordValidationModel.
 */
exports.postPasswordsValidate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/passwords/validate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_PasswordValidationModel,
    response: Roblox_Authentication_Api_Models_PasswordValidationResponse,
    errors: [
        {
            status: 400,
            description: `1: Valid Username and Password are required. Please try again.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/recovery/metadata
 * @summary Get metadata for forgot endpoints
 */
exports.getRecoveryMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/recovery/metadata',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: Roblox_Authentication_Api_Models_RecoveryMetadataResponse,
    errors: [
        {
            status: 503,
            description: `7: The Roblox WeChat API is currently unavailable.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/revert/account
 * @summary Get Revert Account ticket info
 * @param ticket Ticket Guid to revert account.
 */
exports.getRevertAccount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/revert/account',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        ticket: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        ticket: zod_1.z.string(),
    },
    response: Roblox_Authentication_Api_Models_RevertAccountInfoResponse,
    errors: [
        {
            status: 400,
            description: `2: The account revert ticket is not valid`,
        },
        {
            status: 403,
            description: `13: Revert links are disabled for users in the Enhanced Protection Program.`,
        },
        {
            status: 503,
            description: `1: This feature is disabled`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/revert/account
 * @summary Submit Revert Account Request
 * @param body The Roblox.Authentication.Api.Models.RevertAccountSubmitRequest containing the necessary information to revert account.
 */
exports.postRevertAccount = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/revert/account',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_RevertAccountSubmitRequest,
    response: Roblox_Authentication_Api_Models_LoginResponse,
    errors: [
        {
            status: 400,
            description: `2: The account revert ticket is not valid
3: Password is not valid
4: Passwords do not match
5: Password cannot be used
8: The account security ticket is expired.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `0: Unknown
1: This feature is disabled`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/revert/invalidate-tickets
 * @summary Invalidates all account security tickets for the authenticated user.
This endpoint should be called before enrolling in EPP to ensure old revert links cannot be used.
 */
exports.postRevertInvalidateTickets = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/revert/invalidate-tickets',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `1: This feature is disabled`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/session/refresh
 * @summary Logs out user from the current session and create a new one.
 */
exports.postSessionRefresh = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/session/refresh',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/signup
 * @summary Endpoint for signing up a new user
 * @param body Roblox.Authentication.Api.Models.SignupRequest
 */
exports.postSignup = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/signup',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_SignupRequest,
    response: Roblox_Authentication_Api_Models_SignupResponse,
    errors: [
        {
            status: 400,
            description: `Bad request
16: User agreement ids are null.
21: Empty account switch blob required`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: Captcha Failed.
4: Invalid Birthday.
5: Invalid Username.
6: Username already taken.
7: Invalid Password.
8: Password and Username are same.
9: Password is too simple.
10: Email is invalid.
11: Asset is invalid.
12: Too many attempts. Please wait a bit.
17: One time Passcode session was not valid
22: Maximum logged in accounts limit reached.`,
        },
        {
            status: 429,
            description: `3: Too many attempts. Please wait a bit.`,
        },
        {
            status: 500,
            description: `Internal server error
15: Insert acceptances failed.
27: Pre-auth passkey registration failed`,
        },
        {
            status: 503,
            description: `Service unavailable`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/signup/linked
 * @summary Endpoint for signing up a new user, specifically for linked
authentication on PCGDK
 * @param body Roblox.Authentication.Api.Models.SignupRequest
 */
exports.postSignupLinked = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/signup/linked',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_SignupRequest,
    response: Roblox_Authentication_Api_Models_SignupResponse,
    errors: [
        {
            status: 400,
            description: `Bad request
16: User agreement ids are null.
21: Empty account switch blob required`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: Captcha Failed.
4: Invalid Birthday.
5: Invalid Username.
6: Username already taken.
7: Invalid Password.
8: Password and Username are same.
9: Password is too simple.
10: Email is invalid.
11: Asset is invalid.
12: Too many attempts. Please wait a bit.
17: One time Passcode session was not valid
22: Maximum logged in accounts limit reached.`,
        },
        {
            status: 429,
            description: `3: Too many attempts. Please wait a bit.`,
        },
        {
            status: 500,
            description: `Internal server error
15: Insert acceptances failed.
27: Pre-auth passkey registration failed`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/social/:provider/disconnect
 * @summary Removes the given social authentication method from current Roblox user if it is connected.
 * @param body The request body for additional parameters that may be required for disconnect
 * @param provider The social authentication provider, e.g. Facebook
 */
exports.postSocialProviderDisconnect = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/social/:provider/disconnect',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        provider: {
            style: 'simple',
        },
    },
    parameters: {
        provider: zod_1.z.string(),
    },
    body: zod_1.z.object({ Password: zod_1.z.string() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `Bad request
2: Unsupported social provider type.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Forbidden
0: Token Validation Failed
3: Cannot disconnect the only authentication method. Password on account is required.
4: The password provided is invalid.`,
        },
        {
            status: 500,
            description: `Internal server error`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/social/connected-providers
 * @summary Get social network user information if the given social auth method is connected to current user.
 */
exports.getSocialConnectedProviders = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/social/connected-providers',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: Roblox_Authentication_Api_Models_SocialProvidersResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/user/passwords/change
 * @summary Changes the password for the authenticated user.
 * @param body The request model including the current password, and the new password.
 * @description The current password is needed for verification that the password can be changed.
 */
exports.postUserPasswordsChange = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/user/passwords/change',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_PasswordChangeModel,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `Roblox.Web.Authentication.Passwords.PasswordResponseCodes.InvalidCurrentPassword
            OR
            Roblox.Web.Authentication.Passwords.PasswordResponseCodes.InvalidPassword`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Roblox.Web.Authentication.Passwords.PasswordResponseCodes.PinLocked
0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `Roblox.Web.Authentication.Passwords.PasswordResponseCodes.Flooded`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/username
 * @summary Change the user's username
 * @param body The Roblox.Authentication.Api.Models.UsernameChangeRequest
 */
exports.postUsername = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/username',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_UsernameChangeRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `5: You don&#x27;t have enough Robux to change your username.
10: This username is already in use
11: Username not appropriate for Roblox
12: Usernames can be 3 to 20 characters long
13: Usernames can’t start or end with _ and can have at most one _
14: Only a-z, A-Z, 0-9, and _ are allowed
15: Username is null
16: Username might contain private information
17: This username is not available
18: Username is same as current`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: PIN is locked.
2: A verified email is missing
3: Your password is incorrect.
100: Unknown birthday`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.
5: You don&#x27;t have enough Robux to change your username.`,
        },
        {
            status: 503,
            description: `4: The feature is currently not available. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/usernames
 * @summary Gets a list of existing usernames on Roblox based on the query parameters
 * @param username The username
 * @description This endpoint can be expanded in the future to include other query parameters such as "startsWith"
 */
exports.getUsernames = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/usernames',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        username: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        username: zod_1.z.string().optional(),
    },
    response: Roblox_Authentication_Api_Models_UsernamesResponse,
    errors: [],
});
/**
 * @api POST https://auth.roblox.com/v1/usernames/recover
 * @summary Sends an email of all accounts belonging to an email
 * @param body
 */
exports.postUsernamesRecover = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/usernames/recover',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_RecoverUsernameRequest,
    response: Roblox_Authentication_Api_Models_RecoverUsernameResponse,
    errors: [
        {
            status: 400,
            description: `20: Invalid Email
21: Invalid Phone
23: No Account Found`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
11: Too many attempts. Please wait a bit.`,
        },
        {
            status: 500,
            description: `0: An unexpected error occurred.`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/usernames/validate
 * @summary Checks if a username is valid.
 * @param Username
 * @param Birthday
 * @param Context
 */
exports.getUsernamesValidate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/usernames/validate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        Username: {
            style: 'form',
            explode: true,
        },
        Birthday: {
            style: 'form',
            explode: true,
        },
        Context: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        Username: zod_1.z.string(),
        Birthday: zod_1.z.string().datetime({ offset: true }),
        Context: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]),
    },
    response: Roblox_Authentication_Api_Models_UsernameValidationResponse,
    errors: [
        {
            status: 400,
            description: `1: A valid username is required.
2: A valid birthday or authenticated user is required.`,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/usernames/validate
 * @summary Checks if a username is valid.
 * @param body The Roblox.Authentication.Api.Models.UsernameValidationRequest.
 */
exports.postUsernamesValidate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/usernames/validate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_UsernameValidationRequest,
    response: Roblox_Authentication_Api_Models_UsernameValidationResponse,
    errors: [
        {
            status: 400,
            description: `1: A valid username is required.
2: A valid birthday or authenticated user is required.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/validators/email
 * @summary Tries to check if an email is valid
 * @param Email
 */
exports.getValidatorsEmail = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/validators/email',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        Email: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        Email: zod_1.z.string(),
    },
    response: zod_1.z.object({ isEmailValid: zod_1.z.boolean() }),
    errors: [],
});
/**
 * @api GET https://auth.roblox.com/v1/validators/recommendedUsernameFromDisplayName
 * @summary Validates the given display name, and if valid, will convert it to a valid username and return suggested username(s) if available.
 * @param DisplayName
 * @param BirthDay
 */
exports.getValidatorsRecommendedusernamefromdisplayname = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/validators/recommendedUsernameFromDisplayName',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        DisplayName: {
            style: 'form',
            explode: true,
        },
        BirthDay: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        DisplayName: zod_1.z.string(),
        BirthDay: zod_1.z.string().datetime({ offset: true }),
    },
    response: Roblox_Authentication_Api_Models_RecommendedUsernameResponse,
    errors: [],
});
/**
 * @api POST https://auth.roblox.com/v1/validators/recommendedUsernameFromDisplayName
 * @summary Validates the given display name, and if valid, will convert it to a valid username and return suggested username(s) if available.
            
This is a POST request and explicitly does not receive the parameter values from the query
 * @param body
 */
exports.postValidatorsRecommendedusernamefromdisplayname = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/validators/recommendedUsernameFromDisplayName',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_RecommendedUsernameFromDisplayNameRequest,
    response: Roblox_Authentication_Api_Models_RecommendedUsernameResponse,
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/validators/username
 * @summary Tries to get a valid username if the current username is taken
 * @param Username
 * @param BirthDay
 */
exports.getValidatorsUsername = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/validators/username',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        Username: {
            style: 'form',
            explode: true,
        },
        BirthDay: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        Username: zod_1.z.string(),
        BirthDay: zod_1.z.string().datetime({ offset: true }),
    },
    response: Roblox_Authentication_Api_Models_RecommendedUsernameResponse,
    errors: [],
});
/**
 * @api POST https://auth.roblox.com/v1/validators/username
 * @summary Tries to get a valid username if the current username is taken
This is a POST request and explicitly does not receive the parameter values from the query
 * @param body
 */
exports.postValidatorsUsername = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/validators/username',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_RecommendedUsernameRequest,
    response: Roblox_Authentication_Api_Models_RecommendedUsernameResponse,
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/xbox/connection
 * @summary Check if the current user has an Xbox connected.
Also returns the gamertag of the Xbox account if connected.
 */
exports.getXboxConnection = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/xbox/connection',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: Roblox_Authentication_Api_Models_XboxConnectionModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/xbox/disconnect
 * @summary Unlink the current ROBLOX account from the Xbox live account.
 */
exports.postXboxDisconnect = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/xbox/disconnect',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Forbidden
0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://auth.roblox.com/v1/xbox/get-login-consecutive-days
 * @summary Get the consecutive days the xbox user has been logged in.
 */
exports.getXboxGetLoginConsecutiveDays = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/xbox/get-login-consecutive-days',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `36: Invalid Xbox Live Account`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
        },
    ],
});
/**
 * @api POST https://auth.roblox.com/v1/xbox/translate
 * @summary Translate the xbox user to roblox user.
 * @param body
 */
exports.postXboxTranslate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/xbox/translate',
    baseUrl: 'https://auth.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Authentication_Api_Models_XboxTranslateRequest,
    response: Roblox_Authentication_Api_Models_XboxCollectionsOfUserResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `
0: Token Validation Failed`,
        },
    ],
});
