// rozod/lib/endpoints/avatarv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridOutfits = exports.getUsersUseridCurrentlyWearing = exports.getUsersUseridAvatar = exports.getOutfitsUseroutfitidDetails = exports.postOutfitsUseroutfitidDelete = exports.getGameStartInfo = exports.postAvatarSetScales = exports.postAvatarSetPlayerAvatarType = exports.postAvatarSetBodyColors = exports.postAvatarRedrawThumbnail = exports.getAvatarMetadata = exports.getAvatarRules = exports.getAvatar = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_Responses_Avatar_ScaleModel = zod_1.z.object({
    height: zod_1.z.number(),
    width: zod_1.z.number(),
    head: zod_1.z.number(),
    depth: zod_1.z.number(),
    proportion: zod_1.z.number(),
    bodyType: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_BodyColorsModel = zod_1.z.object({
    headColorId: zod_1.z.number().int(),
    torsoColorId: zod_1.z.number().int(),
    rightArmColorId: zod_1.z.number().int(),
    leftArmColorId: zod_1.z.number().int(),
    rightLegColorId: zod_1.z.number().int(),
    leftLegColorId: zod_1.z.number().int(),
});
const Roblox_Api_Avatar_Models_AssetTypeModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AssetPosition = zod_1.z.object({
    X: zod_1.z.number(),
    Y: zod_1.z.number(),
    Z: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetRotation = zod_1.z.object({
    X: zod_1.z.number(),
    Y: zod_1.z.number(),
    Z: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetScale = zod_1.z.object({
    X: zod_1.z.number(),
    Y: zod_1.z.number(),
    Z: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetMetaModelV1 = zod_1.z.object({
    order: zod_1.z.number().int(),
    puffiness: zod_1.z.number(),
    position: Roblox_Api_Avatar_Models_AssetPosition,
    rotation: Roblox_Api_Avatar_Models_AssetRotation,
    scale: Roblox_Api_Avatar_Models_AssetScale,
    headShape: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
        zod_1.z.literal(10),
        zod_1.z.literal(11),
        zod_1.z.literal(12),
        zod_1.z.literal(13),
        zod_1.z.literal(14),
        zod_1.z.literal(15),
        zod_1.z.literal(16),
        zod_1.z.literal(17),
        zod_1.z.literal(18),
        zod_1.z.literal(19),
        zod_1.z.literal(20),
        zod_1.z.literal(21),
        zod_1.z.literal(22),
        zod_1.z.literal(23),
        zod_1.z.literal(24),
        zod_1.z.literal(25),
        zod_1.z.literal(26),
        zod_1.z.literal(27),
        zod_1.z.literal(28),
        zod_1.z.literal(29),
    ]),
    staticFacialAnimation: zod_1.z.boolean(),
    version: zod_1.z.number().int(),
});
const Roblox_Api_Avatar_Models_AssetModelV2 = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    assetType: Roblox_Api_Avatar_Models_AssetTypeModel,
    currentVersionId: zod_1.z.number().int(),
    meta: Roblox_Api_Avatar_Models_AssetMetaModelV1,
    availabilityStatus: zod_1.z.string(),
    expirationTime: zod_1.z.string().datetime({ offset: true }),
    supportsHeadShapes: zod_1.z.boolean(),
});
const Roblox_Api_Avatar_Models_EmoteResponseModel = zod_1.z.object({
    assetId: zod_1.z.number().int(),
    assetName: zod_1.z.string(),
    position: zod_1.z.number().int(),
});
const Roblox_Api_Avatar_Models_AvatarModelV2 = zod_1.z.object({
    scales: Roblox_Web_Responses_Avatar_ScaleModel,
    playerAvatarType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(3)]),
    bodyColors: Roblox_Api_Avatar_Models_BodyColorsModel,
    assets: zod_1.z.array(Roblox_Api_Avatar_Models_AssetModelV2),
    defaultShirtApplied: zod_1.z.boolean(),
    defaultPantsApplied: zod_1.z.boolean(),
    emotes: zod_1.z.array(Roblox_Api_Avatar_Models_EmoteResponseModel),
});
const Roblox_Api_Avatar_Models_ScaleRulesModel = zod_1.z.object({
    min: zod_1.z.number(),
    max: zod_1.z.number(),
    increment: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetTypeRulesModel = zod_1.z.object({
    maxNumber: zod_1.z.number().int(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AccessoryPositionModel = zod_1.z.object({
    xPosition: zod_1.z.number(),
    yPosition: zod_1.z.number(),
    zPosition: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AccessoryRotationModel = zod_1.z.object({
    xRotation: zod_1.z.number(),
    yRotation: zod_1.z.number(),
    zRotation: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AccessoryScaleModel = zod_1.z.object({
    xScale: zod_1.z.number(),
    yScale: zod_1.z.number(),
    zScale: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AccessoryRefinementModel = zod_1.z.object({
    position: Roblox_Api_Avatar_Models_AccessoryPositionModel,
    rotation: Roblox_Api_Avatar_Models_AccessoryRotationModel,
    scale: Roblox_Api_Avatar_Models_AccessoryScaleModel,
});
const Roblox_Api_Avatar_Models_BodyColorModel = zod_1.z.object({
    brickColorId: zod_1.z.number().int(),
    hexColor: zod_1.z.string(),
    name: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_DefaultClothingAssets = zod_1.z.object({
    defaultShirtAssetIds: zod_1.z.array(zod_1.z.number()),
    defaultPantAssetIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Api_Avatar_Models_AvatarRulesModel = zod_1.z.object({
    playerAvatarTypes: zod_1.z.array(zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(3)])),
    scales: Roblox_Api_Avatar_Models_ScaleRulesModel,
    wearableAssetTypes: zod_1.z.array(Roblox_Api_Avatar_Models_AssetTypeRulesModel),
    accessoryRefinementTypes: zod_1.z.array(zod_1.z.number()),
    accessoryRefinementLowerBounds: Roblox_Api_Avatar_Models_AccessoryRefinementModel,
    accessoryRefinementUpperBounds: Roblox_Api_Avatar_Models_AccessoryRefinementModel,
    bodyColorsPalette: zod_1.z.array(Roblox_Api_Avatar_Models_BodyColorModel),
    basicBodyColorsPalette: zod_1.z.array(Roblox_Api_Avatar_Models_BodyColorModel),
    minimumDeltaEBodyColorDifference: zod_1.z.number(),
    proportionsAndBodyTypeEnabledForUser: zod_1.z.boolean(),
    defaultClothingAssetLists: Roblox_Api_Avatar_Models_DefaultClothingAssets,
    bundlesEnabledForUser: zod_1.z.boolean(),
    emotesEnabledForUser: zod_1.z.boolean(),
});
const Roblox_Api_Avatar_Models_AvatarMetadataModel = zod_1.z.object({
    enableDefaultClothingMessage: zod_1.z.boolean(),
    isAvatarScaleEmbeddedInTab: zod_1.z.boolean(),
    isBodyTypeScaleOutOfTab: zod_1.z.boolean(),
    scaleHeightIncrement: zod_1.z.number(),
    scaleWidthIncrement: zod_1.z.number(),
    scaleHeadIncrement: zod_1.z.number(),
    scaleProportionIncrement: zod_1.z.number(),
    scaleBodyTypeIncrement: zod_1.z.number(),
    supportProportionAndBodyType: zod_1.z.boolean(),
    showDefaultClothingMessageOnPageLoad: zod_1.z.boolean(),
    areThreeDeeThumbsEnabled: zod_1.z.boolean(),
    isAvatarWearingApiCallsLockingOnFrontendEnabled: zod_1.z.boolean(),
    isOutfitHandlingOnFrontendEnabled: zod_1.z.boolean(),
    isJustinUiChangesEnabled: zod_1.z.boolean(),
    isCategoryReorgEnabled: zod_1.z.boolean(),
    LCEnabledInEditorAndCatalog: zod_1.z.boolean(),
    isLCCompletelyEnabled: zod_1.z.boolean(),
});
const Roblox_Api_Avatar_Models_UniverseAvatarAssetOverrideResponseModel = zod_1.z.object({
    assetID: zod_1.z.number().int(),
    assetTypeID: zod_1.z.number().int(),
    isPlayerChoice: zod_1.z.boolean(),
});
const Roblox_Api_Avatar_Models_GameStartInfoResponse = zod_1.z.object({
    gameAvatarType: zod_1.z.string(),
    allowCustomAnimations: zod_1.z.string(),
    universeAvatarCollisionType: zod_1.z.string(),
    universeAvatarBodyType: zod_1.z.string(),
    jointPositioningType: zod_1.z.string(),
    message: zod_1.z.string(),
    universeAvatarMinScales: Roblox_Web_Responses_Avatar_ScaleModel,
    universeAvatarMaxScales: Roblox_Web_Responses_Avatar_ScaleModel,
    universeAvatarAssetOverrides: zod_1.z.array(Roblox_Api_Avatar_Models_UniverseAvatarAssetOverrideResponseModel),
    moderationStatus: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_OutfitDetailsModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    universeId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    assets: zod_1.z.array(Roblox_Api_Avatar_Models_AssetModelV2),
    bodyColors: Roblox_Api_Avatar_Models_BodyColorsModel,
    scale: Roblox_Web_Responses_Avatar_ScaleModel,
    playerAvatarType: zod_1.z.string(),
    outfitType: zod_1.z.string(),
    isEditable: zod_1.z.boolean(),
    moderationStatus: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AssetIdListModel = zod_1.z.object({
    assetIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Api_Avatar_Models_OutfitModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    isEditable: zod_1.z.boolean(),
    outfitType: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AvatarFilteredPageResponse_Roblox_Api_Avatar_Models_OutfitModel_ = zod_1.z.object({
    filteredCount: zod_1.z.number().int(),
    data: zod_1.z.array(Roblox_Api_Avatar_Models_OutfitModel),
    total: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Api_Avatar_Models_AvatarApiSuccessResponse = zod_1.z.object({
    success: zod_1.z.boolean(),
});
const Roblox_Api_Avatar_Models_PlayerAvatarTypeModel = zod_1.z.object({
    playerAvatarType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(3)]),
});
/**
 * @api GET https://avatar.roblox.com/v1/avatar
 * @summary Returns details about the authenticated user's avatar.
 * @param Roblox-Place-Id
 */
exports.getAvatar = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/avatar',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: Roblox_Api_Avatar_Models_AvatarModelV2,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://avatar.roblox.com/v1/avatar-rules
 * @summary Returns the business rules related to avatars.
 * @param Roblox-Place-Id
 * @description BodyColorsPalette is a list of valid brickColors you can choose for your avatar.
WearableAssetTypes contains a list of asset types with names, ids, and the maximum number that you can wear at a time.
Does not include packages because they cannot be worn on your avatar directly.
PlayerAvatarTypes are the types of avatars you can choose between.
 */
exports.getAvatarRules = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/avatar-rules',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: Roblox_Api_Avatar_Models_AvatarRulesModel,
    errors: [],
});
/**
 * @api GET https://avatar.roblox.com/v1/avatar/metadata
 * @summary Returns metadata used by the avatar page of the website.
 */
exports.getAvatarMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/avatar/metadata',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    response: Roblox_Api_Avatar_Models_AvatarMetadataModel,
    errors: [],
});
/**
 * @api POST https://avatar.roblox.com/v1/avatar/redraw-thumbnail
 * @summary Requests the authenticated user's thumbnail be redrawn.
 * @param Roblox-Place-Id
 */
exports.postAvatarRedrawThumbnail = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/avatar/redraw-thumbnail',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `1: Redrawing your avatar thumbnail is floodchecked at this time.
1: Redrawing your avatar thumbnail is floodchecked at this time`,
        },
    ],
});
/**
 * @api POST https://avatar.roblox.com/v1/avatar/set-body-colors
 * @summary Sets the authenticated user's body colors.
 * @param body
 * @param Roblox-Place-Id
 */
exports.postAvatarSetBodyColors = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/avatar/set-body-colors',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    body: Roblox_Api_Avatar_Models_BodyColorsModel,
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://avatar.roblox.com/v1/avatar/set-player-avatar-type
 * @summary Sets the authenticated user's player avatar type (e.g. R6 or R15).
 * @param body R6 or R15.
 * @param Roblox-Place-Id
 * @description This is the avatar type chosen on the Avatar page. Some games can override this and force your character to be R6 or R15.
 */
exports.postAvatarSetPlayerAvatarType = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/avatar/set-player-avatar-type',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    body: Roblox_Api_Avatar_Models_PlayerAvatarTypeModel,
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `1: Invalid playerAvatarType. Valid values are: `,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You are not allowed to change player avatar type.`,
        },
    ],
});
/**
 * @api POST https://avatar.roblox.com/v1/avatar/set-scales
 * @summary Sets the authenticated user's scales.
 * @param body
 * @param Roblox-Place-Id
 */
exports.postAvatarSetScales = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/avatar/set-scales',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    body: Roblox_Web_Responses_Avatar_ScaleModel,
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `1: Please pass in the scales JSON`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
3: The user does not have permissions to change scales.`,
        },
    ],
});
/**
 * @api GET https://avatar.roblox.com/v1/game-start-info
 * @summary The server will call this on game server start to request general information about the universe
This is version 1.1, which returns an entry from the UniverseAvatarType enum.
During mixed mode this may return unreliable results.
 * @param universeId
 * @param Roblox-Place-Id
 */
exports.getGameStartInfo = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/game-start-info',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'form',
            explode: true,
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: Roblox_Api_Avatar_Models_GameStartInfoResponse,
    errors: [],
});
/**
 * @api POST https://avatar.roblox.com/v1/outfits/:userOutfitId/delete
 * @summary Deletes the outfit.
 * @param userOutfitId The user outfit id.
 * @param Roblox-Place-Id
 * @description You are only allowed to delete outfits you created.
 */
exports.postOutfitsUseroutfitidDelete = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/outfits/:userOutfitId/delete',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userOutfitId: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        userOutfitId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `1: The specified userOutfitId is invalid!`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You don&#x27;t have permission to delete this outfit.`,
        },
        {
            status: 500,
            description: `3: An error occurred while deleting the outfit.`,
        },
    ],
});
/**
 * @api GET https://avatar.roblox.com/v1/outfits/:userOutfitId/details
 * @summary Gets details about the contents of an outfit.
 * @param userOutfitId The user outfit id.
 * @param Roblox-Place-Id
 */
exports.getOutfitsUseroutfitidDetails = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/outfits/:userOutfitId/details',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userOutfitId: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        userOutfitId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: Roblox_Api_Avatar_Models_OutfitDetailsModel,
    errors: [
        {
            status: 400,
            description: `1: The specified userOutfitId is invalid.
2: The outfit for the specified userOutfit is invalid.`,
        },
        {
            status: 403,
            description: `3: The requester does not have access to the details for the given user outfit.`,
        },
    ],
});
/**
 * @api GET https://avatar.roblox.com/v1/users/:userId/avatar
 * @summary Returns details about a specified user's avatar.
 * @param userId
 * @param Roblox-Place-Id
 * @description Includes assets, bodycolors, and playerAvatarType.
 */
exports.getUsersUseridAvatar = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/avatar',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: Roblox_Api_Avatar_Models_AvatarModelV2,
    errors: [
        {
            status: 400,
            description: `1: The specified user does not exist.
2: An account for the given userId does not exist!`,
        },
    ],
});
/**
 * @api GET https://avatar.roblox.com/v1/users/:userId/currently-wearing
 * @summary Gets a list of asset ids that the user is currently wearing.
 * @param userId The user id.
 * @param Roblox-Place-Id
 */
exports.getUsersUseridCurrentlyWearing = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/currently-wearing',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: Roblox_Api_Avatar_Models_AssetIdListModel,
    errors: [
        {
            status: 400,
            description: `1: The specified user does not exist.
2: An account for the given userId does not exist!`,
        },
    ],
});
/**
 * @api GET https://avatar.roblox.com/v1/users/:userId/outfits
 * @summary Deprecated, user v2.
Gets a list of outfits for the specified user.
 * @param userId The user id.
 * @param outfitType The outfit type being searched for, null will return all outfitTypes.
 * @param page The page number of the current page of requests, default is 1.
 * @param itemsPerPage The max number of outfits that can be returned.
 * @param isEditable Whether the outfits are editable. A null value will lead to no filtering.
 * @param Roblox-Place-Id The placeId of the caller, not required to be passed in.
 */
exports.getUsersUseridOutfits = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/outfits',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        outfitType: {
            style: 'form',
            explode: true,
        },
        page: {
            style: 'form',
            explode: true,
        },
        itemsPerPage: {
            style: 'form',
            explode: true,
        },
        isEditable: {
            style: 'form',
            explode: true,
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        outfitType: zod_1.z.string().optional(),
        page: zod_1.z.number().int().optional().default(1),
        itemsPerPage: zod_1.z.number().int().optional().default(25),
        isEditable: zod_1.z.boolean().optional(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: Roblox_Api_Avatar_Models_AvatarFilteredPageResponse_Roblox_Api_Avatar_Models_OutfitModel_,
    errors: [
        {
            status: 400,
            description: `1: The specified user does not exist.
2: An account for the given userId does not exist!`,
        },
    ],
});
