// rozod/lib/endpoints/avatarv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postOutfitsUseroutfitidUpdate = exports.postOutfitsCreate = exports.patchOutfitsUseroutfitid = exports.getAvatarUsersUseridOutfits = exports.getAvatarUsersUseridAvatar = exports.postAvatarSetWearingAssets = exports.postAvatarSetBodyColors = exports.getAvatarAvatar = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_Responses_Avatar_ScaleModel = zod_1.z.object({
    height: zod_1.z.number(),
    width: zod_1.z.number(),
    head: zod_1.z.number(),
    depth: zod_1.z.number(),
    proportion: zod_1.z.number(),
    bodyType: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_BodyColors3Model = zod_1.z.object({
    headColor3: zod_1.z.string(),
    torsoColor3: zod_1.z.string(),
    rightArmColor3: zod_1.z.string(),
    leftArmColor3: zod_1.z.string(),
    rightLegColor3: zod_1.z.string(),
    leftLegColor3: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AssetTypeModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AssetPosition = zod_1.z.object({
    X: zod_1.z.number(),
    Y: zod_1.z.number(),
    Z: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetRotation = zod_1.z.object({
    X: zod_1.z.number(),
    Y: zod_1.z.number(),
    Z: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetScale = zod_1.z.object({
    X: zod_1.z.number(),
    Y: zod_1.z.number(),
    Z: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetMetaModelV1 = zod_1.z.object({
    order: zod_1.z.number().int(),
    puffiness: zod_1.z.number(),
    position: Roblox_Api_Avatar_Models_AssetPosition,
    rotation: Roblox_Api_Avatar_Models_AssetRotation,
    scale: Roblox_Api_Avatar_Models_AssetScale,
    headShape: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
        zod_1.z.literal(10),
        zod_1.z.literal(11),
        zod_1.z.literal(12),
        zod_1.z.literal(13),
        zod_1.z.literal(14),
        zod_1.z.literal(15),
        zod_1.z.literal(16),
        zod_1.z.literal(17),
        zod_1.z.literal(18),
        zod_1.z.literal(19),
        zod_1.z.literal(20),
        zod_1.z.literal(21),
        zod_1.z.literal(22),
        zod_1.z.literal(23),
        zod_1.z.literal(24),
        zod_1.z.literal(25),
        zod_1.z.literal(26),
        zod_1.z.literal(27),
        zod_1.z.literal(28),
        zod_1.z.literal(29),
    ]),
    staticFacialAnimation: zod_1.z.boolean(),
    version: zod_1.z.number().int(),
});
const Roblox_Api_Avatar_Models_AssetModelV2 = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    assetType: Roblox_Api_Avatar_Models_AssetTypeModel,
    currentVersionId: zod_1.z.number().int(),
    meta: Roblox_Api_Avatar_Models_AssetMetaModelV1,
    availabilityStatus: zod_1.z.string(),
    expirationTime: zod_1.z.string().datetime({ offset: true }),
    supportsHeadShapes: zod_1.z.boolean(),
});
const Roblox_Api_Avatar_Models_EmoteResponseModel = zod_1.z.object({
    assetId: zod_1.z.number().int(),
    assetName: zod_1.z.string(),
    position: zod_1.z.number().int(),
});
const Roblox_Api_Avatar_Models_AvatarModelV3 = zod_1.z.object({
    scales: Roblox_Web_Responses_Avatar_ScaleModel,
    playerAvatarType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(3)]),
    bodyColor3s: Roblox_Api_Avatar_Models_BodyColors3Model,
    assets: zod_1.z.array(Roblox_Api_Avatar_Models_AssetModelV2),
    defaultShirtApplied: zod_1.z.boolean(),
    defaultPantsApplied: zod_1.z.boolean(),
    emotes: zod_1.z.array(Roblox_Api_Avatar_Models_EmoteResponseModel),
});
const Roblox_Api_Avatar_Models_OutfitModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    isEditable: zod_1.z.boolean(),
    outfitType: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AvatarPageResponse_Roblox_Api_Avatar_Models_OutfitModel_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Api_Avatar_Models_OutfitModel),
    paginationToken: zod_1.z.string(),
});
const Roblox_Platform_Avatar_BodyColorsModelV2 = zod_1.z.object({
    headColor3: zod_1.z.string(),
    torsoColor3: zod_1.z.string(),
    rightArmColor3: zod_1.z.string(),
    leftArmColor3: zod_1.z.string(),
    rightLegColor3: zod_1.z.string(),
    leftLegColor3: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AvatarApiSuccessResponse = zod_1.z.object({
    success: zod_1.z.boolean(),
});
const Roblox_Api_Avatar_Models_AssetWearModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    meta: Roblox_Api_Avatar_Models_AssetMetaModelV1,
});
const Roblox_Api_Avatar_Models_WearRequestModel = zod_1.z.object({
    assets: zod_1.z.array(Roblox_Api_Avatar_Models_AssetWearModel),
});
const Roblox_Api_Avatar_Models_WearResponseModel = zod_1.z.object({
    invalidAssets: zod_1.z.array(Roblox_Api_Avatar_Models_AssetModelV2),
    invalidAssetIds: zod_1.z.array(zod_1.z.number()),
    success: zod_1.z.boolean(),
});
const Roblox_Api_Avatar_Models_BodyColorsModel = zod_1.z.object({
    headColorId: zod_1.z.number().int(),
    torsoColorId: zod_1.z.number().int(),
    rightArmColorId: zod_1.z.number().int(),
    leftArmColorId: zod_1.z.number().int(),
    rightLegColorId: zod_1.z.number().int(),
    leftLegColorId: zod_1.z.number().int(),
});
const Roblox_Api_Avatar_Models_OutfitUpdateModelV2 = zod_1.z.object({
    name: zod_1.z.string(),
    bodyColors: Roblox_Api_Avatar_Models_BodyColorsModel,
    assets: zod_1.z.array(Roblox_Api_Avatar_Models_AssetWearModel),
    scale: Roblox_Web_Responses_Avatar_ScaleModel,
    playerAvatarType: zod_1.z.string(),
    outfitType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(4), zod_1.z.literal(5)]),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
/**
 * @api GET https://avatar.roblox.com/v2/avatar/avatar
 * @summary Returns details about the authenticated user's avatar.
 * @param Roblox-Place-Id
 * @param checkAssetAvailability Whether to return assets with availability status.
 */
exports.getAvatarAvatar = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/avatar/avatar',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        'Roblox-Place-Id': {
            style: 'simple',
        },
        checkAssetAvailability: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
        checkAssetAvailability: zod_1.z.boolean().optional(),
    },
    response: Roblox_Api_Avatar_Models_AvatarModelV3,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://avatar.roblox.com/v2/avatar/set-body-colors
 * @summary Sets the authenticated user's body colors.
 * @param body
 * @param Roblox-Place-Id
 */
exports.postAvatarSetBodyColors = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/avatar/set-body-colors',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    body: Roblox_Platform_Avatar_BodyColorsModelV2,
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://avatar.roblox.com/v2/avatar/set-wearing-assets
 * @summary Sets the avatar's current assets to the list.
 * @param body Model of assets to be worn.
 * @param Roblox-Place-Id
 * @description Only allows items that you own, are not expired, and are wearable asset types.
Any assets being worn before this method is called are automatically removed.
 */
exports.postAvatarSetWearingAssets = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/avatar/set-wearing-assets',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    body: Roblox_Api_Avatar_Models_WearRequestModel,
    response: Roblox_Api_Avatar_Models_WearResponseModel,
    errors: [
        {
            status: 400,
            description: `3: Invalid assetId
5: Meta does not apply to specified asset type
7: Required meta is not provided for the specific asset type`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `2: Failed to wear asset.`,
        },
    ],
});
/**
 * @api GET https://avatar.roblox.com/v2/avatar/users/:userId/avatar
 * @summary Returns details about a specified user's avatar.
 * @param userId
 * @param Roblox-Place-Id
 * @param checkAssetAvailability Whether to return assets with availability status.
 * @description Includes assets, bodycolors, and playerAvatarType.
 */
exports.getAvatarUsersUseridAvatar = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/avatar/users/:userId/avatar',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        checkAssetAvailability: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
        checkAssetAvailability: zod_1.z.boolean().optional(),
    },
    response: Roblox_Api_Avatar_Models_AvatarModelV3,
    errors: [
        {
            status: 400,
            description: `1: The specified user does not exist.
2: An account for the given userId does not exist!`,
        },
    ],
});
/**
 * @api GET https://avatar.roblox.com/v2/avatar/users/:userId/outfits
 * @summary Gets a list of outfits for the specified user.
 * @param userId The user id.
 * @param paginationToken The token received from the response to get the next page. For the first request, this value should be empty. Note : If no value is sent the 1st page will be returned.
 * @param outfitType The outfit type being searched for, null will return all outfitTypes.
 * @param page The page number of the current page of requests, default is 1.
 * @param itemsPerPage The max number of outfits that can be returned.
 * @param isEditable Whether the outfits are editable. A null value will lead to no filtering.
 * @param Roblox-Place-Id The placeId of the caller, not required to be passed in.
 */
exports.getAvatarUsersUseridOutfits = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/avatar/users/:userId/outfits',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        paginationToken: {
            style: 'form',
            explode: true,
        },
        outfitType: {
            style: 'form',
            explode: true,
        },
        page: {
            style: 'form',
            explode: true,
        },
        itemsPerPage: {
            style: 'form',
            explode: true,
        },
        isEditable: {
            style: 'form',
            explode: true,
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        paginationToken: zod_1.z.string().optional(),
        outfitType: zod_1.z.string().optional(),
        page: zod_1.z.number().int().optional().default(1),
        itemsPerPage: zod_1.z.number().int().optional().default(25),
        isEditable: zod_1.z.boolean().optional(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    response: Roblox_Api_Avatar_Models_AvatarPageResponse_Roblox_Api_Avatar_Models_OutfitModel_,
    errors: [
        {
            status: 400,
            description: `1: The specified user does not exist.
2: An account for the given userId does not exist!`,
        },
    ],
});
/**
 * @api PATCH https://avatar.roblox.com/v2/outfits/:userOutfitId
 * @summary Updates the contents of an outfit.
 * @param body The updated outfit.
 * @param userOutfitId The user outfit id.
 * @param Roblox-Place-Id
 * @description Fails if the user does not own any of the assetIds or if they are not wearable asset types.
Accepts partial updates.
 */
exports.patchOutfitsUseroutfitid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v2/outfits/:userOutfitId',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userOutfitId: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        userOutfitId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    body: Roblox_Api_Avatar_Models_OutfitUpdateModelV2,
    response: Roblox_Api_Avatar_Models_OutfitModel,
    errors: [
        {
            status: 400,
            description: `1: The specified userOutfit does not exist!
1: Must provide both assetIds and bodyColors in to update outfit contents.
3: Body colors must be valid BrickColor IDs
4: Invalid outfit name
5: Asset is not wearable by you
8: Invalid Player Avatar Type. Valid types are R6 and R15
11: Meta does not apply to specified asset type
12: Meta is required for this specific asset type`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You don&#x27;t have permission to update this outfit.`,
        },
        {
            status: 500,
            description: `6: An error occurred while trying to update the outfit`,
        },
    ],
});
/**
 * @api POST https://avatar.roblox.com/v2/outfits/create
 * @summary Creates a new outfit.
 * @param body The new outfit.
 * @param Roblox-Place-Id
 * @description Fails if any of the assetIds are not owned by the user, or not wearable types.
The name property of the request is optional as one will be auto-generated when the request has a null name.
 */
exports.postOutfitsCreate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/outfits/create',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    body: Roblox_Api_Avatar_Models_OutfitUpdateModelV2,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `3: Body colors must be valid BrickColor IDs
4: Invalid outfit name
5: Asset is not wearable by you and was not added to the outfit
7: Invalid Player Avatar Type. Valid types are R6 and R15
8: Invalid assetIds
9: Meta does not apply to specified asset type
10: Required meta is not provided for the specific asset type`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: You already have the maximum number of outfits`,
        },
        {
            status: 500,
            description: `6: An error occurred while creating the outfit`,
        },
    ],
});
// Patched endpoints removed from Roblox API docs (v6.1.0 compat)
exports.postOutfitsUseroutfitidUpdate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/outfits/:userOutfitId/update',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userOutfitId: { style: 'simple' },
    },
    parameters: { userOutfitId: zod_1.z.number().int() },
    body: Roblox_Api_Avatar_Models_OutfitUpdateModelV2,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The specified userOutfit does not exist!\n3: Body colors must be valid BrickColor IDs\n4: Invalid outfit name\n5: Asset is not wearable by you\n7: Invalid assetIds\n8: Invalid Player Avatar Type. Valid types are R6 and R15`,
        },
        { status: 401, description: `0: Authorization has been denied for this request.` },
        { status: 403, description: `0: Token Validation Failed\n2: You don't have permission to update this outfit.` },
        { status: 500, description: `6: An error occurred while trying to update the outfit` },
    ],
});
