// rozod/lib/endpoints/avatarv3.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postOutfitsCreate = exports.getOutfitsUseroutfitidDetails = exports.patchOutfitsUseroutfitid = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Api_Avatar_Models_AssetTypeModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AssetPosition = zod_1.z.object({
    X: zod_1.z.number(),
    Y: zod_1.z.number(),
    Z: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetRotation = zod_1.z.object({
    X: zod_1.z.number(),
    Y: zod_1.z.number(),
    Z: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetScale = zod_1.z.object({
    X: zod_1.z.number(),
    Y: zod_1.z.number(),
    Z: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_AssetMetaModelV1 = zod_1.z.object({
    order: zod_1.z.number().int(),
    puffiness: zod_1.z.number(),
    position: Roblox_Api_Avatar_Models_AssetPosition,
    rotation: Roblox_Api_Avatar_Models_AssetRotation,
    scale: Roblox_Api_Avatar_Models_AssetScale,
    headShape: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
        zod_1.z.literal(10),
        zod_1.z.literal(11),
        zod_1.z.literal(12),
        zod_1.z.literal(13),
        zod_1.z.literal(14),
        zod_1.z.literal(15),
        zod_1.z.literal(16),
        zod_1.z.literal(17),
        zod_1.z.literal(18),
        zod_1.z.literal(19),
        zod_1.z.literal(20),
        zod_1.z.literal(21),
        zod_1.z.literal(22),
        zod_1.z.literal(23),
        zod_1.z.literal(24),
        zod_1.z.literal(25),
        zod_1.z.literal(26),
        zod_1.z.literal(27),
        zod_1.z.literal(28),
        zod_1.z.literal(29),
    ]),
    staticFacialAnimation: zod_1.z.boolean(),
    version: zod_1.z.number().int(),
});
const Roblox_Api_Avatar_Models_AssetModelV2 = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    assetType: Roblox_Api_Avatar_Models_AssetTypeModel,
    currentVersionId: zod_1.z.number().int(),
    meta: Roblox_Api_Avatar_Models_AssetMetaModelV1,
    availabilityStatus: zod_1.z.string(),
    expirationTime: zod_1.z.string().datetime({ offset: true }),
    supportsHeadShapes: zod_1.z.boolean(),
});
const Roblox_Api_Avatar_Models_BodyColors3Model = zod_1.z.object({
    headColor3: zod_1.z.string(),
    torsoColor3: zod_1.z.string(),
    rightArmColor3: zod_1.z.string(),
    leftArmColor3: zod_1.z.string(),
    rightLegColor3: zod_1.z.string(),
    leftLegColor3: zod_1.z.string(),
});
const Roblox_Web_Responses_Avatar_ScaleModel = zod_1.z.object({
    height: zod_1.z.number(),
    width: zod_1.z.number(),
    head: zod_1.z.number(),
    depth: zod_1.z.number(),
    proportion: zod_1.z.number(),
    bodyType: zod_1.z.number(),
});
const Roblox_Api_Avatar_Models_OutfitDetailsModelV2 = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    assets: zod_1.z.array(Roblox_Api_Avatar_Models_AssetModelV2),
    bodyColor3s: Roblox_Api_Avatar_Models_BodyColors3Model,
    scale: Roblox_Web_Responses_Avatar_ScaleModel,
    playerAvatarType: zod_1.z.string(),
    outfitType: zod_1.z.string(),
    isEditable: zod_1.z.boolean(),
    universeId: zod_1.z.number().int(),
    moderationStatus: zod_1.z.string(),
    bundleId: zod_1.z.number().int(),
    inventoryType: zod_1.z.string(),
});
const Roblox_Api_Avatar_Models_AssetWearModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    meta: Roblox_Api_Avatar_Models_AssetMetaModelV1,
});
const Roblox_Api_Avatar_Models_OutfitUpdateModelV3 = zod_1.z.object({
    name: zod_1.z.string(),
    bodyColor3s: Roblox_Api_Avatar_Models_BodyColors3Model,
    assets: zod_1.z.array(Roblox_Api_Avatar_Models_AssetWearModel),
    scale: Roblox_Web_Responses_Avatar_ScaleModel,
    playerAvatarType: zod_1.z.string(),
    outfitType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(4), zod_1.z.literal(5)]),
});
const Roblox_Api_Avatar_Models_OutfitModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    isEditable: zod_1.z.boolean(),
    outfitType: zod_1.z.string(),
});
/**
 * @api PATCH https://avatar.roblox.com/v3/outfits/:userOutfitId
 * @summary Updates the contents of an outfit.
 * @param body The updated outfit.
 * @param userOutfitId The user outfit id.
 * @param Roblox-Place-Id
 * @description Fails if the user does not own any of the assetIds or if they are not wearable asset types.
Accepts partial updates.
 */
exports.patchOutfitsUseroutfitid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v3/outfits/:userOutfitId',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userOutfitId: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        userOutfitId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    body: Roblox_Api_Avatar_Models_OutfitUpdateModelV3,
    response: Roblox_Api_Avatar_Models_OutfitModel,
    errors: [
        {
            status: 400,
            description: `1: The specified userOutfit does not exist!
1: Must provide both assetIds and bodyColors in to update outfit contents.
3: Body colors must be valid BrickColor IDs
4: Invalid outfit name
5: Asset is not wearable by you
8: Invalid Player Avatar Type. Valid types are R6 and R15
11: Meta does not apply to specified asset type
12: Meta is required for this specific asset type`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You don&#x27;t have permission to update this outfit.`,
        },
        {
            status: 500,
            description: `6: An error occurred while trying to update the outfit`,
        },
    ],
});
/**
 * @api GET https://avatar.roblox.com/v3/outfits/:userOutfitId/details
 * @summary Gets details about the contents of an outfit.
 * @param userOutfitId The user outfit id.
 * @param Roblox-Place-Id
 * @param checkAssetAvailability Whether to return assets with availability status.
 */
exports.getOutfitsUseroutfitidDetails = (0, __1.endpoint)({
    method: 'GET',
    path: '/v3/outfits/:userOutfitId/details',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userOutfitId: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        checkAssetAvailability: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userOutfitId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
        checkAssetAvailability: zod_1.z.boolean().optional(),
    },
    response: Roblox_Api_Avatar_Models_OutfitDetailsModelV2,
    errors: [
        {
            status: 400,
            description: `1: The specified userOutfitId is invalid.
2: The outfit for the specified userOutfit is invalid.`,
        },
        {
            status: 403,
            description: `3: The requester does not have access to the details for the given user outfit.`,
        },
    ],
});
/**
 * @api POST https://avatar.roblox.com/v3/outfits/create
 * @summary Creates a new outfit.
 * @param body The new outfit.
 * @param Roblox-Place-Id
 * @description Fails if any of the assetIds are not owned by the user, or not wearable types.
The name property of the request is optional as one will be auto-generated when the request has a null name.
 */
exports.postOutfitsCreate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v3/outfits/create',
    baseUrl: 'https://avatar.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        'Roblox-Place-Id': {
            style: 'simple',
        },
    },
    parameters: {
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
    },
    body: Roblox_Api_Avatar_Models_OutfitUpdateModelV3,
    response: Roblox_Api_Avatar_Models_OutfitModel,
    errors: [
        {
            status: 400,
            description: `3: Body colors must be valid BrickColor IDs
4: Invalid outfit name
5: Asset is not wearable by you and was not added to the outfit
7: Invalid Player Avatar Type. Valid types are R6 and R15
8: Invalid assetIds
9: Meta does not apply to specified asset type
10: Required meta is not provided for the specific asset type`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: You already have the maximum number of outfits`,
        },
        {
            status: 500,
            description: `6: An error occurred while creating the outfit`,
        },
    ],
});
