// rozod/lib/endpoints/badgesv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridBadgesAwardedDates = exports.getUsersUseridBadgesBadgeidAwardedDate = exports.getUsersUseridBadges = exports.deleteUserBadgesBadgeid = exports.getUniversesUniverseidFreeBadgesQuota = exports.postUniversesUniverseidBadges = exports.getUniversesUniverseidBadges = exports.getBadgesMetadata = exports.postBadgesBadgeidIcon = exports.patchBadgesBadgeid = exports.getBadgesBadgeid = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_Responses_Badges_BadgeAwardStatisticsResponse = zod_1.z.object({
    pastDayAwardedCount: zod_1.z.number().int(),
    awardedCount: zod_1.z.number().int(),
    winRatePercentage: zod_1.z.number(),
});
const Roblox_Badges_Api_UniverseResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    rootPlaceId: zod_1.z.number().int(),
});
const Roblox_Badges_Api_BadgeResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    displayName: zod_1.z.string(),
    displayDescription: zod_1.z.string(),
    enabled: zod_1.z.boolean(),
    iconImageId: zod_1.z.number().int(),
    displayIconImageId: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
    statistics: Roblox_Web_Responses_Badges_BadgeAwardStatisticsResponse,
    awardingUniverse: Roblox_Badges_Api_UniverseResponse,
});
const Roblox_Badges_Api_UpdateBadgeRequest = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    enabled: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Badges_Api_BadgeMetadataResponse = zod_1.z.object({
    badgeCreationPrice: zod_1.z.number().int(),
    maxBadgeNameLength: zod_1.z.number().int(),
    maxBadgeDescriptionLength: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Badges_Api_BadgeResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Badges_Api_BadgeResponse),
});
const universeId_badges_body = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    paymentSourceType: zod_1.z.enum(['User', 'Group']),
    files: zod_1.z.instanceof(File),
    expectedCost: zod_1.z.number().int(),
    isActive: zod_1.z.boolean(),
});
const Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Badges_BadgeAwarderType_ = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.literal('Place'),
    name: zod_1.z.string(),
});
const Roblox_Web_Responses_Badges_BadgeResponseV2 = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    displayName: zod_1.z.string(),
    displayDescription: zod_1.z.string(),
    enabled: zod_1.z.boolean(),
    iconImageId: zod_1.z.number().int(),
    displayIconImageId: zod_1.z.number().int(),
    awarder: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Badges_BadgeAwarderType_,
    statistics: Roblox_Web_Responses_Badges_BadgeAwardStatisticsResponse,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Badges_Api_BadgeCreatorResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    type: zod_1.z.string(),
});
const Roblox_Badges_Api_GetBadgesByUserResponse = zod_1.z.object({
    creator: Roblox_Badges_Api_BadgeCreatorResponse,
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    displayName: zod_1.z.string(),
    displayDescription: zod_1.z.string(),
    enabled: zod_1.z.boolean(),
    iconImageId: zod_1.z.number().int(),
    displayIconImageId: zod_1.z.number().int(),
    awarder: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Badges_BadgeAwarderType_,
    statistics: Roblox_Web_Responses_Badges_BadgeAwardStatisticsResponse,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Badges_Api_GetBadgesByUserResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Badges_Api_GetBadgesByUserResponse),
});
const Roblox_Badges_Api_BadgeAwardResponse = zod_1.z.object({
    badgeId: zod_1.z.number().int(),
    awardedDate: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Badges_Api_BadgeAwardResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Badges_Api_BadgeAwardResponse),
});
const badgeId_icon_body = zod_1.z.object({ Files: zod_1.z.instanceof(File) });
const Roblox_Badges_Api_IconUploadResponse = zod_1.z.object({
    targetId: zod_1.z.number().int(),
});
/**
 * @api GET https://badges.roblox.com/v1/badges/:badgeId
 * @summary Gets badge information by the badge Id.
 * @param badgeId The badge Id.
 */
exports.getBadgesBadgeid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/badges/:badgeId',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        badgeId: {
            style: 'simple',
        },
    },
    parameters: {
        badgeId: zod_1.z.number().int(),
    },
    response: Roblox_Badges_Api_BadgeResponse,
    errors: [
        {
            status: 404,
            description: `1: Badge is invalid or does not exist.
3: The game is invalid or does not exist.`,
        },
    ],
});
/**
 * @api PATCH https://badges.roblox.com/v1/badges/:badgeId
 * @summary Updates badge configuration.
 * @param body The request body.
 * @param badgeId The badge Id.
 */
exports.patchBadgesBadgeid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/badges/:badgeId',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        badgeId: {
            style: 'simple',
        },
    },
    parameters: {
        badgeId: zod_1.z.number().int(),
    },
    body: Roblox_Badges_Api_UpdateBadgeRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `6: Text moderated.
14: Invalid badge name.
15: Invalid badge description.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to manage this badge.`,
        },
        {
            status: 404,
            description: `1: Badge is invalid or does not exist.
3: The game is invalid or does not exist.`,
        },
    ],
});
/**
 * @api POST https://badges.roblox.com/v1/badges/:badgeId/icon
 * @summary Overwrites a badge icon with a new one.
 * @param body
 * @param badgeId The badge Id.
 */
exports.postBadgesBadgeidIcon = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/badges/:badgeId/icon',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        badgeId: {
            style: 'simple',
        },
    },
    parameters: {
        badgeId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ Files: zod_1.z.instanceof(File) }),
    response: zod_1.z.object({ targetId: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `6: Text moderated.
22: Icon file is not present in the request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to manage this badge.`,
        },
        {
            status: 404,
            description: `1: Badge is invalid or does not exist.`,
        },
        {
            status: 429,
            description: `13: Too many requests, try again later.`,
        },
    ],
});
/**
 * @api GET https://badges.roblox.com/v1/badges/metadata
 * @summary Gets metadata about the badges system.
 */
exports.getBadgesMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/badges/metadata',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'json',
    response: Roblox_Badges_Api_BadgeMetadataResponse,
    errors: [],
});
/**
 * @api GET https://badges.roblox.com/v1/universes/:universeId/badges
 * @summary Gets badges by their awarding game.
 * @param universeId The universe Id.
 * @param sortBy The key to sort badges by.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getUniversesUniverseidBadges = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId/badges',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
        sortBy: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        sortBy: zod_1.z.enum(['Rank', 'DateCreated']).optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Badges_Api_BadgeResponse_,
    errors: [
        {
            status: 404,
            description: `3: The game is invalid or does not exist.`,
        },
    ],
});
/**
 * @api POST https://badges.roblox.com/v1/universes/:universeId/badges
 * @summary Creates a new badge.
 * @param body
 * @param universeId The ID of the universe to create the badge for.
 */
exports.postUniversesUniverseidBadges = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/universes/:universeId/badges',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    body: universeId_badges_body,
    response: Roblox_Web_Responses_Badges_BadgeResponseV2,
    errors: [
        {
            status: 400,
            description: `11: The badge icon is invalid.
14: Invalid badge name.
15: Invalid badge description.
16: Payment source is invalid.
18: Expected badge cost is different from the actual badge cost.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: Text moderated.
12: You do not have permission to manage this game&#x27;s badges.
17: Insufficient funds.`,
        },
        {
            status: 404,
            description: `3: The game is invalid or does not exist.`,
        },
        {
            status: 429,
            description: `13: Too many requests, try again later.`,
        },
    ],
});
/**
 * @api GET https://badges.roblox.com/v1/universes/:universeId/free-badges-quota
 * @summary Gets the number of free badges left for the current UTC day by their awarding game.
 * @param universeId The universe Id.
 */
exports.getUniversesUniverseidFreeBadgesQuota = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId/free-badges-quota',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: zod_1.z.number().int(),
    errors: [
        {
            status: 404,
            description: `3: The game is invalid or does not exist.`,
        },
    ],
});
/**
 * @api DELETE https://badges.roblox.com/v1/user/badges/:badgeId
 * @summary Removes a badge from the authenticated user.
 * @param badgeId The badge Id.
 */
exports.deleteUserBadgesBadgeid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/user/badges/:badgeId',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        badgeId: {
            style: 'simple',
        },
    },
    parameters: {
        badgeId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 404,
            description: `1: Badge is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://badges.roblox.com/v1/users/:userId/badges
 * @summary Gets a list of badges a user has been awarded.
 * @param userId The user Id.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getUsersUseridBadges = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/badges',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Badges_Api_GetBadgesByUserResponse_,
    errors: [
        {
            status: 404,
            description: `4: User is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://badges.roblox.com/v1/users/:userId/badges/:badgeId/awarded-date
 * @summary Gets timestamp for when a single badge was awarded to a user.
 * @param userId User id.
 * @param badgeId Badge id.
 */
exports.getUsersUseridBadgesBadgeidAwardedDate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/badges/:badgeId/awarded-date',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        badgeId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        badgeId: zod_1.z.number().int(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 404,
            description: `4: User is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://badges.roblox.com/v1/users/:userId/badges/awarded-dates
 * @summary Gets timestamps for when badges were awarded to a user.
 * @param userId The user Id.
 * @param badgeIds The CSV of badge Ids.
 */
exports.getUsersUseridBadgesAwardedDates = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/badges/awarded-dates',
    baseUrl: 'https://badges.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        badgeIds: {
            style: 'form',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        badgeIds: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Badges_Api_BadgeAwardResponse_,
    errors: [
        {
            status: 400,
            description: `5: Too many badge Ids.`,
        },
        {
            status: 404,
            description: `4: User is invalid or does not exist.`,
        },
    ],
});
