// rozod/lib/endpoints/catalogv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridBundlesBundletype = exports.getUsersUseridBundles = exports.postTopicGetTopics = exports.getSubcategories = exports.getFavoritesUsersUseridFavoritesSubtypeidBundles = exports.getFavoritesUsersUseridFavoritesAssettypeidAssets = exports.deleteFavoritesUsersUseridBundlesBundleidFavorite = exports.postFavoritesUsersUseridBundlesBundleidFavorite = exports.getFavoritesUsersUseridBundlesBundleidFavorite = exports.deleteFavoritesUsersUseridAssetsAssetidFavorite = exports.postFavoritesUsersUseridAssetsAssetidFavorite = exports.getFavoritesUsersUseridAssetsAssetidFavorite = exports.getFavoritesBundlesBundleidCount = exports.getFavoritesAssetsAssetidCount = exports.getCategories = exports.postCatalogItemsDetails = exports.getBundlesDetails = exports.getBundlesBundleidRecommendations = exports.getBundlesBundleidDetails = exports.getAssetsAssetidBundles = exports.getAssetToSubcategory = exports.getAssetToCategory = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Catalog_Api_BundleItemDetailModel = zod_1.z.object({
    owned: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    type: zod_1.z.string(),
    supportsHeadShapes: zod_1.z.boolean(),
    assetType: zod_1.z.number().int(),
});
const Roblox_Catalog_Api_BundleCreatorModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    type: zod_1.z.string(),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Catalog_Api_PremiumPricingModel = zod_1.z.object({
    premiumDiscountPercentage: zod_1.z.number().int(),
    premiumPriceInRobux: zod_1.z.number().int(),
});
const Roblox_Catalog_Api_BundleProductModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.string(),
    isPublicDomain: zod_1.z.boolean(),
    isForSale: zod_1.z.boolean(),
    priceInRobux: zod_1.z.number().int(),
    isFree: zod_1.z.boolean(),
    noPriceText: zod_1.z.string(),
    premiumPricing: Roblox_Catalog_Api_PremiumPricingModel,
});
const Roblox_Catalog_Api_SaleLocation = zod_1.z.object({
    saleLocationType: zod_1.z.enum([
        'NotApplicable',
        'ShopOnly',
        'MyExperiencesOnly',
        'ShopAndMyExperiences',
        'ExperiencesById',
        'ShopAndAllExperiences',
        'ExperiencesDevApiOnly',
        'ShopAndExperiencesById',
    ]),
    saleLocationTypeId: zod_1.z.number().int(),
    universeIds: zod_1.z.array(zod_1.z.number()),
    enabledUniverseIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Catalog_Api_CollectibleItemDetail = zod_1.z.object({
    collectibleItemId: zod_1.z.string(),
    collectibleProductId: zod_1.z.string(),
    price: zod_1.z.number().int(),
    lowestPrice: zod_1.z.number().int(),
    lowestResalePrice: zod_1.z.number().int(),
    totalQuantity: zod_1.z.number().int(),
    unitsAvailable: zod_1.z.number().int(),
    saleLocation: Roblox_Catalog_Api_SaleLocation,
    hasResellers: zod_1.z.boolean(),
    saleStatus: zod_1.z.enum(['Invalid', 'Draft', 'OffSale', 'OnSale', 'PendingSale']),
    quantityLimitPerUser: zod_1.z.number().int(),
    offSaleDeadline: zod_1.z.string().datetime({ offset: true }),
    collectibleItemType: zod_1.z.enum(['Invalid', 'Limited', 'NonLimited']),
    lowestAvailableResaleProductId: zod_1.z.string(),
    lowestAvailableResaleItemInstanceId: zod_1.z.string(),
    resaleRestriction: zod_1.z.enum(['Invalid', 'None', 'Disabled']),
});
const Roblox_Catalog_Api_BundleDetailsModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    bundleType: zod_1.z.string(),
    isRecolorable: zod_1.z.boolean(),
    items: zod_1.z.array(Roblox_Catalog_Api_BundleItemDetailModel),
    creator: Roblox_Catalog_Api_BundleCreatorModel,
    product: Roblox_Catalog_Api_BundleProductModel,
    itemRestrictions: zod_1.z.array(zod_1.z.union([
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
    ])),
    collectibleItemDetail: Roblox_Catalog_Api_CollectibleItemDetail,
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Catalog_Api_BundleDetailsModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Catalog_Api_BundleDetailsModel),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Catalog_Api_BundleDetailsModel_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Catalog_Api_BundleDetailsModel),
});
const Roblox_Catalog_Api_SubcategoryModel = zod_1.z.object({
    subcategory: zod_1.z.string(),
    taxonomy: zod_1.z.string(),
    assetTypeIds: zod_1.z.array(zod_1.z.number()),
    bundleTypeIds: zod_1.z.array(zod_1.z.number()),
    subcategoryId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    shortName: zod_1.z.string(),
});
const Roblox_Catalog_Api_CategoryModel = zod_1.z.object({
    category: zod_1.z.string(),
    taxonomy: zod_1.z.string(),
    assetTypeIds: zod_1.z.array(zod_1.z.number()),
    bundleTypeIds: zod_1.z.array(zod_1.z.number()),
    categoryId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    orderIndex: zod_1.z.number().int(),
    subcategories: zod_1.z.array(Roblox_Catalog_Api_SubcategoryModel),
    isSearchable: zod_1.z.boolean(),
});
const Roblox_Catalog_Api_AssetFavoriteModel = zod_1.z.object({
    assetId: zod_1.z.number().int(),
    userId: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Catalog_Api_BundleFavoriteModel = zod_1.z.object({
    bundleId: zod_1.z.number().int(),
    userId: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Catalog_Api_TimedOption = zod_1.z.object({
    days: zod_1.z.number().int(),
    price: zod_1.z.number().int(),
    selected: zod_1.z.boolean(),
});
const Roblox_Catalog_Api_CatalogSearchDetailedResponseItem = zod_1.z.object({
    id: zod_1.z.number().int(),
    itemType: zod_1.z.enum(['Asset', 'Bundle']),
    assetType: zod_1.z.union([
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
        zod_1.z.literal(10),
        zod_1.z.literal(11),
        zod_1.z.literal(12),
        zod_1.z.literal(13),
        zod_1.z.literal(16),
        zod_1.z.literal(17),
        zod_1.z.literal(18),
        zod_1.z.literal(19),
        zod_1.z.literal(21),
        zod_1.z.literal(22),
        zod_1.z.literal(24),
        zod_1.z.literal(25),
        zod_1.z.literal(26),
        zod_1.z.literal(27),
        zod_1.z.literal(28),
        zod_1.z.literal(29),
        zod_1.z.literal(30),
        zod_1.z.literal(31),
        zod_1.z.literal(32),
        zod_1.z.literal(33),
        zod_1.z.literal(34),
        zod_1.z.literal(35),
        zod_1.z.literal(37),
        zod_1.z.literal(38),
        zod_1.z.literal(39),
        zod_1.z.literal(40),
        zod_1.z.literal(41),
        zod_1.z.literal(42),
        zod_1.z.literal(43),
        zod_1.z.literal(44),
        zod_1.z.literal(45),
        zod_1.z.literal(46),
        zod_1.z.literal(47),
        zod_1.z.literal(48),
        zod_1.z.literal(49),
        zod_1.z.literal(50),
        zod_1.z.literal(51),
        zod_1.z.literal(52),
        zod_1.z.literal(53),
        zod_1.z.literal(54),
        zod_1.z.literal(55),
        zod_1.z.literal(56),
        zod_1.z.literal(59),
        zod_1.z.literal(60),
        zod_1.z.literal(61),
        zod_1.z.literal(62),
        zod_1.z.literal(63),
        zod_1.z.literal(64),
        zod_1.z.literal(65),
        zod_1.z.literal(66),
        zod_1.z.literal(67),
        zod_1.z.literal(68),
        zod_1.z.literal(69),
        zod_1.z.literal(70),
        zod_1.z.literal(71),
        zod_1.z.literal(72),
        zod_1.z.literal(73),
        zod_1.z.literal(74),
        zod_1.z.literal(75),
        zod_1.z.literal(76),
        zod_1.z.literal(77),
        zod_1.z.literal(78),
        zod_1.z.literal(79),
        zod_1.z.literal(80),
        zod_1.z.literal(81),
        zod_1.z.literal(82),
        zod_1.z.literal(83),
        zod_1.z.literal(84),
        zod_1.z.literal(85),
        zod_1.z.literal(86),
        zod_1.z.literal(87),
        zod_1.z.literal(88),
        zod_1.z.literal(89),
        zod_1.z.literal(90),
    ]),
    bundleType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
    isRecolorable: zod_1.z.boolean(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    productId: zod_1.z.number().int(),
    bundledItems: zod_1.z.array(Roblox_Catalog_Api_BundleItemDetailModel),
    itemStatus: zod_1.z.array(zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(7)])),
    itemRestrictions: zod_1.z.array(zod_1.z.union([
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
    ])),
    creatorHasVerifiedBadge: zod_1.z.boolean(),
    creatorType: zod_1.z.string(),
    creatorTargetId: zod_1.z.number().int(),
    creatorName: zod_1.z.string(),
    price: zod_1.z.number().int(),
    lowestPrice: zod_1.z.number().int(),
    lowestResalePrice: zod_1.z.number().int(),
    priceStatus: zod_1.z.string(),
    unitsAvailableForConsumption: zod_1.z.number().int(),
    favoriteCount: zod_1.z.number().int(),
    offSaleDeadline: zod_1.z.string().datetime({ offset: true }),
    collectibleItemId: zod_1.z.string(),
    totalQuantity: zod_1.z.number().int(),
    saleLocationType: zod_1.z.enum([
        'NotApplicable',
        'ShopOnly',
        'MyExperiencesOnly',
        'ShopAndMyExperiences',
        'ExperiencesById',
        'ShopAndAllExperiences',
        'ExperiencesDevApiOnly',
        'ShopAndExperiencesById',
    ]),
    hasResellers: zod_1.z.boolean(),
    isOffSale: zod_1.z.boolean(),
    quantityLimitPerUser: zod_1.z.number().int(),
    supportsHeadShapes: zod_1.z.boolean(),
    timedOptions: zod_1.z.array(Roblox_Catalog_Api_TimedOption),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Catalog_Api_CatalogSearchDetailedResponseItem_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Catalog_Api_CatalogSearchDetailedResponseItem),
});
const Roblox_Catalog_Api_FavoriteBundlesResponse = zod_1.z.object({
    favorites: zod_1.z.array(Roblox_Catalog_Api_BundleDetailsModel),
    moreFavorites: zod_1.z.boolean(),
    nextCursor: zod_1.z.string(),
    previousCursor: zod_1.z.string(),
});
const Roblox_Catalog_Api_OwnedBundleModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    bundleType: zod_1.z.string(),
    creator: Roblox_Catalog_Api_BundleCreatorModel,
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Catalog_Api_OwnedBundleModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Catalog_Api_OwnedBundleModel),
});
const Roblox_Catalog_Api_MultigetItemDetailsRequestItem = zod_1.z.object({
    itemType: zod_1.z.enum(['Asset', 'Bundle']),
    id: zod_1.z.number().int(),
});
const Roblox_Catalog_Api_MultigetItemDetailsRequestModel = zod_1.z.object({
    items: zod_1.z.array(Roblox_Catalog_Api_MultigetItemDetailsRequestItem),
});
const Roblox_Catalog_Api_BundleItemDetailModelV2 = zod_1.z.object({
    owned: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    type: zod_1.z.string(),
    supportsHeadShapes: zod_1.z.boolean(),
    assetType: zod_1.z.number().int(),
});
const Roblox_Catalog_Api_TaxonomyModel = zod_1.z.object({
    taxonomyId: zod_1.z.string(),
    taxonomyName: zod_1.z.string(),
});
const Roblox_Catalog_Api_CatalogSearchDetailedResponseItemV2 = zod_1.z.object({
    bundledItems: zod_1.z.array(Roblox_Catalog_Api_BundleItemDetailModelV2),
    taxonomy: zod_1.z.array(Roblox_Catalog_Api_TaxonomyModel),
    itemCreatedUtc: zod_1.z.string().datetime({ offset: true }),
    id: zod_1.z.number().int(),
    itemType: zod_1.z.enum(['Asset', 'Bundle']),
    assetType: zod_1.z.union([
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
        zod_1.z.literal(10),
        zod_1.z.literal(11),
        zod_1.z.literal(12),
        zod_1.z.literal(13),
        zod_1.z.literal(16),
        zod_1.z.literal(17),
        zod_1.z.literal(18),
        zod_1.z.literal(19),
        zod_1.z.literal(21),
        zod_1.z.literal(22),
        zod_1.z.literal(24),
        zod_1.z.literal(25),
        zod_1.z.literal(26),
        zod_1.z.literal(27),
        zod_1.z.literal(28),
        zod_1.z.literal(29),
        zod_1.z.literal(30),
        zod_1.z.literal(31),
        zod_1.z.literal(32),
        zod_1.z.literal(33),
        zod_1.z.literal(34),
        zod_1.z.literal(35),
        zod_1.z.literal(37),
        zod_1.z.literal(38),
        zod_1.z.literal(39),
        zod_1.z.literal(40),
        zod_1.z.literal(41),
        zod_1.z.literal(42),
        zod_1.z.literal(43),
        zod_1.z.literal(44),
        zod_1.z.literal(45),
        zod_1.z.literal(46),
        zod_1.z.literal(47),
        zod_1.z.literal(48),
        zod_1.z.literal(49),
        zod_1.z.literal(50),
        zod_1.z.literal(51),
        zod_1.z.literal(52),
        zod_1.z.literal(53),
        zod_1.z.literal(54),
        zod_1.z.literal(55),
        zod_1.z.literal(56),
        zod_1.z.literal(59),
        zod_1.z.literal(60),
        zod_1.z.literal(61),
        zod_1.z.literal(62),
        zod_1.z.literal(63),
        zod_1.z.literal(64),
        zod_1.z.literal(65),
        zod_1.z.literal(66),
        zod_1.z.literal(67),
        zod_1.z.literal(68),
        zod_1.z.literal(69),
        zod_1.z.literal(70),
        zod_1.z.literal(71),
        zod_1.z.literal(72),
        zod_1.z.literal(73),
        zod_1.z.literal(74),
        zod_1.z.literal(75),
        zod_1.z.literal(76),
        zod_1.z.literal(77),
        zod_1.z.literal(78),
        zod_1.z.literal(79),
        zod_1.z.literal(80),
        zod_1.z.literal(81),
        zod_1.z.literal(82),
        zod_1.z.literal(83),
        zod_1.z.literal(84),
        zod_1.z.literal(85),
        zod_1.z.literal(86),
        zod_1.z.literal(87),
        zod_1.z.literal(88),
        zod_1.z.literal(89),
        zod_1.z.literal(90),
    ]),
    bundleType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
    isRecolorable: zod_1.z.boolean(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    productId: zod_1.z.number().int(),
    itemStatus: zod_1.z.array(zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(7)])),
    itemRestrictions: zod_1.z.array(zod_1.z.union([
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
    ])),
    creatorHasVerifiedBadge: zod_1.z.boolean(),
    creatorType: zod_1.z.string(),
    creatorTargetId: zod_1.z.number().int(),
    creatorName: zod_1.z.string(),
    price: zod_1.z.number().int(),
    lowestPrice: zod_1.z.number().int(),
    lowestResalePrice: zod_1.z.number().int(),
    priceStatus: zod_1.z.string(),
    unitsAvailableForConsumption: zod_1.z.number().int(),
    favoriteCount: zod_1.z.number().int(),
    offSaleDeadline: zod_1.z.string().datetime({ offset: true }),
    collectibleItemId: zod_1.z.string(),
    totalQuantity: zod_1.z.number().int(),
    saleLocationType: zod_1.z.enum([
        'NotApplicable',
        'ShopOnly',
        'MyExperiencesOnly',
        'ShopAndMyExperiences',
        'ExperiencesById',
        'ShopAndAllExperiences',
        'ExperiencesDevApiOnly',
        'ShopAndExperiencesById',
    ]),
    hasResellers: zod_1.z.boolean(),
    isOffSale: zod_1.z.boolean(),
    quantityLimitPerUser: zod_1.z.number().int(),
    supportsHeadShapes: zod_1.z.boolean(),
    timedOptions: zod_1.z.array(Roblox_Catalog_Api_TimedOption),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Catalog_Api_CatalogSearchDetailedResponseItemV2_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Catalog_Api_CatalogSearchDetailedResponseItemV2),
});
const Roblox_MarketplaceTopicDiscovery_TopicDiscoveryService_V1Beta1_AvatarItem = zod_1.z.object({
    TargetId: zod_1.z.number().int(),
    ItemType: zod_1.z.enum(['Invalid', 'Asset', 'Bundle']),
});
const Roblox_Catalog_Api_TopicRequestModel = zod_1.z.object({
    items: zod_1.z.array(Roblox_MarketplaceTopicDiscovery_TopicDiscoveryService_V1Beta1_AvatarItem),
    selectTopics: zod_1.z.array(zod_1.z.string()),
    inputQuery: zod_1.z.string(),
    maxResult: zod_1.z.number().int(),
    genderType: zod_1.z.enum(['Unknown', 'Male', 'Female']),
});
const Roblox_Catalog_Api_TopicModel = zod_1.z.object({
    displayName: zod_1.z.string(),
    originalTopicName: zod_1.z.string(),
});
const Roblox_MarketplaceTopicDiscovery_TopicDiscoveryService_V1Beta1_Error = zod_1.z.object({
    Message: zod_1.z.string(),
    Code: zod_1.z.number().int(),
});
const Roblox_Catalog_Api_TopicResponse = zod_1.z.object({
    topics: zod_1.z.array(Roblox_Catalog_Api_TopicModel),
    error: Roblox_MarketplaceTopicDiscovery_TopicDiscoveryService_V1Beta1_Error,
});
/**
 * @api GET https://catalog.roblox.com/v1/asset-to-category
 * @summary Lists a mapping for assets to category IDs to convert from inventory ID to catalog ID. Creates a mapping to link 'Get More' button in inventory page to the relevant catalog page.
 */
exports.getAssetToCategory = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/asset-to-category',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.number().int(),
    errors: [],
});
/**
 * @api GET https://catalog.roblox.com/v1/asset-to-subcategory
 * @summary Lists a mapping for assets to subcategory IDs to convert from inventory ID to catalog ID. Creates a mapping to link 'Get More' button in inventory page to the relevant catalog page.
 */
exports.getAssetToSubcategory = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/asset-to-subcategory',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.number().int(),
    errors: [],
});
/**
 * @api GET https://catalog.roblox.com/v1/assets/:assetId/bundles
 * @summary Lists the bundles a particular asset belongs to. Use the Id of the last bundle in the response to get the next page.
 * @param assetId
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getAssetsAssetidBundles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/assets/:assetId/bundles',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Catalog_Api_BundleDetailsModel_,
    errors: [
        {
            status: 400,
            description: `1: Invalid assetId
4: Invalid Cursor.`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/bundles/:bundleId/details
 * @summary Returns details about the given bundleId.
 * @param bundleId
 */
exports.getBundlesBundleidDetails = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/bundles/:bundleId/details',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        bundleId: {
            style: 'simple',
        },
    },
    parameters: {
        bundleId: zod_1.z.number().int(),
    },
    response: Roblox_Catalog_Api_BundleDetailsModel,
    errors: [
        {
            status: 400,
            description: `1: Invalid bundle`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/bundles/:bundleId/recommendations
 * @summary Gets recommendations for a given bundle, bundleId of 0 returns randomized bundles
- Accepts both public and authenticated users.
 * @param bundleId
 * @param numItems The number of recommended items to return.
 */
exports.getBundlesBundleidRecommendations = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/bundles/:bundleId/recommendations',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        bundleId: {
            style: 'simple',
        },
        numItems: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        bundleId: zod_1.z.number().int(),
        numItems: zod_1.z.number().int().optional().default(20),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Catalog_Api_BundleDetailsModel_,
    errors: [
        {
            status: 400,
            description: `1: Invalid bundle
2: Error retrieving bundles
3: Error getting bundle recommendations
4: NumItems exceed maximum`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/bundles/details
 * @summary Returns details about the given bundleIds.
 * @param bundleIds
 */
exports.getBundlesDetails = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/bundles/details',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        bundleIds: {
            style: 'form',
        },
    },
    parameters: {
        bundleIds: zod_1.z.array(zod_1.z.number()),
    },
    response: zod_1.z.array(Roblox_Catalog_Api_BundleDetailsModel),
    errors: [
        {
            status: 400,
            description: `3: Cannot request so many bundles at once.`,
        },
    ],
});
/**
 * @api POST https://catalog.roblox.com/v1/catalog/items/details
 * @summary Returns list of item details.
 * @param body Roblox.Catalog.Api.MultigetItemDetailsRequestModel.
 */
exports.postCatalogItemsDetails = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/catalog/items/details',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Catalog_Api_MultigetItemDetailsRequestModel,
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Catalog_Api_CatalogSearchDetailedResponseItemV2_,
    errors: [
        {
            status: 400,
            description: `2: Invalid count`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
7: User is unauthorized.`,
        },
        {
            status: 429,
            description: `8: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/categories
 * @summary Lists Category Names and their Ids.
 */
exports.getCategories = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/categories',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.array(Roblox_Catalog_Api_CategoryModel),
    errors: [],
});
/**
 * @api GET https://catalog.roblox.com/v1/favorites/assets/:assetId/count
 * @summary Gets the favorite count for the given asset Id.
 * @param assetId
 */
exports.getFavoritesAssetsAssetidCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/favorites/assets/:assetId/count',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
    },
    response: zod_1.z.number().int(),
    errors: [
        {
            status: 400,
            description: `2: Invalid asset Id.`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/favorites/bundles/:bundleId/count
 * @summary Gets the favorite count for the given bundle Id.
 * @param bundleId
 */
exports.getFavoritesBundlesBundleidCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/favorites/bundles/:bundleId/count',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        bundleId: {
            style: 'simple',
        },
    },
    parameters: {
        bundleId: zod_1.z.number().int(),
    },
    response: zod_1.z.number().int(),
    errors: [
        {
            status: 400,
            description: `2: Invalid bundle Id.`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/favorites/users/:userId/assets/:assetId/favorite
 * @summary Gets the favorite model for the asset and user.
 * @param userId
 * @param assetId
 */
exports.getFavoritesUsersUseridAssetsAssetidFavorite = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/favorites/users/:userId/assets/:assetId/favorite',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        assetId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        assetId: zod_1.z.number().int(),
    },
    response: Roblox_Catalog_Api_AssetFavoriteModel,
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
2: Invalid asset Id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://catalog.roblox.com/v1/favorites/users/:userId/assets/:assetId/favorite
 * @summary Create a favorite for an asset by the authenticated user.
 * @param userId
 * @param assetId
 */
exports.postFavoritesUsersUseridAssetsAssetidFavorite = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/favorites/users/:userId/assets/:assetId/favorite',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        assetId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        assetId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
2: Invalid asset Id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: You are not authorized to perform this action.`,
        },
        {
            status: 409,
            description: `3: Asset is already favorited.`,
        },
        {
            status: 429,
            description: `5: This action was floodchecked. Please try again later.`,
        },
    ],
});
/**
 * @api DELETE https://catalog.roblox.com/v1/favorites/users/:userId/assets/:assetId/favorite
 * @summary Delete a favorite for an asset by the authenticated user.
 * @param userId
 * @param assetId
 */
exports.deleteFavoritesUsersUseridAssetsAssetidFavorite = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/favorites/users/:userId/assets/:assetId/favorite',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        assetId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        assetId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
2: Invalid asset Id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: You are not authorized to perform this action.`,
        },
        {
            status: 409,
            description: `4: Asset is already not favorited.`,
        },
        {
            status: 429,
            description: `5: This action was floodchecked. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/favorites/users/:userId/bundles/:bundleId/favorite
 * @summary Gets the favorite model for the bundle and user.
 * @param userId
 * @param bundleId
 */
exports.getFavoritesUsersUseridBundlesBundleidFavorite = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/favorites/users/:userId/bundles/:bundleId/favorite',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        bundleId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        bundleId: zod_1.z.number().int(),
    },
    response: Roblox_Catalog_Api_BundleFavoriteModel,
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
2: Invalid bundle Id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://catalog.roblox.com/v1/favorites/users/:userId/bundles/:bundleId/favorite
 * @summary Create a favorite for the bundle by the authenticated user.
 * @param userId
 * @param bundleId
 */
exports.postFavoritesUsersUseridBundlesBundleidFavorite = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/favorites/users/:userId/bundles/:bundleId/favorite',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        bundleId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        bundleId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
2: Invalid bundle Id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: You are not authorized to perform this action.`,
        },
        {
            status: 409,
            description: `3: Bundle is already favorited.`,
        },
        {
            status: 429,
            description: `5: This action was floodchecked. Please try again later.`,
        },
    ],
});
/**
 * @api DELETE https://catalog.roblox.com/v1/favorites/users/:userId/bundles/:bundleId/favorite
 * @summary Delete favorite for the bundle by the authenticated user.
 * @param userId
 * @param bundleId
 */
exports.deleteFavoritesUsersUseridBundlesBundleidFavorite = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/favorites/users/:userId/bundles/:bundleId/favorite',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        bundleId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        bundleId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
2: Invalid bundle Id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: You are not authorized to perform this action.`,
        },
        {
            status: 409,
            description: `4: Bundle is already not favorited.`,
        },
        {
            status: 429,
            description: `5: This action was floodchecked. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/favorites/users/:userId/favorites/:assetTypeId/assets
 * @summary Lists the marketplace assets favorited by a given user with the given assetTypeId.
 * @param userId
 * @param assetTypeId
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getFavoritesUsersUseridFavoritesAssettypeidAssets = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/favorites/users/:userId/favorites/:assetTypeId/assets',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        assetTypeId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        assetTypeId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(18), zod_1.z.literal(24), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Desc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Catalog_Api_CatalogSearchDetailedResponseItem_,
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
8: Ascending order is not allowed.
11: Invalid asset type id.`,
        },
        {
            status: 500,
            description: `99: Internal server error.`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/favorites/users/:userId/favorites/:subtypeId/bundles
 * @summary Lists the bundles favorited by a given user with the given bundle subtypeId.Switched to EAAS style pagination cursors since July 2024.
 * @param userId
 * @param subtypeId
 * @param itemsPerPage
 * @param cursor
 * @param isPrevious
 */
exports.getFavoritesUsersUseridFavoritesSubtypeidBundles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/favorites/users/:userId/favorites/:subtypeId/bundles',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        subtypeId: {
            style: 'simple',
        },
        itemsPerPage: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        isPrevious: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        subtypeId: zod_1.z.number().int(),
        itemsPerPage: zod_1.z.number().int().optional().default(24),
        cursor: zod_1.z.string().optional(),
        isPrevious: zod_1.z.boolean().optional(),
    },
    response: Roblox_Catalog_Api_FavoriteBundlesResponse,
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
3: Cannot request so many bundles at once.
10: Invalid previous pagination request. Please provide a cursor when isPrevious is true`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `6: You are not authorized to perform this action.`,
        },
        {
            status: 500,
            description: `11: Internal server error. Please check if you have provided correct pagination cursor`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/subcategories
 * @summary Lists Subcategory Names and their Ids.
 */
exports.getSubcategories = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/subcategories',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.number().int(),
    errors: [],
});
/**
 * @api POST https://catalog.roblox.com/v1/topic/get-topics
 * @summary Get topic given TopicRequestModel.
 * @param body
 */
exports.postTopicGetTopics = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/topic/get-topics',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Catalog_Api_TopicRequestModel,
    response: Roblox_Catalog_Api_TopicResponse,
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/users/:userId/bundles
 * @summary Lists the bundles owned by a given user.
 * @param userId
 * @param cursor
 * @param limit
 * @param sortOrder
 */
exports.getUsersUseridBundles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/bundles',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        cursor: zod_1.z.string().optional(),
        limit: zod_1.z.number().int().optional().default(10),
        sortOrder: zod_1.z
            .union([zod_1.z.literal(1), zod_1.z.literal(2)])
            .optional()
            .default(2),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Catalog_Api_OwnedBundleModel_,
    errors: [
        {
            status: 400,
            description: `1: Invalid bundle`,
        },
    ],
});
/**
 * @api GET https://catalog.roblox.com/v1/users/:userId/bundles/:bundleType
 * @param userId
 * @param bundleType
 * @param cursor
 * @param limit
 * @param sortOrder
 */
exports.getUsersUseridBundlesBundletype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/bundles/:bundleType',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        bundleType: {
            style: 'simple',
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        bundleType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
        cursor: zod_1.z.string(),
        limit: zod_1.z.number().int().optional().default(10),
        sortOrder: zod_1.z
            .union([zod_1.z.literal(1), zod_1.z.literal(2)])
            .optional()
            .default(2),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Catalog_Api_OwnedBundleModel_,
    errors: [],
});
