// rozod/lib/endpoints/catalogv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSearchItemsDetails = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Catalog_Api_ElasticsearchDebugInfo = zod_1.z.object({
    elasticsearchQuery: zod_1.z.string(),
    isFromCache: zod_1.z.boolean(),
    indexName: zod_1.z.string(),
    isTerminatedEarly: zod_1.z.boolean(),
    isForceTerminationEnabledByRequest: zod_1.z.boolean(),
    searchResultDataSource: zod_1.z.string(),
    searchResultRelevanceScore: zod_1.z.string(),
    searchResultEngagementScore: zod_1.z.string(),
});
const Roblox_Catalog_Api_BundleItemDetailModelV2 = zod_1.z.object({
    owned: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    type: zod_1.z.string(),
    supportsHeadShapes: zod_1.z.boolean(),
    assetType: zod_1.z.number().int(),
});
const Roblox_Catalog_Api_TaxonomyModel = zod_1.z.object({
    taxonomyId: zod_1.z.string(),
    taxonomyName: zod_1.z.string(),
});
const Roblox_Catalog_Api_TimedOption = zod_1.z.object({
    days: zod_1.z.number().int(),
    price: zod_1.z.number().int(),
    selected: zod_1.z.boolean(),
});
const Roblox_Catalog_Api_CatalogSearchDetailedResponseItemV2 = zod_1.z.object({
    bundledItems: zod_1.z.array(Roblox_Catalog_Api_BundleItemDetailModelV2),
    taxonomy: zod_1.z.array(Roblox_Catalog_Api_TaxonomyModel),
    itemCreatedUtc: zod_1.z.string().datetime({ offset: true }),
    id: zod_1.z.number().int(),
    itemType: zod_1.z.enum(['Asset', 'Bundle']),
    assetType: zod_1.z.union([
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
        zod_1.z.literal(10),
        zod_1.z.literal(11),
        zod_1.z.literal(12),
        zod_1.z.literal(13),
        zod_1.z.literal(16),
        zod_1.z.literal(17),
        zod_1.z.literal(18),
        zod_1.z.literal(19),
        zod_1.z.literal(21),
        zod_1.z.literal(22),
        zod_1.z.literal(24),
        zod_1.z.literal(25),
        zod_1.z.literal(26),
        zod_1.z.literal(27),
        zod_1.z.literal(28),
        zod_1.z.literal(29),
        zod_1.z.literal(30),
        zod_1.z.literal(31),
        zod_1.z.literal(32),
        zod_1.z.literal(33),
        zod_1.z.literal(34),
        zod_1.z.literal(35),
        zod_1.z.literal(37),
        zod_1.z.literal(38),
        zod_1.z.literal(39),
        zod_1.z.literal(40),
        zod_1.z.literal(41),
        zod_1.z.literal(42),
        zod_1.z.literal(43),
        zod_1.z.literal(44),
        zod_1.z.literal(45),
        zod_1.z.literal(46),
        zod_1.z.literal(47),
        zod_1.z.literal(48),
        zod_1.z.literal(49),
        zod_1.z.literal(50),
        zod_1.z.literal(51),
        zod_1.z.literal(52),
        zod_1.z.literal(53),
        zod_1.z.literal(54),
        zod_1.z.literal(55),
        zod_1.z.literal(56),
        zod_1.z.literal(59),
        zod_1.z.literal(60),
        zod_1.z.literal(61),
        zod_1.z.literal(62),
        zod_1.z.literal(63),
        zod_1.z.literal(64),
        zod_1.z.literal(65),
        zod_1.z.literal(66),
        zod_1.z.literal(67),
        zod_1.z.literal(68),
        zod_1.z.literal(69),
        zod_1.z.literal(70),
        zod_1.z.literal(71),
        zod_1.z.literal(72),
        zod_1.z.literal(73),
        zod_1.z.literal(74),
        zod_1.z.literal(75),
        zod_1.z.literal(76),
        zod_1.z.literal(77),
        zod_1.z.literal(78),
        zod_1.z.literal(79),
        zod_1.z.literal(80),
        zod_1.z.literal(81),
        zod_1.z.literal(82),
        zod_1.z.literal(83),
        zod_1.z.literal(84),
        zod_1.z.literal(85),
        zod_1.z.literal(86),
        zod_1.z.literal(87),
        zod_1.z.literal(88),
        zod_1.z.literal(89),
        zod_1.z.literal(90),
    ]),
    bundleType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
    isRecolorable: zod_1.z.boolean(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    productId: zod_1.z.number().int(),
    itemStatus: zod_1.z.array(zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(7)])),
    itemRestrictions: zod_1.z.array(zod_1.z.union([
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
    ])),
    creatorHasVerifiedBadge: zod_1.z.boolean(),
    creatorType: zod_1.z.string(),
    creatorTargetId: zod_1.z.number().int(),
    creatorName: zod_1.z.string(),
    price: zod_1.z.number().int(),
    lowestPrice: zod_1.z.number().int(),
    lowestResalePrice: zod_1.z.number().int(),
    priceStatus: zod_1.z.string(),
    unitsAvailableForConsumption: zod_1.z.number().int(),
    favoriteCount: zod_1.z.number().int(),
    offSaleDeadline: zod_1.z.string().datetime({ offset: true }),
    collectibleItemId: zod_1.z.string(),
    totalQuantity: zod_1.z.number().int(),
    saleLocationType: zod_1.z.enum([
        'NotApplicable',
        'ShopOnly',
        'MyExperiencesOnly',
        'ShopAndMyExperiences',
        'ExperiencesById',
        'ShopAndAllExperiences',
        'ExperiencesDevApiOnly',
        'ShopAndExperiencesById',
    ]),
    hasResellers: zod_1.z.boolean(),
    isOffSale: zod_1.z.boolean(),
    quantityLimitPerUser: zod_1.z.number().int(),
    supportsHeadShapes: zod_1.z.boolean(),
    timedOptions: zod_1.z.array(Roblox_Catalog_Api_TimedOption),
});
const Roblox_Catalog_Api_CatalogSearchPageResponse_Roblox_Catalog_Api_CatalogSearchDetailedResponseItemV2_ = zod_1.z.object({
    keyword: zod_1.z.string(),
    elasticsearchDebugInfo: Roblox_Catalog_Api_ElasticsearchDebugInfo,
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Catalog_Api_CatalogSearchDetailedResponseItemV2),
});
/**
 * @api GET https://catalog.roblox.com/v2/search/items/details
 * @summary Search for catalog items.
 * @param Taxonomy
 * @param AssetTypeIds
 * @param BundleTypeIds
 * @param CategoryFilter
 * @param SortAggregation
 * @param SortType
 * @param CreatorType
 * @param CreatorTargetId
 * @param CreatorName
 * @param MaxPrice
 * @param MinPrice
 * @param Keyword
 * @param IncludeNotForSale
 * @param TriggeredByTopicDiscovery
 * @param SalesTypeFilter
 * @param Topics The input topics format is split by ",". E.g "topics=cat,hat,red".
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 * @description This endpoint is for search by item type ids.
 */
exports.getSearchItemsDetails = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/search/items/details',
    baseUrl: 'https://catalog.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        Taxonomy: {
            style: 'form',
            explode: true,
        },
        AssetTypeIds: {
            style: 'form',
            explode: true,
        },
        BundleTypeIds: {
            style: 'form',
            explode: true,
        },
        CategoryFilter: {
            style: 'form',
            explode: true,
        },
        SortAggregation: {
            style: 'form',
            explode: true,
        },
        SortType: {
            style: 'form',
            explode: true,
        },
        CreatorType: {
            style: 'form',
            explode: true,
        },
        CreatorTargetId: {
            style: 'form',
            explode: true,
        },
        CreatorName: {
            style: 'form',
            explode: true,
        },
        MaxPrice: {
            style: 'form',
            explode: true,
        },
        MinPrice: {
            style: 'form',
            explode: true,
        },
        Keyword: {
            style: 'form',
            explode: true,
        },
        IncludeNotForSale: {
            style: 'form',
            explode: true,
        },
        TriggeredByTopicDiscovery: {
            style: 'form',
            explode: true,
        },
        SalesTypeFilter: {
            style: 'form',
            explode: true,
        },
        Topics: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        Taxonomy: zod_1.z.string(),
        AssetTypeIds: zod_1.z.array(zod_1.z.number()),
        BundleTypeIds: zod_1.z.array(zod_1.z.number()),
        CategoryFilter: zod_1.z.union([
            zod_1.z.literal(0),
            zod_1.z.literal(1),
            zod_1.z.literal(2),
            zod_1.z.literal(3),
            zod_1.z.literal(4),
            zod_1.z.literal(5),
            zod_1.z.literal(6),
        ]),
        SortAggregation: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4), zod_1.z.literal(5)]),
        SortType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4), zod_1.z.literal(5)]),
        CreatorType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]),
        CreatorTargetId: zod_1.z.number().int(),
        CreatorName: zod_1.z.string(),
        MaxPrice: zod_1.z.number().int(),
        MinPrice: zod_1.z.number().int(),
        Keyword: zod_1.z.string(),
        IncludeNotForSale: zod_1.z.boolean(),
        TriggeredByTopicDiscovery: zod_1.z.boolean(),
        SalesTypeFilter: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
        Topics: zod_1.z.string(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(28), zod_1.z.literal(30), zod_1.z.literal(60), zod_1.z.literal(120)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.literal('Desc').optional().default('Desc'),
    },
    response: Roblox_Catalog_Api_CatalogSearchPageResponse_Roblox_Catalog_Api_CatalogSearchDetailedResponseItemV2_,
    errors: [
        {
            status: 400,
            description: `1: Category subcategory selection not supported.
2: Creator id not found.
3: Creator type not found or cannot search by CreatorTargetId with CreatorType.All
4: Genre not found.
5: Sort combination not supported.
6: Invalid price range.
10: StartRequest is invalid.`,
        },
        {
            status: 403,
            description: `7: User is unauthorized.
22: In-experience search is denied for this place or universe.`,
        },
        {
            status: 429,
            description: `8: The flood limit has been exceeded.
8: The flood limit has been exceeded.`,
        },
        {
            status: 503,
            description: `18: Search request timed out`,
        },
    ],
});
