// rozod/lib/endpoints/chatv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postUpdateUserTypingStatus = exports.postStartOneToOneConversation = exports.postStartGroupConversation = exports.postStartCloudEditConversation = exports.postSendMessage = exports.postSendGameLinkMessage = exports.postRenameGroupConversation = exports.postRemoveFromConversation = exports.getMultiGetLatestMessages = exports.getMetadata = exports.postMarkAsSeen = exports.postMarkAsRead = exports.getGetUserConversations = exports.getGetUnreadMessages = exports.getGetUnreadConversations = exports.getGetUnreadConversationCount = exports.getGetRolloutSettings = exports.getGetMessages = exports.getGetConversations = exports.getChatSettings = exports.postAddToConversation = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Chat_Api_Models_ChatSettingsResponse = zod_1.z.object({
    chatEnabled: zod_1.z.boolean(),
    isActiveChatUser: zod_1.z.boolean(),
    isConnectTabEnabled: zod_1.z.boolean(),
});
const Roblox_Chat_Api_Models_ChatParticipant = zod_1.z.object({
    type: zod_1.z.enum(['User', 'System']),
    targetId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Chat_Api_Models_ConversationTitle = zod_1.z.object({
    titleForViewer: zod_1.z.string(),
    isDefaultTitle: zod_1.z.boolean(),
});
const Roblox_Chat_Api_Models_ConversationUniverse = zod_1.z.object({
    universeId: zod_1.z.number().int(),
    rootPlaceId: zod_1.z.number().int(),
});
const Roblox_Chat_Api_Models_Conversation = zod_1.z.object({
    id: zod_1.z.number().int(),
    title: zod_1.z.string(),
    initiator: Roblox_Chat_Api_Models_ChatParticipant,
    hasUnreadMessages: zod_1.z.boolean(),
    participants: zod_1.z.array(Roblox_Chat_Api_Models_ChatParticipant),
    conversationType: zod_1.z.enum(['OneToOneConversation', 'MultiUserConversation', 'CloudEditConversation']),
    conversationTitle: Roblox_Chat_Api_Models_ConversationTitle,
    lastUpdated: zod_1.z.string().datetime({ offset: true }),
    conversationUniverse: Roblox_Chat_Api_Models_ConversationUniverse,
});
const Roblox_Chat_Api_Models_GameLink = zod_1.z.object({
    universeId: zod_1.z.number().int(),
});
const Roblox_Chat_Api_Models_Link = zod_1.z.object({
    type: zod_1.z.literal('Game'),
    game: Roblox_Chat_Api_Models_GameLink,
});
const Roblox_Chat_Api_Models_SetConversationUniverseEventBased = zod_1.z.object({
    actorUserId: zod_1.z.number().int(),
    universeId: zod_1.z.number().int(),
});
const Roblox_Chat_Api_Models_EventBased = zod_1.z.object({
    type: zod_1.z.literal('SetConversationUniverse'),
    setConversationUniverse: Roblox_Chat_Api_Models_SetConversationUniverseEventBased,
});
const Roblox_Chat_Api_Models_ChatMessage = zod_1.z.object({
    id: zod_1.z.string().uuid(),
    senderType: zod_1.z.enum(['User', 'System']),
    sent: zod_1.z.string().datetime({ offset: true }),
    read: zod_1.z.boolean(),
    messageType: zod_1.z.enum(['PlainText', 'Link', 'EventBased']),
    decorators: zod_1.z.array(zod_1.z.string()),
    senderTargetId: zod_1.z.number().int(),
    content: zod_1.z.string(),
    link: Roblox_Chat_Api_Models_Link,
    eventBased: Roblox_Chat_Api_Models_EventBased,
});
const Roblox_Chat_Api_Models_RolloutSettingModel = zod_1.z.object({
    featureName: zod_1.z.enum(['LuaChat', 'ConversationUniverse', 'PlayTogether', 'Party', 'GameLink', 'OldPlayTogether']),
    isRolloutEnabled: zod_1.z.boolean(),
});
const Roblox_Chat_Api_Models_RolloutSettingsResponse = zod_1.z.object({
    rolloutFeatures: zod_1.z.array(Roblox_Chat_Api_Models_RolloutSettingModel),
});
const Roblox_Chat_Api_Models_UnreadConversationCountResponse = zod_1.z.object({
    count: zod_1.z.number().int(),
});
const Roblox_Chat_Api_Models_MultigetConversationMessagesResponse = zod_1.z.object({
    conversationId: zod_1.z.number().int(),
    chatMessages: zod_1.z.array(Roblox_Chat_Api_Models_ChatMessage),
});
const Roblox_Chat_Api_Models_ChatMetadataResponse = zod_1.z.object({
    isChatEnabledByPrivacySetting: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]),
    languageForPrivacySettingUnavailable: zod_1.z.string(),
    maxConversationTitleLength: zod_1.z.number().int(),
    numberOfMembersForPartyChrome: zod_1.z.number().int(),
    partyChromeDisplayTimeStampInterval: zod_1.z.number().int(),
    signalRDisconnectionResponseInMilliseconds: zod_1.z.number().int(),
    typingInChatFromSenderThrottleMs: zod_1.z.number().int(),
    typingInChatForReceiverExpirationMs: zod_1.z.number().int(),
    relativeValueToRecordUiPerformance: zod_1.z.number(),
    isChatDataFromLocalStorageEnabled: zod_1.z.boolean(),
    chatDataFromLocalStorageExpirationSeconds: zod_1.z.number().int(),
    isUsingCacheToLoadFriendsInfoEnabled: zod_1.z.boolean(),
    cachedDataFromLocalStorageExpirationMS: zod_1.z.number().int(),
    senderTypesForUnknownMessageTypeError: zod_1.z.array(zod_1.z.string()),
    isInvalidMessageTypeFallbackEnabled: zod_1.z.boolean(),
    isRespectingMessageTypeEnabled: zod_1.z.boolean(),
    validMessageTypesWhiteList: zod_1.z.array(zod_1.z.string()),
    shouldRespectConversationHasUnreadMessageToMarkAsRead: zod_1.z.boolean(),
    isAliasChatForClientSideEnabled: zod_1.z.boolean(),
    isPlayTogetherForGameCardsEnabled: zod_1.z.boolean(),
    isRoactChatEnabled: zod_1.z.boolean(),
    webChatEventSampleRate: zod_1.z.number().int(),
    isPlatformChatApiEnabled: zod_1.z.boolean(),
});
const Roblox_Chat_Api_Models_AddUsersToConversationRequest = zod_1.z.object({
    participantUserIds: zod_1.z.array(zod_1.z.number()),
    conversationId: zod_1.z.number().int(),
});
const Roblox_Chat_Api_Models_RejectedChatParticipant = zod_1.z.object({
    rejectedReason: zod_1.z.string(),
    type: zod_1.z.enum(['User', 'System']),
    targetId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Chat_Api_Models_AddUserToConversationResponse = zod_1.z.object({
    conversationId: zod_1.z.number().int(),
    rejectedParticipants: zod_1.z.array(Roblox_Chat_Api_Models_RejectedChatParticipant),
    resultType: zod_1.z.literal('Success'),
    statusMessage: zod_1.z.string(),
});
const Roblox_Chat_Api_Models_MarkAsReadRequest = zod_1.z.object({
    conversationId: zod_1.z.number().int(),
    endMessageId: zod_1.z.string(),
});
const Roblox_Chat_Api_Models_MarkAsReadResponse = zod_1.z.object({
    resultType: zod_1.z.literal('Success'),
});
const Roblox_Chat_Api_Models_MarkAsSeenRequest = zod_1.z.object({
    conversationsToMarkSeen: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Chat_Api_Models_MarkAsSeenResponse = zod_1.z.object({
    resultType: zod_1.z.literal('Success'),
});
const Roblox_Chat_Api_Models_RemoveUserFromConversationRequest = zod_1.z.object({
    participantUserId: zod_1.z.number().int(),
    conversationId: zod_1.z.number().int(),
});
const Roblox_Chat_Api_Models_RemoveUserFromConversationResponse = zod_1.z.object({
    conversationId: zod_1.z.number().int(),
    resultType: zod_1.z.literal('Success'),
    statusMessage: zod_1.z.string(),
});
const Roblox_Chat_Api_Models_RenameGroupConversationRequest = zod_1.z.object({
    conversationId: zod_1.z.number().int(),
    newTitle: zod_1.z.string(),
});
const Roblox_Chat_Api_Models_RenameConversationResponse = zod_1.z.object({
    conversationTitle: zod_1.z.string(),
    resultType: zod_1.z.enum(['Success', 'Moderated', 'TextTooLong']),
    title: Roblox_Chat_Api_Models_ConversationTitle,
    statusMessage: zod_1.z.string(),
});
const Roblox_Chat_Api_Models_SendGameLinkChatMessageRequest = zod_1.z.object({
    universeId: zod_1.z.number().int(),
    isExperienceInvite: zod_1.z.boolean(),
    userId: zod_1.z.number().int(),
    placeId: zod_1.z.number().int(),
    conversationId: zod_1.z.number().int(),
    decorators: zod_1.z.array(zod_1.z.string()),
});
const Roblox_Chat_Api_Models_SendLinkChatResponse = zod_1.z.object({
    chatMessageLinkType: zod_1.z.literal('Game'),
    messageId: zod_1.z.string(),
    sent: zod_1.z.string().datetime({ offset: true }),
    messageType: zod_1.z.enum(['PlainText', 'Link', 'EventBased']),
    resultType: zod_1.z.enum(['Success', 'Moderated', 'TextTooLong', 'NoRealtimeConnection']),
    statusMessage: zod_1.z.string(),
});
const Roblox_Chat_Api_Models_SendPlainTextChatMessageRequest = zod_1.z.object({
    message: zod_1.z.string(),
    isExperienceInvite: zod_1.z.boolean(),
    userId: zod_1.z.number().int(),
    conversationId: zod_1.z.number().int(),
    decorators: zod_1.z.array(zod_1.z.string()),
});
const Roblox_Chat_Api_Models_SendPlainTextChatMessageResponse = zod_1.z.object({
    content: zod_1.z.string(),
    filteredForReceivers: zod_1.z.boolean(),
    messageId: zod_1.z.string(),
    sent: zod_1.z.string().datetime({ offset: true }),
    messageType: zod_1.z.enum(['PlainText', 'Link', 'EventBased']),
    resultType: zod_1.z.enum(['Success', 'Moderated', 'TextTooLong', 'NoRealtimeConnection']),
    statusMessage: zod_1.z.string(),
});
const Roblox_Chat_Api_Models_CreateCloudEditConversationRequest = zod_1.z.object({
    placeId: zod_1.z.number().int(),
});
const Roblox_Chat_Api_Models_StartNewConversationResponse = zod_1.z.object({
    conversation: Roblox_Chat_Api_Models_Conversation,
    rejectedParticipants: zod_1.z.array(Roblox_Chat_Api_Models_RejectedChatParticipant),
    resultType: zod_1.z.literal('Success'),
    statusMessage: zod_1.z.string(),
});
const Roblox_Chat_Api_Models_CreateGroupConversationRequest = zod_1.z.object({
    participantUserIds: zod_1.z.array(zod_1.z.number()),
    title: zod_1.z.string(),
});
const Roblox_Chat_Api_Models_CreateOneToOneConversationRequest = zod_1.z.object({
    participantUserId: zod_1.z.number().int(),
});
const Roblox_Chat_Api_Models_UpdateUserTypingStatusRequest = zod_1.z.object({
    conversationId: zod_1.z.number().int(),
    isTyping: zod_1.z.boolean(),
});
const Roblox_Chat_Api_Models_UserVisibleStatusResponse = zod_1.z.object({
    statusMessage: zod_1.z.string(),
});
/**
 * @api POST https://chat.roblox.com/v2/add-to-conversation
 * @summary Adds users to an existing conversation
 * @param body
 */
exports.postAddToConversation = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/add-to-conversation',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Chat_Api_Models_AddUsersToConversationRequest,
    response: Roblox_Chat_Api_Models_AddUserToConversationResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/chat-settings
 * @summary For every authenticated user, the clients hit this endpoint to get the chat related settings. Example : Chat is enabled or not
 */
exports.getChatSettings = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/chat-settings',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    response: Roblox_Chat_Api_Models_ChatSettingsResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/get-conversations
 * @summary Gets conversations for the conversationIds specified in the parameters
 * @param conversationIds Ids of the conversations to be returned
 */
exports.getGetConversations = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/get-conversations',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        conversationIds: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        conversationIds: zod_1.z.array(zod_1.z.number()),
    },
    response: zod_1.z.array(Roblox_Chat_Api_Models_Conversation),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/get-messages
 * @summary Gets messages corresponding to the given conversationId
 * @param conversationId
 * @param pageSize
 * @param exclusiveStartMessageId
 */
exports.getGetMessages = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/get-messages',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        conversationId: {
            style: 'form',
            explode: true,
        },
        pageSize: {
            style: 'form',
            explode: true,
        },
        exclusiveStartMessageId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        conversationId: zod_1.z.number().int(),
        pageSize: zod_1.z.number().int(),
        exclusiveStartMessageId: zod_1.z.string().optional(),
    },
    response: zod_1.z.array(Roblox_Chat_Api_Models_ChatMessage),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/get-rollout-settings
 * @summary Gets the rollout settings for requested feature
 * @param featureNames string array to request feature rollout settings
 */
exports.getGetRolloutSettings = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/get-rollout-settings',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        featureNames: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        featureNames: zod_1.z.array(zod_1.z.string()),
    },
    response: Roblox_Chat_Api_Models_RolloutSettingsResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/get-unread-conversation-count
 * @summary Gets the count of unread conversations
 */
exports.getGetUnreadConversationCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/get-unread-conversation-count',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/get-unread-conversations
 * @summary Gets a list of unread conversations
 * @param pageNumber
 * @param pageSize
 */
exports.getGetUnreadConversations = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/get-unread-conversations',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        pageNumber: {
            style: 'form',
            explode: true,
        },
        pageSize: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        pageNumber: zod_1.z.number().int(),
        pageSize: zod_1.z.number().int(),
    },
    response: zod_1.z.array(Roblox_Chat_Api_Models_Conversation),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/get-unread-messages
 * @summary Gets the unread messages for the conversationIds specified in the parameters
 * @param conversationIds
 * @param pageSize
 */
exports.getGetUnreadMessages = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/get-unread-messages',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        conversationIds: {
            style: 'form',
            explode: true,
        },
        pageSize: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        conversationIds: zod_1.z.array(zod_1.z.number()),
        pageSize: zod_1.z.number().int(),
    },
    response: zod_1.z.array(Roblox_Chat_Api_Models_MultigetConversationMessagesResponse),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/get-user-conversations
 * @summary Gets all conversations associated with the logged-in User
 * @param pageNumber
 * @param pageSize
 */
exports.getGetUserConversations = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/get-user-conversations',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        pageNumber: {
            style: 'form',
            explode: true,
        },
        pageSize: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        pageNumber: zod_1.z.number().int(),
        pageSize: zod_1.z.number().int(),
    },
    response: zod_1.z.array(Roblox_Chat_Api_Models_Conversation),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/mark-as-read
 * @summary Mark the specified conversation messages as read
 * @param body
 */
exports.postMarkAsRead = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/mark-as-read',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Chat_Api_Models_MarkAsReadRequest,
    response: Roblox_Chat_Api_Models_MarkAsReadResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/mark-as-seen
 * @summary Mark the specified conversation messages as seen
 * @param body
 */
exports.postMarkAsSeen = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/mark-as-seen',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Chat_Api_Models_MarkAsSeenRequest,
    response: Roblox_Chat_Api_Models_MarkAsSeenResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/metadata
 * @summary Get Chat metadata.
 */
exports.getMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/metadata',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    response: Roblox_Chat_Api_Models_ChatMetadataResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://chat.roblox.com/v2/multi-get-latest-messages
 * @summary Gets latest messages corresponding to the given list of conversation Ids
 * @param conversationIds
 * @param pageSize
 */
exports.getMultiGetLatestMessages = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/multi-get-latest-messages',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        conversationIds: {
            style: 'form',
            explode: true,
        },
        pageSize: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        conversationIds: zod_1.z.array(zod_1.z.number()),
        pageSize: zod_1.z.number().int(),
    },
    response: zod_1.z.array(Roblox_Chat_Api_Models_MultigetConversationMessagesResponse),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/remove-from-conversation
 * @summary Removes a user from an existing conversation
 * @param body
 */
exports.postRemoveFromConversation = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/remove-from-conversation',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Chat_Api_Models_RemoveUserFromConversationRequest,
    response: Roblox_Chat_Api_Models_RemoveUserFromConversationResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/rename-group-conversation
 * @summary Renames the title of an existing conversation
 * @param body
 */
exports.postRenameGroupConversation = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/rename-group-conversation',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Chat_Api_Models_RenameGroupConversationRequest,
    response: Roblox_Chat_Api_Models_RenameConversationResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/send-game-link-message
 * @param body
 */
exports.postSendGameLinkMessage = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/send-game-link-message',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Chat_Api_Models_SendGameLinkChatMessageRequest,
    response: Roblox_Chat_Api_Models_SendLinkChatResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/send-message
 * @summary Sends a chat message
 * @param body
 */
exports.postSendMessage = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/send-message',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Chat_Api_Models_SendPlainTextChatMessageRequest,
    response: Roblox_Chat_Api_Models_SendPlainTextChatMessageResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/start-cloud-edit-conversation
 * @summary Creates a new cloud edit conversation
 * @param body
 */
exports.postStartCloudEditConversation = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/start-cloud-edit-conversation',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ placeId: zod_1.z.number().int() }),
    response: Roblox_Chat_Api_Models_StartNewConversationResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/start-group-conversation
 * @summary Creates a new group conversation
 * @param body
 */
exports.postStartGroupConversation = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/start-group-conversation',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Chat_Api_Models_CreateGroupConversationRequest,
    response: Roblox_Chat_Api_Models_StartNewConversationResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/start-one-to-one-conversation
 * @summary Creates a new one-on-one conversation
 * @param body
 */
exports.postStartOneToOneConversation = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/start-one-to-one-conversation',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ participantUserId: zod_1.z.number().int() }),
    response: Roblox_Chat_Api_Models_StartNewConversationResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://chat.roblox.com/v2/update-user-typing-status
 * @summary Updates the status for whether a user is typing currently
 * @param body
 */
exports.postUpdateUserTypingStatus = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/update-user-typing-status',
    baseUrl: 'https://chat.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Chat_Api_Models_UpdateUserTypingStatusRequest,
    response: zod_1.z.object({ statusMessage: zod_1.z.string() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
