// rozod/lib/endpoints/clientsettingsv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUserChannel = exports.getOtaVersionBinarytype = exports.getCompressionDictionariesDictionarysha256 = exports.getCompressionDictionaries = exports.getClientVersionBinarytypeChannelChannelname = exports.getClientVersionBinarytype = exports.getAndroidBinariesVersionChannelsChannelname = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_ClientSettings_Api_Models_Response_AndroidBinaryLibraryNames = zod_1.z.object({ engine: zod_1.z.string() });
const Roblox_ClientSettings_Api_Models_Response_AndroidBinaryResponse = zod_1.z.object({
    moduleName: zod_1.z.string(),
    libraryNames: Roblox_ClientSettings_Api_Models_Response_AndroidBinaryLibraryNames,
    supportsAndroidBinaries: zod_1.z.boolean(),
});
const Roblox_ClientSettings_Api_Models_Response_ClientVersionResponse = zod_1.z.object({
    version: zod_1.z.string(),
    clientVersionUpload: zod_1.z.string(),
    bootstrapperVersion: zod_1.z.string(),
    nextClientVersionUpload: zod_1.z.string(),
    nextClientVersion: zod_1.z.string(),
});
const Roblox_ClientSettings_Api_Models_Response_OtaVersionResponse = zod_1.z.object({
    name: zod_1.z.string(),
    version: zod_1.z.string(),
    downloadUrl: zod_1.z.string(),
    isStandalone: zod_1.z.boolean(),
    assetId: zod_1.z.string(),
    assetVersion: zod_1.z.string(),
    maxAppVersion: zod_1.z.string(),
    tryoutName: zod_1.z.string(),
    localAssetURI: zod_1.z.string(),
    isForcedUpdate: zod_1.z.boolean(),
    appStorageResetId: zod_1.z.string(),
    isDevelopmentConfig: zod_1.z.boolean(),
    assetsManifest: zod_1.z.string(),
    versionV2: zod_1.z.number().int(),
});
const Roblox_ClientSettings_Api_Models_Response_UserChannelResponse = zod_1.z.object({
    channelName: zod_1.z.string(),
    channelAssignmentType: zod_1.z.enum([
        'None',
        'PerMille',
        'BoundToPrivateChannel',
        'BoundToPublicChannel',
        'OptedInToBetaProgramWithPrivateChannel',
        'OptedInToBetaProgramWithPublicChannel',
    ]),
    token: zod_1.z.string(),
});
/**
 * @api GET https://clientsettings.roblox.com/v2/android-binaries/:version/channels/:channelName
 * @summary Retrieve the Android binary information for a given version and channel name.
 * @param version The version string of the application. i.e. 2.660.392
 * @param channelName The name of the channel. E.g. ZFlag, ZIntegration
 */
exports.getAndroidBinariesVersionChannelsChannelname = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/android-binaries/:version/channels/:channelName',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        version: {
            style: 'simple',
        },
        channelName: {
            style: 'simple',
        },
    },
    parameters: {
        version: zod_1.z.string(),
        channelName: zod_1.z.string(),
    },
    response: Roblox_ClientSettings_Api_Models_Response_AndroidBinaryResponse,
    errors: [],
});
/**
 * @api GET https://clientsettings.roblox.com/v2/client-version/:binaryType
 * @summary Get client version information for specific binary type
 * @param binaryType Platform(WindowsPlayer, WindowsStudio, MacPlayer or MacStudio) for which we want the latest version
 */
exports.getClientVersionBinarytype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/client-version/:binaryType',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        binaryType: {
            style: 'simple',
        },
    },
    parameters: {
        binaryType: zod_1.z.string(),
    },
    response: Roblox_ClientSettings_Api_Models_Response_ClientVersionResponse,
    errors: [],
});
/**
 * @api GET https://clientsettings.roblox.com/v2/client-version/:binaryType/channel/:channelName
 * @summary Get client version information for specific binary type
 * @param binaryType Platform(WindowsPlayer, WindowsStudio, MacPlayer or MacStudio) for which we want the latest version
 * @param channelName Channel Name
 */
exports.getClientVersionBinarytypeChannelChannelname = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/client-version/:binaryType/channel/:channelName',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        binaryType: {
            style: 'simple',
        },
        channelName: {
            style: 'simple',
        },
    },
    parameters: {
        binaryType: zod_1.z.string(),
        channelName: zod_1.z.string(),
    },
    response: Roblox_ClientSettings_Api_Models_Response_ClientVersionResponse,
    errors: [
        {
            status: 401,
            description: `5: Not authorized to perform this action.`,
        },
    ],
});
/**
 * @api GET https://clientsettings.roblox.com/v2/compression-dictionaries
 * @summary Returns a listing of all known compression dictionaries, including their SHA256 and creation date.
This will be sorted by creation date, with the most recent dictionaries first.
 */
exports.getCompressionDictionaries = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/compression-dictionaries',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://clientsettings.roblox.com/v2/compression-dictionaries/:dictionarySha256
 * @summary Returns the specified compression dictionary as a file download.
 * @param dictionarySha256 The SHA256 of the dictionary we wish to download.
 */
exports.getCompressionDictionariesDictionarysha256 = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/compression-dictionaries/:dictionarySha256',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        dictionarySha256: {
            style: 'simple',
        },
    },
    parameters: {
        dictionarySha256: zod_1.z.string(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://clientsettings.roblox.com/v2/ota-version/:binaryType
 * @summary Get OTA information for a specific binary type with a given version on some channel.
Returns empty list if no updates are found or channel/application with the given version does not exist in CVS.
 * @param binaryType Binary type of the application to get info for
 * @param channel Channel name. If not provided, production is assumed.
 * @param version Application version
 * @param tag Tag to filter results by. Only applicable to non-studio application types.
 * @param name Name to filter results by. Only applicable to non-studio application types.
 */
exports.getOtaVersionBinarytype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/ota-version/:binaryType',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        binaryType: {
            style: 'simple',
        },
        channel: {
            style: 'form',
            explode: true,
        },
        version: {
            style: 'form',
            explode: true,
        },
        tag: {
            style: 'form',
            explode: true,
        },
        name: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        binaryType: zod_1.z.string(),
        channel: zod_1.z.string().optional(),
        version: zod_1.z.string().optional(),
        tag: zod_1.z.string().optional(),
        name: zod_1.z.string().optional(),
    },
    response: zod_1.z.array(Roblox_ClientSettings_Api_Models_Response_OtaVersionResponse),
    errors: [
        {
            status: 400,
            description: `2: Invalid binaryType.
4: Invalid app version.
6: Missing or invalid channel.
7: Unsupported binaryType.`,
        },
        {
            status: 401,
            description: `5: Not authorized to perform this action.`,
        },
    ],
});
/**
 * @api GET https://clientsettings.roblox.com/v2/user-channel
 * @summary Get channel name for currently logged in user
 * @param binaryType
 */
exports.getUserChannel = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/user-channel',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        binaryType: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        binaryType: zod_1.z.string().optional(),
    },
    response: Roblox_ClientSettings_Api_Models_Response_UserChannelResponse,
    errors: [],
});
