// rozod/lib/endpoints/clientsettingsv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMobileClientVersion = exports.getInstallerCdns = exports.getClientVersionBinarytype = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_ClientSettings_Api_Models_Response_ClientVersionResponse = zod_1.z.object({
    version: zod_1.z.string(),
    clientVersionUpload: zod_1.z.string(),
    bootstrapperVersion: zod_1.z.string(),
    nextClientVersionUpload: zod_1.z.string(),
    nextClientVersion: zod_1.z.string(),
});
const Roblox_ClientSettings_Api_Models_Response_MobileClientVersionResponseData = zod_1.z.object({
    UpgradeAction: zod_1.z.string(),
});
const Roblox_ClientSettings_Api_Models_Response_MobileClientVersionResponse = zod_1.z.object({
    activeVersion: zod_1.z.string(),
    upgradeSource: zod_1.z.string(),
    MD5Sum: zod_1.z.string(),
    data: Roblox_ClientSettings_Api_Models_Response_MobileClientVersionResponseData,
});
/**
 * @api GET https://clientsettings.roblox.com/v1/client-version/:binaryType
 * @summary Get client version information for specific binary type
 * @param binaryType Platform(WindowsPlayer, WindowsStudio, MacPlayer or MacStudio) for which we want the latest version
 */
exports.getClientVersionBinarytype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/client-version/:binaryType',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        binaryType: {
            style: 'simple',
        },
    },
    parameters: {
        binaryType: zod_1.z.string(),
    },
    response: Roblox_ClientSettings_Api_Models_Response_ClientVersionResponse,
    errors: [],
});
/**
 * @api GET https://clientsettings.roblox.com/v1/installer-cdns
 * @summary Get information about which CDNs to use for installation.
 */
exports.getInstallerCdns = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/installer-cdns',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://clientsettings.roblox.com/v1/mobile-client-version
 * @summary Get mobile client version information based on app version parameter
 * @param appVersion AppiOSV2.13, AppVersioniOS2.0.1, etc
 */
exports.getMobileClientVersion = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/mobile-client-version',
    baseUrl: 'https://clientsettings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        appVersion: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        appVersion: zod_1.z.string(),
    },
    response: Roblox_ClientSettings_Api_Models_Response_MobileClientVersionResponse,
    errors: [
        {
            status: 400,
            description: `2: Invalid binaryType.`,
        },
    ],
});
