// rozod/lib/endpoints/contactsv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUserTagValidate = exports.postUserTag = exports.postUserGetTags = exports.getContactsMetadata = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Contacts_Api_Models_Response_ContactsMetadataResponseModel = zod_1.z.object({
    multiGetContactsMaxSize: zod_1.z.number().int(),
    multiGetContactsCacheTTLinMS: zod_1.z.number().int(),
});
const Roblox_Contacts_Api_Response_ValidateUserTagResponseModel = zod_1.z.object({
    status: zod_1.z.enum(['Success', 'Moderated', 'TooLong']),
});
const Roblox_Contacts_Api_Request_GetUserTagsRequestModel = zod_1.z.object({
    targetUserIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Contacts_Api_Response_GetUserTagsResponseModel = zod_1.z.object({
    targetUserId: zod_1.z.number().int(),
    targetUserTag: zod_1.z.string(),
});
const Roblox_Contacts_Api_Request_SetUserTagRequestModel = zod_1.z.object({
    targetUserId: zod_1.z.number().int(),
    userTag: zod_1.z.string(),
});
const Roblox_Contacts_Api_Response_SetUserTagResponseModel = zod_1.z.object({
    status: zod_1.z.enum(['Success', 'Moderated']),
});
/**
 * @api GET https://contacts.roblox.com/v1/contacts/metadata
 * @summary Gets contextual information for contacts and usertags
 */
exports.getContactsMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/contacts/metadata',
    baseUrl: 'https://contacts.roblox.com',
    requestFormat: 'json',
    response: Roblox_Contacts_Api_Models_Response_ContactsMetadataResponseModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://contacts.roblox.com/v1/user/get-tags
 * @summary Gets the tags for multiple users
 * @param body
 */
exports.postUserGetTags = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/user/get-tags',
    baseUrl: 'https://contacts.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Contacts_Api_Request_GetUserTagsRequestModel,
    response: zod_1.z.array(Roblox_Contacts_Api_Response_GetUserTagsResponseModel),
    errors: [
        {
            status: 400,
            description: `4: Invalid parameters.
8: Too many user Tags are requested.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 429,
            description: `10: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api POST https://contacts.roblox.com/v1/user/tag
 * @summary Sets the tag for a user
 * @param body The tag receiving userId and the tag itself
 */
exports.postUserTag = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/user/tag',
    baseUrl: 'https://contacts.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Contacts_Api_Request_SetUserTagRequestModel,
    response: Roblox_Contacts_Api_Response_SetUserTagResponseModel,
    errors: [
        {
            status: 400,
            description: `2: The target user is invalid or does not exist.
4: Invalid parameters.
6: The userTag is too long.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
5: The user cannot tag themselves.`,
        },
        {
            status: 429,
            description: `10: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api GET https://contacts.roblox.com/v1/user/tag/validate
 * @summary Validates the tag for a user
 * @param alias The tag to validate
 */
exports.getUserTagValidate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/user/tag/validate',
    baseUrl: 'https://contacts.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        alias: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        alias: zod_1.z.string().optional(),
    },
    response: Roblox_Contacts_Api_Response_ValidateUserTagResponseModel,
    errors: [
        {
            status: 400,
            description: `4: Invalid parameters.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 429,
            description: `10: The flood limit has been exceeded.`,
        },
    ],
});
