// rozod/lib/endpoints/developv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.patchUniversesUniverseidConfiguration = exports.deleteTeamtestPlaceid = exports.patchPlacesPlaceid = exports.getPlacesPlaceid = exports.getAssetsIdVersions = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Api_Develop_AssetVersion = zod_1.z.object({
    Id: zod_1.z.number().int(),
    assetId: zod_1.z.number().int(),
    assetVersionNumber: zod_1.z.number().int(),
    creatorType: zod_1.z.string(),
    creatorTargetId: zod_1.z.number().int(),
    creatingUniverseId: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
    isEqualToCurrentPublishedVersion: zod_1.z.boolean(),
    isPublished: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Api_Develop_AssetVersion_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Api_Develop_AssetVersion),
});
const Roblox_Api_Develop_Models_PlaceModelV2 = zod_1.z.object({
    maxPlayerCount: zod_1.z.number().int(),
    socialSlotType: zod_1.z.string(),
    customSocialSlotsCount: zod_1.z.number().int(),
    allowCopying: zod_1.z.boolean(),
    currentSavedVersion: zod_1.z.number().int(),
    isAllGenresAllowed: zod_1.z.boolean(),
    allowedGearTypes: zod_1.z.array(zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
        zod_1.z.literal(10),
        zod_1.z.literal(11),
        zod_1.z.literal(12),
        zod_1.z.literal(13),
        zod_1.z.literal(14),
        zod_1.z.literal(21),
        zod_1.z.literal(22),
    ])),
    maxPlayersAllowed: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
    id: zod_1.z.number().int(),
    universeId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    isRootPlace: zod_1.z.boolean(),
});
const Roblox_Api_Develop_Models_PlaceConfigurationModelV2 = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    maxPlayerCount: zod_1.z.number().int(),
    socialSlotType: zod_1.z.string(),
    customSocialSlotsCount: zod_1.z.number().int(),
    allowCopying: zod_1.z.boolean(),
    allowedGearTypes: zod_1.z.array(zod_1.z.string()),
    isAllGenresAllowed: zod_1.z.boolean(),
});
const Roblox_Platform_UniverseSettings_UniverseAvatarAssetOverrideResponseModel = zod_1.z.object({
    assetID: zod_1.z.number().int(),
    assetTypeID: zod_1.z.number().int(),
    isPlayerChoice: zod_1.z.boolean(),
});
const Roblox_Web_Responses_Avatar_ScaleModel = zod_1.z.object({
    height: zod_1.z.number(),
    width: zod_1.z.number(),
    head: zod_1.z.number(),
    depth: zod_1.z.number(),
    proportion: zod_1.z.number(),
    bodyType: zod_1.z.number(),
});
const Roblox_UniversePluginPermissionAuthority_Models_UniversePluginPermissions = zod_1.z.object({
    IsThirdPartyTeleportAllowed: zod_1.z.boolean(),
    IsThirdPartyAssetAllowed: zod_1.z.boolean(),
    IsThirdPartyPurchaseAllowed: zod_1.z.boolean(),
    IsClientTeleportAllowed: zod_1.z.boolean(),
});
const Roblox_Api_Develop_Models_UniverseSettingsRequestV2 = zod_1.z.object({
    allowPrivateServers: zod_1.z.boolean(),
    privateServerPrice: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    universeAvatarType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
    universeAnimationType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2)]),
    universeCollisionType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2)]),
    universeJointPositioningType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2)]),
    engineAvatarSettings: zod_1.z.string(),
    isArchived: zod_1.z.boolean(),
    isFriendsOnly: zod_1.z.boolean(),
    genre: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
        zod_1.z.literal(10),
        zod_1.z.literal(11),
        zod_1.z.literal(12),
        zod_1.z.literal(13),
        zod_1.z.literal(14),
    ]),
    playableDevices: zod_1.z.array(zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4), zod_1.z.literal(5)])),
    isForSale: zod_1.z.boolean(),
    price: zod_1.z.number().int(),
    universeAvatarAssetOverrides: zod_1.z.array(Roblox_Platform_UniverseSettings_UniverseAvatarAssetOverrideResponseModel),
    universeAvatarMinScales: Roblox_Web_Responses_Avatar_ScaleModel,
    universeAvatarMaxScales: Roblox_Web_Responses_Avatar_ScaleModel,
    studioAccessToApisAllowed: zod_1.z.boolean(),
    permissions: Roblox_UniversePluginPermissionAuthority_Models_UniversePluginPermissions,
    optInRegions: zod_1.z.array(zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1)])),
    optOutRegions: zod_1.z.array(zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1)])),
    isMeshTextureApiAccessAllowed: zod_1.z.boolean(),
    isRewardedOnDemandAdsAllowed: zod_1.z.boolean(),
    fiatBasePriceId: zod_1.z.string(),
    fiatProductChangeType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
});
const Roblox_Api_Develop_Models_UniverseModerationPolicyStatus = zod_1.z.object({
    region: zod_1.z.enum(['Unknown', 'China']),
    status: zod_1.z.string(),
});
const Roblox_Api_Develop_Models_UniverseSettingsResponseV2 = zod_1.z.object({
    allowPrivateServers: zod_1.z.boolean(),
    privateServerPrice: zod_1.z.number().int(),
    optInRegions: zod_1.z.array(Roblox_Api_Develop_Models_UniverseModerationPolicyStatus),
    isMeshTextureApiAccessAllowed: zod_1.z.boolean(),
    isRewardedOnDemandAdsAllowed: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    universeAvatarType: zod_1.z.enum(['MorphToR6', 'PlayerChoice', 'MorphToR15']),
    universeAnimationType: zod_1.z.enum(['Standard', 'PlayerChoice']),
    universeCollisionType: zod_1.z.enum(['InnerBox', 'OuterBox']),
    universeJointPositioningType: zod_1.z.enum(['Standard', 'ArtistIntent']),
    engineAvatarSettings: zod_1.z.string(),
    isArchived: zod_1.z.boolean(),
    isFriendsOnly: zod_1.z.boolean(),
    genre: zod_1.z.enum([
        'All',
        'Tutorial',
        'Scary',
        'TownAndCity',
        'War',
        'Funny',
        'Fantasy',
        'Adventure',
        'SciFi',
        'Pirate',
        'FPS',
        'RPG',
        'Sports',
        'Ninja',
        'WildWest',
    ]),
    playableDevices: zod_1.z.array(zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4), zod_1.z.literal(5)])),
    isForSale: zod_1.z.boolean(),
    price: zod_1.z.number().int(),
    universeAvatarAssetOverrides: zod_1.z.array(Roblox_Platform_UniverseSettings_UniverseAvatarAssetOverrideResponseModel),
    universeAvatarMinScales: Roblox_Web_Responses_Avatar_ScaleModel,
    universeAvatarMaxScales: Roblox_Web_Responses_Avatar_ScaleModel,
    studioAccessToApisAllowed: zod_1.z.boolean(),
    permissions: Roblox_UniversePluginPermissionAuthority_Models_UniversePluginPermissions,
    isForSaleInFiat: zod_1.z.boolean(),
    fiatBasePriceId: zod_1.z.string(),
    fiatModerationStatus: zod_1.z.enum(['Invalid', 'NotModerated', 'Pending', 'Approved', 'Rejected']),
    eligibleForTextChatMigration: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
/**
 * @api GET https://develop.roblox.com/v2/assets/:id/versions
 * @summary Retrieves asset information for the specified asset ID. The authenticated user must be able to manage the asset
or granted by package permission.
 * @param id The ID of the asset.Roblox.Platform.Assets.IAsset
 * @param Roblox-Place-Id The ID of the place.Roblox.Platform.Assets.IPlace
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder Sort by version number, default is desc.
 */
exports.getAssetsIdVersions = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/assets/:id/versions',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        id: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        id: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Desc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Api_Develop_AssetVersion_,
    errors: [],
});
/**
 * @api GET https://develop.roblox.com/v2/places/:placeId
 * @summary Gets the place configuration for the place with the id placeId
 * @param placeId The place id for the place to be updated.
 */
exports.getPlacesPlaceid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/places/:placeId',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        placeId: {
            style: 'simple',
        },
    },
    parameters: {
        placeId: zod_1.z.number().int(),
    },
    response: Roblox_Api_Develop_Models_PlaceModelV2,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Authenticated user is not authorized to manage this place.`,
        },
        {
            status: 404,
            description: `placeId Place not found.`,
        },
    ],
});
/**
 * @api PATCH https://develop.roblox.com/v2/places/:placeId
 * @summary Updates the place configuration for the place with the id placeId
 * @param body
 * @param placeId The place id for the place to be updated.
 */
exports.patchPlacesPlaceid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v2/places/:placeId',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        placeId: {
            style: 'simple',
        },
    },
    parameters: {
        placeId: zod_1.z.number().int(),
    },
    body: Roblox_Api_Develop_Models_PlaceConfigurationModelV2,
    response: Roblox_Api_Develop_Models_PlaceModelV2,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Authenticated user is not authorized to manage this place.
0: Token Validation Failed`,
        },
        {
            status: 404,
            description: `placeId Place not found.`,
        },
    ],
});
/**
 * @api DELETE https://develop.roblox.com/v2/teamtest/:placeId
 * @summary Close a game instance that is being used for team testing
 * @param placeId The Id of the place we are setting the metadata for.
 * @param gameId the Guid of the game instance System.Guid
 */
exports.deleteTeamtestPlaceid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v2/teamtest/:placeId',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        placeId: {
            style: 'simple',
        },
        gameId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        placeId: zod_1.z.number().int(),
        gameId: zod_1.z.string().uuid(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api PATCH https://develop.roblox.com/v2/universes/:universeId/configuration
 * @summary Update universe settings for an owned universe.
V2 Contains data for avatar scale and asset override.
 * @param body The Roblox.Api.Develop.Models.UniverseSettingsRequest model.
 * @param universeId The universeId.
 */
exports.patchUniversesUniverseidConfiguration = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v2/universes/:universeId/configuration',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    body: Roblox_Api_Develop_Models_UniverseSettingsRequestV2,
    response: Roblox_Api_Develop_Models_UniverseSettingsResponseV2,
    errors: [
        {
            status: 400,
            description: `1: The universe does not exist.
3: Invalid UniverseAvatarType.
4: Invalid UniverseScaleType.
5: Invalid UniverseAnimationType.
6: Invalid UniverseCollisionType.
7: New universe name or description has been rejected.
8: New universe name is too long.
10: Invalid UniverseBodyType.
11: Invalid UniverseJointPositioningType.
12: The universe has no root place.
15: Price is required when isForSale is true.
16: This game cannot be offered for sale because it is not public.
17: This game cannot be offered for sale because it has private servers enabled.
18: The game price is outside of the allowed range.
19: Invalid genre.
20: The request body is missing.
21: Invalid device type.
22: Invalid asset type.
23: Invalid value, the min must be less than or equal to the max
24: Invalid scale value
28: OptIn/Out Regions Not Supported.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You are not authorized to configure this universe.
14: You are not authorized to sell games.
29: Luobu app terms of service user agreement is missing.
30: Unknown error while updating Opt in out region.`,
        },
        {
            status: 409,
            description: `9: Failed to shutdown all intances of game after changing AvatarType. The change has been reverted.`,
        },
    ],
});
