// rozod/lib/endpoints/developv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUserUniverses = exports.getUserGroupsCanmanagegamesoritems = exports.getUserGroupsCanmanage = exports.getUniversesUserPublicPublishEligibility = exports.getUniversesMultigetTeamcreate = exports.getUniversesMultigetPermissions = exports.getUniversesMultiget = exports.deleteUniversesUniverseidTeamcreateMemberships = exports.patchUniversesUniverseidTeamcreate = exports.getUniversesUniverseidTeamcreate = exports.getUniversesUniverseidPlaces = exports.getUniversesUniverseidPermissions = exports.postUniversesUniverseidDeactivate = exports.getUniversesUniverseidConfigurationVipServers = exports.patchUniversesUniverseidConfiguration = exports.getUniversesUniverseidConfiguration = exports.getUniversesUniverseidActivationEligibility = exports.postUniversesUniverseidActivate = exports.getUniversesUniverseid = exports.patchPluginsPluginid = exports.getPlugins = exports.getPlacesPlaceidTeamcreateActive_sessionMembers = exports.patchPlacesPlaceid = exports.postPlacesPlaceid = exports.getGroupsGroupidUniverses = exports.getGametemplates = exports.getAssetsVoting = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Api_Develop_Models_Response_AssetVotingModel = zod_1.z.object({
    assetId: zod_1.z.number().int(),
    hasUserVoted: zod_1.z.boolean(),
    canUserVote: zod_1.z.boolean(),
    shouldShowVotes: zod_1.z.boolean(),
    upVotes: zod_1.z.number().int(),
    downVotes: zod_1.z.number().int(),
    reasonForNotAbleToVote: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_Response_AssetVotingModel_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Api_Develop_Models_Response_AssetVotingModel),
});
const Roblox_Api_Develop_Models_UniverseModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    isArchived: zod_1.z.boolean(),
    rootPlaceId: zod_1.z.number().int(),
    isActive: zod_1.z.boolean(),
    privacyType: zod_1.z.string(),
    creatorType: zod_1.z.string(),
    creatorTargetId: zod_1.z.number().int(),
    creatorName: zod_1.z.string(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Api_Develop_Models_GameTemplateModel = zod_1.z.object({
    gameTemplateType: zod_1.z.string(),
    hasTutorials: zod_1.z.boolean(),
    universe: Roblox_Api_Develop_Models_UniverseModel,
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_GameTemplateModel_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Api_Develop_Models_GameTemplateModel),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Api_Develop_Models_UniverseModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Api_Develop_Models_UniverseModel),
});
const Roblox_Web_Responses_Users_SkinnyUserResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Users_SkinnyUserResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Web_Responses_Users_SkinnyUserResponse),
});
const Roblox_Web_Responses_Plugins_PluginResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    commentsEnabled: zod_1.z.boolean(),
    versionId: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Plugins_PluginResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Web_Responses_Plugins_PluginResponse),
});
const Roblox_Api_Develop_Models_ActivationEligibilityResponse = zod_1.z.object({
    isEligible: zod_1.z.boolean(),
    maturityRated: zod_1.z.boolean(),
    isUserEligibleForPublicPublish: zod_1.z.boolean(),
    remainingPublicPublishCount: zod_1.z.number().int(),
    isPublicPublish: zod_1.z.boolean(),
    isPublishToExistingUniverse: zod_1.z.boolean(),
});
const Roblox_Api_Develop_Models_UniverseSettingsResponse = zod_1.z.object({
    allowPrivateServers: zod_1.z.boolean(),
    privateServerPrice: zod_1.z.number().int(),
    isMeshTextureApiAccessAllowed: zod_1.z.boolean(),
    isRewardedOnDemandAdsAllowed: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    universeAvatarType: zod_1.z.enum(['MorphToR6', 'PlayerChoice', 'MorphToR15']),
    universeScaleType: zod_1.z.enum(['NoScales', 'AllScales']),
    universeAnimationType: zod_1.z.enum(['Standard', 'PlayerChoice']),
    universeCollisionType: zod_1.z.enum(['InnerBox', 'OuterBox']),
    universeBodyType: zod_1.z.enum(['Standard', 'PlayerChoice']),
    universeJointPositioningType: zod_1.z.enum(['Standard', 'ArtistIntent']),
    isArchived: zod_1.z.boolean(),
    isFriendsOnly: zod_1.z.boolean(),
    genre: zod_1.z.enum([
        'All',
        'Tutorial',
        'Scary',
        'TownAndCity',
        'War',
        'Funny',
        'Fantasy',
        'Adventure',
        'SciFi',
        'Pirate',
        'FPS',
        'RPG',
        'Sports',
        'Ninja',
        'WildWest',
    ]),
    playableDevices: zod_1.z.array(zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4), zod_1.z.literal(5)])),
    isForSale: zod_1.z.boolean(),
    price: zod_1.z.number().int(),
    isStudioAccessToApisAllowed: zod_1.z.boolean(),
    privacyType: zod_1.z.string(),
    isForSaleInFiat: zod_1.z.boolean(),
    fiatBasePriceId: zod_1.z.string(),
    fiatModerationStatus: zod_1.z.enum(['Invalid', 'NotModerated', 'Pending', 'Approved', 'Rejected']),
});
const Roblox_Api_Develop_Models_UniverseSettingsRequest = zod_1.z.object({
    name: zod_1.z.string(),
    universeAvatarType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
    universeScaleType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2)]),
    universeAnimationType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2)]),
    universeCollisionType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2)]),
    universeBodyType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2)]),
    universeJointPositioningType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2)]),
    isArchived: zod_1.z.boolean(),
    isFriendsOnly: zod_1.z.boolean(),
    genre: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
        zod_1.z.literal(9),
        zod_1.z.literal(10),
        zod_1.z.literal(11),
        zod_1.z.literal(12),
        zod_1.z.literal(13),
        zod_1.z.literal(14),
    ]),
    playableDevices: zod_1.z.array(zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4), zod_1.z.literal(5)])),
    isForSale: zod_1.z.boolean(),
    price: zod_1.z.number().int(),
    isMeshTextureApiAccessAllowed: zod_1.z.boolean(),
    isRewardedOnDemandAdsAllowed: zod_1.z.boolean(),
    fiatBasePriceId: zod_1.z.string(),
    fiatProductChangeType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
});
const Roblox_Api_Develop_Models_PrivateServerDetailsResponse = zod_1.z.object({
    isEnabled: zod_1.z.boolean(),
    price: zod_1.z.number().int(),
    activeServersCount: zod_1.z.number().int(),
    activeSubscriptionsCount: zod_1.z.number().int(),
});
const Roblox_Api_Develop_Models_UniversePermissionsModel = zod_1.z.object({
    canManage: zod_1.z.boolean(),
    canCloudEdit: zod_1.z.boolean(),
});
const Roblox_Api_Develop_Models_IPlaceModel = zod_1.z.object({});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Api_Develop_Models_IPlaceModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Api_Develop_Models_IPlaceModel),
});
const Roblox_Api_Develop_Models_Response_TeamCreateSettingsResponse = zod_1.z.object({
    isEnabled: zod_1.z.boolean(),
});
const Roblox_Api_Develop_Models_UpdateTeamCreateSettingsRequest = zod_1.z.object({
    isEnabled: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_UniverseModel_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Api_Develop_Models_UniverseModel),
});
const Roblox_Api_Develop_Models_UniverseIdPermissionsModel = zod_1.z.object({
    universeId: zod_1.z.number().int(),
    canManage: zod_1.z.boolean(),
    canCloudEdit: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_UniverseIdPermissionsModel_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Api_Develop_Models_UniverseIdPermissionsModel),
});
const Roblox_Api_Develop_Models_UniverseTeamCreateSettingsModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    isEnabled: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_UniverseTeamCreateSettingsModel_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Api_Develop_Models_UniverseTeamCreateSettingsModel),
});
const Roblox_Api_Develop_Models_UserPublicPublishEligibilityResponse = zod_1.z.object({
    isEligible: zod_1.z.boolean(),
    hasTransactions: zod_1.z.enum(['Incomplete', 'NotRequired', 'Completed']),
    idVerified: zod_1.z.enum(['Incomplete', 'NotRequired', 'Completed']),
    hasDevex: zod_1.z.enum(['Incomplete', 'NotRequired', 'Completed']),
});
const Roblox_Api_Develop_Models_GroupModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_GroupModel_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Api_Develop_Models_GroupModel),
});
const Roblox_Api_Develop_Models_PlaceConfigurationModel = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string(),
});
const Roblox_Api_Develop_Models_PlaceModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    universeId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
});
const Roblox_Develop_Api_UpdatePluginRequest = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    commentsEnabled: zod_1.z.boolean(),
});
const Roblox_Api_Develop_Models_TeamCreateMembershipRequest = zod_1.z.object({
    userId: zod_1.z.number().int(),
});
/**
 * @api GET https://develop.roblox.com/v1/assets/voting
 * @summary Gets the voting information of the given assets
 * @param assetIds The ids of the Roblox.Platform.Assets.IAsset.
 */
exports.getAssetsVoting = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/assets/voting',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetIds: {
            style: 'form',
        },
    },
    parameters: {
        assetIds: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_Response_AssetVotingModel_,
    errors: [],
});
/**
 * @api GET https://develop.roblox.com/v1/gametemplates
 * @summary Gets a page of templates that can be used to start off making games.
 * @description Templates subject to change without notice.
Sort order of templates specified by Roblox.
 */
exports.getGametemplates = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/gametemplates',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_GameTemplateModel_,
    errors: [],
});
/**
 * @api GET https://develop.roblox.com/v1/groups/:groupId/universes
 * @summary Gets a list of universes for the given group.
 * @param groupId The group id.
 * @param isArchived Whether or not to return archived games.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder Sorted by universeId
 */
exports.getGroupsGroupidUniverses = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/universes',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        isArchived: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        isArchived: zod_1.z.boolean().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Api_Develop_Models_UniverseModel_,
    errors: [
        {
            status: 400,
            description: `Invalid groupId.`,
        },
    ],
});
/**
 * @api POST https://develop.roblox.com/v1/places/:placeId
 * @summary Updates the place configuration for the place with the id placeId
 * @param body
 * @param placeId The place id for the place to be updated.
 * @description Currently the only supported functionality for updating the configuration is around Name, and Description.
 */
exports.postPlacesPlaceid = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/places/:placeId',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        placeId: {
            style: 'simple',
        },
    },
    parameters: {
        placeId: zod_1.z.number().int(),
    },
    body: Roblox_Api_Develop_Models_PlaceConfigurationModel,
    response: Roblox_Api_Develop_Models_PlaceModel,
    errors: [
        {
            status: 400,
            description: `placeId is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Authenticated user is not authorized to manage this place.
0: Token Validation Failed`,
        },
    ],
});
/**
 * @api PATCH https://develop.roblox.com/v1/places/:placeId
 * @summary Updates the place configuration for the place with the id placeId
 * @param body
 * @param placeId The place id for the place to be updated.
 * @description Currently the only supported functionality for updating the configuration is around Name, and Description.
 */
exports.patchPlacesPlaceid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/places/:placeId',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        placeId: {
            style: 'simple',
        },
    },
    parameters: {
        placeId: zod_1.z.number().int(),
    },
    body: Roblox_Api_Develop_Models_PlaceConfigurationModel,
    response: Roblox_Api_Develop_Models_PlaceModel,
    errors: [
        {
            status: 400,
            description: `placeId is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Authenticated user is not authorized to manage this place.
0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/places/:placeId/teamcreate/active_session/members
 * @summary List of users in the active Team Create session
 * @param placeId The place Id.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 */
exports.getPlacesPlaceidTeamcreateActive_sessionMembers = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/places/:placeId/teamcreate/active_session/members',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        placeId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        placeId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Users_SkinnyUserResponse_,
    errors: [
        {
            status: 400,
            description: `1: The universe is invalid.
5: The place is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `2: Not authorized to perform this action.
4: TeamCreate on universe is disabled.`,
        },
        {
            status: 404,
            description: `0: An unknown error occurred.`,
        },
        {
            status: 500,
            description: `6: Multiple active sessions in a Team Create place.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/plugins
 * @summary Gets plugin details by ids.
 * @param pluginIds The plugin ids.
 */
exports.getPlugins = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/plugins',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        pluginIds: {
            style: 'form',
        },
    },
    parameters: {
        pluginIds: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Plugins_PluginResponse_,
    errors: [
        {
            status: 400,
            description: `1: Too many ids.
2: The format of the ids are invalid.`,
        },
    ],
});
/**
 * @api PATCH https://develop.roblox.com/v1/plugins/:pluginId
 * @summary Updates a plugin.
 * @param body The Roblox.Develop.Api.UpdatePluginRequest.
 * @param pluginId The id of the plugin.
 */
exports.patchPluginsPluginid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/plugins/:pluginId',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        pluginId: {
            style: 'simple',
        },
    },
    parameters: {
        pluginId: zod_1.z.number().int(),
    },
    body: Roblox_Develop_Api_UpdatePluginRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `5: Description too long.
6: Text moderated.
7: Invalid name.
8: The request body is missing.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
4: Insufficient permissions.`,
        },
        {
            status: 404,
            description: `3: The id is invalid.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/:universeId
 * @summary Gets a Roblox.Api.Develop.Models.UniverseModel.
 * @param universeId The Universe id.
 */
exports.getUniversesUniverseid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: Roblox_Api_Develop_Models_UniverseModel,
    errors: [
        {
            status: 400,
            description: `1: The universe does not exist.`,
        },
    ],
});
/**
 * @api POST https://develop.roblox.com/v1/universes/:universeId/activate
 * @summary Activates a universes.
 * @param universeId The universe id.
 */
exports.postUniversesUniverseidActivate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/universes/:universeId/activate',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The universe does not exist.
2: This universe does not have a root place.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
3: You are not authorized to configure this universe.
6: The root place for this universe is under review and can not be activated.
7: Creator already has the maximum number of places active.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/:universeId/activation-eligibility
 * @summary Returns the result of various checks for a user's eligibility to activate a given universe from private to public universeId for authenticated user
 * @param universeId The universe id.
 */
exports.getUniversesUniverseidActivationEligibility = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId/activation-eligibility',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: Roblox_Api_Develop_Models_ActivationEligibilityResponse,
    errors: [],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/:universeId/configuration
 * @summary Get settings for an owned universe.
 * @param universeId The universe Id.
 */
exports.getUniversesUniverseidConfiguration = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId/configuration',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: Roblox_Api_Develop_Models_UniverseSettingsResponse,
    errors: [
        {
            status: 400,
            description: `1: The universe does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `2: You are not authorized to configure this universe.`,
        },
    ],
});
/**
 * @api PATCH https://develop.roblox.com/v1/universes/:universeId/configuration
 * @summary Update universe settings for an owned universe.
 * @param body The Roblox.Api.Develop.Models.UniverseSettingsRequest model.
 * @param universeId The universeId.
 */
exports.patchUniversesUniverseidConfiguration = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/universes/:universeId/configuration',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    body: Roblox_Api_Develop_Models_UniverseSettingsRequest,
    response: Roblox_Api_Develop_Models_UniverseSettingsResponse,
    errors: [
        {
            status: 400,
            description: `1: The universe does not exist.
3: Invalid UniverseAvatarType.
4: Invalid UniverseScaleType.
5: Invalid UniverseAnimationType.
6: Invalid UniverseCollisionType.
7: New universe name or description has been rejected.
8: New universe name is too long.
10: Invalid UniverseBodyType.
11: Invalid UniverseJointPositioningType.
12: The universe has no root place.
15: Price is required when isForSale is true.
16: This game cannot be offered for sale because it is not public.
17: This game cannot be offered for sale because it has private servers enabled.
18: The game price is outside of the allowed range.
19: Invalid genre.
20: The request body is missing.
21: Invalid device type.
22: Invalid asset type.
23: Invalid value, the min must be less than or equal to the max
24: Invalid scale value`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You are not authorized to configure this universe.
14: You are not authorized to sell games.`,
        },
        {
            status: 409,
            description: `9: Failed to shutdown all intances of game after changing AvatarType. The change has been reverted.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/:universeId/configuration/vip-servers
 * @summary Get settings for an owned universe's VIP servers.
 * @param universeId The universe Id.
 */
exports.getUniversesUniverseidConfigurationVipServers = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId/configuration/vip-servers',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: Roblox_Api_Develop_Models_PrivateServerDetailsResponse,
    errors: [
        {
            status: 400,
            description: `1: The universe does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `2: You are not authorized to configure this universe.`,
        },
    ],
});
/**
 * @api POST https://develop.roblox.com/v1/universes/:universeId/deactivate
 * @summary Deactivates a universe.
 * @param universeId The universe id.
 */
exports.postUniversesUniverseidDeactivate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/universes/:universeId/deactivate',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The universe does not exist.
2: This universe does not have a root place.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
3: You are not authorized to configure this universe.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/:universeId/permissions
 * @summary Returns list of granted and declined permissions related to the universe with the id universeId for authenticated user
 * @param universeId The universe id.
 */
exports.getUniversesUniverseidPermissions = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId/permissions',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: Roblox_Api_Develop_Models_UniversePermissionsModel,
    errors: [
        {
            status: 400,
            description: `1: The universe does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/:universeId/places
 * @summary Gets a list of places for a universe.
 * @param universeId The asset id.
 * @param isUniverseCreation
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder Sorted by placeId
 */
exports.getUniversesUniverseidPlaces = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId/places',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
        isUniverseCreation: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        isUniverseCreation: zod_1.z.boolean().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Api_Develop_Models_IPlaceModel_,
    errors: [],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/:universeId/teamcreate
 * @summary Gets TeamCreate settings for an Roblox.Platform.Universes.IUniverse.
 * @param universeId The universe Id.
 */
exports.getUniversesUniverseidTeamcreate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId/teamcreate',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({ isEnabled: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `Roblox.Api.Develop.ResponseEnums.TeamCreateErrors.InvalidUniverse`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Roblox.Api.Develop.ResponseEnums.TeamCreateErrors.Unauthorized`,
        },
    ],
});
/**
 * @api PATCH https://develop.roblox.com/v1/universes/:universeId/teamcreate
 * @summary Edit team create settings for a universe.
 * @param body The request body containing the team create settings.
 * @param universeId The universe Id.
 * @description Enables, or disables team create for a universe.
 */
exports.patchUniversesUniverseidTeamcreate = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/universes/:universeId/teamcreate',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ isEnabled: zod_1.z.boolean() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `Roblox.Api.Develop.ResponseEnums.TeamCreateErrors.InvalidUniverse`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Roblox.Api.Develop.ResponseEnums.TeamCreateErrors.Unauthorized
0: Token Validation Failed`,
        },
    ],
});
/**
 * @api DELETE https://develop.roblox.com/v1/universes/:universeId/teamcreate/memberships
 * @summary Removes a user from a TeamCreate permissions list.
 * @param body The request body.
 * @param universeId The universe Id.
 */
exports.deleteUniversesUniverseidTeamcreateMemberships = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/universes/:universeId/teamcreate/memberships',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ userId: zod_1.z.number().int() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `Roblox.Api.Develop.ResponseEnums.TeamCreateErrors.InvalidUniverse OR Roblox.Api.Develop.ResponseEnums.TeamCreateErrors.InvalidUser`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Roblox.Api.Develop.ResponseEnums.TeamCreateErrors.TeamCreateDisabled
0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/multiget
 * @summary Gets a list of universes.
 * @param ids The universe IDs to get. Limit 100.
 * @description If a universe can not be found for a given ID (such as -1) it will be skipped.
 */
exports.getUniversesMultiget = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/multiget',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        ids: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        ids: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_UniverseModel_,
    errors: [
        {
            status: 400,
            description: `8: No universe IDs sent to get.
9: Too many universe IDs sent to get, the limit is: `,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/multiget/permissions
 * @summary Returns an array of granted and declined permissions related to the universes with the ids in ids for the authenticated user.
 * @param ids The universe ids.
 * @description If a universe can not be found for a given ID (such as -1) it will be skipped.
 */
exports.getUniversesMultigetPermissions = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/multiget/permissions',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        ids: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        ids: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_UniverseIdPermissionsModel_,
    errors: [
        {
            status: 400,
            description: `8: No universe IDs sent to get.
9: Too many universe IDs sent to get, the limit is: `,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/multiget/teamcreate
 * @summary Gets TeamCreate settings for multiple universes specified by Ids
 * @param ids The universe Ids.
 */
exports.getUniversesMultigetTeamcreate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/multiget/teamcreate',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        ids: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        ids: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_UniverseTeamCreateSettingsModel_,
    errors: [
        {
            status: 400,
            description: `Roblox.Api.Develop.ResponseEnums.TeamCreateErrors.TooManyUniverseIdsSent`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/universes/user-public-publish-eligibility
 * @summary Returns the result of various checks for a user's eligibility to publish a public universe
 */
exports.getUniversesUserPublicPublishEligibility = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/user-public-publish-eligibility',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    response: Roblox_Api_Develop_Models_UserPublicPublishEligibilityResponse,
    errors: [],
});
/**
 * @api GET https://develop.roblox.com/v1/user/groups/canmanage
 * @summary Gets a list of Groups that a user can manage.
 */
exports.getUserGroupsCanmanage = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/user/groups/canmanage',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_GroupModel_,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/user/groups/canmanagegamesoritems
 * @summary Gets a list of groups a user can manage games or items for.
 */
exports.getUserGroupsCanmanagegamesoritems = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/user/groups/canmanagegamesoritems',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Api_Develop_Models_GroupModel_,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://develop.roblox.com/v1/user/universes
 * @summary Gets a list of universes for the authenticated user.
 * @param isArchived Whether or not to return archived games.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder Sorted by universeId
 */
exports.getUserUniverses = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/user/universes',
    baseUrl: 'https://develop.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        isArchived: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        isArchived: zod_1.z.boolean().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Api_Develop_Models_UniverseModel_,
    errors: [
        {
            status: 400,
            description: `cursor is not valid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
