// rozod/lib/endpoints/economycreatorstatsv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUniversesUniverseidStats = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_EconomyCreatorStats_Api_Models_StatisticsResponse = zod_1.z.object({
    dataGranularity: zod_1.z.enum(['Hourly', 'Daily', 'Monthly']),
    data: zod_1.z.array(zod_1.z.array(zod_1.z.number())),
});
/**
 * @api GET https://economycreatorstats.roblox.com/v1/universes/:universeId/stats
 * @summary Get statistics data for a universe.
 * @param universeId The universe id.
 * @param Type
 * @param StartTime
 * @param EndTime
 */
exports.getUniversesUniverseidStats = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universes/:universeId/stats',
    baseUrl: 'https://economycreatorstats.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
        Type: {
            style: 'form',
            explode: true,
        },
        StartTime: {
            style: 'form',
            explode: true,
        },
        EndTime: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        Type: zod_1.z.enum(['PremiumUpsells', 'PremiumVisits']),
        StartTime: zod_1.z.string().datetime({ offset: true }),
        EndTime: zod_1.z.string().datetime({ offset: true }),
    },
    response: Roblox_EconomyCreatorStats_Api_Models_StatisticsResponse,
    errors: [
        {
            status: 400,
            description: `1: The Universe is invalid.
3: Too many data points requested.
4: The requested data type is not known.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
2: Not authorized to perform this action.`,
        },
    ],
});
