// rozod/lib/endpoints/engagementpayoutsv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUniversePayoutHistory = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_EngagementPayouts_Api_PayoutResponseModel = zod_1.z.object({
    engagementScore: zod_1.z.number(),
    payoutInRobux: zod_1.z.number().int(),
    payoutType: zod_1.z.string(),
    eligibilityType: zod_1.z.string(),
});
/**
 * @api GET https://engagementpayouts.roblox.com/v1/universe-payout-history
 * @summary Gets the engagement payout history for a specific universe and a given date range, specified by start and end dates.
 * @param universeId The ID of the universe in question.
 * @param startDate The first date in the range, specified as yyyy-MM-dd.
 * @param endDate The last date in the range, specified as yyyy-MM-dd.
 */
exports.getUniversePayoutHistory = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/universe-payout-history',
    baseUrl: 'https://engagementpayouts.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'form',
            explode: true,
        },
        startDate: {
            style: 'form',
            explode: true,
        },
        endDate: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        startDate: zod_1.z.string(),
        endDate: zod_1.z.string(),
    },
    response: Roblox_EngagementPayouts_Api_PayoutResponseModel,
    errors: [
        {
            status: 400,
            description: `1: InvalidUniverseId
2: InvalidStartDate
3: InvalidEndDate
4: InvalidDateRange
5: Forbidden
6: TooManyDays`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
