// rozod/lib/endpoints/followingsv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridUniversesUniverseidStatus = exports.deleteUsersUseridUniversesUniverseid = exports.postUsersUseridUniversesUniverseid = exports.getUsersUseridUniverses = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Followings_Api_Models_UserFollowingUniverseResponse = zod_1.z.object({
    universeId: zod_1.z.number().int(),
    userId: zod_1.z.number().int(),
});
const Roblox_Followings_Api_Models_UserFollowingUniverseStatusResponse = zod_1.z.object({
    UniverseId: zod_1.z.number().int(),
    UserId: zod_1.z.number().int(),
    CanFollow: zod_1.z.boolean(),
    IsFollowing: zod_1.z.boolean(),
    FollowingCountByType: zod_1.z.number().int(),
    FollowingLimitByType: zod_1.z.number().int(),
});
/**
 * @api GET https://followings.roblox.com/v1/users/:userId/universes
 * @summary Gets all the followings between a user with userId and universes
 * @param userId
 */
exports.getUsersUseridUniverses = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/universes',
    baseUrl: 'https://followings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.array(Roblox_Followings_Api_Models_UserFollowingUniverseResponse),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `User is not authorized for this action.`,
        },
    ],
});
/**
 * @api POST https://followings.roblox.com/v1/users/:userId/universes/:universeId
 * @summary Creates the following between a user with userId and universe with universeId
 * @param userId
 * @param universeId
 */
exports.postUsersUseridUniversesUniverseid = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/universes/:universeId',
    baseUrl: 'https://followings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        universeId: zod_1.z.number().int(),
    },
    response: Roblox_Followings_Api_Models_UserFollowingUniverseResponse,
    errors: [
        {
            status: 400,
            description: `The user has reached the limit of number of followed universes.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `User is not authorized for this action.
0: Token Validation Failed`,
        },
    ],
});
/**
 * @api DELETE https://followings.roblox.com/v1/users/:userId/universes/:universeId
 * @summary Deletes the following between a user with userId and universe with universeId
 * @param userId
 * @param universeId
 */
exports.deleteUsersUseridUniversesUniverseid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/users/:userId/universes/:universeId',
    baseUrl: 'https://followings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        universeId: zod_1.z.number().int(),
    },
    response: Roblox_Followings_Api_Models_UserFollowingUniverseResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `User is not authorized for this action.
0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://followings.roblox.com/v1/users/:userId/universes/:universeId/status
 * @summary Gets the status of a following relationship between a user and a universe.
 * @param userId
 * @param universeId
 */
exports.getUsersUseridUniversesUniverseidStatus = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/universes/:universeId/status',
    baseUrl: 'https://followings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        universeId: zod_1.z.number().int(),
    },
    response: Roblox_Followings_Api_Models_UserFollowingUniverseStatusResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `User is not authorized for this action.`,
        },
    ],
});
