// rozod/lib/endpoints/followingsv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridUniverses = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Followings_Api_Models_FollowsByTypeResponse = zod_1.z.object({
    followerType: zod_1.z.enum(['Invalid', 'User']),
    followerId: zod_1.z.number().int(),
    sourceType: zod_1.z.enum(['Invalid', 'Universe']),
    followedSources: zod_1.z.string().datetime({ offset: true }),
});
/**
 * @api GET https://followings.roblox.com/v2/users/:userId/universes
 * @summary Gets all universes followed by a user.
 * @param userId The user ID.
 */
exports.getUsersUseridUniverses = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/users/:userId/universes',
    baseUrl: 'https://followings.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Followings_Api_Models_FollowsByTypeResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `User is not authorized for this action.`,
        },
    ],
});
