// rozod/lib/endpoints/friendsv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUserUseridMultigetAreTrustedFriends = exports.getMyTrustedFriendsUseridStatus = exports.getUsersUseridFriendsStatuses = exports.getUsersUseridFriendsSearch = exports.getUsersUseridFriendsOnline = exports.getUsersUseridFriendsInactive = exports.getUsersUseridFriendsFind = exports.getUsersUseridFriendsCount = exports.getUsersUseridFriends = exports.postUsersTargetuseridUnfriend = exports.postUsersTargetuseridUnfollow = exports.postUsersTargetuseridRequestFriendship = exports.getUsersTargetuseridFollowingsCount = exports.getUsersTargetuseridFollowings = exports.getUsersTargetuseridFollowersCount = exports.getUsersTargetuseridFollowers = exports.postUsersTargetuseridFollow = exports.postUsersSenderuseridAcceptFriendRequestWithToken = exports.postUsersRequesteruseridDeclineFriendRequest = exports.postUsersRequesteruseridAcceptFriendRequest = exports.postUserFriendRequestsDeclineAll = exports.getUserFriendRequestsCount = exports.postUserFollowingExists = exports.postUserUseridMultigetAreFriends = exports.getMyNewFriendRequestsCount = exports.deleteMyNewFriendRequests = exports.getMyFriendsRequests = exports.postMyFriendsRefreshQrSession = exports.getMyFriendsCount = exports.getMyFriendsUseridCheckQrSession = exports.getMetadata = exports.postContactsTargetcontactidRequestFriendship = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Friends_Api_Models_Response_FriendsPageMetadataResponse = zod_1.z.object({
    isFriendsFilterBarEnabled: zod_1.z.boolean(),
    isFriendsPageSortExperimentEnabled: zod_1.z.boolean(),
    isFriendsUserDataStoreCacheEnabled: zod_1.z.boolean(),
    frequentFriendSortRollout: zod_1.z.number().int(),
    userName: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Friends_Api_FriendsCountResponse = zod_1.z.object({
    count: zod_1.z.number().int(),
});
const Roblox_Friends_Api_FriendRequest = zod_1.z.object({
    sentAt: zod_1.z.string().datetime({ offset: true }),
    senderId: zod_1.z.number().int(),
    sourceUniverseId: zod_1.z.number().int(),
    originSourceType: zod_1.z.enum([
        'Unknown',
        'PlayerSearch',
        'QrCode',
        'InGame',
        'UserProfile',
        'QqContactImporter',
        'WeChatContactImporter',
        'ProfileShare',
        'PhoneContactImporter',
        'FriendRecommendations',
    ]),
    contactName: zod_1.z.string(),
    senderNickname: zod_1.z.string(),
});
const Roblox_Friends_Api_FriendRequestResponse = zod_1.z.object({
    friendRequest: Roblox_Friends_Api_FriendRequest,
    mutualFriendsList: zod_1.z.array(zod_1.z.string()),
    hasVerifiedBadge: zod_1.z.boolean(),
    description: zod_1.z.string(),
    created: zod_1.z.string().datetime({ offset: true }),
    isBanned: zod_1.z.boolean(),
    externalAppDisplayName: zod_1.z.string(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Friends_Api_FriendRequestResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Friends_Api_FriendRequestResponse),
});
const Roblox_Friends_Api_Models_Response_NewFriendRequestsCountResponse = zod_1.z.object({ count: zod_1.z.number().int() });
const Roblox_Friends_Api_PendingFriendRequestCountModel = zod_1.z.object({
    count: zod_1.z.number().int(),
});
const Roblox_Friends_Api_Models_Response_UserResponse = zod_1.z.object({
    isOnline: zod_1.z.boolean(),
    presenceType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
    isDeleted: zod_1.z.boolean(),
    friendFrequentScore: zod_1.z.number().int(),
    friendFrequentRank: zod_1.z.number().int(),
    hasVerifiedBadge: zod_1.z.boolean(),
    description: zod_1.z.string(),
    created: zod_1.z.string().datetime({ offset: true }),
    isBanned: zod_1.z.boolean(),
    externalAppDisplayName: zod_1.z.string(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Friends_Api_Models_Response_UserResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Friends_Api_Models_Response_UserResponse),
});
const Roblox_Friends_Api_FollowCountResponse = zod_1.z.object({
    count: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Friends_Api_Models_Response_UserResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Friends_Api_Models_Response_UserResponse),
});
const Roblox_Friends_Api_Models_Response_FriendResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Paging_CursoredPagedResult_Roblox_Friends_Api_Models_Response_FriendResponse_ = zod_1.z.object({
    PreviousCursor: zod_1.z.string(),
    PageItems: zod_1.z.array(Roblox_Friends_Api_Models_Response_FriendResponse),
    NextCursor: zod_1.z.string(),
    HasMore: zod_1.z.boolean(),
});
const Roblox_Friends_Api_Models_Response_UserPresenceResponseModel = zod_1.z.object({
    UserPresenceType: zod_1.z.string(),
    UserLocationType: zod_1.z.string(),
    lastLocation: zod_1.z.string(),
    placeId: zod_1.z.number().int(),
    rootPlaceId: zod_1.z.number().int(),
    gameInstanceId: zod_1.z.string().uuid(),
    universeId: zod_1.z.number().int(),
    lastOnline: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Friends_Api_Models_Response_UserPresenceResponse = zod_1.z.object({
    userPresence: Roblox_Friends_Api_Models_Response_UserPresenceResponseModel,
    sortScore: zod_1.z.number(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Friends_Api_Models_Response_UserPresenceResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Friends_Api_Models_Response_UserPresenceResponse),
});
const Roblox_Friends_Api_FriendStatusResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    status: zod_1.z.enum(['NotFriends', 'Friends', 'RequestSent', 'RequestReceived']),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Friends_Api_FriendStatusResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Friends_Api_FriendStatusResponse),
});
const Roblox_Friends_Api_CaptchaStatusResponseModel = zod_1.z.object({
    success: zod_1.z.boolean(),
    isCaptchaRequired: zod_1.z.boolean(),
});
const Roblox_Friends_Api_Models_Response_RefreshQrSessionResponse = zod_1.z.object({
    Success: zod_1.z.boolean(),
});
const Roblox_Friends_Api_MultigetAreFriendsRequestModel = zod_1.z.object({
    targetUserIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Friends_Api_MultigetAreFriendsResponse = zod_1.z.object({
    friendsId: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Friends_Api_FollowingExistsRequestModel = zod_1.z.object({
    targetUserIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Friends_Api_Models_Response_FollowingExistsResponse = zod_1.z.object({
    isFollowing: zod_1.z.boolean(),
    isFollowed: zod_1.z.boolean(),
    userId: zod_1.z.number().int(),
});
const Roblox_Friends_Api_Models_Response_FollowingExistsResponseModel = zod_1.z.object({
    followings: zod_1.z.array(Roblox_Friends_Api_Models_Response_FollowingExistsResponse),
});
const Roblox_Friends_Api_Models_Response_DeclineAllFriendRequestsResponse = zod_1.z.object({ backgrounded: zod_1.z.boolean() });
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Friends_Api_Models_Request_FriendingTokenRequestModel = zod_1.z.object({
    friendingToken: zod_1.z.string(),
});
const Roblox_Web_Captcha_Models_Request_CaptchaTokenRequest = zod_1.z.object({
    captchaId: zod_1.z.string(),
    captchaToken: zod_1.z.string(),
    captchaProvider: zod_1.z.string(),
    challengeId: zod_1.z.string(),
});
const Roblox_Friends_Api_FriendshipRequestModel = zod_1.z.object({
    friendshipOriginSourceType: zod_1.z.enum([
        'Unknown',
        'PlayerSearch',
        'QrCode',
        'InGame',
        'UserProfile',
        'QqContactImporter',
        'WeChatContactImporter',
        'ProfileShare',
        'PhoneContactImporter',
        'FriendRecommendations',
    ]),
    senderNickname: zod_1.z.string(),
});
const Roblox_Friends_Api_Models_Response_ClearNewFriendRequestResponse = zod_1.z.object({ status: zod_1.z.boolean() });
/**
 * @api POST https://friends.roblox.com/v1/contacts/:targetContactId/request-friendship
 * @summary Send a contact friend request to target user
 * @param targetContactId The target contact Id for friend request
 */
exports.postContactsTargetcontactidRequestFriendship = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/contacts/:targetContactId/request-friendship',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        targetContactId: {
            style: 'simple',
        },
    },
    parameters: {
        targetContactId: zod_1.z.string(),
    },
    response: Roblox_Friends_Api_CaptchaStatusResponseModel,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
5: The target user is already a friend.
6: Invalid parameters.
7: The user cannot be friends with itself.
31: User with max friends sent friend request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user is banned from performing operation.
3: The user is blocked from performing this action.
14: The user has not passed the captcha.`,
        },
        {
            status: 429,
            description: `9: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/metadata
 * @param targetUserId
 */
exports.getMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/metadata',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        targetUserId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        targetUserId: zod_1.z.number().int().optional(),
    },
    response: Roblox_Friends_Api_Models_Response_FriendsPageMetadataResponse,
    errors: [],
});
/**
 * @api GET https://friends.roblox.com/v1/my/friends/:userId/check-qr-session
 * @param userId user Id that shows the qr code
 */
exports.getMyFriendsUseridCheckQrSession = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/my/friends/:userId/check-qr-session',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.boolean(),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/my/friends/count
 * @summary Get the number of friends a user has
 */
exports.getMyFriendsCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/my/friends/count',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/my/friends/refresh-qr-session
 */
exports.postMyFriendsRefreshQrSession = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/my/friends/refresh-qr-session',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ Success: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/my/friends/requests
 * @summary Get all users that friend requests with targetUserId using exclusive start paging
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sessionId Optional session identifier.
 * @param friendRequestSort
 */
exports.getMyFriendsRequests = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/my/friends/requests',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sessionId: {
            style: 'form',
            explode: true,
        },
        friendRequestSort: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        limit: zod_1.z.number().int().optional().default(10),
        cursor: zod_1.z.string().optional(),
        sessionId: zod_1.z.string().optional(),
        friendRequestSort: zod_1.z
            .union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)])
            .optional()
            .default(1),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Friends_Api_FriendRequestResponse_,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
6: Invalid parameters.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `2: The user is banned from performing operation.
3: The user is blocked from performing this action.`,
        },
        {
            status: 429,
            description: `9: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api DELETE https://friends.roblox.com/v1/my/new-friend-requests
 */
exports.deleteMyNewFriendRequests = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/my/new-friend-requests',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ status: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/my/new-friend-requests/count
 */
exports.getMyNewFriendRequestsCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/my/new-friend-requests/count',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/user/:userId/multiget-are-friends
 * @summary Check if the requesting user is friends with the specified users.
 * @param body The user ids to check against with the requesting user.
 * @param userId The requesting userId.
 */
exports.postUserUseridMultigetAreFriends = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/user/:userId/multiget-are-friends',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_Friends_Api_MultigetAreFriendsRequestModel,
    response: Roblox_Friends_Api_MultigetAreFriendsResponse,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/user/following-exists
 * @summary Returns whether or not the current user is following each userId in a list of userIds
 * @param body The userIds potentially being followed
 */
exports.postUserFollowingExists = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/user/following-exists',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Friends_Api_FollowingExistsRequestModel,
    response: Roblox_Friends_Api_Models_Response_FollowingExistsResponseModel,
    errors: [
        {
            status: 400,
            description: `0: An invalid userId was passed in.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `1: Followers are disabled at this time.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/user/friend-requests/count
 * @summary Return the number of pending friend requests.
 */
exports.getUserFriendRequestsCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/user/friend-requests/count',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/user/friend-requests/decline-all
 * @summary Decline all pending friend requests for the authenticated user.
 */
exports.postUserFriendRequestsDeclineAll = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/user/friend-requests/decline-all',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ backgrounded: zod_1.z.boolean() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/users/:requesterUserId/accept-friend-request
 * @summary Accept a friend request.
 * @param requesterUserId The user Id of the requester
 */
exports.postUsersRequesteruseridAcceptFriendRequest = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:requesterUserId/accept-friend-request',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        requesterUserId: {
            style: 'simple',
        },
    },
    parameters: {
        requesterUserId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
10: The friend request does not exist.
11: The current users friends limit has been exceeded.
12: The target users friends limit has been exceeded.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
3: The user is blocked from performing this action.`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/users/:requesterUserId/decline-friend-request
 * @summary Decline a friend request.
 * @param requesterUserId The user Id of the requester
 */
exports.postUsersRequesteruseridDeclineFriendRequest = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:requesterUserId/decline-friend-request',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        requesterUserId: {
            style: 'simple',
        },
    },
    parameters: {
        requesterUserId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
10: The friend request does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/users/:senderUserId/accept-friend-request-with-token
 * @summary Accept a friend request with an Off Network Friending token.
 * @param body
 * @param senderUserId The user id of the sender of the off network friend request
 */
exports.postUsersSenderuseridAcceptFriendRequestWithToken = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:senderUserId/accept-friend-request-with-token',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        senderUserId: {
            style: 'simple',
        },
    },
    parameters: {
        senderUserId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ friendingToken: zod_1.z.string() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
3: The user is blocked from performing this action.
5: The target user is already a friend.
6: Invalid parameters.
7: The user cannot be friends with itself.
11: The current users friends limit has been exceeded.
12: The target users friends limit has been exceeded.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/users/:targetUserId/follow
 * @summary Creates the following between a user and user with targetUserId
 * @param body
 * @param targetUserId
 */
exports.postUsersTargetuseridFollow = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:targetUserId/follow',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        targetUserId: {
            style: 'simple',
        },
    },
    parameters: {
        targetUserId: zod_1.z.number().int(),
    },
    body: Roblox_Web_Captcha_Models_Request_CaptchaTokenRequest.optional(),
    response: Roblox_Friends_Api_CaptchaStatusResponseModel,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
6: Invalid parameters.
8: The user cannot follow itself.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user is banned from performing operation.
3: The user is blocked from performing this action.
14: The user has not passed the captcha.`,
        },
        {
            status: 429,
            description: `9: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:targetUserId/followers
 * @summary Get all users that follow user with targetUserId in page response format
 * @param targetUserId
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getUsersTargetuseridFollowers = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:targetUserId/followers',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        targetUserId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        targetUserId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(18), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Friends_Api_Models_Response_UserResponse_,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
6: Invalid parameters.`,
        },
        {
            status: 403,
            description: `2: The user is banned from performing operation.
3: The user is blocked from performing this action.`,
        },
        {
            status: 429,
            description: `9: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:targetUserId/followers/count
 * @summary Get the number of following a user has
 * @param targetUserId
 */
exports.getUsersTargetuseridFollowersCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:targetUserId/followers/count',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        targetUserId: {
            style: 'simple',
        },
    },
    parameters: {
        targetUserId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:targetUserId/followings
 * @summary Get all users that user with targetUserId is following in page response format
 * @param targetUserId
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getUsersTargetuseridFollowings = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:targetUserId/followings',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        targetUserId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        targetUserId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(18), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Friends_Api_Models_Response_UserResponse_,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
6: Invalid parameters.`,
        },
        {
            status: 403,
            description: `2: The user is banned from performing operation.
3: The user is blocked from performing this action.`,
        },
        {
            status: 429,
            description: `9: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:targetUserId/followings/count
 * @summary Get the number of following a user has
 * @param targetUserId
 */
exports.getUsersTargetuseridFollowingsCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:targetUserId/followings/count',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        targetUserId: {
            style: 'simple',
        },
    },
    parameters: {
        targetUserId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/users/:targetUserId/request-friendship
 * @summary Send a friend request to target user
 * @param body The source which the friend request originated from
 * @param targetUserId The target user Id for friend request
 */
exports.postUsersTargetuseridRequestFriendship = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:targetUserId/request-friendship',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        targetUserId: {
            style: 'simple',
        },
    },
    parameters: {
        targetUserId: zod_1.z.number().int(),
    },
    body: Roblox_Friends_Api_FriendshipRequestModel,
    response: Roblox_Friends_Api_CaptchaStatusResponseModel,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
5: The target user is already a friend.
6: Invalid parameters.
7: The user cannot be friends with itself.
10: The friend request does not exist.
13: The users are not in the same game.
31: User with max friends sent friend request.
35: Invalid nickname.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user is banned from performing operation.
3: The user is blocked from performing this action.
14: The user has not passed the captcha.`,
        },
        {
            status: 429,
            description: `9: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/users/:targetUserId/unfollow
 * @summary Deletes the following between a user and user with targetUserId
 * @param targetUserId
 */
exports.postUsersTargetuseridUnfollow = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:targetUserId/unfollow',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        targetUserId: {
            style: 'simple',
        },
    },
    parameters: {
        targetUserId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
6: Invalid parameters.
8: The user cannot follow itself.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user is banned from performing operation.
3: The user is blocked from performing this action.
14: The user has not passed the captcha.`,
        },
        {
            status: 429,
            description: `9: The flood limit has been exceeded.`,
        },
    ],
});
/**
 * @api POST https://friends.roblox.com/v1/users/:targetUserId/unfriend
 * @summary Unfriend a user
 * @param targetUserId The target user id to unfriend
 */
exports.postUsersTargetuseridUnfriend = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:targetUserId/unfriend',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        targetUserId: {
            style: 'simple',
        },
    },
    parameters: {
        targetUserId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:userId/friends
 * @summary Get list of all friends for the specified user.
 * @param userId The user Id to get the friends for.
 * @param userSort Specifies how to sort the returned friends.
 */
exports.getUsersUseridFriends = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/friends',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        userSort: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        userSort: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]).optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Friends_Api_Models_Response_UserResponse_,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:userId/friends/count
 * @summary Get the number of friends a user has
 * @param userId
 */
exports.getUsersUseridFriendsCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/friends/count',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:userId/friends/find
 * @summary Get a paginated list of all friends for the specified user.
 * @param userId The user Id to get the friends for.
 * @param userSort Specifies how to sort the returned friends.
 * @param cursor The paging cursor for the previous or next page.
 * @param limit The number of results per request.
 * @param findFriendsType
 */
exports.getUsersUseridFriendsFind = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/friends/find',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        userSort: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        findFriendsType: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        userSort: zod_1.z
            .union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)])
            .optional()
            .default(2),
        cursor: zod_1.z.string().optional(),
        limit: zod_1.z.number().int().optional().default(50),
        findFriendsType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1)]).optional(),
    },
    response: Roblox_Paging_CursoredPagedResult_Roblox_Friends_Api_Models_Response_FriendResponse_,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
6: Invalid parameters.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:userId/friends/inactive
 * @summary Get list of inactive friends for the specified user.
 * @param userId The user Id to get the friends for.
 */
exports.getUsersUseridFriendsInactive = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/friends/inactive',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Friends_Api_Models_Response_UserResponse_,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
6: Invalid parameters.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:userId/friends/online
 * @summary Get list of all online friends for the specified user.
 * @param userId The user Id to get the friends for.
 * @param userSort The sort order to return the friends.
 */
exports.getUsersUseridFriendsOnline = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/friends/online',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        userSort: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        userSort: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]).optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Friends_Api_Models_Response_UserPresenceResponse_,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
6: Invalid parameters.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:userId/friends/search
 * @summary Search for friends by name using a text query.
 * @param userId The user Id to get the friends for.
 * @param query The string to search names of friends for.
 * @param cursor The paging cursor for the previous or next page.
 * @param limit The number of results per request.
 */
exports.getUsersUseridFriendsSearch = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/friends/search',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        query: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        query: zod_1.z.string().optional(),
        cursor: zod_1.z.string().optional(),
        limit: zod_1.z.number().int().optional().default(20),
    },
    response: Roblox_Paging_CursoredPagedResult_Roblox_Friends_Api_Models_Response_FriendResponse_,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
6: Invalid parameters.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://friends.roblox.com/v1/users/:userId/friends/statuses
 * @summary Gets a list of friend statuses of specified users against the specified user.
 * @param userId The user to check the friend statuses against.
 * @param userIds
 */
exports.getUsersUseridFriendsStatuses = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/friends/statuses',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        userIds: {
            style: 'form',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        userIds: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Friends_Api_FriendStatusResponse_,
    errors: [
        {
            status: 400,
            description: `1: The target user is invalid or does not exist.
15: Too many ids.
16: Invalid ids.`,
        },
    ],
});
// Patched endpoints removed from Roblox API docs (v6.1.0 compat)
const Patch_TrustedFriendStatusResponse = zod_1.z.object({
    status: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4), zod_1.z.literal(5), zod_1.z.literal(6)]),
});
const Patch_MultigetAreTrustedFriendsResponse = zod_1.z.object({
    trustedFriendsId: zod_1.z.array(zod_1.z.number()),
});
exports.getMyTrustedFriendsUseridStatus = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/my/trusted-friends/:userId/status',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: { userId: { style: 'simple' } },
    parameters: { userId: zod_1.z.number().int() },
    response: Patch_TrustedFriendStatusResponse,
    errors: [
        { status: 400, description: `1: The target user is invalid or does not exist.` },
        { status: 401, description: `0: Authorization has been denied for this request.` },
    ],
});
exports.getUserUseridMultigetAreTrustedFriends = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/user/:userId/multiget-are-trusted-friends',
    baseUrl: 'https://friends.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: { style: 'simple' },
        userIds: { style: 'form' },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        userIds: zod_1.z.array(zod_1.z.number()),
    },
    response: Patch_MultigetAreTrustedFriendsResponse,
    errors: [{ status: 400, description: `1: The target user is invalid or does not exist.` }],
});
