// rozod/lib/endpoints/gamesv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getGroupsGroupidGames = exports.getGroupsGroupidGamesv2 = exports.getUsersUseridGames = exports.getUsersUseridFavoriteGames = exports.getGamesUniverseidMedia = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_Responses_Games_GameMediaItemResponseV2 = zod_1.z.object({
    assetTypeId: zod_1.z.number().int(),
    assetType: zod_1.z.string(),
    imageId: zod_1.z.number().int(),
    videoHash: zod_1.z.string(),
    videoTitle: zod_1.z.string(),
    approved: zod_1.z.boolean(),
    altText: zod_1.z.string(),
    videoId: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Games_GameMediaItemResponseV2_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Web_Responses_Games_GameMediaItemResponseV2),
});
const Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Core_CreatorType_ = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.enum(['User', 'Group']),
    name: zod_1.z.string(),
});
const Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Assets_AssetType_ = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.enum([
        'Image',
        'TShirt',
        'Audio',
        'Mesh',
        'Lua',
        'HTML',
        'Text',
        'Hat',
        'Place',
        'Model',
        'Shirt',
        'Pants',
        'Decal',
        'Avatar',
        'Head',
        'Face',
        'Gear',
        'Badge',
        'GroupEmblem',
        'Animation',
        'Arms',
        'Legs',
        'Torso',
        'RightArm',
        'LeftArm',
        'LeftLeg',
        'RightLeg',
        'Package',
        'YouTubeVideo',
        'GamePass',
        'App',
        'Code',
        'Plugin',
        'SolidModel',
        'MeshPart',
        'HairAccessory',
        'FaceAccessory',
        'NeckAccessory',
        'ShoulderAccessory',
        'FrontAccessory',
        'BackAccessory',
        'WaistAccessory',
        'ClimbAnimation',
        'DeathAnimation',
        'FallAnimation',
        'IdleAnimation',
        'JumpAnimation',
        'RunAnimation',
        'SwimAnimation',
        'WalkAnimation',
        'PoseAnimation',
        'LocalizationTableManifest',
        'LocalizationTableTranslation',
        'EmoteAnimation',
        'Video',
        'TexturePack',
        'TShirtAccessory',
        'ShirtAccessory',
        'PantsAccessory',
        'JacketAccessory',
        'SweaterAccessory',
        'ShortsAccessory',
        'LeftShoeAccessory',
        'RightShoeAccessory',
        'DressSkirtAccessory',
        'FontFamily',
        'FontFace',
        'MeshHiddenSurfaceRemoval',
        'EyebrowAccessory',
        'EyelashAccessory',
        'MoodAnimation',
        'DynamicHead',
        'CodeSnippet',
        'AdsVideo',
        'OtaUpdate',
        'Screenshot',
        'RuntimePropertySet',
        'StorePreviewVideo',
        'GamePreviewVideo',
        'CreatorExperienceConfig',
        'FaceMakeup',
        'LipMakeup',
        'EyeMakeup',
    ]),
    name: zod_1.z.string(),
});
const Roblox_Web_Responses_Games_GameFavoriteResponseModel = zod_1.z.object({
    price: zod_1.z.number().int(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    creator: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Core_CreatorType_,
    rootPlace: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Assets_AssetType_,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
    placeVisits: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Web_Responses_Games_GameFavoriteResponseModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Web_Responses_Games_GameFavoriteResponseModel),
});
const Roblox_Web_Responses_Games_GameResponseV2 = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    creator: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Core_CreatorType_,
    rootPlace: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Assets_AssetType_,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
    placeVisits: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Web_Responses_Games_GameResponseV2_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Web_Responses_Games_GameResponseV2),
});
/**
 * @api GET https://games.roblox.com/v2/games/:universeId/media
 * @summary Get the game media data
 * @param universeId The id of the universe we get media data from.
 * @param fetchAllExperienceRelatedMedia to tell if the API query is to fetch all related media for this experience
 */
exports.getGamesUniverseidMedia = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/games/:universeId/media',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
        fetchAllExperienceRelatedMedia: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        fetchAllExperienceRelatedMedia: zod_1.z.boolean().optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Games_GameMediaItemResponseV2_,
    errors: [
        {
            status: 400,
            description: `3: The universe&#x27;s root place is invalid.`,
        },
        {
            status: 404,
            description: `2: The requested universe does not exist.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v2/users/:userId/favorite/games
 * @summary Gets users favorite games.
 * @param userId The user Id.
 * @param accessFilter Filtering option via access level.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getUsersUseridFavoriteGames = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/users/:userId/favorite/games',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        accessFilter: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        accessFilter: zod_1.z
            .union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(4)])
            .optional()
            .default(2),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Desc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Web_Responses_Games_GameFavoriteResponseModel_,
    errors: [],
});
/**
 * @api GET https://games.roblox.com/v2/users/:userId/games
 * @summary Gets games created by the specified user.
 * @param userId The user Id.
 * @param accessFilter Filtering option via access level.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getUsersUseridGames = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/users/:userId/games',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        accessFilter: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        accessFilter: zod_1.z
            .union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(4)])
            .optional()
            .default(2),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Web_Responses_Games_GameResponseV2_,
    errors: [],
});
/**
 * @api GET https://games.roblox.com/v2/groups/:groupId/gamesV2
 * @summary Gets games created by the specified group. (gamesV2 path - removed from docs)
 * @param groupId The group Id.
 * @param accessFilter Filtering option via access level.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidGamesv2 = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/groups/:groupId/gamesV2',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: { style: 'simple' },
        accessFilter: { style: 'form', explode: true },
        limit: { style: 'form', explode: true },
        cursor: { style: 'form', explode: true },
        sortOrder: { style: 'form', explode: true },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        accessFilter: zod_1.z
            .union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(4)])
            .optional()
            .default(1),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Web_Responses_Games_GameResponseV2_,
    errors: [
        { status: 500, description: `0: Compliance Context service is unavailable.` },
        { status: 501, description: `1: Code path is not Implemented.` },
    ],
});
/**
 * @api GET https://games.roblox.com/v2/groups/:groupId/games
 * @summary Gets games created by the specified group.
 * @param groupId The group Id.
 * @param accessFilter Filtering option via access level.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidGames = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/groups/:groupId/games',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        accessFilter: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        accessFilter: zod_1.z
            .union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(4)])
            .optional()
            .default(1),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Web_Responses_Games_GameResponseV2_,
    errors: [
        {
            status: 403,
            description: `3: Not authorized.`,
        },
        {
            status: 500,
            description: `0: Compliance Context service is unavailable.`,
        },
    ],
});
