// rozod/lib/endpoints/gamejoinv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postTeamCreatePreemptive = exports.postTeamCreate = exports.postPlayWithUser = exports.getJoinTeamTest = exports.postJoinReservedGame = exports.postJoinPrivateGame = exports.postJoinPlayTogetherGame = exports.postJoinGameInstance = exports.postJoinGame = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_GameJoin_Api_GameJoinRequest = zod_1.z.object({
    isoContext: zod_1.z.string(),
    gameIdToExclude: zod_1.z.string().uuid(),
    eventId: zod_1.z.string(),
    gameJoinAttemptId: zod_1.z.string().uuid(),
    placeId: zod_1.z.number().int(),
    gamerTag: zod_1.z.string(),
    isPlayTogetherGame: zod_1.z.boolean(),
    browserTrackerId: zod_1.z.number().int(),
    isTeleport: zod_1.z.boolean(),
    isQueueAllowedOverride: zod_1.z.boolean(),
    isImmersiveAdsTeleport: zod_1.z.boolean(),
    channelName: zod_1.z.string(),
    joinOrigin: zod_1.z.string(),
    partyId: zod_1.z.string().uuid(),
});
const Roblox_Web_GameJoin_StatusData_CreatorExperienceBanData = zod_1.z.object({
    startTime: zod_1.z.string().datetime({ offset: true }),
    durationSeconds: zod_1.z.number().int(),
    displayReason: zod_1.z.string(),
    displayReasonTextFilterStatus: zod_1.z.number().int(),
    isInherited: zod_1.z.boolean(),
});
const Roblox_Web_GameJoin_StatusData = zod_1.z.object({
    creatorExperienceBan: Roblox_Web_GameJoin_StatusData_CreatorExperienceBanData,
});
const Roblox_Web_GameLaunch_ConnectionFlow_ServerConnection = zod_1.z.object({
    Address: zod_1.z.string(),
    Port: zod_1.z.number().int(),
});
const Roblox_Web_GameLaunch_ConnectionFlow_UdmuxEndpoint = zod_1.z.object({
    Address: zod_1.z.string(),
    Port: zod_1.z.number().int(),
});
const Roblox_Web_GameLaunch_ConnectionFlow_JoinInformation = zod_1.z.object({
    ClientPort: zod_1.z.number().int(),
    MachineAddress: zod_1.z.string(),
    ServerPort: zod_1.z.number().int(),
    ServerConnections: zod_1.z.array(Roblox_Web_GameLaunch_ConnectionFlow_ServerConnection),
    UdmuxEndpoints: zod_1.z.array(Roblox_Web_GameLaunch_ConnectionFlow_UdmuxEndpoint),
    DirectServerReturn: zod_1.z.boolean(),
    TokenGenAlgorithm: zod_1.z.number().int(),
    PepperId: zod_1.z.number().int(),
    TokenValue: zod_1.z.string(),
    PingUrl: zod_1.z.string(),
    PingInterval: zod_1.z.number().int(),
    UserName: zod_1.z.string(),
    DisplayName: zod_1.z.string(),
    HasVerifiedBadge: zod_1.z.boolean(),
    SeleniumTestMode: zod_1.z.boolean(),
    UserId: zod_1.z.number().int(),
    RobloxLocale: zod_1.z.string(),
    GameLocale: zod_1.z.string(),
    SuperSafeChat: zod_1.z.boolean(),
    FlexibleChatEnabled: zod_1.z.boolean(),
    CharacterAppearance: zod_1.z.string(),
    ClientTicket: zod_1.z.string(),
    GameId: zod_1.z.string(),
    PlaceId: zod_1.z.number().int(),
    BaseUrl: zod_1.z.string(),
    ChatStyle: zod_1.z.string(),
    CreatorId: zod_1.z.number().int(),
    CreatorTypeEnum: zod_1.z.string(),
    MembershipType: zod_1.z.string(),
    AccountAge: zod_1.z.number().int(),
    CookieStoreFirstTimePlayKey: zod_1.z.string(),
    CookieStoreFiveMinutePlayKey: zod_1.z.string(),
    CookieStoreEnabled: zod_1.z.boolean(),
    IsUnknownOrUnder13: zod_1.z.boolean(),
    GameChatType: zod_1.z.string(),
    WhoCanWhisperChatWithMeInExperiences: zod_1.z.string(),
    SessionId: zod_1.z.string(),
    AnalyticsSessionId: zod_1.z.string(),
    DataCenterId: zod_1.z.number().int(),
    UniverseId: zod_1.z.number().int(),
    FollowUserId: zod_1.z.number().int(),
    characterAppearanceId: zod_1.z.number().int(),
    CountryCode: zod_1.z.string(),
    AlternateName: zod_1.z.string(),
    RandomSeed1: zod_1.z.string().regex(/^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/),
    ClientPublicKeyData: zod_1.z.string(),
    RccVersion: zod_1.z.string(),
    ChannelName: zod_1.z.string(),
    VerifiedAMP: zod_1.z.number().int(),
    PrivateServerOwnerID: zod_1.z.number().int(),
    PrivateServerID: zod_1.z.string(),
    EventId: zod_1.z.string(),
    EphemeralEarlyPubKey: zod_1.z.string(),
    PartyId: zod_1.z.string(),
    ShowRobloxTranslations: zod_1.z.boolean(),
    MatchmakingAttributes: zod_1.z.string(),
    TranslationDisplayMode: zod_1.z.string(),
});
const Roblox_GameJoin_Api_GameJoinResponse = zod_1.z.object({
    jobId: zod_1.z.string(),
    status: zod_1.z.number().int(),
    statusData: Roblox_Web_GameJoin_StatusData,
    joinScriptUrl: zod_1.z.string(),
    authenticationUrl: zod_1.z.string(),
    authenticationTicket: zod_1.z.string(),
    message: zod_1.z.string(),
    joinScript: Roblox_Web_GameLaunch_ConnectionFlow_JoinInformation,
    queuePosition: zod_1.z.number().int(),
});
const Roblox_GameJoin_Api_JoinGameInstanceRequest = zod_1.z.object({
    gameId: zod_1.z.string().uuid(),
    cId: zod_1.z.string(),
    gameJoinAttemptId: zod_1.z.string().uuid(),
    placeId: zod_1.z.number().int(),
    gamerTag: zod_1.z.string(),
    isPlayTogetherGame: zod_1.z.boolean(),
    browserTrackerId: zod_1.z.number().int(),
    isTeleport: zod_1.z.boolean(),
    isQueueAllowedOverride: zod_1.z.boolean(),
    isImmersiveAdsTeleport: zod_1.z.boolean(),
    channelName: zod_1.z.string(),
    joinOrigin: zod_1.z.string(),
    partyId: zod_1.z.string().uuid(),
});
const Roblox_GameJoin_Api_JoinPlayTogetherGameRequest = zod_1.z.object({
    conversationId: zod_1.z.number().int(),
    gameJoinAttemptId: zod_1.z.string().uuid(),
    placeId: zod_1.z.number().int(),
    gamerTag: zod_1.z.string(),
    isPlayTogetherGame: zod_1.z.boolean(),
    browserTrackerId: zod_1.z.number().int(),
    isTeleport: zod_1.z.boolean(),
    isQueueAllowedOverride: zod_1.z.boolean(),
    isImmersiveAdsTeleport: zod_1.z.boolean(),
    channelName: zod_1.z.string(),
    joinOrigin: zod_1.z.string(),
    partyId: zod_1.z.string().uuid(),
});
const Roblox_GameJoin_Api_JoinPrivateGameRequest = zod_1.z.object({
    accessCode: zod_1.z.string(),
    linkCode: zod_1.z.string(),
    gameJoinAttemptId: zod_1.z.string().uuid(),
    placeId: zod_1.z.number().int(),
    gamerTag: zod_1.z.string(),
    isPlayTogetherGame: zod_1.z.boolean(),
    browserTrackerId: zod_1.z.number().int(),
    isTeleport: zod_1.z.boolean(),
    isQueueAllowedOverride: zod_1.z.boolean(),
    isImmersiveAdsTeleport: zod_1.z.boolean(),
    channelName: zod_1.z.string(),
    joinOrigin: zod_1.z.string(),
    partyId: zod_1.z.string().uuid(),
});
const Roblox_GameJoin_Api_JoinReservedGameRequest = zod_1.z.object({
    accessCode: zod_1.z.string(),
    cId: zod_1.z.string(),
    gameJoinAttemptId: zod_1.z.string().uuid(),
    placeId: zod_1.z.number().int(),
    gamerTag: zod_1.z.string(),
    isPlayTogetherGame: zod_1.z.boolean(),
    browserTrackerId: zod_1.z.number().int(),
    isTeleport: zod_1.z.boolean(),
    isQueueAllowedOverride: zod_1.z.boolean(),
    isImmersiveAdsTeleport: zod_1.z.boolean(),
    channelName: zod_1.z.string(),
    joinOrigin: zod_1.z.string(),
    partyId: zod_1.z.string().uuid(),
});
const Roblox_GameJoin_Api_PlayWithUserRequest = zod_1.z.object({
    userIdToFollow: zod_1.z.number().int(),
    gameJoinAttemptId: zod_1.z.string().uuid(),
    placeId: zod_1.z.number().int(),
    gamerTag: zod_1.z.string(),
    isPlayTogetherGame: zod_1.z.boolean(),
    browserTrackerId: zod_1.z.number().int(),
    isTeleport: zod_1.z.boolean(),
    isQueueAllowedOverride: zod_1.z.boolean(),
    isImmersiveAdsTeleport: zod_1.z.boolean(),
    channelName: zod_1.z.string(),
    joinOrigin: zod_1.z.string(),
    partyId: zod_1.z.string().uuid(),
});
const Roblox_GameJoin_Api_TeamCreateRequest = zod_1.z.object({
    gameJoinAttemptId: zod_1.z.string().uuid(),
    placeId: zod_1.z.number().int(),
    gamerTag: zod_1.z.string(),
    isPlayTogetherGame: zod_1.z.boolean(),
    browserTrackerId: zod_1.z.number().int(),
    isTeleport: zod_1.z.boolean(),
    isQueueAllowedOverride: zod_1.z.boolean(),
    isImmersiveAdsTeleport: zod_1.z.boolean(),
    channelName: zod_1.z.string(),
    joinOrigin: zod_1.z.string(),
    partyId: zod_1.z.string().uuid(),
});
const Roblox_GameJoin_Api_TeamCreateResponse = zod_1.z.object({
    status: zod_1.z.number().int(),
    message: zod_1.z.string(),
    settings: zod_1.z.object({}),
});
/**
 * @api POST https://gamejoin.roblox.com/v1/join-game
 * @summary Endpoint to join a game
 * @param body The Roblox.GameJoin.Api.GameJoinRequest in JSON format
 */
exports.postJoinGame = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/join-game',
    baseUrl: 'https://gamejoin.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_GameJoin_Api_GameJoinRequest,
    response: Roblox_GameJoin_Api_GameJoinResponse,
    errors: [
        {
            status: 400,
            description: `-1: An unknown error occurred.`,
        },
        {
            status: 429,
            description: `3: Too many requests, please slow down.`,
        },
    ],
});
/**
 * @api POST https://gamejoin.roblox.com/v1/join-game-instance
 * @summary Endpoint to join a particular game instance
 * @param body The Roblox.GameJoin.Api.JoinGameInstanceRequest in JSON format
 */
exports.postJoinGameInstance = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/join-game-instance',
    baseUrl: 'https://gamejoin.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_GameJoin_Api_JoinGameInstanceRequest,
    response: Roblox_GameJoin_Api_GameJoinResponse,
    errors: [
        {
            status: 400,
            description: `-1: An unknown error occurred.`,
        },
        {
            status: 429,
            description: `3: Too many requests, please slow down.`,
        },
    ],
});
/**
 * @api POST https://gamejoin.roblox.com/v1/join-play-together-game
 * @summary Endpoint to join play together game. DEPRECATED
 * @param body The Roblox.GameJoin.Api.JoinPlayTogetherGameRequest in JSON format
 */
exports.postJoinPlayTogetherGame = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/join-play-together-game',
    baseUrl: 'https://gamejoin.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_GameJoin_Api_JoinPlayTogetherGameRequest,
    response: Roblox_GameJoin_Api_GameJoinResponse,
    errors: [
        {
            status: 400,
            description: `-1: An unknown error occurred.`,
        },
        {
            status: 429,
            description: `3: Too many requests, please slow down.`,
        },
    ],
});
/**
 * @api POST https://gamejoin.roblox.com/v1/join-private-game
 * @summary Endpoint to join a private game
 * @param body The Roblox.GameJoin.Api.JoinPrivateGameRequest in JSON format
 */
exports.postJoinPrivateGame = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/join-private-game',
    baseUrl: 'https://gamejoin.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_GameJoin_Api_JoinPrivateGameRequest,
    response: Roblox_GameJoin_Api_GameJoinResponse,
    errors: [
        {
            status: 400,
            description: `-1: An unknown error occurred.`,
        },
        {
            status: 429,
            description: `3: Too many requests, please slow down.`,
        },
    ],
});
/**
 * @api POST https://gamejoin.roblox.com/v1/join-reserved-game
 * @summary Endpoint to join a reserved game
 * @param body The Roblox.GameJoin.Api.JoinReservedGameRequest in JSON format
 */
exports.postJoinReservedGame = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/join-reserved-game',
    baseUrl: 'https://gamejoin.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_GameJoin_Api_JoinReservedGameRequest,
    response: Roblox_GameJoin_Api_GameJoinResponse,
    errors: [
        {
            status: 400,
            description: `-1: An unknown error occurred.`,
        },
        {
            status: 429,
            description: `3: Too many requests, please slow down.`,
        },
    ],
});
/**
 * @api GET https://gamejoin.roblox.com/v1/join-team-test
 * @summary Endpoint to join a team test.
 * @param PlaceId The Target Place Id for Team Test.
 * @param PlaceVersion The Target Place Version for Team Test.
 */
exports.getJoinTeamTest = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/join-team-test',
    baseUrl: 'https://gamejoin.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        PlaceId: {
            style: 'form',
            explode: true,
        },
        PlaceVersion: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        PlaceId: zod_1.z.number().int(),
        PlaceVersion: zod_1.z.number().int(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api POST https://gamejoin.roblox.com/v1/play-with-user
 * @summary Endpoint to play with user
 * @param body The Roblox.GameJoin.Api.PlayWithUserRequest in JSON format
 */
exports.postPlayWithUser = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/play-with-user',
    baseUrl: 'https://gamejoin.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_GameJoin_Api_PlayWithUserRequest,
    response: Roblox_GameJoin_Api_GameJoinResponse,
    errors: [
        {
            status: 400,
            description: `-1: An unknown error occurred.`,
        },
        {
            status: 429,
            description: `3: Too many requests, please slow down.`,
        },
    ],
});
/**
 * @api POST https://gamejoin.roblox.com/v1/team-create
 * @summary Endpoint to join team create session
 * @param body The Roblox.GameJoin.Api.TeamCreateRequest in JSON format
 * @description Xsrf protection disabled because it will only be used on Roblox clients. Adding an extra
round-trip would impact client performance.
 */
exports.postTeamCreate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/team-create',
    baseUrl: 'https://gamejoin.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_GameJoin_Api_TeamCreateRequest,
    response: Roblox_GameJoin_Api_TeamCreateResponse,
    errors: [
        {
            status: 400,
            description: `-1: An unknown error occurred.`,
        },
        {
            status: 403,
            description: `1: Request is not authorized from specified origin.
2: User is invalid or does not exist.`,
        },
        {
            status: 429,
            description: `3: Too many requests, please slow down.`,
        },
    ],
});
/**
 * @api POST https://gamejoin.roblox.com/v1/team-create-preemptive
 * @summary Endpoint to preemptively start team create session.
 * @param body The Roblox.GameJoin.Api.TeamCreateRequest in JSON format
 * @description Since the request origin might not be Roblox software, this endpoint implements a different set of security rules.
More specifically, it requires Xsrf protection and disables user-agent check so that it can be called from browsers.
 */
exports.postTeamCreatePreemptive = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/team-create-preemptive',
    baseUrl: 'https://gamejoin.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_GameJoin_Api_TeamCreateRequest,
    response: Roblox_GameJoin_Api_TeamCreateResponse,
    errors: [
        {
            status: 400,
            description: `-1: An unknown error occurred.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: Request is not authorized from specified origin.
2: User is invalid or does not exist.`,
        },
        {
            status: 429,
            description: `3: Too many requests, please slow down.`,
        },
    ],
});
