// rozod/lib/endpoints/gamesv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postGamesVipServersUniverseid = exports.patchVipServersIdSubscription = exports.patchVipServersIdPermissions = exports.patchVipServersId = exports.getVipServersId = exports.getVipServerCanInviteUserid = exports.getPrivateServersMyPrivateServers = exports.getGamesVotes = exports.getGamesUniverseidVotesUser = exports.getGamesUniverseidVotes = exports.patchGamesUniverseidUserVotes = exports.getGamesUniverseidGamePasses = exports.getPrivateServersEnabledInUniverseUniverseid = exports.getGamesRecommendationsGameUniverseid = exports.getGamesMultigetPlayabilityStatus = exports.getGamesMultigetPlaceDetails = exports.getGamesGamesProductInfo = exports.getGamesUniverseidMedia = exports.getGamesUniverseidFavoritesCount = exports.postGamesUniverseidFavorites = exports.getGamesUniverseidFavorites = exports.getGamesPlaceidServersServertype = exports.getGamesPlaceidPrivateServers = exports.getGames = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Games_Api_Models_Response_GameCreator = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    type: zod_1.z.string(),
    isRNVAccount: zod_1.z.boolean(),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Games_Api_Models_Response_RefundPolicy = zod_1.z.object({
    policyText: zod_1.z.string(),
    learnMoreBaseUrl: zod_1.z.string(),
    locale: zod_1.z.string(),
    articleId: zod_1.z.string(),
});
const Roblox_Games_Api_Models_Response_GameDetailResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    rootPlaceId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    sourceName: zod_1.z.string(),
    sourceDescription: zod_1.z.string(),
    creator: Roblox_Games_Api_Models_Response_GameCreator,
    price: zod_1.z.number().int(),
    allowedGearGenres: zod_1.z.array(zod_1.z.string()),
    allowedGearCategories: zod_1.z.array(zod_1.z.string()),
    isGenreEnforced: zod_1.z.boolean(),
    copyingAllowed: zod_1.z.boolean(),
    playing: zod_1.z.number().int(),
    visits: zod_1.z.number().int(),
    maxPlayers: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
    studioAccessToApisAllowed: zod_1.z.boolean(),
    createVipServersAllowed: zod_1.z.boolean(),
    universeAvatarType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
    genre: zod_1.z.string(),
    genre_l1: zod_1.z.string(),
    genre_l2: zod_1.z.string(),
    untranslated_genre_l1: zod_1.z.string(),
    isAllGenre: zod_1.z.boolean(),
    isFavoritedByUser: zod_1.z.boolean(),
    favoritedCount: zod_1.z.number().int(),
    licenseDescription: zod_1.z.string(),
    refundLink: zod_1.z.string(),
    localizedFiatPrice: zod_1.z.string(),
    refundPolicy: Roblox_Games_Api_Models_Response_RefundPolicy,
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Games_Api_Models_Response_GameDetailResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Games_Api_Models_Response_GameDetailResponse),
});
const Roblox_Games_Api_GameServerPlayerResponse = zod_1.z.object({
    playerToken: zod_1.z.string(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Games_Api_Models_Response_VerifiedBadgeUserResponse = zod_1.z.object({
    hasVerifiedBadge: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Web_Responses_Games_GameServerResponse = zod_1.z.object({
    id: zod_1.z.string().uuid(),
    maxPlayers: zod_1.z.number().int(),
    playing: zod_1.z.number().int(),
    playerTokens: zod_1.z.array(zod_1.z.string()),
    players: zod_1.z.array(Roblox_Games_Api_GameServerPlayerResponse),
    fps: zod_1.z.number(),
    ping: zod_1.z.number().int(),
    name: zod_1.z.string(),
    vipServerId: zod_1.z.number().int(),
    accessCode: zod_1.z.string().uuid(),
    owner: Roblox_Games_Api_Models_Response_VerifiedBadgeUserResponse,
});
const Roblox_Games_Api_Models_Response_GetPrivateServerListResponse = zod_1.z.object({
    gameJoinRestricted: zod_1.z.boolean(),
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Web_Responses_Games_GameServerResponse),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Web_Responses_Games_GameServerResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Web_Responses_Games_GameServerResponse),
});
const Roblox_Games_Api_Models_Response_GameFavoriteResponse = zod_1.z.object({
    isFavorited: zod_1.z.boolean(),
});
const Roblox_Games_Api_Models_Request_GameFavoritesRequest = zod_1.z.object({
    isFavorited: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Games_Api_Models_Response_GameFavoritesCountResponse = zod_1.z.object({
    favoritesCount: zod_1.z.number().int(),
});
const Roblox_Games_Api_Models_Response_GameMediaItem = zod_1.z.object({
    id: zod_1.z.number().int(),
    assetTypeId: zod_1.z.number().int(),
    assetType: zod_1.z.string(),
    imageId: zod_1.z.number().int(),
    videoHash: zod_1.z.string(),
    videoTitle: zod_1.z.string(),
    approved: zod_1.z.boolean(),
    altText: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Games_Api_Models_Response_GameMediaItem_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Games_Api_Models_Response_GameMediaItem),
});
const Roblox_Games_Api_Models_Response_PurchaseData = zod_1.z.object({
    localizedFiatPrice: zod_1.z.string(),
    basePriceId: zod_1.z.string(),
});
const Roblox_Games_Api_Models_Response_GameProductResponse = zod_1.z.object({
    universeId: zod_1.z.number().int(),
    isForSale: zod_1.z.boolean(),
    productId: zod_1.z.number().int(),
    price: zod_1.z.number().int(),
    sellerId: zod_1.z.number().int(),
    fiatPurchaseData: Roblox_Games_Api_Models_Response_PurchaseData,
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Games_Api_Models_Response_GameProductResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Games_Api_Models_Response_GameProductResponse),
});
const Roblox_Games_Api_Models_Response_PlaceDetails = zod_1.z.object({
    placeId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    sourceName: zod_1.z.string(),
    sourceDescription: zod_1.z.string(),
    url: zod_1.z.string(),
    builder: zod_1.z.string(),
    builderId: zod_1.z.number().int(),
    hasVerifiedBadge: zod_1.z.boolean(),
    isPlayable: zod_1.z.boolean(),
    reasonProhibited: zod_1.z.string(),
    universeId: zod_1.z.number().int(),
    universeRootPlaceId: zod_1.z.number().int(),
    price: zod_1.z.number().int(),
    imageToken: zod_1.z.string(),
    fiatPurchaseData: Roblox_Games_Api_Models_Response_PurchaseData,
});
const Roblox_Games_Api_Models_Response_PlayableUxTreatmentData = zod_1.z.object({
    titleText: zod_1.z.string(),
    bodyText: zod_1.z.string(),
    primaryActionText: zod_1.z.string(),
    secondaryActionText: zod_1.z.string(),
});
const Roblox_Games_Api_Models_Response_PlayableUxTreatment = zod_1.z.object({
    treatment: zod_1.z.string(),
    data: Roblox_Games_Api_Models_Response_PlayableUxTreatmentData,
});
const Roblox_Games_Api_Models_Response_PlayabilityStatusResponse = zod_1.z.object({
    playabilityStatus: zod_1.z.enum([
        'UnplayableOtherReason',
        'Playable',
        'GuestProhibited',
        'GameUnapproved',
        'IncorrectConfiguration',
        'UniverseRootPlaceIsPrivate',
        'InsufficientPermissionFriendsOnly',
        'InsufficientPermissionGroupOnly',
        'DeviceRestricted',
        'UnderReview',
        'PurchaseRequired',
        'AccountRestricted',
        'TemporarilyUnavailable',
        'PlaceHasNoPublishedVersion',
        'ComplianceBlocked',
        'ContextualPlayabilityRegionalAvailability',
        'ContextualPlayabilityRegionalCompliance',
        'ContextualPlayabilityAgeRecommendationParentalControls',
        'ContextualPlayabilityExperienceBlockedParentalControls',
        'ContextualPlayabilityAgeGated',
        'ContextualPlayabilityUnverifiedSeventeenPlusUser',
        'FiatPurchaseRequired',
        'FiatPurchaseDeviceRestricted',
        'ContextualPlayabilityUnrated',
        'ContextualPlayabilityAgeGatedByDescriptor',
        'ContextualPlayabilityGeneral',
    ]),
    isPlayable: zod_1.z.boolean(),
    universeId: zod_1.z.number().int(),
    unplayableDisplayText: zod_1.z.string(),
    playableUxTreatment: Roblox_Games_Api_Models_Response_PlayableUxTreatment,
});
const Roblox_Games_Api_Models_Response_GameResponseModel = zod_1.z.object({
    creatorId: zod_1.z.number().int(),
    creatorName: zod_1.z.string(),
    creatorType: zod_1.z.string(),
    creatorHasVerifiedBadge: zod_1.z.boolean(),
    totalUpVotes: zod_1.z.number().int(),
    totalDownVotes: zod_1.z.number().int(),
    universeId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    placeId: zod_1.z.number().int(),
    playerCount: zod_1.z.number().int(),
    imageToken: zod_1.z.string(),
    isSponsored: zod_1.z.boolean(),
    nativeAdData: zod_1.z.string(),
    isShowSponsoredLabel: zod_1.z.boolean(),
    price: zod_1.z.number().int(),
    analyticsIdentifier: zod_1.z.string(),
    gameDescription: zod_1.z.string(),
    genre: zod_1.z.string(),
    minimumAge: zod_1.z.number().int(),
    ageRecommendationDisplayName: zod_1.z.string(),
    canonicalUrlPath: zod_1.z.string(),
});
const Roblox_Games_Api_Models_Response_GameRecommendationsResponse = zod_1.z.object({
    games: zod_1.z.array(Roblox_Games_Api_Models_Response_GameResponseModel),
    nextPaginationKey: zod_1.z.string(),
});
const Roblox_Games_Api_Models_Response_PrivateServersEnabledInUniverseResponse = zod_1.z.object({
    privateServersEnabled: zod_1.z.boolean(),
});
/**
 * @api GET https://games.roblox.com/v1/games
 * @summary Gets a list of games' detail
 * @param universeIds A list of universe Ids. Cannot exceed a maximum of 50 IDs.
 * @param languageCode The HTML language code [optional].
 */
exports.getGames = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeIds: {
            style: 'form',
        },
        languageCode: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        universeIds: zod_1.z.array(zod_1.z.number()),
        languageCode: zod_1.z.string().optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Games_Api_Models_Response_GameDetailResponse_,
    errors: [
        {
            status: 400,
            description: `8: The universe IDs specified are invalid.
9: Too many universe IDs were requested.`,
        },
        {
            status: 429,
            description: `4: Too many requests have been made.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/games/:placeId/private-servers
 * @summary Get list of private servers user can access for given game id.
 * @param placeId The Id of the place we are geting the private server list for.
 * @param excludeFriendServers
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGamesPlaceidPrivateServers = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/:placeId/private-servers',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        placeId: {
            style: 'simple',
        },
        excludeFriendServers: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        placeId: zod_1.z.number().int(),
        excludeFriendServers: zod_1.z.boolean().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Games_Api_Models_Response_GetPrivateServerListResponse,
    errors: [
        {
            status: 400,
            description: `1: The place is invalid.
7: Guest users are not allowed.`,
        },
        {
            status: 404,
            description: `1: The place is invalid.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/games/:placeId/servers/:serverType
 * @summary Get the game server list
 * @param placeId The Id of the place we are geting the server list for.
 * @param serverType The type of the server we geting the server list for.
 * @param sortOrder The sort order of the servers.
 * @param excludeFullGames Exclude full servers.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 */
exports.getGamesPlaceidServersServertype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/:placeId/servers/:serverType',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        placeId: {
            style: 'simple',
        },
        serverType: {
            style: 'simple',
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
        excludeFullGames: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        placeId: zod_1.z.number().int(),
        serverType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1)]),
        sortOrder: zod_1.z
            .union([zod_1.z.literal(1), zod_1.z.literal(2)])
            .optional()
            .default(2),
        excludeFullGames: zod_1.z.boolean().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Web_Responses_Games_GameServerResponse_,
    errors: [
        {
            status: 400,
            description: `1: The place is invalid.
6: The server type is invalid. For fetching private servers, please use https://games.roblox.com/v1/games/{placeId}/private-servers.
7: Guest users are not allowed.`,
        },
        {
            status: 404,
            description: `1: The place is invalid.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/games/:universeId/favorites
 * @summary Returns if a game was marked as favorite for the authenticated user
 * @param universeId The Id of the universe.
 */
exports.getGamesUniverseidFavorites = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/:universeId/favorites',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({ isFavorited: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `3: The universe&#x27;s root place is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 404,
            description: `2: The requested universe does not exist.`,
        },
    ],
});
/**
 * @api POST https://games.roblox.com/v1/games/:universeId/favorites
 * @summary Favors (or unfavors) a game for the authenticated user
 * @param body Request data.
 * @param universeId The Id of the universe.
 */
exports.postGamesUniverseidFavorites = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/games/:universeId/favorites',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ isFavorited: zod_1.z.boolean() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `3: The universe&#x27;s root place is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
11: You are not authorized to perform this action.`,
        },
        {
            status: 404,
            description: `2: The requested universe does not exist.`,
        },
        {
            status: 429,
            description: `4: Too many requests have been made.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/games/:universeId/favorites/count
 * @summary Get the favorites count of a specific game.
 * @param universeId The Id of the universe.
 */
exports.getGamesUniverseidFavoritesCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/:universeId/favorites/count',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({ favoritesCount: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `3: The universe&#x27;s root place is invalid.`,
        },
        {
            status: 404,
            description: `2: The requested universe does not exist.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/games/:universeId/media
 * @summary Get the game media data
 * @param universeId The id of the universe we get media data from.
 */
exports.getGamesUniverseidMedia = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/:universeId/media',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Games_Api_Models_Response_GameMediaItem_,
    errors: [
        {
            status: 400,
            description: `3: The universe&#x27;s root place is invalid.`,
        },
        {
            status: 404,
            description: `2: The requested universe does not exist.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/games/games-product-info
 * @summary Gets a list of games' product info, used to purchase a game
 * @param universeIds A list of universe Ids. Cannot exceed a maximum of 100 IDs.
 */
exports.getGamesGamesProductInfo = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/games-product-info',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeIds: {
            style: 'form',
        },
    },
    parameters: {
        universeIds: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Games_Api_Models_Response_GameProductResponse_,
    errors: [
        {
            status: 400,
            description: `8: The universe IDs specified are invalid.
9: Too many universe IDs were requested.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/games/multiget-place-details
 * @summary Get place details
 * @param placeIds List of placeId to uniquely Identify a place
 */
exports.getGamesMultigetPlaceDetails = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/multiget-place-details',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        placeIds: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        placeIds: zod_1.z.array(zod_1.z.number()),
    },
    response: zod_1.z.array(Roblox_Games_Api_Models_Response_PlaceDetails),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/games/multiget-playability-status
 * @summary Gets a list of universe playability statuses for the authenticated user
 * @param universeIds A list of universe Ids. Cannot exceed a maximum of 50 IDs.
 */
exports.getGamesMultigetPlayabilityStatus = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/multiget-playability-status',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeIds: {
            style: 'form',
        },
    },
    parameters: {
        universeIds: zod_1.z.array(zod_1.z.number()),
    },
    response: zod_1.z.array(Roblox_Games_Api_Models_Response_PlayabilityStatusResponse),
    errors: [
        {
            status: 400,
            description: `8: The universe IDs specified are invalid.
9: Too many universe IDs were requested.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/games/recommendations/game/:universeId
 * @summary Get games recommendations based on a given universe
 * @param universeId The universe to base recommendations on
 * @param PaginationKey The key of a page, which includes the start row index and all other necessary information to query the data.
This parameter is usually not needed for the first page.
 * @param MaxRows The requested number of rows.
 * @param IsTruncatedResultsEnabled Truncated Results
 */
exports.getGamesRecommendationsGameUniverseid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/recommendations/game/:universeId',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
        PaginationKey: {
            style: 'form',
            explode: true,
        },
        MaxRows: {
            style: 'form',
            explode: true,
        },
        IsTruncatedResultsEnabled: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        PaginationKey: zod_1.z.string(),
        MaxRows: zod_1.z.number().int(),
        IsTruncatedResultsEnabled: zod_1.z.boolean(),
    },
    response: Roblox_Games_Api_Models_Response_GameRecommendationsResponse,
    errors: [
        {
            status: 400,
            description: `1: The pagination key is invalid.`,
        },
        {
            status: 404,
            description: `2: The requested universe does not exist.`,
        },
    ],
});
/**
 * @api GET https://games.roblox.com/v1/private-servers/enabled-in-universe/:universeId
 * @summary Checks if the private servers are enabled in the specified universe.
 * @param universeId
 */
exports.getPrivateServersEnabledInUniverseUniverseid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/private-servers/enabled-in-universe/:universeId',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({ privateServersEnabled: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `8: The universe IDs specified are invalid.`,
        },
    ],
});
// Patched endpoints removed from Roblox API docs (v6.1.0 compat)
const Patch_GamePassResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
    productId: zod_1.z.number().int(),
    price: zod_1.z.number().int(),
    sellerName: zod_1.z.string(),
    sellerId: zod_1.z.number().int(),
    isOwned: zod_1.z.boolean(),
});
const Patch_ApiPageResponse_GamePassResponse = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Patch_GamePassResponse),
});
const Patch_GameVoteResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    upVotes: zod_1.z.number().int(),
    downVotes: zod_1.z.number().int(),
});
const Patch_UserGameVoteResponse = zod_1.z.object({
    canVote: zod_1.z.boolean(),
    userVote: zod_1.z.boolean(),
    reasonForNotVoteable: zod_1.z.string(),
});
const Patch_ApiArrayResponse_GameVoteResponse = zod_1.z.object({
    data: zod_1.z.array(Patch_GameVoteResponse),
});
const Patch_MyPrivateServersData = zod_1.z.object({
    active: zod_1.z.boolean(),
    universeId: zod_1.z.number().int(),
    placeId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    ownerId: zod_1.z.number().int(),
    ownerName: zod_1.z.string(),
    priceInRobux: zod_1.z.number().int(),
    privateServerId: zod_1.z.number().int(),
    expirationDate: zod_1.z.string().datetime({ offset: true }),
    willRenew: zod_1.z.boolean(),
    universeName: zod_1.z.string(),
});
const Patch_MyPrivateServersResponse = zod_1.z.object({
    nextPageCursor: zod_1.z.string(),
    previousPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Patch_MyPrivateServersData),
});
const Patch_VipServerCanInviteResponse = zod_1.z.object({
    canInvite: zod_1.z.boolean(),
    doesOwnerPrivacyRestrictJoins: zod_1.z.boolean(),
    inviteResponseType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
});
const Patch_PlaceResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
});
const Patch_GameResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    rootPlace: Patch_PlaceResponse,
});
const Patch_VipServerSubscriptionResponse = zod_1.z.object({
    active: zod_1.z.boolean(),
    expired: zod_1.z.boolean(),
    expirationDate: zod_1.z.string().datetime({ offset: true }),
    price: zod_1.z.number().int(),
    canRenew: zod_1.z.boolean(),
    hasInsufficientFunds: zod_1.z.boolean(),
    hasRecurringProfile: zod_1.z.boolean(),
    hasPriceChanged: zod_1.z.boolean(),
});
const Patch_SkinnyUserResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Patch_VipServerPermissionsResponse = zod_1.z.object({
    clanAllowed: zod_1.z.boolean(),
    enemyClanId: zod_1.z.number().int(),
    friendsAllowed: zod_1.z.boolean(),
    users: zod_1.z.array(Patch_SkinnyUserResponse),
});
const Patch_VipServerVoiceSettingsResponse = zod_1.z.object({
    enabled: zod_1.z.boolean(),
});
const Patch_VipServerResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    game: Patch_GameResponse,
    joinCode: zod_1.z.string(),
    active: zod_1.z.boolean(),
    subscription: Patch_VipServerSubscriptionResponse,
    permissions: Patch_VipServerPermissionsResponse,
    voiceSettings: Patch_VipServerVoiceSettingsResponse,
    link: zod_1.z.string(),
});
const Patch_VipServerUpdateRequest = zod_1.z.object({
    name: zod_1.z.string(),
    newJoinCode: zod_1.z.boolean(),
    active: zod_1.z.boolean(),
});
const Patch_CreateVipServersRequest = zod_1.z.object({
    name: zod_1.z.string(),
    expectedPrice: zod_1.z.number().int(),
    isPurchaseConfirmed: zod_1.z.boolean(),
});
const Patch_VipServerUpdatePermissionsRequest = zod_1.z.object({
    clanAllowed: zod_1.z.boolean(),
    enemyClanId: zod_1.z.number().int(),
    friendsAllowed: zod_1.z.boolean(),
    usersToAdd: zod_1.z.array(zod_1.z.number()),
    usersToRemove: zod_1.z.array(zod_1.z.number()),
});
const Patch_VipServerUpdateSubscriptionRequest = zod_1.z.object({
    active: zod_1.z.boolean(),
    price: zod_1.z.number().int(),
});
exports.getGamesUniverseidGamePasses = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/:universeId/game-passes',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: { style: 'simple' },
        limit: { style: 'form', explode: true },
        sortOrder: { style: 'form', explode: true },
        cursor: { style: 'form', explode: true },
    },
    parameters: {
        universeId: zod_1.z.number().int(),
        limit: zod_1.z.number().int(),
        sortOrder: zod_1.z
            .union([zod_1.z.literal(1), zod_1.z.literal(2)])
            .optional()
            .default(1),
        cursor: zod_1.z.string().optional(),
    },
    response: Patch_ApiPageResponse_GamePassResponse,
    errors: [
        { status: 400, description: `2: The universe's root place is invalid.` },
        { status: 404, description: `1: The requested universe does not exist.` },
    ],
});
exports.patchGamesUniverseidUserVotes = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/games/:universeId/user-votes',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { body: {}, universeId: { style: 'simple' } },
    parameters: { universeId: zod_1.z.number().int() },
    body: zod_1.z.object({ vote: zod_1.z.boolean() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: The universe's root place is invalid.\n3: The asset is not voteable.\n4: The requested vote is invalid.`,
        },
        { status: 401, description: `0: Authorization has been denied for this request.` },
        {
            status: 403,
            description: `0: Token Validation Failed\n6: The user needs to play the game before vote.\n7: The user needs additional permission to vote.`,
        },
        { status: 404, description: `1: The requested universe does not exist.` },
        {
            status: 429,
            description: `5: Too many attempts to vote. Please try again later.\n10: Internal service busy. Please try again later.`,
        },
        { status: 500, description: `0: An unknown error occurred.` },
    ],
});
exports.getGamesUniverseidVotes = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/:universeId/votes',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { universeId: { style: 'simple' } },
    parameters: { universeId: zod_1.z.number().int() },
    response: Patch_GameVoteResponse,
    errors: [
        { status: 400, description: `2: The universe's root place is invalid.\n3: The asset is not voteable.` },
        { status: 404, description: `1: The requested universe does not exist.` },
        { status: 429, description: `10: Internal service busy. Please try again later.` },
        { status: 500, description: `0: An unknown error occurred.` },
    ],
});
exports.getGamesUniverseidVotesUser = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/:universeId/votes/user',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { universeId: { style: 'simple' } },
    parameters: { universeId: zod_1.z.number().int() },
    response: Patch_UserGameVoteResponse,
    errors: [
        { status: 400, description: `2: The universe's root place is invalid.\n3: The asset is not voteable.` },
        { status: 401, description: `0: Authorization has been denied for this request.` },
        { status: 404, description: `1: The requested universe does not exist.` },
        { status: 429, description: `10: Internal service busy. Please try again later.` },
        { status: 500, description: `0: An unknown error occurred.` },
    ],
});
exports.getGamesVotes = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/games/votes',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { universeIds: { style: 'form' } },
    parameters: { universeIds: zod_1.z.array(zod_1.z.number()) },
    response: Patch_ApiArrayResponse_GameVoteResponse,
    errors: [
        {
            status: 400,
            description: `3: The asset is not voteable.\n8: No universe IDs were specified.\n9: Too many universe IDs were requested.`,
        },
        { status: 429, description: `10: Internal service busy. Please try again later.` },
        { status: 500, description: `0: An unknown error occurred.` },
    ],
});
exports.getPrivateServersMyPrivateServers = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/private-servers/my-private-servers',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        privateServersTab: { style: 'form', explode: true },
        itemsPerPage: { style: 'form', explode: true },
        cursor: { style: 'form', explode: true },
    },
    parameters: {
        privateServersTab: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1)]).optional(),
        itemsPerPage: zod_1.z.number().int().optional().default(25),
        cursor: zod_1.z.string().optional(),
    },
    response: Patch_MyPrivateServersResponse,
    errors: [
        { status: 400, description: `39: Invalid cursor provided in the request.` },
        { status: 401, description: `0: Authorization has been denied for this request.` },
    ],
});
exports.getVipServerCanInviteUserid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/vip-server/can-invite/:userId',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { userId: { style: 'simple' } },
    parameters: { userId: zod_1.z.number().int() },
    response: Patch_VipServerCanInviteResponse,
    errors: [
        { status: 401, description: `0: Authorization has been denied for this request.` },
        { status: 404, description: `19: The user is does not exist.` },
    ],
});
exports.getVipServersId = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/vip-servers/:id',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { id: { style: 'simple' } },
    parameters: { id: zod_1.z.number().int() },
    response: Patch_VipServerResponse,
    errors: [
        { status: 400, description: `8: The creator of this game has disabled private servers for this game.` },
        { status: 401, description: `0: Authorization has been denied for this request.` },
        { status: 403, description: `2: You are not the owner of this private server.` },
        {
            status: 404,
            description: `1: The private server is invalid or does not exist.\n4: The universe does not exist.`,
        },
    ],
});
exports.patchVipServersId = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/vip-servers/:id',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { body: {}, id: { style: 'simple' } },
    parameters: { id: zod_1.z.number().int() },
    body: Patch_VipServerUpdateRequest,
    response: Patch_VipServerResponse,
    errors: [
        {
            status: 400,
            description: `8: The creator of this game has disabled private servers for this game.\n10: The name of the private server is either empty or too long.`,
        },
        { status: 401, description: `0: Authorization has been denied for this request.` },
        {
            status: 403,
            description: `0: Token Validation Failed\n2: You are not the owner of this private server.\n11: You cannot activate a cancelled private server.\n12: The game must not be Friends Only to allow private servers.\n13: Join Link can be generated only for active private servers.`,
        },
        {
            status: 404,
            description: `1: The private server is invalid or does not exist.\n4: The universe does not exist.`,
        },
        { status: 429, description: `3: Please wait a few minutes before configuring your private server again.` },
    ],
});
exports.patchVipServersIdPermissions = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/vip-servers/:id/permissions',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { body: {}, id: { style: 'simple' } },
    parameters: { id: zod_1.z.number().int() },
    body: Patch_VipServerUpdatePermissionsRequest,
    response: Patch_VipServerPermissionsResponse,
    errors: [
        {
            status: 400,
            description: `6: You cannot add so many players to the private server's invite list.\n8: The creator of this game has disabled private servers for this game.\n12: The game must not be Friends Only to allow private servers.\n28: You may only add or remove valid players to your private server's invite list.\n29: You may only add or remove players when your private server is active.`,
        },
        { status: 401, description: `0: Authorization has been denied for this request.` },
        { status: 403, description: `0: Token Validation Failed\n2: You are not the owner of this private server.` },
        {
            status: 404,
            description: `1: The private server is invalid or does not exist.\n4: The universe does not exist.`,
        },
    ],
});
exports.patchVipServersIdSubscription = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/vip-servers/:id/subscription',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { body: {}, id: { style: 'simple' } },
    parameters: { id: zod_1.z.number().int() },
    body: Patch_VipServerUpdateSubscriptionRequest,
    response: Patch_VipServerSubscriptionResponse,
    errors: [
        {
            status: 400,
            description: `8: The creator of this game has disabled private servers for this game.\n32: You do not have enough funds available to renew the subscription for this private server.`,
        },
        { status: 401, description: `0: Authorization has been denied for this request.` },
        {
            status: 403,
            description: `0: Token Validation Failed\n2: You are not the owner of this private server.\n21: You may not configure a cancelled private server. Please renew your subscription before configuring.`,
        },
        {
            status: 404,
            description: `1: The private server is invalid or does not exist.\n4: The universe does not exist.`,
        },
        { status: 429, description: `3: Please wait a few minutes before configuring your private server again.` },
    ],
});
exports.postGamesVipServersUniverseid = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/games/vip-servers/:universeId',
    baseUrl: 'https://games.roblox.com',
    requestFormat: 'json',
    serializationMethod: { body: {}, universeId: { style: 'simple' } },
    parameters: { universeId: zod_1.z.number().int() },
    body: Patch_CreateVipServersRequest,
    response: Roblox_Web_Responses_Games_GameServerResponse,
    errors: [
        {
            status: 400,
            description: `15: The price for purchasing this private server has changed. Please refresh the page and try again.`,
        },
        { status: 401, description: `0: Authorization has been denied for this request.` },
        { status: 403, description: `0: Token Validation Failed` },
        { status: 404, description: `4: The universe does not exist.` },
        {
            status: 500,
            description: `16: We are having a problem completing your purchase. Please try again in a few minutes.`,
        },
    ],
});
