// rozod/lib/endpoints/gameinternationalizationv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSupportedLanguagesGamesGameid = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_GameInternationalization_Api_Language = zod_1.z.object({
    name: zod_1.z.string(),
    nativeName: zod_1.z.string(),
    languageCode: zod_1.z.string(),
});
const Roblox_Localization_Client_LanguageFamily = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    nativeName: zod_1.z.string(),
    languageCode: zod_1.z.string(),
    isRightToLeft: zod_1.z.boolean(),
});
const Roblox_Localization_Client_SupportedLocale = zod_1.z.object({
    id: zod_1.z.number().int(),
    locale: zod_1.z.enum([
        'en_us',
        'es_es',
        'fr_fr',
        'id_id',
        'it_it',
        'ja_jp',
        'ko_kr',
        'ru_ru',
        'th_th',
        'tr_tr',
        'vi_vn',
        'pt_br',
        'de_de',
        'zh_cn',
        'zh_tw',
        'bg_bg',
        'bn_bd',
        'cs_cz',
        'da_dk',
        'el_gr',
        'et_ee',
        'fi_fi',
        'hi_in',
        'hr_hr',
        'hu_hu',
        'ka_ge',
        'kk_kz',
        'km_kh',
        'lt_lt',
        'lv_lv',
        'ms_my',
        'my_mm',
        'nb_no',
        'nl_nl',
        'fil_ph',
        'pl_pl',
        'ro_ro',
        'uk_ua',
        'si_lk',
        'sk_sk',
        'sl_sl',
        'sq_al',
        'bs_ba',
        'sr_rs',
        'sv_se',
        'zh_cjv',
        'ar_001',
        'en_gb',
        'pt_pt',
        'es_mx',
        'fr_ca',
    ]),
    localeCode: zod_1.z.string(),
    name: zod_1.z.string(),
    nativeName: zod_1.z.string(),
    language: Roblox_Localization_Client_LanguageFamily,
});
const Roblox_GameInternationalization_Api_LanguageWithLocales = zod_1.z.object({
    languageFamily: Roblox_GameInternationalization_Api_Language,
    childLocales: zod_1.z.array(Roblox_Localization_Client_SupportedLocale),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_GameInternationalization_Api_LanguageWithLocales_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_GameInternationalization_Api_LanguageWithLocales),
});
/**
 * @api GET https://gameinternationalization.roblox.com/v2/supported-languages/games/:gameId
 * @summary Get the supported languages for a game.
 * @param gameId The id of the game.
 */
exports.getSupportedLanguagesGamesGameid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/supported-languages/games/:gameId',
    baseUrl: 'https://gameinternationalization.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_GameInternationalization_Api_LanguageWithLocales_,
    errors: [
        {
            status: 400,
            description: `14: Invalid game id`,
        },
        {
            status: 503,
            description: `17: Feature is disabled`,
        },
    ],
});
