// rozod/lib/endpoints/groupsv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridGroupsRoles = exports.postGroupsGroupidWallPosts = exports.getGroupsGroupidWallPosts = exports.getGroupsGroupidRelationshipsGrouprelationshiptypeRequests = exports.getGroupsGroupidRelationshipsGrouprelationshiptype = exports.getGroupsGroupidExperiences = exports.getGroups = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Web_Responses_Groups_GroupOwnerType_ = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.literal('User'),
    name: zod_1.z.string(),
});
const Roblox_Web_Responses_Groups_GroupResponseV2 = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    owner: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Web_Responses_Groups_GroupOwnerType_,
    memberCount: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Groups_GroupResponseV2_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Web_Responses_Groups_GroupResponseV2),
});
const Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Core_CreatorType_ = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.enum(['User', 'Group']),
    name: zod_1.z.string(),
});
const Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Assets_AssetType_ = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.enum([
        'Image',
        'TShirt',
        'Audio',
        'Mesh',
        'Lua',
        'HTML',
        'Text',
        'Hat',
        'Place',
        'Model',
        'Shirt',
        'Pants',
        'Decal',
        'Avatar',
        'Head',
        'Face',
        'Gear',
        'Badge',
        'GroupEmblem',
        'Animation',
        'Arms',
        'Legs',
        'Torso',
        'RightArm',
        'LeftArm',
        'LeftLeg',
        'RightLeg',
        'Package',
        'YouTubeVideo',
        'GamePass',
        'App',
        'Code',
        'Plugin',
        'SolidModel',
        'MeshPart',
        'HairAccessory',
        'FaceAccessory',
        'NeckAccessory',
        'ShoulderAccessory',
        'FrontAccessory',
        'BackAccessory',
        'WaistAccessory',
        'ClimbAnimation',
        'DeathAnimation',
        'FallAnimation',
        'IdleAnimation',
        'JumpAnimation',
        'RunAnimation',
        'SwimAnimation',
        'WalkAnimation',
        'PoseAnimation',
        'LocalizationTableManifest',
        'LocalizationTableTranslation',
        'EmoteAnimation',
        'Video',
        'TexturePack',
        'TShirtAccessory',
        'ShirtAccessory',
        'PantsAccessory',
        'JacketAccessory',
        'SweaterAccessory',
        'ShortsAccessory',
        'LeftShoeAccessory',
        'RightShoeAccessory',
        'DressSkirtAccessory',
        'FontFamily',
        'FontFace',
        'MeshHiddenSurfaceRemoval',
        'EyebrowAccessory',
        'EyelashAccessory',
        'MoodAnimation',
        'DynamicHead',
        'CodeSnippet',
        'AdsVideo',
        'OtaUpdate',
        'Screenshot',
        'RuntimePropertySet',
        'StorePreviewVideo',
        'GamePreviewVideo',
        'CreatorExperienceConfig',
        'FaceMakeup',
        'LipMakeup',
        'EyeMakeup',
    ]),
    name: zod_1.z.string(),
});
const Roblox_Groups_Api_Models_Response_GroupExperienceResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    creator: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Core_CreatorType_,
    rootPlace: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Platform_Assets_AssetType_,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
    placeVisits: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_Models_Response_GroupExperienceResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_Models_Response_GroupExperienceResponse),
});
const Roblox_Groups_Api_Models_Response_UserModel = zod_1.z.object({
    buildersClubMembershipType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
    hasVerifiedBadge: zod_1.z.boolean(),
    userId: zod_1.z.number().int(),
    username: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Groups_Api_ShoutResponse = zod_1.z.object({
    body: zod_1.z.string(),
    poster: Roblox_Groups_Api_Models_Response_UserModel,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Groups_Api_GroupDetailResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    owner: Roblox_Groups_Api_Models_Response_UserModel,
    shout: Roblox_Groups_Api_ShoutResponse,
    memberCount: zod_1.z.number().int(),
    isBuildersClubOnly: zod_1.z.boolean(),
    publicEntryAllowed: zod_1.z.boolean(),
    isLocked: zod_1.z.boolean(),
    hasVerifiedBadge: zod_1.z.boolean(),
    hasSocialModules: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_GroupDetailResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_GroupDetailResponse),
});
const Roblox_Groups_Api_Models_Response_GroupRelationshipsV2Response = zod_1.z.object({
    groupId: zod_1.z.number().int(),
    relationshipType: zod_1.z.enum(['Allies', 'Enemies']),
    groupResponses: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_GroupDetailResponse_,
});
const Roblox_Groups_Api_GroupRoleResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    rank: zod_1.z.number().int(),
    memberCount: zod_1.z.number().int(),
    isBase: zod_1.z.boolean(),
});
const Roblox_Groups_Api_UserGroupRoleResponse = zod_1.z.object({
    user: Roblox_Groups_Api_Models_Response_UserModel,
    role: Roblox_Groups_Api_GroupRoleResponse,
});
const Roblox_Groups_Api_GroupWallPostV2Model = zod_1.z.object({
    id: zod_1.z.number().int(),
    poster: Roblox_Groups_Api_UserGroupRoleResponse,
    body: zod_1.z.string(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_GroupWallPostV2Model_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_GroupWallPostV2Model),
});
const Roblox_Groups_Api_CreateWallPostRequest = zod_1.z.object({
    body: zod_1.z.string(),
    captchaId: zod_1.z.string(),
    captchaToken: zod_1.z.string(),
    captchaProvider: zod_1.z.string(),
    challengeId: zod_1.z.string(),
});
const Roblox_Web_Responses_Groups_GroupBasicResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    memberCount: zod_1.z.number().int(),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Web_Responses_Groups_GroupRoleBasicResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    rank: zod_1.z.number().int(),
});
const Roblox_Groups_Api_GroupNotificationPreferenceData = zod_1.z.object({
    type: zod_1.z.enum([
        'AnnouncementCreatedNotification',
        'ForumPostCreatedNotification',
        'ForumCommentCreatedNotification',
        'ForumCommentReplyCreatedNotification',
        'ForumSubscriberNotification',
    ]),
    enabled: zod_1.z.boolean(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
});
const Roblox_Groups_Api_GroupMembershipResponse = zod_1.z.object({
    group: Roblox_Web_Responses_Groups_GroupBasicResponse,
    role: Roblox_Web_Responses_Groups_GroupRoleBasicResponse,
    isNotificationsEnabled: zod_1.z.boolean(),
    notificationPreferences: zod_1.z.array(Roblox_Groups_Api_GroupNotificationPreferenceData),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupMembershipResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Groups_Api_GroupMembershipResponse),
});
/**
 * @api GET https://groups.roblox.com/v2/groups
 * @summary Multi-get groups information by Ids.
 * @param groupIds The group Ids.
 * @description If a group comes back as null, it will not be returned in the response.
 */
exports.getGroups = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/groups',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupIds: {
            style: 'form',
        },
    },
    parameters: {
        groupIds: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Groups_GroupResponseV2_,
    errors: [
        {
            status: 400,
            description: `2: Too many ids in request.
3: Ids could not be parsed from request.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v2/groups/:groupId/experiences
 * @summary Gets experiences created by the specified group id.
 * @param groupId The group Id
 * @param accessFilter The access type of the experiences.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidExperiences = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/groups/:groupId/experiences',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        accessFilter: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        accessFilter: zod_1.z
            .union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(4)])
            .optional()
            .default(1),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_Models_Response_GroupExperienceResponse_,
    errors: [
        {
            status: 501,
            description: `47: Code path is not implemented.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v2/groups/:groupId/relationships/:groupRelationshipType
 * @summary Gets a group's relationships with cursor-based pagination.
 * @param groupId The group Id.
 * @param groupRelationshipType The group relationship type, enemies or allies.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidRelationshipsGrouprelationshiptype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/groups/:groupId/relationships/:groupRelationshipType',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Groups_Api_Models_Response_GroupRelationshipsV2Response,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
4: Group relationship type or request type is invalid.
8: Invalid or missing pagination parameters`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v2/groups/:groupId/relationships/:groupRelationshipType/requests
 * @summary Gets a group's relationship requests with cursor-based pagination.
 * @param groupId The group Id.
 * @param groupRelationshipType The group relationship type of the request, only allies are supported.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidRelationshipsGrouprelationshiptypeRequests = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/groups/:groupId/relationships/:groupRelationshipType/requests',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Groups_Api_Models_Response_GroupRelationshipsV2Response,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
4: Group relationship type or request type is invalid.
8: Invalid or missing pagination parameters`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `5: You don&#x27;t have permission to manage this group&#x27;s relationships.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v2/groups/:groupId/wall/posts
 * @summary Gets a list of group wall posts.
 * @param groupId The group id.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder Sorted by group wall post Id
 */
exports.getGroupsGroupidWallPosts = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/groups/:groupId/wall/posts',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_GroupWallPostV2Model_,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 403,
            description: `2: You do not have permission to access this group wall.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v2/groups/:groupId/wall/posts
 * @summary Creates a post on a group wall
 * @param body The Roblox.Groups.Api.CreateWallPostRequest.
 * @param groupId The group id.
 */
exports.postGroupsGroupidWallPosts = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/groups/:groupId/wall/posts',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_CreateWallPostRequest,
    response: Roblox_Groups_Api_GroupWallPostV2Model,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
5: Your post was empty, white space, or more than 500 characters.
9: The provided content was moderated.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to access this group wall.`,
        },
        {
            status: 429,
            description: `4: You are posting too fast, please try again in a few minutes.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v2/users/:userId/groups/roles
 * @summary Gets a list of all group roles for groups the specified user is in.
 * @param userId The user id.
 * @param includeLocked
 * @param includeNotificationPreferences
 * @param discoveryType
 */
exports.getUsersUseridGroupsRoles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/users/:userId/groups/roles',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        includeLocked: {
            style: 'form',
            explode: true,
        },
        includeNotificationPreferences: {
            style: 'form',
            explode: true,
        },
        discoveryType: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        includeLocked: zod_1.z.boolean(),
        includeNotificationPreferences: zod_1.z.boolean(),
        discoveryType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1)]),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupMembershipResponse_,
    errors: [
        {
            status: 400,
            description: `3: The user is invalid or does not exist.`,
        },
    ],
});
