// rozod/lib/endpoints/groupsv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getGroupsGroupidSettings = exports.postGroupsGroupidRolesetsCreate = exports.patchGroupsGroupidRolesetsRolesetid = exports.deleteGroupsGroupidRolesetsRolesetid = exports.getGroupsGroupidRolesPermissions = exports.getGroupsGroupidRolesGuestPermissions = exports.getGroupsGroupidRolesRolesetidUsers = exports.patchGroupsGroupidRolesRolesetidPermissions = exports.getGroupsGroupidRolesRolesetidPermissions = exports.getGroupsGroupidRoles = exports.deleteGroupsGroupidRelationshipsGrouprelationshiptypeRequestsRelatedgroupid = exports.postGroupsGroupidRelationshipsGrouprelationshiptypeRequestsRelatedgroupid = exports.deleteGroupsGroupidRelationshipsGrouprelationshiptypeRequests = exports.postGroupsGroupidRelationshipsGrouprelationshiptypeRequests = exports.getGroupsGroupidRelationshipsGrouprelationshiptypeRequests = exports.deleteGroupsGroupidRelationshipsGrouprelationshiptypeRelatedgroupid = exports.postGroupsGroupidRelationshipsGrouprelationshiptypeRelatedgroupid = exports.getGroupsGroupidRelationshipsGrouprelationshiptype = exports.postGroupsGroupidPayoutsRecurring = exports.postGroupsGroupidPayouts = exports.getGroupsGroupidPayouts = exports.getGroupsGroupidPayoutRestriction = exports.patchGroupsGroupidNotificationPreference = exports.getGroupsGroupidNameHistory = exports.patchGroupsGroupidName = exports.getGroupsGroupidMembership = exports.deleteGroupsGroupidJoinRequestsUsersUserid = exports.postGroupsGroupidJoinRequestsUsersUserid = exports.getGroupsGroupidJoinRequestsUsersUserid = exports.deleteGroupsGroupidJoinRequests = exports.postGroupsGroupidJoinRequests = exports.getGroupsGroupidJoinRequests = exports.getGroupsGroupidEmotes = exports.patchGroupsGroupidDescription = exports.getGroupsGroupidConfiguration = exports.postGroupsGroupidClaimOwnership = exports.postGroupsGroupidChangeOwner = exports.patchGroupsGroupidBlockedKeywordsKeywordid = exports.deleteGroupsGroupidBlockedKeywordsKeywordid = exports.postGroupsGroupidBlockedKeywords = exports.getGroupsGroupidBlockedKeywords = exports.deleteGroupsGroupidBansUserid = exports.postGroupsGroupidBansUserid = exports.getGroupsGroupidBansUserid = exports.getGroupsGroupidBans = exports.getGroupsGroupidAuditLog = exports.getGroupsGroupid = exports.deleteFeaturedContentEvent = exports.postFeaturedContentEvent = exports.getFeaturedContentEvent = void 0;
exports.getUsersUseridGroupsRoles = exports.getUsersUseridGroupsPrimaryRole = exports.getUsersUseridFriendsGroupsRoles = exports.deleteUserGroupsPrimary = exports.postUserGroupsPrimary = exports.getUserGroupsPending = exports.getRoles = exports.getGroupsSearchMetadata = exports.getGroupsSearchLookup = exports.getGroupsSearch = exports.postGroupsPolicies = exports.getGroupsMetadata = exports.patchGroupsIcon = exports.postGroupsCreate = exports.getGroupsConfigurationMetadata = exports.deleteGroupsGroupidWallUsersUseridPosts = exports.postGroupsGroupidWallSubscribe = exports.deleteGroupsGroupidWallPostsPostid = exports.postGroupsGroupidWallPosts = exports.getGroupsGroupidWallPosts = exports.getGroupsGroupidUsersUseridPermissions = exports.patchGroupsGroupidUsersUserid = exports.deleteGroupsGroupidUsersUserid = exports.postGroupsGroupidUsers = exports.getGroupsGroupidUsers = exports.patchGroupsGroupidStatus = exports.patchGroupsGroupidSocialLinksSociallinkid = exports.deleteGroupsGroupidSocialLinksSociallinkid = exports.postGroupsGroupidSocialLinks = exports.getGroupsGroupidSocialLinks = exports.patchGroupsGroupidSettings = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Groups_Client_GroupFeaturedContentResponse = zod_1.z.object({
    groupId: zod_1.z.number().int(),
    contentType: zod_1.z.string(),
    contentId: zod_1.z.string(),
});
const Roblox_Groups_Api_Models_Response_UserModel = zod_1.z.object({
    buildersClubMembershipType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
    hasVerifiedBadge: zod_1.z.boolean(),
    userId: zod_1.z.number().int(),
    username: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Groups_Api_ShoutResponse = zod_1.z.object({
    body: zod_1.z.string(),
    poster: Roblox_Groups_Api_Models_Response_UserModel,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Groups_Api_GroupDetailResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    owner: Roblox_Groups_Api_Models_Response_UserModel,
    shout: Roblox_Groups_Api_ShoutResponse,
    memberCount: zod_1.z.number().int(),
    isBuildersClubOnly: zod_1.z.boolean(),
    publicEntryAllowed: zod_1.z.boolean(),
    isLocked: zod_1.z.boolean(),
    hasVerifiedBadge: zod_1.z.boolean(),
    hasSocialModules: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupRoleResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    rank: zod_1.z.number().int(),
    memberCount: zod_1.z.number().int(),
    isBase: zod_1.z.boolean(),
});
const Roblox_Groups_Api_UserGroupRoleResponse = zod_1.z.object({
    user: Roblox_Groups_Api_Models_Response_UserModel,
    role: Roblox_Groups_Api_GroupRoleResponse,
});
const Roblox_Groups_Api_Models_Response_GroupAuditLogResponseItem = zod_1.z.object({
    actor: Roblox_Groups_Api_UserGroupRoleResponse,
    actionType: zod_1.z.string(),
    description: zod_1.z.object({}),
    created: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Groups_Api_GroupAuditLogPageResponse_Roblox_Groups_Api_Models_Response_GroupAuditLogResponseItem_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_Models_Response_GroupAuditLogResponseItem),
});
const Roblox_Groups_Api_GroupBanMemberResponse = zod_1.z.object({
    user: Roblox_Groups_Api_Models_Response_UserModel,
    actingUser: Roblox_Groups_Api_UserGroupRoleResponse,
    created: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_GroupBanMemberResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_GroupBanMemberResponse),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Groups_Client_BlockedKeywordModel = zod_1.z.object({
    id: zod_1.z.string(),
    keyword: zod_1.z.string(),
    createdBy: zod_1.z.number().int(),
    isPrivate: zod_1.z.boolean(),
    createdAt: zod_1.z.string().datetime({ offset: true }),
    updatedAt: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Groups_Api_BlockedKeywordPageResponse_Roblox_Groups_Client_BlockedKeywordModel_ = zod_1.z.object({
    totalActiveKeywordsCount: zod_1.z.number().int(),
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Client_BlockedKeywordModel),
});
const Roblox_Groups_Api_CreateBlockedKeywordsRequest = zod_1.z.object({
    keywords: zod_1.z.string(),
    isPrivate: zod_1.z.boolean(),
});
const Roblox_Groups_Client_CreateBlockedKeywordsResponse = zod_1.z.object({
    createdKeywords: zod_1.z.array(Roblox_Groups_Client_BlockedKeywordModel),
    hadModeratedKeywords: zod_1.z.boolean(),
    hadDuplicateKeywords: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupConfigurationDetailsResponse = zod_1.z.object({
    groupId: zod_1.z.number().int(),
    emblemId: zod_1.z.number().int(),
    coverPhotoId: zod_1.z.number().int(),
});
const Roblox_Groups_Client_EmoteModel = zod_1.z.object({
    id: zod_1.z.string(),
    name: zod_1.z.string(),
});
const Roblox_Groups_Client_EmoteSetModel = zod_1.z.object({
    id: zod_1.z.string(),
    name: zod_1.z.string(),
    emotes: zod_1.z.array(Roblox_Groups_Client_EmoteModel),
});
const Roblox_Groups_Client_GetGroupEmoteSetsResponse = zod_1.z.object({
    emoteSets: zod_1.z.array(Roblox_Groups_Client_EmoteSetModel),
});
const Roblox_Groups_Api_GroupJoinRequestResponse = zod_1.z.object({
    requester: Roblox_Groups_Api_Models_Response_UserModel,
    created: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_GroupJoinRequestResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_GroupJoinRequestResponse),
});
const Roblox_Groups_Api_MembersRequest = zod_1.z.object({
    UserIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Groups_Api_GroupPostsPermissionsModel = zod_1.z.object({
    viewWall: zod_1.z.boolean(),
    postToWall: zod_1.z.boolean(),
    deleteFromWall: zod_1.z.boolean(),
    viewStatus: zod_1.z.boolean(),
    postToStatus: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupForumsPermissionsModel = zod_1.z.object({
    viewForums: zod_1.z.boolean(),
    manageCategories: zod_1.z.boolean(),
    createPosts: zod_1.z.boolean(),
    removePosts: zod_1.z.boolean(),
    lockPosts: zod_1.z.boolean(),
    pinPosts: zod_1.z.boolean(),
    createComments: zod_1.z.boolean(),
    removeComments: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupContentModerationPermissionsModel = zod_1.z.object({
    manageKeywordBlockList: zod_1.z.boolean(),
    viewKeywordBlockList: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupMembershipPermissionsModel = zod_1.z.object({
    changeRank: zod_1.z.boolean(),
    inviteMembers: zod_1.z.boolean(),
    removeMembers: zod_1.z.boolean(),
    banMembers: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupManagementPermissionsModel = zod_1.z.object({
    manageRelationships: zod_1.z.boolean(),
    manageClan: zod_1.z.boolean(),
    viewAuditLogs: zod_1.z.boolean(),
    bypassSlowmode: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupEconomyPermissionsModel = zod_1.z.object({
    spendGroupFunds: zod_1.z.boolean(),
    advertiseGroup: zod_1.z.boolean(),
    createItems: zod_1.z.boolean(),
    manageItems: zod_1.z.boolean(),
    addGroupPlaces: zod_1.z.boolean(),
    manageGroupGames: zod_1.z.boolean(),
    viewGroupPayouts: zod_1.z.boolean(),
    viewAnalytics: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupOpenCloudPermissionsModel = zod_1.z.object({
    useCloudAuthentication: zod_1.z.boolean(),
    administerCloudAuthentication: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupPermissionsModel = zod_1.z.object({
    groupPostsPermissions: Roblox_Groups_Api_GroupPostsPermissionsModel,
    groupForumsPermissions: Roblox_Groups_Api_GroupForumsPermissionsModel,
    groupContentModerationPermissions: Roblox_Groups_Api_GroupContentModerationPermissionsModel,
    groupMembershipPermissions: Roblox_Groups_Api_GroupMembershipPermissionsModel,
    groupManagementPermissions: Roblox_Groups_Api_GroupManagementPermissionsModel,
    groupEconomyPermissions: Roblox_Groups_Api_GroupEconomyPermissionsModel,
    groupOpenCloudPermissions: Roblox_Groups_Api_GroupOpenCloudPermissionsModel,
});
const Roblox_Groups_Api_GroupChannelPermissionsModel = zod_1.z.object({
    channelId: zod_1.z.string(),
    groupForumsPermissions: Roblox_Groups_Api_GroupForumsPermissionsModel,
});
const Roblox_Groups_Api_GroupNotificationPreferenceData = zod_1.z.object({
    type: zod_1.z.enum([
        'AnnouncementCreatedNotification',
        'ForumPostCreatedNotification',
        'ForumCommentCreatedNotification',
        'ForumCommentReplyCreatedNotification',
        'ForumSubscriberNotification',
    ]),
    enabled: zod_1.z.boolean(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
});
const Roblox_Groups_Api_GroupMembershipMetadataResponse = zod_1.z.object({
    groupId: zod_1.z.number().int(),
    isPrimary: zod_1.z.boolean(),
    isPendingJoin: zod_1.z.boolean(),
    userRole: Roblox_Groups_Api_UserGroupRoleResponse,
    permissions: Roblox_Groups_Api_GroupPermissionsModel,
    channelPermissions: zod_1.z.array(Roblox_Groups_Api_GroupChannelPermissionsModel),
    areGroupGamesVisible: zod_1.z.boolean(),
    areGroupFundsVisible: zod_1.z.boolean(),
    areEnemiesAllowed: zod_1.z.boolean(),
    canConfigure: zod_1.z.boolean(),
    isNotificationsEnabled: zod_1.z.boolean(),
    notificationPreferences: zod_1.z.array(Roblox_Groups_Api_GroupNotificationPreferenceData),
    isBannedFromGroup: zod_1.z.boolean(),
    canViewMemberList: zod_1.z.boolean(),
});
const Roblox_Groups_Api_Models_Response_GroupNameHistoryResponseItem = zod_1.z.object({
    name: zod_1.z.string(),
    created: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_Models_Response_GroupNameHistoryResponseItem_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_Models_Response_GroupNameHistoryResponseItem),
});
const Roblox_Groups_Api_GroupPayoutRestrictionResponse = zod_1.z.object({
    canUseRecurringPayout: zod_1.z.boolean(),
    canUseOneTimePayout: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupPayoutResponse = zod_1.z.object({
    user: Roblox_Groups_Api_Models_Response_UserModel,
    percentage: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupPayoutResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Groups_Api_GroupPayoutResponse),
});
const Roblox_Groups_Api_PayoutRecipientRequest = zod_1.z.object({
    recipientId: zod_1.z.number().int(),
    recipientType: zod_1.z.enum(['User', 'Group']),
    amount: zod_1.z.number().int(),
});
const Roblox_Groups_Api_PayoutRequest = zod_1.z.object({
    PayoutType: zod_1.z.enum(['FixedAmount', 'Percentage']),
    Recipients: zod_1.z.array(Roblox_Groups_Api_PayoutRecipientRequest),
    IdempotencyKey: zod_1.z.string(),
});
const Roblox_Groups_Api_OneTimePayoutResponse = zod_1.z.object({
    status: zod_1.z.enum(['NotHeld', 'Held']),
});
const Roblox_Groups_Api_GroupRelationshipsResponse = zod_1.z.object({
    groupId: zod_1.z.number().int(),
    relationshipType: zod_1.z.enum(['Allies', 'Enemies']),
    totalGroupCount: zod_1.z.number().int(),
    relatedGroups: zod_1.z.array(Roblox_Groups_Api_GroupDetailResponse),
    nextRowIndex: zod_1.z.number().int(),
});
const Roblox_Groups_Api_RelationshipsRequest = zod_1.z.object({
    GroupIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Groups_Api_GroupAllRolesResponse = zod_1.z.object({
    groupId: zod_1.z.number().int(),
    roles: zod_1.z.array(Roblox_Groups_Api_GroupRoleResponse),
});
const Roblox_Groups_Api_GroupPermissionsResponse = zod_1.z.object({
    groupId: zod_1.z.number().int(),
    role: Roblox_Groups_Api_GroupRoleResponse,
    permissions: Roblox_Groups_Api_GroupPermissionsModel,
});
const Roblox_Groups_Api_UpdatePermissionsRequest = zod_1.z.object({
    permissions: zod_1.z.object({}),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_Models_Response_UserModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_Models_Response_UserModel),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupPermissionsResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Groups_Api_GroupPermissionsResponse),
});
const Roblox_Groups_Api_GroupSettingsResponse = zod_1.z.object({
    isApprovalRequired: zod_1.z.boolean(),
    isBuildersClubRequired: zod_1.z.boolean(),
    areEnemiesAllowed: zod_1.z.boolean(),
    areGroupFundsVisible: zod_1.z.boolean(),
    areGroupGamesVisible: zod_1.z.boolean(),
    isGroupNameChangeEnabled: zod_1.z.boolean(),
    verificationLevel: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
    accountTenureRequirement: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
    ]),
    slowmode: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
    isMemberListVisibleToPublic: zod_1.z.boolean(),
});
const Roblox_Groups_Api_UpdateGroupSettingsRequest = zod_1.z.object({
    isApprovalRequired: zod_1.z.boolean(),
    areEnemiesAllowed: zod_1.z.boolean(),
    areGroupFundsVisible: zod_1.z.boolean(),
    areGroupGamesVisible: zod_1.z.boolean(),
    verificationLevel: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
    accountTenureRequirement: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
    ]),
    slowmode: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
    isMemberListVisibleToPublic: zod_1.z.boolean(),
});
const Roblox_Groups_Api_SocialLinkResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.enum([
        'Facebook',
        'Twitter',
        'YouTube',
        'Twitch',
        'GooglePlus',
        'Discord',
        'RobloxGroup',
        'Amazon',
        'Guilded',
    ]),
    url: zod_1.z.string(),
    title: zod_1.z.string(),
});
const Roblox_Groups_Api_GetSocialLinkResponse = zod_1.z.object({
    data: zod_1.z.array(Roblox_Groups_Api_SocialLinkResponse),
    socialLinksVerificationStatus: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
});
const Roblox_Groups_Api_SocialLinkRequest = zod_1.z.object({
    type: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
        zod_1.z.literal(8),
    ]),
    url: zod_1.z.string(),
    title: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_UserGroupRoleResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_UserGroupRoleResponse),
});
const Roblox_Groups_Api_JoinGroupRequest = zod_1.z.object({
    sessionId: zod_1.z.string(),
    redemptionToken: zod_1.z.string(),
    captchaId: zod_1.z.string(),
    captchaToken: zod_1.z.string(),
    captchaProvider: zod_1.z.string(),
    challengeId: zod_1.z.string(),
});
const Roblox_Groups_Api_Models_Response_GroupWallPostModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    poster: Roblox_Groups_Api_Models_Response_UserModel,
    body: zod_1.z.string(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_Models_Response_GroupWallPostModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_Models_Response_GroupWallPostModel),
});
const Roblox_Groups_Api_CreateWallPostRequest = zod_1.z.object({
    body: zod_1.z.string(),
    captchaId: zod_1.z.string(),
    captchaToken: zod_1.z.string(),
    captchaProvider: zod_1.z.string(),
    challengeId: zod_1.z.string(),
});
const Roblox_Groups_Api_GroupConfigurationResponse = zod_1.z.object({
    nameMaxLength: zod_1.z.number().int(),
    descriptionMaxLength: zod_1.z.number().int(),
    iconMaxFileSizeMb: zod_1.z.number().int(),
    coverPhotoMaxFileSizeMb: zod_1.z.number().int(),
    validCoverPhotoDimensions: zod_1.z.string(),
    cost: zod_1.z.number().int(),
    isUsingTwoStepWebviewComponent: zod_1.z.boolean(),
});
const Roblox_Groups_Api_RecurringPayoutsConfigurationResponse = zod_1.z.object({
    maxPayoutPartners: zod_1.z.number().int(),
});
const Roblox_Groups_Api_RoleConfigurationResponse = zod_1.z.object({
    nameMaxLength: zod_1.z.number().int(),
    descriptionMaxLength: zod_1.z.number().int(),
    limit: zod_1.z.number().int(),
    cost: zod_1.z.number().int(),
    minRank: zod_1.z.number().int(),
    maxRank: zod_1.z.number().int(),
});
const Roblox_Groups_Api_GroupNameChangeConfigurationResponse = zod_1.z.object({
    cost: zod_1.z.number().int(),
    cooldownInDays: zod_1.z.number().int(),
    ownershipCooldownInDays: zod_1.z.number().int(),
});
const Roblox_Groups_Api_GroupConfigurationDisplayOptionsResponse = zod_1.z.object({
    groupConfiguration: Roblox_Groups_Api_GroupConfigurationResponse,
    recurringPayoutsConfiguration: Roblox_Groups_Api_RecurringPayoutsConfigurationResponse,
    roleConfiguration: Roblox_Groups_Api_RoleConfigurationResponse,
    groupNameChangeConfiguration: Roblox_Groups_Api_GroupNameChangeConfigurationResponse,
    isPremiumPayoutsEnabled: zod_1.z.boolean(),
    isDefaultEmblemPolicyEnabled: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupsDisplayOptionsResponse = zod_1.z.object({
    groupLimit: zod_1.z.number().int(),
    currentGroupCount: zod_1.z.number().int(),
    groupStatusMaxLength: zod_1.z.number().int(),
    groupPostMaxLength: zod_1.z.number().int(),
    isGroupWallNotificationsEnabled: zod_1.z.boolean(),
    groupWallNotificationsSubscribeIntervalInMilliseconds: zod_1.z.number().int(),
    areProfileGroupsHidden: zod_1.z.boolean(),
    isGroupDetailsPolicyEnabled: zod_1.z.boolean(),
    showPreviousGroupNames: zod_1.z.boolean(),
    areGroupBansEnabled: zod_1.z.boolean(),
    canEnableGroupNotifications: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupSearchResponseItem = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    memberCount: zod_1.z.number().int(),
    previousName: zod_1.z.string(),
    publicEntryAllowed: zod_1.z.boolean(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupSearchPageResponse = zod_1.z.object({
    keyword: zod_1.z.string(),
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Groups_Api_GroupSearchResponseItem),
});
const Roblox_Web_Responses_Groups_GroupBasicResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    memberCount: zod_1.z.number().int(),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Groups_GroupBasicResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Web_Responses_Groups_GroupBasicResponse),
});
const Roblox_Groups_Api_GroupSearchMetadataResponse = zod_1.z.object({
    SuggestedGroupKeywords: zod_1.z.array(zod_1.z.string()),
    ShowFriendsGroupsSort: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupRoleDetailResponse = zod_1.z.object({
    groupId: zod_1.z.number().int(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    rank: zod_1.z.number().int(),
    memberCount: zod_1.z.number().int(),
    isBase: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupRoleDetailResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Groups_Api_GroupRoleDetailResponse),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupDetailResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Groups_Api_GroupDetailResponse),
});
const Roblox_Groups_Api_GroupMembershipDetailResponse = zod_1.z.object({
    group: Roblox_Groups_Api_GroupDetailResponse,
    role: Roblox_Groups_Api_GroupRoleResponse,
    isPrimaryGroup: zod_1.z.boolean(),
    isNotificationsEnabled: zod_1.z.boolean(),
    notificationPreferences: zod_1.z.array(Roblox_Groups_Api_GroupNotificationPreferenceData),
});
const Roblox_Groups_Api_UserGroupMembershipResponse = zod_1.z.object({
    user: Roblox_Groups_Api_Models_Response_UserModel,
    groups: zod_1.z.array(Roblox_Groups_Api_GroupMembershipDetailResponse),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_UserGroupMembershipResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Groups_Api_UserGroupMembershipResponse),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupMembershipDetailResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Groups_Api_GroupMembershipDetailResponse),
});
const Roblox_Groups_Api_ChangeOwnerRequest = zod_1.z.object({
    userId: zod_1.z.number().int(),
});
const Roblox_Groups_Api_Models_Request_CreateRoleSetRequest = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    rank: zod_1.z.number().int(),
    usingGroupFunds: zod_1.z.boolean(),
});
const groups_create_body = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    publicGroup: zod_1.z.boolean(),
    buildersClubMembersOnly: zod_1.z.boolean(),
    Files: zod_1.z.instanceof(File),
});
const Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Web_Responses_Groups_GroupOwnerType_ = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.literal('User'),
    name: zod_1.z.string(),
});
const Roblox_Web_Responses_Groups_GroupResponseV2 = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    owner: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Web_Responses_Groups_GroupOwnerType_,
    memberCount: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
    hasVerifiedBadge: zod_1.z.boolean(),
});
const Roblox_Groups_Api_GroupPolicyRequest = zod_1.z.object({
    groupIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Groups_Api_GroupPolicyResponse = zod_1.z.object({
    canViewGroup: zod_1.z.boolean(),
    groupId: zod_1.z.number().int(),
});
const Roblox_Groups_Api_GroupPoliciesResponse = zod_1.z.object({
    groups: zod_1.z.array(Roblox_Groups_Api_GroupPolicyResponse),
});
const Roblox_Groups_Api_PrimaryGroupRequest = zod_1.z.object({
    groupId: zod_1.z.number().int(),
});
const Roblox_Groups_Api_UpdateBlockedKeywordRequest = zod_1.z.object({
    keyword: zod_1.z.string(),
    isPrivate: zod_1.z.boolean(),
});
const Roblox_Groups_Api_UpdateGroupDescriptionRequest = zod_1.z.object({
    description: zod_1.z.string(),
});
const Roblox_Groups_Api_GroupDescriptionResponse = zod_1.z.object({
    newDescription: zod_1.z.string(),
});
const Roblox_Groups_Api_UpdateGroupNameRequest = zod_1.z.object({ name: zod_1.z.string() });
const Roblox_Groups_Api_UpdateGroupNameResponse = zod_1.z.object({
    newName: zod_1.z.string(),
});
const Roblox_Groups_Api_UpdateGroupNotificationPreferenceRequest = zod_1.z.object({
    notificationsEnabled: zod_1.z.boolean(),
    type: zod_1.z.enum([
        'AnnouncementCreatedNotification',
        'ForumPostCreatedNotification',
        'ForumCommentCreatedNotification',
        'ForumCommentReplyCreatedNotification',
        'ForumSubscriberNotification',
    ]),
});
const Roblox_Groups_Api_Models_Request_UpdateRoleSetRequest = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    rank: zod_1.z.number().int(),
});
const Roblox_Groups_Api_PostGroupStatusRequest = zod_1.z.object({
    message: zod_1.z.string(),
});
const Roblox_Groups_Api_UpdateUserRoleRequest = zod_1.z.object({
    roleId: zod_1.z.number().int(),
});
const groups_icon_body = zod_1.z.object({ Files: zod_1.z.instanceof(File) });
/**
 * @api GET https://groups.roblox.com/v1/featured-content/event
 * @summary Gets the featured event for a group
 * @param groupId The group Id.
 */
exports.getFeaturedContentEvent = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/featured-content/event',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Client_GroupFeaturedContentResponse,
    errors: [],
});
/**
 * @api POST https://groups.roblox.com/v1/featured-content/event
 * @summary Sets the featured event for a group
 * @param groupId The group Id.
 * @param eventId The event Id.
 */
exports.postFeaturedContentEvent = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/featured-content/event',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'form',
            explode: true,
        },
        eventId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        eventId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Client_GroupFeaturedContentResponse,
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed
3: User is not authorized to set featured content for this group.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/featured-content/event
 * @summary Deletes the featured event for a group
 * @param groupId The group Id.
 * @param eventId The event Id.
 */
exports.deleteFeaturedContentEvent = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/featured-content/event',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'form',
            explode: true,
        },
        eventId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        eventId: zod_1.z.number().int(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 403,
            description: `0: Token Validation Failed
3: User is not authorized to set featured content for this group.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId
 * @summary Gets group information
 * @param groupId The group Id.
 */
exports.getGroupsGroupid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupDetailResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/audit-log
 * @summary Gets the Group's audit log
 * @param groupId The id of the group the user is in.
 * @param actionType Filter for specific type of action performed
 * @param userId Filter for specific user id
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidAuditLog = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/audit-log',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        actionType: {
            style: 'form',
            explode: true,
        },
        userId: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        actionType: zod_1.z
            .enum([
            'DeletePost',
            'RemoveMember',
            'AcceptJoinRequest',
            'DeclineJoinRequest',
            'PostStatus',
            'ChangeRank',
            'BuyAd',
            'SendAllyRequest',
            'CreateEnemy',
            'AcceptAllyRequest',
            'DeclineAllyRequest',
            'DeleteAlly',
            'DeleteEnemy',
            'AddGroupPlace',
            'RemoveGroupPlace',
            'CreateItems',
            'ConfigureItems',
            'SpendGroupFunds',
            'ChangeOwner',
            'Delete',
            'AdjustCurrencyAmounts',
            'Abandon',
            'Claim',
            'Rename',
            'ChangeDescription',
            'InviteToClan',
            'KickFromClan',
            'CancelClanInvite',
            'BuyClan',
            'CreateGroupAsset',
            'UpdateGroupAsset',
            'ConfigureGroupAsset',
            'RevertGroupAsset',
            'CreateGroupDeveloperProduct',
            'ConfigureGroupGame',
            'CreateGroupDeveloperSubscriptionProduct',
            'Lock',
            'Unlock',
            'CreateGamePass',
            'CreateBadge',
            'ConfigureBadge',
            'SavePlace',
            'PublishPlace',
            'UpdateRolesetRank',
            'UpdateRolesetData',
            'BanMember',
            'UnbanMember',
            'CreateForumCategory',
            'UpdateForumCategory',
            'ArchiveForumCategory',
            'DeleteForumCategory',
            'DeleteForumPost',
            'DeleteForumComment',
            'PinForumPost',
            'UnpinForumPost',
            'LockForumPost',
            'UnlockForumPost',
            'CreateRoleset',
            'DeleteRoleset',
            'CreateCommerceProduct',
            'SetCommerceProductActive',
            'ArchiveCommerceProduct',
            'AcceptCommerceProductBundlingFee',
            'SetCommerceProductInactive',
            'ConnectMerchant',
            'DisconnectMerchant',
            'JoinGroup',
            'LeaveGroup',
            'UpdateGroupIcon',
            'UpdateGroupCoverPhoto',
            'AssignRole',
            'UnassignRole',
        ])
            .optional(),
        userId: zod_1.z.number().int().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Groups_Api_GroupAuditLogPageResponse_Roblox_Groups_Api_Models_Response_GroupAuditLogResponseItem_,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
8: Invalid or missing pagination parameters`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `23: Insufficient permissions to complete the request.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/bans
 * @summary Gets the bans for the group
 * @param groupId The group Id.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidBans = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/bans',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_GroupBanMemberResponse_,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `19: You have insufficient permissions for this request.`,
        },
        {
            status: 404,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 405,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/bans/:userId
 * @summary Fetch the group ban of a user from a group
 * @param groupId The group Id.
 * @param userId The user Id.
 */
exports.getGroupsGroupidBansUserid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/bans/:userId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupBanMemberResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `19: You have insufficient permissions for this request.`,
        },
        {
            status: 404,
            description: `1: The group is invalid or does not exist.
3: The user is invalid or does not exist.`,
        },
        {
            status: 405,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/bans/:userId
 * @summary Bans a user from a group
 * @param groupId The group Id.
 * @param userId The Id of the user being banned.
 */
exports.postGroupsGroupidBansUserid = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/bans/:userId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupBanMemberResponse,
    errors: [
        {
            status: 400,
            description: `28: The user is already banned from the group.
31: You cannot perform this action against the group owner.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
19: You have insufficient permissions for this request.`,
        },
        {
            status: 404,
            description: `1: The group is invalid or does not exist.
3: The user is invalid or does not exist.
3: The user is invalid or does not exist.
15: User is not a member of the group.`,
        },
        {
            status: 405,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/bans/:userId
 * @summary Unbans a user from a group
 * @param groupId The group Id.
 * @param userId The Id of the user being unbanned.
 */
exports.deleteGroupsGroupidBansUserid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/bans/:userId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `29: The user is not banned from the group.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
19: You have insufficient permissions for this request.`,
        },
        {
            status: 404,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 405,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/blocked-keywords
 * @summary Retrieves a paginated list of blocked keywords for a specific Group.
 * @param groupId
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidBlockedKeywords = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/blocked-keywords',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Groups_Api_BlockedKeywordPageResponse_Roblox_Groups_Client_BlockedKeywordModel_,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
6: Limit is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `9: User is invalid or does not exist.
10: Insufficient permissions to complete the request.`,
        },
        {
            status: 405,
            description: `31: Service is currently unavailable.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/blocked-keywords
 * @summary Adds new blocked keyword(s) to the specified Group.
 * @param body
 * @param groupId
 */
exports.postGroupsGroupidBlockedKeywords = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/blocked-keywords',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_CreateBlockedKeywordsRequest,
    response: Roblox_Groups_Client_CreateBlockedKeywordsResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
2: One or more keywords are invalid.
4: Invalid request.
12: The provided content was moderated.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
8: Insufficient permissions to complete the request.
9: User is invalid or does not exist.
10: Insufficient permissions to complete the request.`,
        },
        {
            status: 405,
            description: `31: Service is currently unavailable.`,
        },
        {
            status: 409,
            description: `11: There was a conflict in your request.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/blocked-keywords/:keywordId
 * @summary Deletes a specific blocked keyword from the specified Group.
 * @param groupId
 * @param keywordId
 */
exports.deleteGroupsGroupidBlockedKeywordsKeywordid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/blocked-keywords/:keywordId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        keywordId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        keywordId: zod_1.z.string(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
3: KeywordId is invalid.
4: Invalid request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
9: User is invalid or does not exist.
10: Insufficient permissions to complete the request.`,
        },
        {
            status: 404,
            description: `7: Not found.`,
        },
        {
            status: 405,
            description: `31: Service is currently unavailable.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/blocked-keywords/:keywordId
 * @summary Updates an existing blocked keyword for the specified Group.
 * @param body
 * @param groupId
 * @param keywordId
 */
exports.patchGroupsGroupidBlockedKeywordsKeywordid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/blocked-keywords/:keywordId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
        keywordId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        keywordId: zod_1.z.string(),
    },
    body: Roblox_Groups_Api_UpdateBlockedKeywordRequest,
    response: Roblox_Groups_Client_BlockedKeywordModel,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
3: KeywordId is invalid.
4: Invalid request.
12: The provided content was moderated.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
8: Insufficient permissions to complete the request.
9: User is invalid or does not exist.
10: Insufficient permissions to complete the request.`,
        },
        {
            status: 404,
            description: `7: Not found.`,
        },
        {
            status: 405,
            description: `31: Service is currently unavailable.`,
        },
        {
            status: 409,
            description: `11: There was a conflict in your request.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/change-owner
 * @summary Changes the group owner to another user.
 * @param body The request.
 * @param groupId The group Id.
 */
exports.postGroupsGroupidChangeOwner = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/change-owner',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ userId: zod_1.z.number().int() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
3: The user is invalid or does not exist.
15: User is not a member of the group.
16: The user does not have the necessary level of premium membership.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
17: You are not authorized to change the owner of this group.
25: 2-Step Verification is required to make further transactions. Go to Settings &gt; Security to complete 2-Step Verification.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/claim-ownership
 * @summary Claims ownership of the group as the authenticated user
 * @param groupId The group Id.
 */
exports.postGroupsGroupidClaimOwnership = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/claim-ownership',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
11: You are not authorized to claim this group
12: This group already has an owner
13: Too many attempts to claim groups. Please try again later.`,
        },
        {
            status: 503,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/configuration
 * @summary Gets group configuration information
 * @param groupId The group Id.
 */
exports.getGroupsGroupidConfiguration = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/configuration',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupConfigurationDetailsResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
15: User is not a member of the group.`,
        },
        {
            status: 403,
            description: `1: Group is invalid or does not exist.
23: Insufficient permissions to complete the request.`,
        },
        {
            status: 503,
            description: `31: Service is currently unavailable.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/description
 * @summary Updates the groups description
 * @param body The Roblox.Groups.Api.UpdateGroupDescriptionRequest.
 * @param groupId The id of the group the user is in.
 */
exports.patchGroupsGroupidDescription = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/description',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ description: zod_1.z.string() }),
    response: zod_1.z.object({ newDescription: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
29: Your group description was empty.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
18: The description is too long.
23: Insufficient permissions to complete the request.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/emotes
 * @summary Gets a groups emote sets.
 * @param groupId
 */
exports.getGroupsGroupidEmotes = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/emotes',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Client_GetGroupEmoteSetsResponse,
    errors: [],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/join-requests
 * @summary Gets a page of Group Join Requests for a group.
 * @param groupId The group id.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidJoinRequests = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/join-requests',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_GroupJoinRequestResponse_,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `19: You have insufficient permissions for this request.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/join-requests
 * @summary Batch accepts group join requests
 * @param body The Roblox.Groups.Api.MembersRequest.
 * @param groupId The group id.
 */
exports.postGroupsGroupidJoinRequests = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/join-requests',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_MembersRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
3: The user is invalid or does not exist.
20: The group join request is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: You are already in the maximum number of groups.
19: You have insufficient permissions for this request.`,
        },
        {
            status: 500,
            description: `0: Something went wrong.`,
        },
        {
            status: 503,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/join-requests
 * @summary Batch declines group join requests
 * @param body The Roblox.Groups.Api.MembersRequest.
 * @param groupId The group id.
 */
exports.deleteGroupsGroupidJoinRequests = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/join-requests',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_MembersRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
3: The user is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/join-requests/users/:userId
 * @summary Gets a group join request by userId.
 * @param groupId The group Id.
 * @param userId The user Id.
 */
exports.getGroupsGroupidJoinRequestsUsersUserid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/join-requests/users/:userId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupJoinRequestResponse,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `19: You have insufficient permissions for this request.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/join-requests/users/:userId
 * @summary Accepts a group join request.
 * @param groupId The group Id.
 * @param userId The user Id.
 */
exports.postGroupsGroupidJoinRequestsUsersUserid = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/join-requests/users/:userId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
3: The user is invalid or does not exist.
20: The group join request is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: You are already in the maximum number of groups.
19: You have insufficient permissions for this request.`,
        },
        {
            status: 503,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/join-requests/users/:userId
 * @summary Declines/cancels a group join request.
 * @param groupId The group Id.
 * @param userId The user Id.
 */
exports.deleteGroupsGroupidJoinRequestsUsersUserid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/join-requests/users/:userId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
3: The user is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
4: You do not have permission to manage this member.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/membership
 * @summary Gets group membership information in the context of the authenticated user
 * @param groupId The group Id.
 * @param includeNotificationPreferences
 */
exports.getGroupsGroupidMembership = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/membership',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        includeNotificationPreferences: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        includeNotificationPreferences: zod_1.z.boolean(),
    },
    response: Roblox_Groups_Api_GroupMembershipMetadataResponse,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/name
 * @summary Updates the group's name.
 * @param body The Roblox.Groups.Api.UpdateGroupNameRequest.
 * @param groupId The id of the group the user is in.
 * @description This endpoint will charge Robux for the group rename.
 */
exports.patchGroupsGroupidName = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/name',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ name: zod_1.z.string() }),
    response: zod_1.z.object({ newName: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
13: The name is invalid.
19: The name is too long.
20: The name has been taken.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
12: Insufficient Robux funds.
14: The name is moderated.
23: Insufficient permissions to complete the request.
38: Your account must be verified in order to change your group&#x27;s name.
39: The group ownership was changed too recently.`,
        },
        {
            status: 409,
            description: `36: The name was changed too recently.
37: The name was in use too recently.`,
        },
        {
            status: 413,
            description: `0: Unknown error.`,
        },
        {
            status: 429,
            description: `17: Too many requests.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/name-history
 * @summary Gets the Group's name change history.
 * @param groupId The id of the group.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidNameHistory = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/name-history',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_Models_Response_GroupNameHistoryResponseItem_,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
        {
            status: 403,
            description: `23: Insufficient permissions to complete the request.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/notification-preference
 * @summary Updates the group's settings
 * @param body Roblox.Groups.Api.UpdateGroupSettingsRequest
 * @param groupId The id of the group the user is in.
 */
exports.patchGroupsGroupidNotificationPreference = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/notification-preference',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_UpdateGroupNotificationPreferenceRequest,
    response: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/payout-restriction
 * @summary Gets a value indicating whether the group can use payout feature
 * @param groupId The group id.
 */
exports.getGroupsGroupidPayoutRestriction = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/payout-restriction',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupPayoutRestrictionResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `9: You don&#x27;t have permission to view this group&#x27;s payouts.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/payouts
 * @summary Gets a list of the group payout percentages
 * @param groupId The group id.
 */
exports.getGroupsGroupidPayouts = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/payouts',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupPayoutResponse_,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `9: You don&#x27;t have permission to view this group&#x27;s payouts.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/payouts
 * @summary Pays out a user in Robux.
 * @param body The Roblox.Groups.Api.PayoutRequest.
 * @param groupId The group Id.
 */
exports.postGroupsGroupidPayouts = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/payouts',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_PayoutRequest,
    response: Roblox_Groups_Api_OneTimePayoutResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
12: Insufficient Robux funds.
24: Invalid payout type.
25: The amount is invalid.
26: Too many recipients.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
23: Insufficient permissions to complete the request.
28: Group has paid out too recently. Please wait and try again.
35: 2-Step Verification is required to make further transactions. Go to Settings &gt; Security to complete 2-Step Verification.`,
        },
        {
            status: 503,
            description: `22: The feature is disabled.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/payouts/recurring
 * @summary Updates recurring payouts.
 * @param body The Roblox.Groups.Api.PayoutRequest.
 * @param groupId The group Id.
 * @description This endpoint will remove any recipients not sent in the request.
If a recipient in the request is not a valid member in the group they will not be added to the recurring payouts.
 */
exports.postGroupsGroupidPayoutsRecurring = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/payouts/recurring',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_PayoutRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
24: Invalid payout type.
25: The amount is invalid.
26: Too many recipients.
27: The recipients are invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
12: Insufficient Robux funds.
28: Group has paid out too recently. Please wait and try again.
35: 2-Step Verification is required to make further transactions. Go to Settings &gt; Security to complete 2-Step Verification.`,
        },
        {
            status: 503,
            description: `22: The feature is disabled.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/relationships/:groupRelationshipType
 * @summary Gets a group's relationships
 * @param groupId The group Id.
 * @param groupRelationshipType The group relationship type, enemies or allies.
 * @param StartRowIndex The start index of the page request
 * @param MaxRows The maximum number of rows for the page request, should be at least 1.
 */
exports.getGroupsGroupidRelationshipsGrouprelationshiptype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/relationships/:groupRelationshipType',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
        StartRowIndex: {
            style: 'form',
            explode: true,
        },
        MaxRows: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
        StartRowIndex: zod_1.z.number().int(),
        MaxRows: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupRelationshipsResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
4: Group relationship type or request type is invalid.
8: Invalid or missing pagination parameters`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/relationships/:groupRelationshipType/:relatedGroupId
 * @summary Create a group relationship.
 * @param groupId The group id.
 * @param groupRelationshipType The group relationship type, enemies or allies.
 * @param relatedGroupId The id of the group you want to create a relationship with.
 */
exports.postGroupsGroupidRelationshipsGrouprelationshiptypeRelatedgroupid = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/relationships/:groupRelationshipType/:relatedGroupId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
        relatedGroupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
        relatedGroupId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group relationship type or request type is invalid.
2: Invalid group.
3: Target group is invalid or does not exist.
4: Your group cannot establish a relationship with itself.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
5: Your group does not allow enemy declarations.
6: Other group does not allow enemy declarations.
7: Your group already has a relationship with the target group.
8: You are blocked from communicating with this user.
9: Insufficient permissions.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/relationships/:groupRelationshipType/:relatedGroupId
 * @summary Deletes a group relationship.
 * @param groupId The group id.
 * @param groupRelationshipType The group relationship type, enemies or allies.
 * @param relatedGroupId The id of the group you want to delete the relationship with.
 */
exports.deleteGroupsGroupidRelationshipsGrouprelationshiptypeRelatedgroupid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/relationships/:groupRelationshipType/:relatedGroupId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
        relatedGroupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
        relatedGroupId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: Invalid group.
3: Target group is invalid or does not exist.
11: Relationship does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
8: You are blocked from communicating with this user.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/relationships/:groupRelationshipType/requests
 * @summary Gets a group's relationship requests
 * @param groupId The group Id.
 * @param groupRelationshipType The group relationship type of the request, only allies are supported.
 * @param StartRowIndex The start index of the page request
 * @param MaxRows The maximum number of rows for the page request, should be at least 1.
 */
exports.getGroupsGroupidRelationshipsGrouprelationshiptypeRequests = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/relationships/:groupRelationshipType/requests',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
        StartRowIndex: {
            style: 'form',
            explode: true,
        },
        MaxRows: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
        StartRowIndex: zod_1.z.number().int(),
        MaxRows: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupRelationshipsResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
4: Group relationship type or request type is invalid.
8: Invalid or missing pagination parameters`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `5: You don&#x27;t have permission to manage this group&#x27;s relationships.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/relationships/:groupRelationshipType/requests
 * @summary Batch accepts group affiliate requests
 * @param body The Roblox.Groups.Api.RelationshipsRequest.
 * @param groupId The group id.
 * @param groupRelationshipType The type of group relationship being made
 */
exports.postGroupsGroupidRelationshipsGrouprelationshiptypeRequests = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/relationships/:groupRelationshipType/requests',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
    },
    body: Roblox_Groups_Api_RelationshipsRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/relationships/:groupRelationshipType/requests
 * @summary Batch declines group affiliate requests
 * @param body The Roblox.Groups.Api.RelationshipsRequest.
 * @param groupId The group id.
 * @param groupRelationshipType The type of group relationship being made
 */
exports.deleteGroupsGroupidRelationshipsGrouprelationshiptypeRequests = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/relationships/:groupRelationshipType/requests',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
    },
    body: Roblox_Groups_Api_RelationshipsRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/relationships/:groupRelationshipType/requests/:relatedGroupId
 * @summary Accepts a group relationship request.
 * @param groupId The group id.
 * @param groupRelationshipType The group relationship type, enemies or allies, only allies are supported.
 * @param relatedGroupId The id of the group you want to accept the relationship request with.
 */
exports.postGroupsGroupidRelationshipsGrouprelationshiptypeRequestsRelatedgroupid = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/relationships/:groupRelationshipType/requests/:relatedGroupId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
        relatedGroupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
        relatedGroupId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group relationship type or request type is invalid.
2: Invalid group.
3: Target group is invalid or does not exist.
10: Relationship request does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
9: Insufficient permissions.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/relationships/:groupRelationshipType/requests/:relatedGroupId
 * @summary Declines a group relationship request.
 * @param groupId The group id.
 * @param groupRelationshipType The group relationship type, enemies or allies.
 * @param relatedGroupId The id of the group you want to accept the relationship request with.
 */
exports.deleteGroupsGroupidRelationshipsGrouprelationshiptypeRequestsRelatedgroupid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/relationships/:groupRelationshipType/requests/:relatedGroupId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        groupRelationshipType: {
            style: 'simple',
        },
        relatedGroupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        groupRelationshipType: zod_1.z.string(),
        relatedGroupId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group relationship type or request type is invalid.
2: Invalid group.
3: Target group is invalid or does not exist.
10: Relationship request does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
9: Insufficient permissions.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/roles
 * @summary Gets a list of the rolesets in a group.
 * @param groupId The group id.
 */
exports.getGroupsGroupidRoles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/roles',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupAllRolesResponse,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/roles/:roleSetId/permissions
 * @summary Gets the permissions for a group's roleset. The authorized user must either be the group owner or the roleset being requested, except for guest roles, which can be viewed by all (members and guests).
 * @param groupId The group id.
 * @param roleSetId The group's role set id.
 */
exports.getGroupsGroupidRolesRolesetidPermissions = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/roles/:roleSetId/permissions',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        roleSetId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        roleSetId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupPermissionsResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
2: The roleset is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `3: You are not authorized to view/edit permissions for this role.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/roles/:roleSetId/permissions
 * @summary Updates the permissions for a group's roleset. The authorized user must be the group owner.
 * @param body The request.
 * @param groupId The group's id.
 * @param roleSetId The roleset's id.
 */
exports.patchGroupsGroupidRolesRolesetidPermissions = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/roles/:roleSetId/permissions',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
        roleSetId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        roleSetId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ permissions: zod_1.z.object({}) }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
2: The roleset is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
3: You are not authorized to view/edit permissions for this role.
4: This role&#x27;s permissions can not be modified.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/roles/:roleSetId/users
 * @summary Gets a list of users in a group for a specific roleset.
 * @param groupId The group id.
 * @param roleSetId The group's role set id.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidRolesRolesetidUsers = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/roles/:roleSetId/users',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        roleSetId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        roleSetId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_Models_Response_UserModel_,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 403,
            description: `2: The roleset is invalid or does not exist.
35: You do not have permission to view this group&#x27;s member list.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/roles/guest/permissions
 * @summary Gets the permissions for a group's guest roleset. These can be viewed by all (members and guests) users.
 * @param groupId The group id.
 */
exports.getGroupsGroupidRolesGuestPermissions = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/roles/guest/permissions',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupPermissionsResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/roles/permissions
 * @summary Gets all permissions for each role
 * @param groupId The group id.
 */
exports.getGroupsGroupidRolesPermissions = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/roles/permissions',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupPermissionsResponse_,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/rolesets/:rolesetId
 * @summary Deletes existing group roleset.
 * @param groupId The group Id.
 * @param rolesetId The roleset Id.
 */
exports.deleteGroupsGroupidRolesetsRolesetid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/rolesets/:rolesetId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        rolesetId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        rolesetId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `10: This group does not exist.
15: This role does not exist.
17: Cannot remove any more roles
18: Cannot delete this role.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
9: You do not have permissions to perform this action.
16: There are users in this role.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/rolesets/:rolesetId
 * @summary Updates existing group roleset.
 * @param body The Roblox.Groups.Api.Models.Request.UpdateRoleSetRequest.
 * @param groupId The group Id.
 * @param rolesetId The roleset Id.
 */
exports.patchGroupsGroupidRolesetsRolesetid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/rolesets/:rolesetId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
        rolesetId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        rolesetId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_Models_Request_UpdateRoleSetRequest,
    response: Roblox_Groups_Api_GroupRoleResponse,
    errors: [
        {
            status: 400,
            description: `5: Role name already exists.
6: Rank value is out of bounds.
7: The role name is too long.
8: The role description is too long.
10: This group does not exist.
14: Role name can not be empty.
15: This role does not exist.
19: Cannot update Guest role.
20: Cannot update Owner role rank.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
9: You do not have permissions to perform this action.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/rolesets/create
 * @summary Creates new group roleset.
 * @param body The Roblox.Groups.Api.Models.Request.CreateRoleSetRequest.
 * @param groupId The group Id.
 */
exports.postGroupsGroupidRolesetsCreate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/rolesets/create',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_Models_Request_CreateRoleSetRequest,
    response: Roblox_Groups_Api_GroupRoleResponse,
    errors: [
        {
            status: 400,
            description: `0: Something went wrong.
3: You do not have enough funds to purchase this role.
5: Role name already exists.
6: Rank value is out of bounds.
7: The role name is too long.
8: The role description is too long.
10: This group does not exist.
11: Failed to process payment to purchase role.
12: Limit for roles have been reached on this group.
14: Role name can not be empty.
15: This role does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
9: You do not have permissions to perform this action.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/settings
 * @summary Gets the Group's settings
 * @param groupId The id of the group the user is in.
 */
exports.getGroupsGroupidSettings = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/settings',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupSettingsResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `23: Insufficient permissions to complete the request.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/settings
 * @summary Updates the group's settings
 * @param body Roblox.Groups.Api.UpdateGroupSettingsRequest
 * @param groupId The id of the group the user is in.
 */
exports.patchGroupsGroupidSettings = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/settings',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_UpdateGroupSettingsRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
23: Insufficient permissions to complete the request.`,
        },
        {
            status: 503,
            description: `31: Service is currently unavailable.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/social-links
 * @summary Get social link data associated with a group
 * @param groupId The Id of the game
 */
exports.getGroupsGroupidSocialLinks = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/social-links',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GetSocialLinkResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `13: Only users who are over thirteen years of age may edit social links.`,
        },
        {
            status: 404,
            description: `11: Social links cannot be processed as this time.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/social-links
 * @summary Posts a social links
 * @param body The Roblox.Groups.Api.SocialLinkRequest
 * @param groupId The id of the group
 */
exports.postGroupsGroupidSocialLinks = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/social-links',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_SocialLinkRequest,
    response: Roblox_Groups_Api_SocialLinkResponse,
    errors: [
        {
            status: 400,
            description: `3: The social link title is too long.
4: The social link title cannot be empty.
5: The social link url cannot be empty.
7: The request was null.
9: The social link type is invalid.
12: The social link title was moderated.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to configure this social link.`,
        },
        {
            status: 404,
            description: `8: The requested group or social link was not found.`,
        },
        {
            status: 503,
            description: `11: Social links cannot be processed as this time.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/social-links/:socialLinkId
 * @summary Deletes a social link
 * @param groupId The id of the game you are editing, required for permissions checking
 * @param socialLinkId The id of the social link
 */
exports.deleteGroupsGroupidSocialLinksSociallinkid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/social-links/:socialLinkId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        socialLinkId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        socialLinkId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
10: The social link is not for a group.
15: The social link id doesn&#x27;t match the group id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to configure this social link.
13: Only users who are over thirteen years of age may edit social links.`,
        },
        {
            status: 404,
            description: `11: Social links cannot be processed as this time.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/social-links/:socialLinkId
 * @summary Updates a social link
 * @param body The Roblox.Groups.Api.SocialLinkRequest.
 * @param groupId The id of the universe
 * @param socialLinkId The id of the social link being updated
 */
exports.patchGroupsGroupidSocialLinksSociallinkid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/social-links/:socialLinkId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
        socialLinkId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        socialLinkId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_SocialLinkRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
3: The social link title is too long.
4: The social link title cannot be empty.
5: The social link url cannot be empty.
6: The social link url was improperly formatted.
7: The request was null.
8: The requested group or social link was not found.
9: The social link type is invalid.
10: The social link is not for a group.
12: The social link title was moderated.
16: A social link with this type already exists on this group.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to configure this social link.`,
        },
        {
            status: 404,
            description: `11: Social links cannot be processed as this time.`,
        },
        {
            status: 503,
            description: `11: Social links cannot be processed as this time.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/status
 * @summary Sets group status
 * @param body The Roblox.Groups.Api.PostGroupStatusRequest.
 * @param groupId The group Id.
 */
exports.patchGroupsGroupidStatus = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/status',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ message: zod_1.z.string() }),
    response: Roblox_Groups_Api_ShoutResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
7: Missing group status content.
32: Description was filtered.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: You are not authorized to set the status of this group`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/users
 * @summary Gets a list of users in a group.
 * @param groupId The group id.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getGroupsGroupidUsers = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/users',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_UserGroupRoleResponse_,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 403,
            description: `35: You do not have permission to view this group&#x27;s member list.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/users
 * @summary Joins a group
 * @param body Only supplied when captcha has been solved.
 * @param groupId The group Id.
 * @param Roblox-Place-Id The place ID of the experience the player is in.
 * @param Roblox-Game-Id The player's current game Id.
 * @param Roblox-Session-Id The player's current session Id.
 */
exports.postGroupsGroupidUsers = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/users',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
        'Roblox-Place-Id': {
            style: 'simple',
        },
        'Roblox-Game-Id': {
            style: 'simple',
        },
        'Roblox-Session-Id': {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        'Roblox-Place-Id': zod_1.z.number().int().optional(),
        'Roblox-Game-Id': zod_1.z.string().optional(),
        'Roblox-Session-Id': zod_1.z.string().optional(),
    },
    body: Roblox_Groups_Api_JoinGroupRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: You are already in the maximum number of groups.
9: You do not have the builders club membership necessary to join this group.
14: You cannot join a closed group.
33: You do not have the required verification level to join this group.
34: You do not have the required account tenure to join this group.`,
        },
        {
            status: 409,
            description: `7: You have already requested to join this group.
8: You are already a member of this group.`,
        },
        {
            status: 429,
            description: `10: Too many attempts to join the group. Please try again later.`,
        },
        {
            status: 503,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/users/:userId
 * @summary Removes a user from a group
 * @param groupId The group Id.
 * @param userId The Id of the user being removed.
 */
exports.deleteGroupsGroupidUsersUserid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/users/:userId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
3: The user is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
4: You do not have permission to manage this member.
25: 2-Step Verification is required to make further transactions. Go to Settings &gt; Security to complete 2-Step Verification.`,
        },
        {
            status: 503,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/:groupId/users/:userId
 * @summary Updates a users role in a group.
 * @param body The Roblox.Groups.Api.UpdateUserRoleRequest.
 * @param groupId The id of the group the user is in.
 * @param userId The id of the user being updated.
 */
exports.patchGroupsGroupidUsersUserid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/:groupId/users/:userId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ roleId: zod_1.z.number().int() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
2: The roleset is invalid or does not exist.
3: The user is invalid or does not exist.
23: You cannot change your own role.
26: You cannot change the user&#x27;s role to the same role.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
4: You do not have permission to manage this member.`,
        },
        {
            status: 503,
            description: `18: The operation is temporarily unavailable. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/users/:userId/permissions
 * @summary Gets the permissions a user has in a group. Only available to group owner and RCC
 * @param groupId The group id.
 * @param userId The user id.
 */
exports.getGroupsGroupidUsersUseridPermissions = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/users/:userId/permissions',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupPermissionsResponse,
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
        {
            status: 403,
            description: `3: You are not authorized to view/edit permissions for this role.`,
        },
        {
            status: 404,
            description: `3: The user is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/:groupId/wall/posts
 * @summary Gets a list of group wall posts.
 * @param groupId The group id.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder Sorted by group wall post Id
 */
exports.getGroupsGroupidWallPosts = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/:groupId/wall/posts',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Groups_Api_Models_Response_GroupWallPostModel_,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 403,
            description: `2: You do not have permission to access this group wall.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/wall/posts
 * @summary Creates a post on a group wall
 * @param body The Roblox.Groups.Api.CreateWallPostRequest.
 * @param groupId The group id.
 */
exports.postGroupsGroupidWallPosts = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/wall/posts',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: Roblox_Groups_Api_CreateWallPostRequest,
    response: Roblox_Groups_Api_Models_Response_GroupWallPostModel,
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
5: Your post was empty, white space, or more than 500 characters.
9: The provided content was moderated.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to access this group wall.`,
        },
        {
            status: 405,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 429,
            description: `4: You are posting too fast, please try again in a few minutes.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/wall/posts/:postId
 * @summary Deletes a group wall post.
 * @param groupId The group id.
 * @param postId The group wall post id.
 */
exports.deleteGroupsGroupidWallPostsPostid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/wall/posts/:postId',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        postId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        postId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
3: The group wall post id is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to access this group wall.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/:groupId/wall/subscribe
 * @summary Subscribes the authenticated user to notifications of group wall events.
 * @param groupId The group id.
 */
exports.postGroupsGroupidWallSubscribe = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/:groupId/wall/subscribe',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to access this group wall.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/groups/:groupId/wall/users/:userId/posts
 * @summary Deletes all group wall posts made by a specific user.
 * @param groupId The group id.
 * @param userId The user id.
 */
exports.deleteGroupsGroupidWallUsersUseridPosts = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/groups/:groupId/wall/users/:userId/posts',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupId: {
            style: 'simple',
        },
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The group is invalid or does not exist.
6: The user specified is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to access this group wall.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/configuration/metadata
 * @summary Gets Group configuration contextual information
 */
exports.getGroupsConfigurationMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/configuration/metadata',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    response: Roblox_Groups_Api_GroupConfigurationDisplayOptionsResponse,
    errors: [],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/create
 * @summary Creates a new group.
 * @param body
 * @description This endpoint will charge Robux for the group purchase.
Accepts "icon" and "coverPhoto" in Files object. Defaults to first file if "icon" is not present.
Http status code 413 is thrown when the group icon file size is too large.
 */
exports.postGroupsCreate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/create',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: groups_create_body,
    response: Roblox_Web_Responses_Groups_GroupResponseV2,
    errors: [
        {
            status: 400,
            description: `13: The name is invalid.
15: The group icon is invalid.
16: The group icon is missing from the request.
18: The description is too long.
19: The name is too long.
20: The name has been taken.
46: Invalid file format or dimensions for group cover photo.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
10: User must have builders club membership.
11: User is in maximum number of groups.
12: Insufficient Robux funds.
14: The name is moderated.`,
        },
        {
            status: 409,
            description: `37: The name was in use too recently.`,
        },
        {
            status: 413,
            description: `0: Unknown error.`,
        },
        {
            status: 429,
            description: `17: Too many requests.`,
        },
        {
            status: 503,
            description: `21: Group creation is currently disabled.`,
        },
    ],
});
/**
 * @api PATCH https://groups.roblox.com/v1/groups/icon
 * @summary Updates the group icon.
 * @param body
 * @param groupId The group Id.
 */
exports.patchGroupsIcon = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/groups/icon',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        groupId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ Files: zod_1.z.instanceof(File) }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.
16: The group icon is missing from the request.
17: Too many requests.
30: Invalid file type for group icon.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
23: Insufficient permissions to complete the request.`,
        },
        {
            status: 413,
            description: `0: Unknown error.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/metadata
 * @summary Gets Groups contextual information:
Max number of groups a user can be part of.
Current number of groups a user is a member of.
Whether to show/hide certain features based on device type.
 */
exports.getGroupsMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/metadata',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    response: Roblox_Groups_Api_GroupsDisplayOptionsResponse,
    errors: [],
});
/**
 * @api POST https://groups.roblox.com/v1/groups/policies
 * @summary Gets group policy info used for compliance.
 * @param body
 */
exports.postGroupsPolicies = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/groups/policies',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Groups_Api_GroupPolicyRequest,
    response: Roblox_Groups_Api_GroupPoliciesResponse,
    errors: [
        {
            status: 400,
            description: `1: Too many ids in request.
2: Ids could not be parsed from request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/search
 * @summary Search for groups by keyword.
 * @param keyword The keyword or phrase to use as the search parameter.
 * @param prioritizeExactMatch Whether or not to prioritize the exact match for the keyword (optional, defaults to false).
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 */
exports.getGroupsSearch = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/search',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        keyword: {
            style: 'form',
            explode: true,
        },
        prioritizeExactMatch: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        keyword: zod_1.z.string(),
        prioritizeExactMatch: zod_1.z.boolean().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
    },
    response: Roblox_Groups_Api_GroupSearchPageResponse,
    errors: [
        {
            status: 400,
            description: `2: Search term not appropriate for Roblox.
3: Search term was left empty.
4: Search terms can be 2 to 50 characters long.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/search/lookup
 * @summary Looks up groups by a name. Prioritizes an exact match as the first result.
 * @param groupName The group name.
 * @description Should only be used for direct lookups where a user is inputting a group name, shouldn't be used for search pages.
 */
exports.getGroupsSearchLookup = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/search/lookup',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        groupName: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        groupName: zod_1.z.string(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Web_Responses_Groups_GroupBasicResponse_,
    errors: [
        {
            status: 400,
            description: `1: Name is missing or has invalid characters.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/groups/search/metadata
 * @summary Get suggested groups and other miscellaneous information needed for the group/join page like flags
 * @description Although there is no reason for this to require an authenticated user right now, in the future,
we will use coco to return different suggested groups based upon that user's request context
 */
exports.getGroupsSearchMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/groups/search/metadata',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    response: Roblox_Groups_Api_GroupSearchMetadataResponse,
    errors: [
        {
            status: 404,
            description: `5: No Localized Version of group search category exists`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/roles
 * @summary Gets the Roles by their ids.
 * @param ids A list of role ids
 */
exports.getRoles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/roles',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        ids: {
            style: 'form',
        },
    },
    parameters: {
        ids: zod_1.z.array(zod_1.z.number()),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupRoleDetailResponse_,
    errors: [
        {
            status: 400,
            description: `1: Ids could not be parsed from request.
2: Too many ids in request.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/user/groups/pending
 * @summary Gets groups that the authenticated user has requested to join
 */
exports.getUserGroupsPending = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/user/groups/pending',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupDetailResponse_,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://groups.roblox.com/v1/user/groups/primary
 * @summary Sets the authenticated user's primary group
 * @param body The request body containing the group id.
 */
exports.postUserGroupsPrimary = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/user/groups/primary',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ groupId: zod_1.z.number().int() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Group is invalid or does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You aren&#x27;t a member of the group specified.`,
        },
    ],
});
/**
 * @api DELETE https://groups.roblox.com/v1/user/groups/primary
 * @summary Removes the authenticated user's primary group
 */
exports.deleteUserGroupsPrimary = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/user/groups/primary',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/users/:userId/friends/groups/roles
 * @summary Gets a list of all groups the specified users' friends are in.
 * @param userId The user id.
 */
exports.getUsersUseridFriendsGroupsRoles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/friends/groups/roles',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_UserGroupMembershipResponse_,
    errors: [
        {
            status: 400,
            description: `3: The user is invalid or does not exist.`,
        },
        {
            status: 403,
            description: `3: The user is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/users/:userId/groups/primary/role
 * @summary Gets a user's primary group.
 * @param userId The user id.
 */
exports.getUsersUseridGroupsPrimaryRole = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/groups/primary/role',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Groups_Api_GroupMembershipDetailResponse,
    errors: [
        {
            status: 400,
            description: `4: User is invalid or does not exist.`,
        },
    ],
});
/**
 * @api GET https://groups.roblox.com/v1/users/:userId/groups/roles
 * @summary Gets a list of all group roles for groups the specified user is in.
 * @param userId The user id.
 * @param includeLocked
 * @param includeNotificationPreferences
 * @param discoveryType
 */
exports.getUsersUseridGroupsRoles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/groups/roles',
    baseUrl: 'https://groups.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        includeLocked: {
            style: 'form',
            explode: true,
        },
        includeNotificationPreferences: {
            style: 'form',
            explode: true,
        },
        discoveryType: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        includeLocked: zod_1.z.boolean().optional(),
        includeNotificationPreferences: zod_1.z.boolean().optional(),
        discoveryType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1)]).optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Groups_Api_GroupMembershipDetailResponse_,
    errors: [
        {
            status: 400,
            description: `3: The user is invalid or does not exist.`,
        },
    ],
});
