// rozod/lib/endpoints/inventoryv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridInventoryAssettypeid = exports.getUsersUseridInventory = exports.deleteInventoryAssetAssetid = exports.getAssetsAssetidOwners = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Agents_AgentType_ = zod_1.z.object({
    id: zod_1.z.number().int(),
    type: zod_1.z.enum(['User', 'Group']),
    name: zod_1.z.string(),
});
const Roblox_Inventory_Api_V2_AssetOwnerResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    collectibleItemInstanceId: zod_1.z.string(),
    serialNumber: zod_1.z.number().int(),
    owner: Roblox_Web_Responses_RelatedEntityTypeResponse_Roblox_Agents_AgentType_,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_V2_AssetOwnerResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Inventory_Api_V2_AssetOwnerResponse),
});
const Roblox_Inventory_Api_V2_UserAssetItemModelV2 = zod_1.z.object({
    assetId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    assetType: zod_1.z.enum([
        'Image',
        'TShirt',
        'Audio',
        'Mesh',
        'Lua',
        'HTML',
        'Text',
        'Hat',
        'Place',
        'Model',
        'Shirt',
        'Pants',
        'Decal',
        'Avatar',
        'Head',
        'Face',
        'Gear',
        'Badge',
        'GroupEmblem',
        'Animation',
        'Arms',
        'Legs',
        'Torso',
        'RightArm',
        'LeftArm',
        'LeftLeg',
        'RightLeg',
        'Package',
        'YouTubeVideo',
        'GamePass',
        'App',
        'Code',
        'Plugin',
        'SolidModel',
        'MeshPart',
        'HairAccessory',
        'FaceAccessory',
        'NeckAccessory',
        'ShoulderAccessory',
        'FrontAccessory',
        'BackAccessory',
        'WaistAccessory',
        'ClimbAnimation',
        'DeathAnimation',
        'FallAnimation',
        'IdleAnimation',
        'JumpAnimation',
        'RunAnimation',
        'SwimAnimation',
        'WalkAnimation',
        'PoseAnimation',
        'LocalizationTableManifest',
        'LocalizationTableTranslation',
        'EmoteAnimation',
        'Video',
        'TexturePack',
        'TShirtAccessory',
        'ShirtAccessory',
        'PantsAccessory',
        'JacketAccessory',
        'SweaterAccessory',
        'ShortsAccessory',
        'LeftShoeAccessory',
        'RightShoeAccessory',
        'DressSkirtAccessory',
        'FontFamily',
        'FontFace',
        'MeshHiddenSurfaceRemoval',
        'EyebrowAccessory',
        'EyelashAccessory',
        'MoodAnimation',
        'DynamicHead',
        'CodeSnippet',
        'AdsVideo',
        'OtaUpdate',
        'Screenshot',
        'RuntimePropertySet',
        'StorePreviewVideo',
        'GamePreviewVideo',
        'CreatorExperienceConfig',
        'FaceMakeup',
        'LipMakeup',
        'EyeMakeup',
    ]),
    created: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_V2_UserAssetItemModelV2_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Inventory_Api_V2_UserAssetItemModelV2),
});
const Roblox_Inventory_Api_Models_UserModel = zod_1.z.object({
    userId: zod_1.z.number().int(),
    username: zod_1.z.string(),
    buildersClubMembershipType: zod_1.z.enum(['None', 'BC', 'TBC', 'OBC', 'RobloxPremium']),
});
const Roblox_Inventory_Api_Models_InventoryItemModel = zod_1.z.object({
    expireAt: zod_1.z.string().datetime({ offset: true }),
    userAssetId: zod_1.z.number().int(),
    assetId: zod_1.z.number().int(),
    assetName: zod_1.z.string(),
    collectibleItemId: zod_1.z.string(),
    collectibleItemInstanceId: zod_1.z.string(),
    serialNumber: zod_1.z.number().int(),
    owner: Roblox_Inventory_Api_Models_UserModel,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_Models_InventoryItemModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Inventory_Api_Models_InventoryItemModel),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
/**
 * @api GET https://inventory.roblox.com/v2/assets/:assetId/owners
 * @summary Gets a list of owners of an asset.
 * @param assetId The asset id.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder Sorted by userAssetId
 */
exports.getAssetsAssetidOwners = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/assets/:assetId/owners',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_V2_AssetOwnerResponse_,
    errors: [
        {
            status: 400,
            description: `1: The asset id is invalid.`,
        },
        {
            status: 403,
            description: `2: You do not have permission to view the owners of this asset.`,
        },
    ],
});
/**
 * @api DELETE https://inventory.roblox.com/v2/inventory/asset/:assetId
 * @summary Give up an asset owned by the authenticated user.
Assets that are created by Roblox user or are limited edition are not eligible for deletion
and will return NotEligibleForDelete.
 * @param assetId ID of the asset to delete.
 */
exports.deleteInventoryAssetAssetid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v2/inventory/asset/:assetId',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
4: You are not authorized.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You don&#x27;t own the specified item.
3: The item is not allowed to be deleted.`,
        },
        {
            status: 404,
            description: `1: The item does not exist.`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api GET https://inventory.roblox.com/v2/users/:userId/inventory
 * @summary Get user's inventory by multiple Roblox.Platform.Assets.AssetType.
 * @param userId The inventory owner's userId.
 * @param assetTypes The asset types to query.
 * @param filterDisapprovedAssets Filters moderated assets when enabled.
 * @param showApprovedOnly Filters moderated assets and assets pending review when enabled.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 * @description GamePass and Badges not allowed.
 */
exports.getUsersUseridInventory = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/users/:userId/inventory',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        assetTypes: {
            style: 'form',
        },
        filterDisapprovedAssets: {
            style: 'form',
            explode: true,
        },
        showApprovedOnly: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        assetTypes: zod_1.z.array(zod_1.z.object({})),
        filterDisapprovedAssets: zod_1.z.boolean().optional(),
        showApprovedOnly: zod_1.z.boolean().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_V2_UserAssetItemModelV2_,
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
2: Invalid asset type Id.`,
        },
        {
            status: 403,
            description: `3: Insufficient permission.
4: You are not authorized to view this user&#x27;s inventory.`,
        },
    ],
});
/**
 * @api GET https://inventory.roblox.com/v2/users/:userId/inventory/:assetTypeId
 * @summary Gets user's inventory based on specific asset type
 * @param userId The user Id of the inventory owner
 * @param assetTypeId The asset type Id of the items to get
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getUsersUseridInventoryAssettypeid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/users/:userId/inventory/:assetTypeId',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        assetTypeId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        assetTypeId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_Models_InventoryItemModel_,
    errors: [
        {
            status: 400,
            description: `1: Invalid user Id.
2: Invalid asset type Id.`,
        },
        {
            status: 403,
            description: `3: Insufficient permission.
4: You are not authorized to view this user&#x27;s inventory.`,
        },
    ],
});
