// rozod/lib/endpoints/inventoryv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridPlacesInventory = exports.getUsersUseridItemsItemtypeItemtargetidIsOwned = exports.getUsersUseridItemsItemtypeItemtargetid = exports.getUsersUseridCategoriesFavorites = exports.getUsersUseridCategories = exports.getUsersUseridCanViewInventory = exports.getUsersUseridAssetsCollectibles = exports.getPackagesPackageidAssets = exports.deleteCollectionsItemsItemtypeItemtargetid = exports.postCollectionsItemsItemtypeItemtargetid = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Inventory_Api_Models_AssetIdListModel = zod_1.z.object({
    assetIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_Inventory_Api_Models_CollectibleUserAssetModel = zod_1.z.object({
    userAssetId: zod_1.z.number().int(),
    serialNumber: zod_1.z.number().int(),
    assetId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    recentAveragePrice: zod_1.z.number().int(),
    originalPrice: zod_1.z.number().int(),
    assetStock: zod_1.z.number().int(),
    buildersClubMembershipType: zod_1.z.enum(['None', 'BC', 'TBC', 'OBC', 'RobloxPremium']),
    isOnHold: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_Models_CollectibleUserAssetModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Inventory_Api_Models_CollectibleUserAssetModel),
});
const Roblox_Inventory_Api_Models_CanViewInventoryResponse = zod_1.z.object({
    canView: zod_1.z.boolean(),
});
const Roblox_Inventory_Api_Models_AssetsExplorerCategoryItemModel = zod_1.z.object({
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
    filter: zod_1.z.string(),
    id: zod_1.z.number().int(),
    type: zod_1.z.enum(['AssetType', 'Bundle', 'Outfit', 'Set', 'Avatar']),
    categoryType: zod_1.z.string(),
});
const Roblox_Inventory_Api_Models_AssetsExplorerCategoryModel = zod_1.z.object({
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
    categoryType: zod_1.z.string(),
    items: zod_1.z.array(Roblox_Inventory_Api_Models_AssetsExplorerCategoryItemModel),
});
const Roblox_Inventory_Api_Models_CategoriesModel = zod_1.z.object({
    categories: zod_1.z.array(Roblox_Inventory_Api_Models_AssetsExplorerCategoryModel),
});
const Roblox_Inventory_Api_Models_IItemModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    type: zod_1.z.enum(['Asset', 'GamePass', 'Badge', 'Bundle', 'Avatar']),
    instanceId: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_Models_IItemModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Inventory_Api_Models_IItemModel),
});
const Roblox_Inventory_Api_Models_CreatorModel = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    type: zod_1.z.enum(['User', 'Group']),
});
const Roblox_Inventory_Api_Models_PlaceModel = zod_1.z.object({
    universeId: zod_1.z.number().int(),
    placeId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    creator: Roblox_Inventory_Api_Models_CreatorModel,
    priceInRobux: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_Models_PlaceModel_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Inventory_Api_Models_PlaceModel),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
/**
 * @api POST https://inventory.roblox.com/v1/collections/items/:itemType/:itemTargetId
 * @summary Adds an item to the appropriate collection
 * @param itemType Type of the item (i.e. Asset, Bundle)
 * @param itemTargetId ID of the item
 */
exports.postCollectionsItemsItemtypeItemtargetid = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/collections/items/:itemType/:itemTargetId',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        itemType: {
            style: 'simple',
        },
        itemTargetId: {
            style: 'simple',
        },
    },
    parameters: {
        itemType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
        itemTargetId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The item type does not exist.
2: The asset does not exist.
3: The bundle does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
4: You don&#x27;t own the specified item.
5: Assets of this type are not allowed in collections.
6: Items of this type are not allowed in collections.
7: The item is already in the collection.
9: The collection is full.`,
        },
    ],
});
/**
 * @api DELETE https://inventory.roblox.com/v1/collections/items/:itemType/:itemTargetId
 * @summary Removes an item to the appropriate collection
 * @param itemType Type of the item (i.e. Asset, Bundle)
 * @param itemTargetId ID of the item
 */
exports.deleteCollectionsItemsItemtypeItemtargetid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/collections/items/:itemType/:itemTargetId',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        itemType: {
            style: 'simple',
        },
        itemTargetId: {
            style: 'simple',
        },
    },
    parameters: {
        itemType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
        itemTargetId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The item type does not exist.
2: The asset does not exist.
3: The bundle does not exist.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
8: The item is not in the collection.`,
        },
    ],
});
/**
 * @api GET https://inventory.roblox.com/v1/packages/:packageId/assets
 * @summary Given a package ID, returns the list of asset IDs for that package
 * @param packageID The asset ID of the package
 */
exports.getPackagesPackageidAssets = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/packages/:packageId/assets',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        packageID: {
            style: 'simple',
        },
    },
    parameters: {
        packageID: zod_1.z.number().int(),
    },
    response: Roblox_Inventory_Api_Models_AssetIdListModel,
    errors: [],
});
/**
 * @api GET https://inventory.roblox.com/v1/users/:userId/assets/collectibles
 * @summary Gets all collectible assets owned by the specified user.
 * @param userId The userid of the owner of the collectibles.
 * @param assetType The asset type for the collectibles you're trying to get.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder Sorted by userAssetId
 */
exports.getUsersUseridAssetsCollectibles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/assets/collectibles',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        assetType: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        assetType: zod_1.z
            .union([
            zod_1.z.literal(1),
            zod_1.z.literal(2),
            zod_1.z.literal(3),
            zod_1.z.literal(4),
            zod_1.z.literal(5),
            zod_1.z.literal(6),
            zod_1.z.literal(7),
            zod_1.z.literal(8),
            zod_1.z.literal(9),
            zod_1.z.literal(10),
            zod_1.z.literal(11),
            zod_1.z.literal(12),
            zod_1.z.literal(13),
            zod_1.z.literal(16),
            zod_1.z.literal(17),
            zod_1.z.literal(18),
            zod_1.z.literal(19),
            zod_1.z.literal(21),
            zod_1.z.literal(22),
            zod_1.z.literal(24),
            zod_1.z.literal(25),
            zod_1.z.literal(26),
            zod_1.z.literal(27),
            zod_1.z.literal(28),
            zod_1.z.literal(29),
            zod_1.z.literal(30),
            zod_1.z.literal(31),
            zod_1.z.literal(32),
            zod_1.z.literal(33),
            zod_1.z.literal(34),
            zod_1.z.literal(35),
            zod_1.z.literal(37),
            zod_1.z.literal(38),
            zod_1.z.literal(39),
            zod_1.z.literal(40),
            zod_1.z.literal(41),
            zod_1.z.literal(42),
            zod_1.z.literal(43),
            zod_1.z.literal(44),
            zod_1.z.literal(45),
            zod_1.z.literal(46),
            zod_1.z.literal(47),
            zod_1.z.literal(48),
            zod_1.z.literal(49),
            zod_1.z.literal(50),
            zod_1.z.literal(51),
            zod_1.z.literal(52),
            zod_1.z.literal(53),
            zod_1.z.literal(54),
            zod_1.z.literal(55),
            zod_1.z.literal(56),
            zod_1.z.literal(59),
            zod_1.z.literal(60),
            zod_1.z.literal(61),
            zod_1.z.literal(62),
            zod_1.z.literal(63),
            zod_1.z.literal(64),
            zod_1.z.literal(65),
            zod_1.z.literal(66),
            zod_1.z.literal(67),
            zod_1.z.literal(68),
            zod_1.z.literal(69),
            zod_1.z.literal(70),
            zod_1.z.literal(71),
            zod_1.z.literal(72),
            zod_1.z.literal(73),
            zod_1.z.literal(74),
            zod_1.z.literal(75),
            zod_1.z.literal(76),
            zod_1.z.literal(77),
            zod_1.z.literal(78),
            zod_1.z.literal(79),
            zod_1.z.literal(80),
            zod_1.z.literal(81),
            zod_1.z.literal(82),
            zod_1.z.literal(83),
            zod_1.z.literal(84),
            zod_1.z.literal(85),
            zod_1.z.literal(86),
            zod_1.z.literal(87),
            zod_1.z.literal(88),
            zod_1.z.literal(89),
            zod_1.z.literal(90),
        ])
            .optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_Models_CollectibleUserAssetModel_,
    errors: [
        {
            status: 400,
            description: `The specified asset type(s) are invalid.`,
        },
        {
            status: 403,
            description: `The specified user&#x27;s inventory is hidden.`,
        },
    ],
});
/**
 * @api GET https://inventory.roblox.com/v1/users/:userId/can-view-inventory
 * @summary Gets whether the specified user's inventory can be viewed.
 * @param userId The user identifier.
 */
exports.getUsersUseridCanViewInventory = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/can-view-inventory',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({ canView: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `1: The specified user does not exist!`,
        },
    ],
});
/**
 * @api GET https://inventory.roblox.com/v1/users/:userId/categories
 * @summary Return inventory categories for a user
 * @param userId
 */
exports.getUsersUseridCategories = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/categories',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Inventory_Api_Models_CategoriesModel,
    errors: [],
});
/**
 * @api GET https://inventory.roblox.com/v1/users/:userId/categories/favorites
 * @summary Return favorites categories for a user
 * @param userId
 */
exports.getUsersUseridCategoriesFavorites = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/categories/favorites',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Inventory_Api_Models_CategoriesModel,
    errors: [],
});
/**
 * @api GET https://inventory.roblox.com/v1/users/:userId/items/:itemType/:itemTargetId
 * @summary Gets owned items of the specified item type. Game Servers can make requests for any user, but can only make requests for game passes that belong to the place sending the request.
Place creators can make requests as if they were the Game Server.
 * @param userId ID of the user in question
 * @param itemType Type of the item in question (i.e. Asset, GamePass, Badge, Bundle)
 * @param itemTargetId ID of the item in question
 */
exports.getUsersUseridItemsItemtypeItemtargetid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/items/:itemType/:itemTargetId',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        itemType: {
            style: 'simple',
        },
        itemTargetId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        itemType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
        itemTargetId: zod_1.z.number().int(),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_Models_IItemModel_,
    errors: [
        {
            status: 400,
            description: `1: The specified user does not exist!
5: The specified game pass does not exist! Are you using the new game pass ID?
6: The specified item type does not exist.
7: The specified Asset does not exist!
10: The specified asset is not a badge!
12: The specified bundle does not exist!`,
        },
    ],
});
/**
 * @api GET https://inventory.roblox.com/v1/users/:userId/items/:itemType/:itemTargetId/is-owned
 * @summary Gets whether a user owns an item of type itemType with id itemTargetId.
 * @param userId ID of the user in question
 * @param itemType Type of the item in question (i.e. Asset, GamePass, Badge, Bundle)
 * @param itemTargetId ID of the item in question
 */
exports.getUsersUseridItemsItemtypeItemtargetidIsOwned = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/items/:itemType/:itemTargetId/is-owned',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        itemType: {
            style: 'simple',
        },
        itemTargetId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        itemType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
        itemTargetId: zod_1.z.number().int(),
    },
    response: zod_1.z.boolean(),
    errors: [
        {
            status: 400,
            description: `1: The specified user does not exist!
5: The specified game pass does not exist! Are you using the new game pass ID?
6: The specified item type does not exist.
7: The specified Asset does not exist!
10: The specified asset is not a badge!
12: The specified bundle does not exist!`,
        },
    ],
});
/**
 * @api GET https://inventory.roblox.com/v1/users/:userId/places/inventory
 * @summary Gets Created, MyGames, or OtherGames places inventory for a user
 * @param userId
 * @param placesTab
 * @param itemsPerPage
 * @param cursor
 */
exports.getUsersUseridPlacesInventory = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/places/inventory',
    baseUrl: 'https://inventory.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        placesTab: {
            style: 'form',
            explode: true,
        },
        itemsPerPage: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        placesTab: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4), zod_1.z.literal(5)]),
        itemsPerPage: zod_1.z.number().int(),
        cursor: zod_1.z.number().int(),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Inventory_Api_Models_PlaceModel_,
    errors: [
        {
            status: 400,
            description: `6: Invalid request`,
        },
        {
            status: 403,
            description: `3: Insufficient permission.`,
        },
    ],
});
