// rozod/lib/endpoints/itemconfigurationv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getTagsPrefixSearch = exports.getTags = exports.getCreationsGetAssets = exports.postCreationsGetAssetDetails = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_ItemConfiguration_Api_AssetCreationsResponse = zod_1.z.object({
    assetId: zod_1.z.number().int(),
    name: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_ItemConfiguration_Api_AssetCreationsResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_ItemConfiguration_Api_AssetCreationsResponse),
});
const Roblox_ItemConfiguration_Api_AssetCreationsDetailsRequest = zod_1.z.object({
    AssetIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_ItemConfiguration_Api_PriceConfigurationModel = zod_1.z.object({
    priceInRobux: zod_1.z.number().int(),
    premiumDiscountPercentage: zod_1.z.number().int(),
    premiumPriceInRobux: zod_1.z.number().int(),
    priceOffset: zod_1.z.number().int(),
});
const Roblox_ItemConfiguration_Api_ReleaseConfigurationResponseModel = zod_1.z.object({
    saleAvailabilityLocations: zod_1.z.array(zod_1.z.enum(['Undefined', 'Catalog', 'AllUniverses', 'MyUniverses'])),
});
const Roblox_ItemConfiguration_Api_AssetCreationsDetailsResponse = zod_1.z.object({
    assetId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    status: zod_1.z.enum([
        'Unknown',
        'ReviewPending',
        'Moderated',
        'ReviewApproved',
        'OnSale',
        'OffSale',
        'DelayedRelease',
        'Free',
    ]),
    description: zod_1.z.string(),
    creatorType: zod_1.z.string(),
    creatorTargetId: zod_1.z.number().int(),
    price: zod_1.z.number().int(),
    priceConfiguration: Roblox_ItemConfiguration_Api_PriceConfigurationModel,
    isArchived: zod_1.z.boolean(),
    assetType: zod_1.z.string(),
    releaseConfiguration: Roblox_ItemConfiguration_Api_ReleaseConfigurationResponseModel,
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
    isDelisted: zod_1.z.boolean(),
    isCreatedForBundle: zod_1.z.boolean(),
});
/**
 * @api POST https://itemconfiguration.roblox.com/v1/creations/get-asset-details
 * @summary Gets the asset status and other configuration details for the given assetIds list.
 * @param body
 */
exports.postCreationsGetAssetDetails = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/creations/get-asset-details',
    baseUrl: 'https://itemconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_ItemConfiguration_Api_AssetCreationsDetailsRequest,
    response: zod_1.z.array(Roblox_ItemConfiguration_Api_AssetCreationsDetailsResponse),
    errors: [
        {
            status: 400,
            description: `1: Missing AssetIds parameters
2: Invalid asset Ids`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 414,
            description: `3: Too many asset Ids`,
        },
        {
            status: 429,
            description: `9: Flood Limit Exceeded`,
        },
        {
            status: 503,
            description: `6: Service Unavailable`,
        },
    ],
});
/**
 * @api GET https://itemconfiguration.roblox.com/v1/creations/get-assets
 * @summary Gets the user created asset information filtered by the given asset type.
 * @param assetType
 * @param isArchived
 * @param groupId
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 */
exports.getCreationsGetAssets = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/creations/get-assets',
    baseUrl: 'https://itemconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetType: {
            style: 'form',
            explode: true,
        },
        isArchived: {
            style: 'form',
            explode: true,
        },
        groupId: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        assetType: zod_1.z.string(),
        isArchived: zod_1.z.boolean().optional(),
        groupId: zod_1.z.number().int().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_ItemConfiguration_Api_AssetCreationsResponse_,
    errors: [
        {
            status: 400,
            description: `5: Invalid assetType
10: Invalid Asset Category`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `7: User does not have necessary permissions for group
8: Asset type does not have necessary permissions for group`,
        },
        {
            status: 429,
            description: `9: Flood Limit Exceeded`,
        },
        {
            status: 503,
            description: `6: Service Unavailable`,
        },
    ],
});
// Patched endpoints removed from Roblox API docs (v6.1.0 compat)
const Roblox_ItemConfiguration_Api_TagDetails = zod_1.z.object({
    tagId: zod_1.z.string(),
    name: zod_1.z.string(),
    localizedDisplayName: zod_1.z.string(),
    status: zod_1.z.enum(['Success', 'MissingItem']),
});
const Patch_ApiArrayResponse_TagDetails = zod_1.z.object({
    data: zod_1.z.array(Roblox_ItemConfiguration_Api_TagDetails),
});
exports.getTags = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/tags',
    baseUrl: 'https://itemconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: { tagIds: { style: 'form' } },
    parameters: { tagIds: zod_1.z.array(zod_1.z.string()) },
    response: Patch_ApiArrayResponse_TagDetails,
    errors: [
        { status: 400, description: `1: No tag Ids requested\n2: Too many tag Ids requested` },
        { status: 429, description: `3: Too many requests` },
    ],
});
exports.getTagsPrefixSearch = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/tags/prefix-search',
    baseUrl: 'https://itemconfiguration.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        prefix: { style: 'form', explode: true },
        numberOfResults: { style: 'form', explode: true },
    },
    parameters: {
        prefix: zod_1.z.string(),
        numberOfResults: zod_1.z.number().int(),
    },
    response: Patch_ApiArrayResponse_TagDetails,
    errors: [
        { status: 400, description: `5: The given prefix is invalid\n6: The number of results requested is invalid` },
        { status: 401, description: `0: Authorization has been denied for this request.` },
        { status: 403, description: `4: This endpoint is not yet enabled for the current user` },
        { status: 429, description: `3: Too many requests` },
    ],
});
