// rozod/lib/endpoints/localev1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getLocalesUserLocalizationLocusSupportedLocales = exports.getLocalesUserLocale = exports.getLocalesSupportedLocalesForCreators = exports.getLocalesSupportedLocales = exports.postLocalesSetUserSupportedLocale = exports.postLocalesSetShowRobloxTranslations = exports.getLocales = exports.getCountryRegions = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Locale_Api_CountryRegion = zod_1.z.object({
    code: zod_1.z.string(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Locale_Api_CountryRegionListResponse = zod_1.z.object({
    countryRegionList: zod_1.z.array(Roblox_Locale_Api_CountryRegion),
});
const Roblox_Locale_Api_Language = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    nativeName: zod_1.z.string(),
    languageCode: zod_1.z.string(),
    isRightToLeft: zod_1.z.boolean(),
});
const Roblox_Locale_Api_SupportedLocale = zod_1.z.object({
    id: zod_1.z.number().int(),
    locale: zod_1.z.string(),
    name: zod_1.z.string(),
    nativeName: zod_1.z.string(),
    language: Roblox_Locale_Api_Language,
});
const Roblox_Locale_Api_SupportedLocaleLocus = zod_1.z.object({
    locale: Roblox_Locale_Api_SupportedLocale,
    isEnabledForFullExperience: zod_1.z.boolean(),
    isEnabledForSignupAndLogin: zod_1.z.boolean(),
    isEnabledForInGameUgc: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Locale_Api_SupportedLocaleLocus_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Locale_Api_SupportedLocaleLocus),
});
const Roblox_Locale_Api_SupportedLocalesResponse = zod_1.z.object({
    supportedLocales: zod_1.z.array(Roblox_Locale_Api_SupportedLocale),
});
const Roblox_Locale_Api_UserLocaleResponse = zod_1.z.object({
    supportedLocale: Roblox_Locale_Api_SupportedLocale,
    nativeLanguage: Roblox_Locale_Api_Language,
});
const Roblox_Locale_Api_UserLocalizationLocusLocalesResponse = zod_1.z.object({
    signupAndLogin: Roblox_Locale_Api_SupportedLocale,
    generalExperience: Roblox_Locale_Api_SupportedLocale,
    ugc: Roblox_Locale_Api_SupportedLocale,
    showRobloxTranslations: zod_1.z.boolean(),
});
const Roblox_Locale_Api_SetShowRobloxTranslationsRequest = zod_1.z.object({
    showRobloxTranslations: zod_1.z.boolean(),
});
const Roblox_Locale_Api_SuccessResponse = zod_1.z.object({ success: zod_1.z.boolean() });
const Roblox_Locale_Api_SetSupportedLocaleForUserRequest = zod_1.z.object({
    supportedLocaleCode: zod_1.z.string(),
});
/**
 * @api GET https://locale.roblox.com/v1/country-regions
 * @summary Get list of country regions sorted by localized name
 * @param locale
 */
exports.getCountryRegions = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/country-regions',
    baseUrl: 'https://locale.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        locale: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        locale: zod_1.z.string().optional(),
    },
    response: Roblox_Locale_Api_CountryRegionListResponse,
    errors: [
        {
            status: 400,
            description: `2: Invalid supported locale code.`,
        },
        {
            status: 403,
            description: `7: Feature is disabled`,
        },
    ],
});
/**
 * @api GET https://locale.roblox.com/v1/locales
 * @summary Get list of Supported locales with user locus information.
 * @param displayValueLocale
 */
exports.getLocales = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/locales',
    baseUrl: 'https://locale.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        displayValueLocale: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        displayValueLocale: zod_1.z.string().optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Locale_Api_SupportedLocaleLocus_,
    errors: [
        {
            status: 403,
            description: `Feature is turned off temporary`,
        },
        {
            status: 500,
            description: `Internal server error`,
        },
    ],
});
/**
 * @api POST https://locale.roblox.com/v1/locales/set-show-roblox-translations
 * @summary Sets whether translations suggested by Roblox will be shown to the user.
 * @param body Whether to show Roblox-suggested translations
 */
exports.postLocalesSetShowRobloxTranslations = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/locales/set-show-roblox-translations',
    baseUrl: 'https://locale.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ showRobloxTranslations: zod_1.z.boolean() }),
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 401,
            description: `Unauthorized
0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Feature is turned off temporary
0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `Internal server error`,
        },
    ],
});
/**
 * @api POST https://locale.roblox.com/v1/locales/set-user-supported-locale
 * @summary Sets user's supported locale.
Null supported locale will clear out user's supported locale (set users' supported locale to null)
 * @param body Supported locale code that needs to be set for user
 */
exports.postLocalesSetUserSupportedLocale = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/locales/set-user-supported-locale',
    baseUrl: 'https://locale.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ supportedLocaleCode: zod_1.z.string() }),
    response: zod_1.z.object({ success: zod_1.z.boolean() }),
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 401,
            description: `Unauthorized
0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `Feature is turned off temporary
0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `Internal server error`,
        },
    ],
});
/**
 * @api GET https://locale.roblox.com/v1/locales/supported-locales
 * @summary Get list of supported locales sorted by the Native Name property.
 */
exports.getLocalesSupportedLocales = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/locales/supported-locales',
    baseUrl: 'https://locale.roblox.com',
    requestFormat: 'json',
    response: Roblox_Locale_Api_SupportedLocalesResponse,
    errors: [
        {
            status: 403,
            description: `Feature is turned off temporary`,
        },
        {
            status: 500,
            description: `Internal server error`,
        },
    ],
});
/**
 * @api GET https://locale.roblox.com/v1/locales/supported-locales-for-creators
 * @param displayValueLocale
 */
exports.getLocalesSupportedLocalesForCreators = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/locales/supported-locales-for-creators',
    baseUrl: 'https://locale.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        displayValueLocale: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        displayValueLocale: zod_1.z.string().optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Locale_Api_SupportedLocaleLocus_,
    errors: [],
});
/**
 * @api GET https://locale.roblox.com/v1/locales/user-locale
 * @summary Gets user locale. If user is absent returns, locale from http request object.
 */
exports.getLocalesUserLocale = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/locales/user-locale',
    baseUrl: 'https://locale.roblox.com',
    requestFormat: 'json',
    response: Roblox_Locale_Api_UserLocaleResponse,
    errors: [
        {
            status: 500,
            description: `Internal server error`,
        },
    ],
});
/**
 * @api GET https://locale.roblox.com/v1/locales/user-localization-locus-supported-locales
 * @summary Gets each of a user's localization locus supported locales. A localization locus supported locale is a page (or group of pages) that
have been defined by the International team which need independent locale support.
If the user is null we will attempt to return the locales appropriate for the user's device language.
 */
exports.getLocalesUserLocalizationLocusSupportedLocales = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/locales/user-localization-locus-supported-locales',
    baseUrl: 'https://locale.roblox.com',
    requestFormat: 'json',
    response: Roblox_Locale_Api_UserLocalizationLocusLocalesResponse,
    errors: [
        {
            status: 500,
            description: `Internal server error`,
        },
    ],
});
