// rozod/lib/endpoints/localizationtablesv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getLocalizationTableTablesTableidEntryCount = exports.postLocalizationTableTablesTableidEntriesTranslationHistory = exports.postLocalizationTableTablesTableidEntriesTranslationFeedback = exports.getLocalizationTableTablesTableidEntries = exports.patchLocalizationTableTablesTableid = exports.getLocalizationTableTablesTableid = exports.getLocalizationTableTablesAssetid = exports.postLocalizationTableTables = exports.getLocalizationTableMetadata = exports.getLocalizationTableLimits = exports.getLocalizationTableAvailableLanguages = exports.getAutolocalizationMetadata = exports.patchAutolocalizationGamesGameidSettings = exports.patchAutolocalizationGamesGameidAutolocalizationtable = exports.postAutolocalizationGamesGameidAutolocalizationtable = exports.patchAutoLocalizationTableGamesGameidIngestion = exports.postAutoLocalizationTableGamesGameidAutoScrapeCleanupRequest = exports.postAutoLocalizationTableGamesGameidAssetsGenerationRequest = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_LocalizationTables_Api_AutoLocalizationMetadataResponse = zod_1.z.object({
    isReactVersionEnabledForAutoLocalizationSettings: zod_1.z.boolean(),
    isTabbedUIEnabledForConfigureLocalizationPage: zod_1.z.boolean(),
    isAutomaticTranslationToggleUIEnabled: zod_1.z.boolean(),
    isAutomaticTranslationQuotaUIEnabled: zod_1.z.boolean(),
});
const Roblox_LocalizationTables_Api_Language = zod_1.z.object({
    name: zod_1.z.string(),
    nativeName: zod_1.z.string(),
    languageCode: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_LocalizationTables_Api_Language_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_LocalizationTables_Api_Language),
});
const Roblox_LocalizationTables_Api_EntryOperationLimits = zod_1.z.object({
    maxContextLength: zod_1.z.number().int(),
    maxKeyLength: zod_1.z.number().int(),
    maxSourceLength: zod_1.z.number().int(),
    maxExampleLength: zod_1.z.number().int(),
    maxGameLocationPathLength: zod_1.z.number().int(),
});
const Roblox_LocalizationTables_Api_TableOperationLimits = zod_1.z.object({
    maxEntriesPerUpdate: zod_1.z.number().int(),
});
const Roblox_LocalizationTables_Api_GetLimitsResponse = zod_1.z.object({
    entryOperationLimits: Roblox_LocalizationTables_Api_EntryOperationLimits,
    tableOperationLimits: Roblox_LocalizationTables_Api_TableOperationLimits,
});
const Roblox_LocalizationTables_Api_LocalizationTablesMetadataResponse = zod_1.z.object({
    isBulkUploadFeatureEnabled: zod_1.z.boolean(),
    isCsvDownloadEnabled: zod_1.z.boolean(),
    isAccessToTranslationMetaDataEnabled: zod_1.z.boolean(),
    isTranslationManagementRedirectionEnabled: zod_1.z.boolean(),
    isUntranslatedFilterEnabled: zod_1.z.boolean(),
    isAutomaticTranslationFilterEnabled: zod_1.z.boolean(),
});
const Roblox_LocalizationTables_Api_GetTableResponse = zod_1.z.object({
    id: zod_1.z.string().uuid(),
    name: zod_1.z.string(),
    ownerType: zod_1.z.enum(['User', 'Group']),
    ownerId: zod_1.z.number().int(),
    assetId: zod_1.z.number().int(),
});
const Roblox_LocalizationTables_Api_EntryIdentifier = zod_1.z.object({
    key: zod_1.z.string(),
    context: zod_1.z.string(),
    source: zod_1.z.string(),
    entryFormat: zod_1.z.enum(['Invalid', 'Legacy', 'Icu']),
});
const Roblox_InGameContentTables_Client_GameLocation = zod_1.z.object({
    path: zod_1.z.string(),
});
const Roblox_LocalizationTables_Api_EntryMetadata = zod_1.z.object({
    example: zod_1.z.string(),
    gameLocations: zod_1.z.array(Roblox_InGameContentTables_Client_GameLocation),
    entryFormat: zod_1.z.enum(['Invalid', 'Legacy', 'Icu']),
});
const Roblox_Localizationtables_Localizationtables_V1_ChangeAgent = zod_1.z.object({
    ChangeAgentType: zod_1.z.enum(['Invalid', 'User', 'Automation', 'Default']),
    Id: zod_1.z.string(),
    OptionalIdCase: zod_1.z.enum(['None', 'Id']),
});
const Roblox_LocalizationTables_Api_PatchTranslation = zod_1.z.object({
    locale: zod_1.z.string(),
    translationText: zod_1.z.string(),
    delete: zod_1.z.boolean(),
    changeAgent: Roblox_Localizationtables_Localizationtables_V1_ChangeAgent,
    updatedTime: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_LocalizationTables_Api_PatchEntry = zod_1.z.object({
    identifier: Roblox_LocalizationTables_Api_EntryIdentifier,
    metadata: Roblox_LocalizationTables_Api_EntryMetadata,
    translations: zod_1.z.array(Roblox_LocalizationTables_Api_PatchTranslation),
    delete: zod_1.z.boolean(),
});
const Roblox_LocalizationTables_Api_UpdateTableContentsRequest = zod_1.z.object({
    name: zod_1.z.string(),
    entries: zod_1.z.array(Roblox_LocalizationTables_Api_PatchEntry),
});
const Roblox_LocalizationTables_Api_Error = zod_1.z.object({
    errorCode: zod_1.z.number().int(),
    errorMessage: zod_1.z.string(),
});
const Roblox_LocalizationTables_Api_Translator = zod_1.z.object({
    id: zod_1.z.number().int(),
    agentType: zod_1.z.enum(['User', 'Automation']),
});
const Roblox_LocalizationTables_Api_Translation = zod_1.z.object({
    locale: zod_1.z.string(),
    translationText: zod_1.z.string(),
    translator: Roblox_LocalizationTables_Api_Translator,
    updatedTime: zod_1.z.string().datetime({ offset: true }),
    feedbackCount: zod_1.z.number().int(),
});
const Roblox_LocalizationTables_Api_FailedEntry = zod_1.z.object({
    error: Roblox_LocalizationTables_Api_Error,
    identifier: Roblox_LocalizationTables_Api_EntryIdentifier,
    metadata: Roblox_LocalizationTables_Api_EntryMetadata,
    translations: zod_1.z.array(Roblox_LocalizationTables_Api_Translation),
    createdTime: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_LocalizationTables_Api_ModifiedEntry = zod_1.z.object({
    identifier: Roblox_LocalizationTables_Api_EntryIdentifier,
    translations: zod_1.z.array(Roblox_LocalizationTables_Api_Translation),
});
const Roblox_LocalizationTables_Api_UpdateTableContentsResponse = zod_1.z.object({
    failedEntriesAndTranslations: zod_1.z.array(Roblox_LocalizationTables_Api_FailedEntry),
    modifiedEntriesAndTranslations: zod_1.z.array(Roblox_LocalizationTables_Api_ModifiedEntry),
});
const Roblox_LocalizationTables_Api_Entry = zod_1.z.object({
    identifier: Roblox_LocalizationTables_Api_EntryIdentifier,
    metadata: Roblox_LocalizationTables_Api_EntryMetadata,
    translations: zod_1.z.array(Roblox_LocalizationTables_Api_Translation),
    createdTime: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_LocalizationTables_Api_GetTableEntriesPagedResponse = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_LocalizationTables_Api_Entry),
});
const Roblox_LocalizationTables_Api_GetTableEntryCountResponse = zod_1.z.object({
    id: zod_1.z.string().uuid(),
    entryCount: zod_1.z.number().int(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_LocalizationTables_Api_RaiseEventForAutoScrapedEntriesCleanupRequest = zod_1.z.object({
    maxAgeForFlush: zod_1.z.string(),
});
const Roblox_LocalizationTables_Api_GameAutolocalizationInformationResponse = zod_1.z.object({
    isAutolocalizationEnabled: zod_1.z.boolean(),
    isAutomaticEntriesSettingEnabled: zod_1.z.boolean(),
    isAutomaticEntriesDeletionEnabled: zod_1.z.boolean(),
    shouldUseLocalizationTable: zod_1.z.boolean(),
    autoLocalizationTableId: zod_1.z.string().uuid(),
    sourceLanguage: zod_1.z.string(),
    assetId: zod_1.z.number().int(),
});
const Roblox_LocalizationTables_Api_SetAutolocalizationTableForGameRequest = zod_1.z.object({ tableId: zod_1.z.string().uuid() });
const Roblox_LocalizationTables_Api_CreateTableRequest = zod_1.z.object({
    name: zod_1.z.string(),
    ownerType: zod_1.z.enum(['User', 'Group']),
    ownerId: zod_1.z.number().int(),
});
const Roblox_LocalizationTables_Api_CreateTableResponse = zod_1.z.object({
    id: zod_1.z.string().uuid(),
    assetId: zod_1.z.number().int(),
});
const Roblox_LocalizationTables_Api_EntryIdentifierWithTranslation = zod_1.z.object({
    translation: Roblox_LocalizationTables_Api_Translation,
    key: zod_1.z.string(),
    context: zod_1.z.string(),
    source: zod_1.z.string(),
    entryFormat: zod_1.z.enum(['Invalid', 'Legacy', 'Icu']),
});
const Roblox_LocalizationTables_Api_GetTableEntriesTranslationFeedbackRequest = zod_1.z.object({
    sourceLocale: zod_1.z.string(),
    entries: zod_1.z.array(Roblox_LocalizationTables_Api_EntryIdentifierWithTranslation),
});
const Roblox_LocalizationTables_Api_EntryTranslationFeedback = zod_1.z.object({
    identifier: Roblox_LocalizationTables_Api_EntryIdentifierWithTranslation,
    feedbackCount: zod_1.z.number().int(),
    playerSuggestionText: zod_1.z.array(zod_1.z.string()),
    reasons: zod_1.z.array(zod_1.z.enum(['None', 'Untranslated', 'Inaccurate', 'SpellingOrGrammar', 'Inappropriate'])),
    robloxSuggestionText: zod_1.z.string(),
});
const Roblox_LocalizationTables_Api_GetTableEntriesTranslationFeedbackResponse = zod_1.z.object({
    tableId: zod_1.z.string().uuid(),
    entries: zod_1.z.array(Roblox_LocalizationTables_Api_EntryTranslationFeedback),
});
const Roblox_LocalizationTables_Api_CursorEntryIdentifier = zod_1.z.object({
    cursor: zod_1.z.string(),
    identifier: Roblox_LocalizationTables_Api_EntryIdentifier,
    count: zod_1.z.number().int(),
    sortOrder: zod_1.z.enum(['Asc', 'Desc']),
});
const Roblox_LocalizationTables_Api_GetTableEntriesTranslationHistoryRequest = zod_1.z.object({
    locale: zod_1.z.string(),
    entries: zod_1.z.array(Roblox_LocalizationTables_Api_CursorEntryIdentifier),
});
const Roblox_LocalizationTables_Api_TranslationHistory = zod_1.z.object({
    translationText: zod_1.z.string(),
    translator: Roblox_LocalizationTables_Api_Translator,
    created: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_LocalizationTables_Api_EntryTranslationHistoryPaged = zod_1.z.object({
    identifier: Roblox_LocalizationTables_Api_EntryIdentifier,
    history: zod_1.z.array(Roblox_LocalizationTables_Api_TranslationHistory),
    nextCursor: zod_1.z.string(),
});
const Roblox_LocalizationTables_Api_FailedEntryTranslationHistoryPaged = zod_1.z.object({
    identifier: Roblox_LocalizationTables_Api_EntryIdentifier,
    count: zod_1.z.number().int(),
    error: Roblox_LocalizationTables_Api_Error,
});
const Roblox_LocalizationTables_Api_GetTableEntriesTranslationHistoryResponse = zod_1.z.object({
    tableId: zod_1.z.string().uuid(),
    locale: zod_1.z.string(),
    entries: zod_1.z.array(Roblox_LocalizationTables_Api_EntryTranslationHistoryPaged),
    failedEntries: zod_1.z.array(Roblox_LocalizationTables_Api_FailedEntryTranslationHistoryPaged),
});
const Roblox_LocalizationTables_Api_MatchedEntry = zod_1.z.object({
    source: zod_1.z.string(),
    matchedParamIndex: zod_1.z.number().int(),
});
const Roblox_LocalizationTables_Api_AutoScrapeEntryMetadata = zod_1.z.object({
    text: zod_1.z.string(),
    userId: zod_1.z.number().int(),
    osPlatform: zod_1.z.string(),
    sessionId: zod_1.z.string().uuid(),
    matchedEntry: Roblox_LocalizationTables_Api_MatchedEntry,
});
const Roblox_LocalizationTables_Api_AutoScrapeEntry = zod_1.z.object({
    context: zod_1.z.string(),
    source: zod_1.z.string(),
    screenshot: zod_1.z.string(),
    meta: Roblox_LocalizationTables_Api_AutoScrapeEntryMetadata,
});
const Roblox_LocalizationTables_Api_IngestContentMetadataPlaceInformation = zod_1.z.object({
    placeId: zod_1.z.number().int(),
    placeVersionNumber: zod_1.z.number().int(),
});
const Roblox_LocalizationTables_Api_IngestContentMetadata = zod_1.z.object({
    placeInformation: Roblox_LocalizationTables_Api_IngestContentMetadataPlaceInformation,
});
const Roblox_LocalizationTables_Api_IngestAutoScrapedContentForGameRequest = zod_1.z.object({
    entries: zod_1.z.array(Roblox_LocalizationTables_Api_AutoScrapeEntry),
    metadata: Roblox_LocalizationTables_Api_IngestContentMetadata,
});
const Roblox_LocalizationTables_Api_SetAutolocalizationSettingsForGameRequest = zod_1.z.object({
    isAutolocalizationEnabled: zod_1.z.boolean(),
    isAutomaticEntriesSettingEnabled: zod_1.z.boolean(),
    isAutomaticEntriesDeletionsEnabled: zod_1.z.boolean(),
    shouldUseLocalizationTable: zod_1.z.boolean(),
});
/**
 * @api POST https://localizationtables.roblox.com/v1/auto-localization-table/games/:gameId/assets-generation-request
 * @summary Generates localization asset of a game.
 * @param gameId The game id.
 */
exports.postAutoLocalizationTableGamesGameidAssetsGenerationRequest = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/auto-localization-table/games/:gameId/assets-generation-request',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `3: Invalid table id.
14: Invalid game id
29: You do not have permission to generate asset for this table.
32: LocalizationTable is not available for the game.
34: Actor provided is invalid`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `24: Too many attempts.Please try again later.`,
        },
    ],
});
/**
 * @api POST https://localizationtables.roblox.com/v1/auto-localization-table/games/:gameId/auto-scrape-cleanup-request
 * @summary Enqueues an event to flush the auto scraped entries which doesn't have translations.
 * @param body
 * @param gameId The id of the game.
 */
exports.postAutoLocalizationTableGamesGameidAutoScrapeCleanupRequest = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/auto-localization-table/games/:gameId/auto-scrape-cleanup-request',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ maxAgeForFlush: zod_1.z.string() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `14: Invalid game id
31: You do not have permission to flush auto scraped entries asset for this game.
32: LocalizationTable is not available for the game.
34: Actor provided is invalid`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `33: Too many attempts to flush the game.Please try again later.`,
        },
    ],
});
/**
 * @api PATCH https://localizationtables.roblox.com/v1/auto-localization-table/games/:gameId/ingestion
 * @summary Ingests entries for auto localization. Needs to be an authorized user.
 * @param body The request body.
 * @param gameId The game id.
 */
exports.patchAutoLocalizationTableGamesGameidIngestion = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/auto-localization-table/games/:gameId/ingestion',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    body: Roblox_LocalizationTables_Api_IngestAutoScrapedContentForGameRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `10: Maximum entries exceeded. Please keep the number of entries per request below the maximum.
13: Request body can&#x27;t be null
16: Entries can&#x27;t be null or empty
34: Actor provided is invalid`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `17: Feature is disabled`,
        },
    ],
});
/**
 * @api POST https://localizationtables.roblox.com/v1/autolocalization/games/:gameId/autolocalizationtable
 * @param gameId
 */
exports.postAutolocalizationGamesGameidAutolocalizationtable = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/autolocalization/games/:gameId/autolocalizationtable',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    response: Roblox_LocalizationTables_Api_GameAutolocalizationInformationResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api PATCH https://localizationtables.roblox.com/v1/autolocalization/games/:gameId/autolocalizationtable
 * @param body
 * @param gameId
 */
exports.patchAutolocalizationGamesGameidAutolocalizationtable = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/autolocalization/games/:gameId/autolocalizationtable',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ tableId: zod_1.z.string().uuid() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api PATCH https://localizationtables.roblox.com/v1/autolocalization/games/:gameId/settings
 * @summary Sets a game's auto-localization related settings
 * @param body The request body.
 * @param gameId The id of the game.
 */
exports.patchAutolocalizationGamesGameidSettings = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/autolocalization/games/:gameId/settings',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    body: Roblox_LocalizationTables_Api_SetAutolocalizationSettingsForGameRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `14: Invalid game id
61: IsAutomaticEntriesSettingEnabled can only be enabled if IsAutolocalizationEnabled is also enabled.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
18: You do not have permission to manage this game`,
        },
        {
            status: 503,
            description: `17: Feature is disabled`,
        },
    ],
});
/**
 * @api GET https://localizationtables.roblox.com/v1/autolocalization/metadata
 * @summary Metadata for AutoLocalization Configuration
 */
exports.getAutolocalizationMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/autolocalization/metadata',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    response: Roblox_LocalizationTables_Api_AutoLocalizationMetadataResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://localizationtables.roblox.com/v1/localization-table/available-languages
 * @summary Gets the supported language codes that can be used by localization tables.
 */
exports.getLocalizationTableAvailableLanguages = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/localization-table/available-languages',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_LocalizationTables_Api_Language_,
    errors: [],
});
/**
 * @api GET https://localizationtables.roblox.com/v1/localization-table/limits
 * @summary Get limits for translation table entries operations
 */
exports.getLocalizationTableLimits = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/localization-table/limits',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    response: Roblox_LocalizationTables_Api_GetLimitsResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://localizationtables.roblox.com/v1/localization-table/metadata
 * @summary Get metadata for localization UI
 */
exports.getLocalizationTableMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/localization-table/metadata',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    response: Roblox_LocalizationTables_Api_LocalizationTablesMetadataResponse,
    errors: [],
});
/**
 * @api POST https://localizationtables.roblox.com/v1/localization-table/tables
 * @summary Creates a Localization Table with the given data.
Note that this endpoint simply creates a table and does not associate it with any universe, so if intending to use this to create tables usable in experience more setup will be needed to grant those experiences access.
 * @param body
 */
exports.postLocalizationTableTables = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/localization-table/tables',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_LocalizationTables_Api_CreateTableRequest,
    response: Roblox_LocalizationTables_Api_CreateTableResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to get this table.`,
        },
    ],
});
/**
 * @api GET https://localizationtables.roblox.com/v1/localization-table/tables/:assetId
 * @summary Get table information by the assetId of the table.
 * @param assetId The asset id associated with the table.
 */
exports.getLocalizationTableTablesAssetid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/localization-table/tables/:assetId',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        assetId: {
            style: 'simple',
        },
    },
    parameters: {
        assetId: zod_1.z.number().int(),
    },
    response: Roblox_LocalizationTables_Api_GetTableResponse,
    errors: [
        {
            status: 400,
            description: `12: Invalid asset id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `2: You do not have permission to get this table.`,
        },
    ],
});
/**
 * @api GET https://localizationtables.roblox.com/v1/localization-table/tables/:tableId
 * @summary Get table information by the id of the table.
 * @param tableId
 */
exports.getLocalizationTableTablesTableid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/localization-table/tables/:tableId',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        tableId: {
            style: 'simple',
        },
    },
    parameters: {
        tableId: zod_1.z.string().uuid(),
    },
    response: Roblox_LocalizationTables_Api_GetTableResponse,
    errors: [
        {
            status: 400,
            description: `3: Invalid table id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `2: You do not have permission to get this table.`,
        },
    ],
});
/**
 * @api PATCH https://localizationtables.roblox.com/v1/localization-table/tables/:tableId
 * @summary Updates the tables contents based on what is provided.
 * @param body The metadata object is optional.
 * @param tableId The table guid for the table to update.
 * @param gameId The game id.
 */
exports.patchLocalizationTableTablesTableid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/localization-table/tables/:tableId',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        tableId: {
            style: 'simple',
        },
        gameId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        tableId: zod_1.z.string().uuid(),
        gameId: zod_1.z.number().int().optional(),
    },
    body: Roblox_LocalizationTables_Api_UpdateTableContentsRequest,
    response: Roblox_LocalizationTables_Api_UpdateTableContentsResponse,
    errors: [
        {
            status: 400,
            description: `3: Invalid table id.
4: Table does not exist.
10: Maximum entries exceeded. Please keep the number of entries per request below the maximum.
13: Request body can&#x27;t be null
14: Invalid game id`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
6: You do not have permission to create this table.`,
        },
        {
            status: 503,
            description: `17: Feature is disabled`,
        },
    ],
});
/**
 * @api GET https://localizationtables.roblox.com/v1/localization-table/tables/:tableId/entries
 * @summary Gets a batch of entries for a table.
 * @param tableId
 * @param cursor If null, there are no more entries in the table and you've reached the last page.
 * @param gameId
 * @param entryFormat
 */
exports.getLocalizationTableTablesTableidEntries = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/localization-table/tables/:tableId/entries',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        tableId: {
            style: 'simple',
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        gameId: {
            style: 'form',
            explode: true,
        },
        entryFormat: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        tableId: zod_1.z.string().uuid(),
        cursor: zod_1.z.string().optional(),
        gameId: zod_1.z.number().int().optional(),
        entryFormat: zod_1.z.enum(['Invalid', 'Legacy', 'Icu']).optional().default('Legacy'),
    },
    response: Roblox_LocalizationTables_Api_GetTableEntriesPagedResponse,
    errors: [
        {
            status: 400,
            description: `3: Invalid table id.
14: Invalid game id`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `2: You do not have permission to get this table.`,
        },
    ],
});
/**
 * @api POST https://localizationtables.roblox.com/v1/localization-table/tables/:tableId/entries/translation-feedback
 * @summary Gets the translation feedback for each entry passed in.
 * @param body A request body containing all relevant data for entry translation feedback lookup.
 * @param tableId The entries' tableId.
 * @param gameId The game id.
 */
exports.postLocalizationTableTablesTableidEntriesTranslationFeedback = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/localization-table/tables/:tableId/entries/translation-feedback',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        tableId: {
            style: 'simple',
        },
        gameId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        tableId: zod_1.z.string().uuid(),
        gameId: zod_1.z.number().int().optional(),
    },
    body: Roblox_LocalizationTables_Api_GetTableEntriesTranslationFeedbackRequest,
    response: Roblox_LocalizationTables_Api_GetTableEntriesTranslationFeedbackResponse,
    errors: [
        {
            status: 400,
            description: `3: Invalid table id.
13: Request body can&#x27;t be null
14: Invalid game id
16: Entries can&#x27;t be null or empty
35: The entries provided are invalid
37: Invalid locale code.
38: Invalid entry identifier.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to get this table.`,
        },
        {
            status: 429,
            description: `24: Too many attempts.Please try again later.`,
        },
        {
            status: 503,
            description: `17: Feature is disabled`,
        },
    ],
});
/**
 * @api POST https://localizationtables.roblox.com/v1/localization-table/tables/:tableId/entries/translation-history
 * @summary Gets the translation history for each entry passed in.
 * @param body A request body containing all relevant data for entry history lookup.
 * @param tableId The entries' tableId.
 * @param gameId The game id.
 */
exports.postLocalizationTableTablesTableidEntriesTranslationHistory = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/localization-table/tables/:tableId/entries/translation-history',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        tableId: {
            style: 'simple',
        },
        gameId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        tableId: zod_1.z.string().uuid(),
        gameId: zod_1.z.number().int().optional(),
    },
    body: Roblox_LocalizationTables_Api_GetTableEntriesTranslationHistoryRequest,
    response: Roblox_LocalizationTables_Api_GetTableEntriesTranslationHistoryResponse,
    errors: [
        {
            status: 400,
            description: `3: Invalid table id.
13: Request body can&#x27;t be null
14: Invalid game id
16: Entries can&#x27;t be null or empty
35: The entries provided are invalid
37: Invalid locale code.
38: Invalid entry identifier.
39: Count should be at least 1.
45: Invalid exclusive start id.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: You do not have permission to get this table.`,
        },
        {
            status: 429,
            description: `24: Too many attempts.Please try again later.`,
        },
        {
            status: 503,
            description: `17: Feature is disabled`,
        },
    ],
});
/**
 * @api GET https://localizationtables.roblox.com/v1/localization-table/tables/:tableId/entry-count
 * @summary Gets the number of entries in the specified table
 * @param tableId The table id
 * @param gameId The game id
 * @param entryFormat
 */
exports.getLocalizationTableTablesTableidEntryCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/localization-table/tables/:tableId/entry-count',
    baseUrl: 'https://localizationtables.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        tableId: {
            style: 'simple',
        },
        gameId: {
            style: 'form',
            explode: true,
        },
        entryFormat: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        tableId: zod_1.z.string().uuid(),
        gameId: zod_1.z.number().int().optional(),
        entryFormat: zod_1.z.enum(['Invalid', 'Legacy', 'Icu']).optional().default('Invalid'),
    },
    response: Roblox_LocalizationTables_Api_GetTableEntryCountResponse,
    errors: [
        {
            status: 400,
            description: `3: Invalid table id.
14: Invalid game id`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `2: You do not have permission to get this table.`,
        },
    ],
});
