// rozod/lib/endpoints/matchmakingv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMatchmakingUniverseUniverseidFeatureFlags = exports.getMatchmakingServerAttributesUniverseid = exports.patchMatchmakingServerAttributeAttributeid = exports.deleteMatchmakingServerAttributeAttributeid = exports.postMatchmakingServerAttribute = exports.getMatchmakingScoringConfigurationsUniverseidPlaces = exports.getMatchmakingScoringConfigurationsUniverseid = exports.deleteMatchmakingScoringConfigurationPlacePlaceid = exports.postMatchmakingScoringConfigurationPlace = exports.getMatchmakingScoringConfigurationDefaultWeights = exports.patchMatchmakingScoringConfigurationScoringconfigurationidSignalsSignalname = exports.deleteMatchmakingScoringConfigurationScoringconfigurationidSignalsSignalname = exports.postMatchmakingScoringConfigurationScoringconfigurationidSignals = exports.patchMatchmakingScoringConfigurationScoringconfigurationid = exports.deleteMatchmakingScoringConfigurationScoringconfigurationid = exports.getMatchmakingScoringConfigurationScoringconfigurationid = exports.postMatchmakingScoringConfiguration = exports.getMatchmakingPlayerAttributesUniverseid = exports.patchMatchmakingPlayerAttributeAttributeid = exports.deleteMatchmakingPlayerAttributeAttributeid = exports.postMatchmakingPlayerAttribute = exports.postGameInstancesShutdownAll = exports.postGameInstancesShutdown = exports.postClientStatus = exports.getClientStatus = void 0;
const zod_1 = require("zod");
const __1 = require("..");
/**
 * @api GET https://matchmaking.roblox.com/v1/client-status
 * @summary Get the client-status
 */
exports.getClientStatus = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/client-status',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api POST https://matchmaking.roblox.com/v1/client-status
 * @summary Set the client-status
 */
exports.postClientStatus = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/client-status',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api POST https://matchmaking.roblox.com/v1/game-instances/shutdown
 * @summary Shutdown game instances.
 */
exports.postGameInstancesShutdown = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/game-instances/shutdown',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api POST https://matchmaking.roblox.com/v1/game-instances/shutdown-all
 * @summary Shutdown all game instances.
 */
exports.postGameInstancesShutdownAll = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/game-instances/shutdown-all',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api POST https://matchmaking.roblox.com/v1/matchmaking/player-attribute
 * @summary Create a PlayerAttributeDefinition.
 */
exports.postMatchmakingPlayerAttribute = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/matchmaking/player-attribute',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api DELETE https://matchmaking.roblox.com/v1/matchmaking/player-attribute/:attributeId
 * @summary Delete the PlayerAttributeDefinition specified by attributeId.
 * @param attributeId
 */
exports.deleteMatchmakingPlayerAttributeAttributeid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/matchmaking/player-attribute/:attributeId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        attributeId: {
            style: 'simple',
        },
    },
    parameters: {
        attributeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api PATCH https://matchmaking.roblox.com/v1/matchmaking/player-attribute/:attributeId
 * @summary Update the PlayerAttributeDefinition specified by attributeId.
 * @param attributeId
 */
exports.patchMatchmakingPlayerAttributeAttributeid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/matchmaking/player-attribute/:attributeId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        attributeId: {
            style: 'simple',
        },
    },
    parameters: {
        attributeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://matchmaking.roblox.com/v1/matchmaking/player-attributes/:universeId
 * @summary List all PlayerAttributeDefinitions of a universe.
 * @param universeId
 */
exports.getMatchmakingPlayerAttributesUniverseid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/matchmaking/player-attributes/:universeId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api POST https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration
 * @summary Creates a matchmaking scoring configuration.
 */
exports.postMatchmakingScoringConfiguration = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/matchmaking/scoring-configuration',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration/:scoringConfigurationId
 * @summary Updates a matchmaking scoring configuration.
 * @param scoringConfigurationId
 */
exports.getMatchmakingScoringConfigurationScoringconfigurationid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/matchmaking/scoring-configuration/:scoringConfigurationId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        scoringConfigurationId: {
            style: 'simple',
        },
    },
    parameters: {
        scoringConfigurationId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api DELETE https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration/:scoringConfigurationId
 * @summary Deletes a matchmaking scoring configuration.
 * @param scoringConfigurationId
 */
exports.deleteMatchmakingScoringConfigurationScoringconfigurationid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/matchmaking/scoring-configuration/:scoringConfigurationId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        scoringConfigurationId: {
            style: 'simple',
        },
    },
    parameters: {
        scoringConfigurationId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api PATCH https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration/:scoringConfigurationId
 * @summary Updates a matchmaking scoring configuration.
 * @param scoringConfigurationId
 */
exports.patchMatchmakingScoringConfigurationScoringconfigurationid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/matchmaking/scoring-configuration/:scoringConfigurationId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        scoringConfigurationId: {
            style: 'simple',
        },
    },
    parameters: {
        scoringConfigurationId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api POST https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration/:scoringConfigurationId/signals
 * @summary Creates a matchmaking scoring configuration signal.
 * @param scoringConfigurationId
 */
exports.postMatchmakingScoringConfigurationScoringconfigurationidSignals = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/matchmaking/scoring-configuration/:scoringConfigurationId/signals',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        scoringConfigurationId: {
            style: 'simple',
        },
    },
    parameters: {
        scoringConfigurationId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api DELETE https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration/:scoringConfigurationId/signals/:signalName
 * @summary Deletes a matchmaking scoring configuration custom signal.
 * @param scoringConfigurationId
 * @param signalName
 */
exports.deleteMatchmakingScoringConfigurationScoringconfigurationidSignalsSignalname = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/matchmaking/scoring-configuration/:scoringConfigurationId/signals/:signalName',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        scoringConfigurationId: {
            style: 'simple',
        },
        signalName: {
            style: 'simple',
        },
    },
    parameters: {
        scoringConfigurationId: zod_1.z.unknown(),
        signalName: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api PATCH https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration/:scoringConfigurationId/signals/:signalName
 * @summary Updates a matchmaking scoring configuration signal.
 * @param scoringConfigurationId
 * @param signalName
 */
exports.patchMatchmakingScoringConfigurationScoringconfigurationidSignalsSignalname = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/matchmaking/scoring-configuration/:scoringConfigurationId/signals/:signalName',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        scoringConfigurationId: {
            style: 'simple',
        },
        signalName: {
            style: 'simple',
        },
    },
    parameters: {
        scoringConfigurationId: zod_1.z.unknown(),
        signalName: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration/default-weights
 */
exports.getMatchmakingScoringConfigurationDefaultWeights = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/matchmaking/scoring-configuration/default-weights',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api POST https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration/place
 * @summary Sets a matchmaking scoring configuration for a place.
 */
exports.postMatchmakingScoringConfigurationPlace = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/matchmaking/scoring-configuration/place',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api DELETE https://matchmaking.roblox.com/v1/matchmaking/scoring-configuration/place/:placeId
 * @summary Removes the matchmaking scoring configuration for a place.
 * @param placeId
 */
exports.deleteMatchmakingScoringConfigurationPlacePlaceid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/matchmaking/scoring-configuration/place/:placeId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        placeId: {
            style: 'simple',
        },
    },
    parameters: {
        placeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://matchmaking.roblox.com/v1/matchmaking/scoring-configurations/:universeId
 * @summary List all matchmaking scoring configurations for a universe.
 * @param universeId
 */
exports.getMatchmakingScoringConfigurationsUniverseid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/matchmaking/scoring-configurations/:universeId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://matchmaking.roblox.com/v1/matchmaking/scoring-configurations/:universeId/places
 * @summary List all places with a matchmaking scoring configuration for a universe.
 * @param universeId
 */
exports.getMatchmakingScoringConfigurationsUniverseidPlaces = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/matchmaking/scoring-configurations/:universeId/places',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api POST https://matchmaking.roblox.com/v1/matchmaking/server-attribute
 * @summary Create a ServerAttributeDefinition.
 */
exports.postMatchmakingServerAttribute = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/matchmaking/server-attribute',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api DELETE https://matchmaking.roblox.com/v1/matchmaking/server-attribute/:attributeId
 * @summary Delete the ServerAttributeDefinition specified by attributeId.
 * @param attributeId
 */
exports.deleteMatchmakingServerAttributeAttributeid = (0, __1.endpoint)({
    method: 'DELETE',
    path: '/v1/matchmaking/server-attribute/:attributeId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        attributeId: {
            style: 'simple',
        },
    },
    parameters: {
        attributeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api PATCH https://matchmaking.roblox.com/v1/matchmaking/server-attribute/:attributeId
 * @summary Update the ServerAttributeDefinition specified by attributeId.
 * @param attributeId
 */
exports.patchMatchmakingServerAttributeAttributeid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/matchmaking/server-attribute/:attributeId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        attributeId: {
            style: 'simple',
        },
    },
    parameters: {
        attributeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://matchmaking.roblox.com/v1/matchmaking/server-attributes/:universeId
 * @summary List all ServerAttributeDefinitions of a universe.
 * @param universeId
 */
exports.getMatchmakingServerAttributesUniverseid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/matchmaking/server-attributes/:universeId',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://matchmaking.roblox.com/v1/matchmaking/universe/:universeId/feature-flags
 * @summary Gets feature flags for a customized matchmaking for a given universe.
 * @param universeId
 */
exports.getMatchmakingUniverseUniverseidFeatureFlags = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/matchmaking/universe/:universeId/feature-flags',
    baseUrl: 'https://matchmaking.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeId: {
            style: 'simple',
        },
    },
    parameters: {
        universeId: zod_1.z.unknown(),
    },
    response: zod_1.z.void(),
    errors: [],
});
