// rozod/lib/endpoints/metricsv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getThumbnailsMetadata = exports.postThumbnailsLoad = exports.postBundleMetricsReport = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Metrics_Api_ThumbnailLoadMetadataResponse = zod_1.z.object({
    logRatio: zod_1.z.number(),
});
const Roblox_Metrics_Api_RecordBundleLoadRequest = zod_1.z.object({
    bundleUrl: zod_1.z.string().url(),
    bundleName: zod_1.z.string(),
    loadTimeInMilliseconds: zod_1.z.number().int(),
    cdnProviderName: zod_1.z.string(),
    loadState: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]),
    bundleContentType: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Metrics_Api_RecordThumbnailLoadRequest = zod_1.z.object({
    duration: zod_1.z.number().int(),
    loadState: zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)]),
    thumbnailType: zod_1.z.string(),
});
/**
 * @api POST https://metrics.roblox.com/v1/bundle-metrics/report
 * @summary Records bundle load successes.
 * @param body The Roblox.Metrics.Api.RecordBundleLoadRequest.
 * @description Xsrf protection disabled because this endpoint is supposed to be used
to record static content. It's possible XSRF can't be retried if:
- The XSRF retry JavaScript fails to load (or fails to execute)
- CSS fails to load before JavaScript finishes loading.
 */
exports.postBundleMetricsReport = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/bundle-metrics/report',
    baseUrl: 'https://metrics.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Metrics_Api_RecordBundleLoadRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: The bundle url is invalid.`,
        },
    ],
});
/**
 * @api POST https://metrics.roblox.com/v1/thumbnails/load
 * @summary Records the time it takes for a thumbnail to get loaded in the UI.
 * @param body The Roblox.Metrics.Api.RecordThumbnailLoadRequest.
 */
exports.postThumbnailsLoad = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/thumbnails/load',
    baseUrl: 'https://metrics.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Metrics_Api_RecordThumbnailLoadRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: A required parameter is missing from the request`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://metrics.roblox.com/v1/thumbnails/metadata
 * @summary Get metadata related to logging thumbnail load metrics.
e.g. ThumbnailLoadLoggingRatio
 */
exports.getThumbnailsMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/thumbnails/metadata',
    baseUrl: 'https://metrics.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ logRatio: zod_1.z.number() }),
    errors: [],
});
