// rozod/lib/endpoints/notificationsv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getStreamNotificationsUnreadCount = exports.getStreamNotificationsMetadata = exports.getStreamNotificationsGetRecent = exports.getStreamNotificationsGetLatestGameUpdates = exports.postStreamNotificationsClearUnread = exports.postPushNotificationsRegisterIosPushkit = exports.postPushNotificationsRegisterIosNative = exports.postPushNotificationsRegisterAndroidNative = exports.getPushNotificationsMetadata = exports.getPushNotificationsGetDestinations = exports.getPushNotificationsGetCurrentDeviceDestination = exports.postPushNotificationsDeregisterCurrentDeviceIosPushkit = exports.postPushNotificationsDeregisterCurrentDevice = exports.postPushNotificationsDeregisterAllDevices = exports.getPushNotificationsChromeManifest = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Api_Notifications_Models_ChromeManifestModel = zod_1.z.object({
    name: zod_1.z.string(),
    gcm_sender_id: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_NotificationUser = zod_1.z.object({
    name: zod_1.z.string(),
    userId: zod_1.z.number().int(),
});
const Roblox_Api_Notifications_Models_UserPushDestination = zod_1.z.object({
    user: Roblox_Api_Notifications_Models_NotificationUser,
    name: zod_1.z.string(),
    notificationToken: zod_1.z.string(),
    supportsUpdateNotifications: zod_1.z.boolean(),
    userPushNotificationDestinationId: zod_1.z.number().int(),
    application: zod_1.z.string(),
    platform: zod_1.z.enum([
        'ChromeOnDesktop',
        'AndroidNative',
        'FirefoxOnDesktop',
        'IOSNative',
        'AndroidAmazon',
        'IOSTencent',
        'AndroidTencentService',
        'IOSPushKit',
    ]),
});
const Roblox_Api_Notifications_Models_GetCurrentPushDestinationResponseModel = zod_1.z.object({
    destination: Roblox_Api_Notifications_Models_UserPushDestination,
    statusMessage: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_GetPushDestinationsResponseModel = zod_1.z.object({
    destinations: zod_1.z.array(Roblox_Api_Notifications_Models_UserPushDestination),
    statusMessage: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_PushNotificationClientMetadata = zod_1.z.object({
    notificationId: zod_1.z.string().uuid(),
    type: zod_1.z.string(),
    detail: zod_1.z.object({}),
    fallbackDelivered: zod_1.z.boolean(),
});
const Roblox_Api_Notifications_Models_GetMetadataResponseModel = zod_1.z.object({
    metadata: Roblox_Api_Notifications_Models_PushNotificationClientMetadata,
    statusMessage: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_GameUpdateNotificationModel = zod_1.z.object({
    universeId: zod_1.z.number().int(),
    rootPlaceId: zod_1.z.number().int(),
    createdOn: zod_1.z.string().datetime({ offset: true }),
    createdOnKey: zod_1.z.string(),
    content: zod_1.z.string(),
    universeName: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_NotificationStreamEntriesModel = zod_1.z.object({
    id: zod_1.z.string().uuid(),
    notificationSourceType: zod_1.z.enum([
        'Test',
        'FriendRequestReceived',
        'FriendRequestAccepted',
        'PartyInviteReceived',
        'PartyMemberJoined',
        'ChatNewMessage',
        'PrivateMessageReceived',
        'UserAddedToPrivateServerWhiteList',
        'ConversationUniverseChanged',
        'TeamCreateInvite',
        'GameUpdate',
        'DeveloperMetricsAvailable',
        'GroupJoinRequestAccepted',
        'Sendr',
        'ExperienceInvitation',
    ]),
    eventDate: zod_1.z.string().datetime({ offset: true }),
    timestamp: zod_1.z.string(),
    isInteracted: zod_1.z.boolean(),
    metadataCollection: zod_1.z.array(zod_1.z.object({})),
    eventCount: zod_1.z.number().int(),
    content: zod_1.z.object({}).passthrough(),
});
const Roblox_Api_Notifications_Models_ResponseModels_NotificationStreamMetadataResponse = zod_1.z.object({
    bannerDismissTimeSpan: zod_1.z.number().int(),
    signalRDisconnectionResponseInMilliseconds: zod_1.z.number().int(),
    canLaunchGameFromGameUpdate: zod_1.z.boolean(),
});
const Roblox_Api_Notifications_Models_UnreadStreamNotificationsModel = zod_1.z.object({
    unreadNotifications: zod_1.z.number().int(),
    statusMessage: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_SuccessResponseModel = zod_1.z.object({
    statusMessage: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_RegisterAndroidRequestModel = zod_1.z.object({
    notificationToken: zod_1.z.string(),
    authorizeForUser: zod_1.z.boolean(),
    oldNotificationToken: zod_1.z.string(),
    deviceName: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_PushNotificationRegistration = zod_1.z.object({
    userPushNotificationDestinationId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    notificationToken: zod_1.z.string(),
    application: zod_1.z.string(),
    platform: zod_1.z.enum([
        'ChromeOnDesktop',
        'AndroidNative',
        'FirefoxOnDesktop',
        'IOSNative',
        'AndroidAmazon',
        'IOSTencent',
        'AndroidTencentService',
        'IOSPushKit',
    ]),
});
const Roblox_Api_Notifications_Models_RegistrationResponseModel = zod_1.z.object({
    registration: Roblox_Api_Notifications_Models_PushNotificationRegistration,
    statusMessage: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_RegisterIOSNativeRequestModel = zod_1.z.object({
    notificationToken: zod_1.z.string(),
    destinationIdentifier: zod_1.z.string(),
    authorizeForUser: zod_1.z.boolean(),
    oldNotificationToken: zod_1.z.string(),
    deviceName: zod_1.z.string(),
});
const Roblox_Api_Notifications_Models_RegisterIOSPushKitRequestModel = zod_1.z.object({
    notificationToken: zod_1.z.string(),
    destinationIdentifier: zod_1.z.string(),
    authorizeForUser: zod_1.z.boolean(),
    oldNotificationToken: zod_1.z.string(),
    deviceName: zod_1.z.string(),
});
/**
 * @api GET https://notifications.roblox.com/v2/push-notifications/chrome-manifest
 * @summary Get Chrome Manifest to link GCM project to Chrome Browser
 */
exports.getPushNotificationsChromeManifest = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/push-notifications/chrome-manifest',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    response: Roblox_Api_Notifications_Models_ChromeManifestModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://notifications.roblox.com/v2/push-notifications/deregister-all-devices
 * @summary De-register all devices to disable push notifications
 */
exports.postPushNotificationsDeregisterAllDevices = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/push-notifications/deregister-all-devices',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ statusMessage: zod_1.z.string() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://notifications.roblox.com/v2/push-notifications/deregister-current-device
 * @summary De-register current device to disable push notifications
 */
exports.postPushNotificationsDeregisterCurrentDevice = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/push-notifications/deregister-current-device',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ statusMessage: zod_1.z.string() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://notifications.roblox.com/v2/push-notifications/deregister-current-device-ios-pushkit
 * @summary De-register current device to disable pushkit notifications
 */
exports.postPushNotificationsDeregisterCurrentDeviceIosPushkit = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/push-notifications/deregister-current-device-ios-pushkit',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ statusMessage: zod_1.z.string() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://notifications.roblox.com/v2/push-notifications/get-current-device-destination
 * @summary Gets the current device destination
 */
exports.getPushNotificationsGetCurrentDeviceDestination = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/push-notifications/get-current-device-destination',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    response: Roblox_Api_Notifications_Models_GetCurrentPushDestinationResponseModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://notifications.roblox.com/v2/push-notifications/get-destinations
 * @summary Gets valid destinations associated with the signed user
 */
exports.getPushNotificationsGetDestinations = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/push-notifications/get-destinations',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    response: Roblox_Api_Notifications_Models_GetPushDestinationsResponseModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://notifications.roblox.com/v2/push-notifications/metadata
 * @summary Gets the corresponding metadata for the specified notification
 * @param notificationToken Token for the notification
 * @param notificationId Id of the specified notification
 */
exports.getPushNotificationsMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/push-notifications/metadata',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        notificationToken: {
            style: 'form',
            explode: true,
        },
        notificationId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        notificationToken: zod_1.z.string(),
        notificationId: zod_1.z.string().uuid(),
    },
    response: Roblox_Api_Notifications_Models_GetMetadataResponseModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://notifications.roblox.com/v2/push-notifications/register-android-native
 * @summary Register Android Native for push notifications
 * @param body
 */
exports.postPushNotificationsRegisterAndroidNative = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/push-notifications/register-android-native',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Api_Notifications_Models_RegisterAndroidRequestModel,
    response: Roblox_Api_Notifications_Models_RegistrationResponseModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://notifications.roblox.com/v2/push-notifications/register-ios-native
 * @summary Registers IOS device for push notifications
 * @param body
 */
exports.postPushNotificationsRegisterIosNative = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/push-notifications/register-ios-native',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Api_Notifications_Models_RegisterIOSNativeRequestModel,
    response: Roblox_Api_Notifications_Models_RegistrationResponseModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://notifications.roblox.com/v2/push-notifications/register-ios-pushkit
 * @summary Registers IOS device for pushkit notifications
 * @param body
 */
exports.postPushNotificationsRegisterIosPushkit = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/push-notifications/register-ios-pushkit',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Api_Notifications_Models_RegisterIOSPushKitRequestModel,
    response: Roblox_Api_Notifications_Models_RegistrationResponseModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://notifications.roblox.com/v2/stream-notifications/clear-unread
 * @summary Clears the unread Notification stream count
 */
exports.postStreamNotificationsClearUnread = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/stream-notifications/clear-unread',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ statusMessage: zod_1.z.string() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://notifications.roblox.com/v2/stream-notifications/get-latest-game-updates
 * @summary Get the latest non aggregated Game Updates sent to the logged in user
 * @param universeIds List of universe IDs
 * @param sinceDateTime For retrieving only updates that created after a time point.
 */
exports.getStreamNotificationsGetLatestGameUpdates = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/stream-notifications/get-latest-game-updates',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        universeIds: {
            style: 'form',
            explode: true,
        },
        sinceDateTime: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        universeIds: zod_1.z.array(zod_1.z.number()),
        sinceDateTime: zod_1.z.string().datetime({ offset: true }).optional(),
    },
    response: zod_1.z.array(Roblox_Api_Notifications_Models_GameUpdateNotificationModel),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://notifications.roblox.com/v2/stream-notifications/get-recent
 * @summary Gets the recent entries from the notification stream
 * @param startIndex Index to start the entries from. (Optional : Defaults to 0 which means the most recent entry)
 * @param maxRows Number of entries to be returned. (Optional : Defaults to 10 entries)
 */
exports.getStreamNotificationsGetRecent = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/stream-notifications/get-recent',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        startIndex: {
            style: 'form',
            explode: true,
        },
        maxRows: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        startIndex: zod_1.z.number().int().optional(),
        maxRows: zod_1.z.number().int().optional().default(10),
    },
    response: zod_1.z.array(Roblox_Api_Notifications_Models_NotificationStreamEntriesModel),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://notifications.roblox.com/v2/stream-notifications/metadata
 * @summary Get Notification Stream metadata.
 */
exports.getStreamNotificationsMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/stream-notifications/metadata',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    response: Roblox_Api_Notifications_Models_ResponseModels_NotificationStreamMetadataResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://notifications.roblox.com/v2/stream-notifications/unread-count
 * @summary Gets the count of unread Notification stream entries
 */
exports.getStreamNotificationsUnreadCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/stream-notifications/unread-count',
    baseUrl: 'https://notifications.roblox.com',
    requestFormat: 'json',
    response: Roblox_Api_Notifications_Models_UnreadStreamNotificationsModel,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
