// rozod/lib/endpoints/premiumfeaturesv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridValidateMembership = exports.getUsersUseridPremiumUpsellPrecheck = void 0;
const zod_1 = require("zod");
const __1 = require("..");
/**
 * @api GET https://premiumfeatures.roblox.com/v1/users/:userId/premium-upsell-precheck
 * @summary Premium upsell precheck
 * @param userId User ID
 * @param universeId
 * @param placeId
 */
exports.getUsersUseridPremiumUpsellPrecheck = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/premium-upsell-precheck',
    baseUrl: 'https://premiumfeatures.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        universeId: {
            style: 'form',
            explode: true,
        },
        placeId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        universeId: zod_1.z.number().int(),
        placeId: zod_1.z.number().int(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://premiumfeatures.roblox.com/v1/users/:userId/validate-membership
 * @summary Get if a user has a Premium membership
 * @param userId User ID
 */
exports.getUsersUseridValidateMembership = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/validate-membership',
    baseUrl: 'https://premiumfeatures.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: zod_1.z.void(),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
