// rozod/lib/endpoints/presencev1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postPresenceUsers = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const UserPresenceRequest = zod_1.z.object({
    userIds: zod_1.z.array(zod_1.z.number()).nullable(),
});
const presence_users_body = UserPresenceRequest;
const PresenceType = zod_1.z.union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3)]);
const UserPresence = zod_1.z.object({
    userPresenceType: PresenceType,
    lastLocation: zod_1.z.string().nullable(),
    placeId: zod_1.z.number().int().nullable(),
    rootPlaceId: zod_1.z.number().int().nullable(),
    gameId: zod_1.z.string().uuid().nullable(),
    universeId: zod_1.z.number().int().nullable(),
    userId: zod_1.z.number().int(),
});
const UserPresencesResponse = zod_1.z.object({
    userPresences: zod_1.z.array(UserPresence).nullable(),
});
const Error = zod_1.z.object({
    code: zod_1.z.number().int(),
    message: zod_1.z.string().nullable(),
});
const ErrorResponse = zod_1.z.object({ errors: zod_1.z.array(Error).nullable() });
/**
 * @api POST https://presence.roblox.com/v1/presence/users
 * @param body
 */
exports.postPresenceUsers = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/presence/users',
    baseUrl: 'https://presence.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: presence_users_body,
    response: UserPresencesResponse,
    errors: [
        {
            status: 400,
            description: `Bad Request`,
        },
        {
            status: 403,
            description: `Forbidden`,
        },
        {
            status: 429,
            description: `Too Many Requests`,
        },
    ],
});
