// rozod/lib/endpoints/privatemessagesv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMessagesUnreadCount = exports.postMessagesUnarchive = exports.postMessagesMarkUnread = exports.postMessagesMarkRead = exports.postMessagesArchive = exports.getMessagesMessageid = exports.getMessages = exports.getAnnouncementsMetadata = exports.getAnnouncements = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_PrivateMessages_Api_Models_VerifiedSkinnyUserResponse = zod_1.z.object({
    hasVerifiedBadge: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_PrivateMessages_Api_Models_AnnouncementsDetailsResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    sender: Roblox_PrivateMessages_Api_Models_VerifiedSkinnyUserResponse,
    subject: zod_1.z.string(),
    body: zod_1.z.string(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_PrivateMessages_Api_Models_GetAnnouncementsResponse = zod_1.z.object({
    collection: zod_1.z.array(Roblox_PrivateMessages_Api_Models_AnnouncementsDetailsResponse),
    totalCollectionSize: zod_1.z.number().int(),
});
const Roblox_PrivateMessages_Api_Models_AnnouncementsMetadataResponse = zod_1.z.object({
    numOfAnnouncements: zod_1.z.number().int(),
});
const Roblox_PrivateMessages_Api_Models_MessageDetailsResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    sender: Roblox_PrivateMessages_Api_Models_VerifiedSkinnyUserResponse,
    recipient: Roblox_PrivateMessages_Api_Models_VerifiedSkinnyUserResponse,
    subject: zod_1.z.string(),
    body: zod_1.z.string(),
    created: zod_1.z.string().datetime({ offset: true }),
    updated: zod_1.z.string().datetime({ offset: true }),
    isRead: zod_1.z.boolean(),
    isSystemMessage: zod_1.z.boolean(),
    isReportAbuseDisplayed: zod_1.z.boolean(),
});
const Roblox_PrivateMessages_Api_Models_GetMessagesResponse = zod_1.z.object({
    collection: zod_1.z.array(Roblox_PrivateMessages_Api_Models_MessageDetailsResponse),
    totalCollectionSize: zod_1.z.number().int(),
    totalPages: zod_1.z.number().int(),
    pageNumber: zod_1.z.number().int(),
});
const Roblox_PrivateMessages_Api_Models_UnreadMessagesCountResponse = zod_1.z.object({
    count: zod_1.z.number().int(),
});
const Roblox_PrivateMessages_Api_Models_BatchMessagesRequest = zod_1.z.object({
    messageIds: zod_1.z.array(zod_1.z.number()),
});
const Roblox_PrivateMessages_Api_Models_FailedMessageResponse = zod_1.z.object({
    messageId: zod_1.z.number().int(),
    errorMessage: zod_1.z.string(),
});
const Roblox_PrivateMessages_Api_Models_BatchMessagesResponse = zod_1.z.object({
    failedMessages: zod_1.z.array(Roblox_PrivateMessages_Api_Models_FailedMessageResponse),
});
/**
 * @api GET https://privatemessages.roblox.com/v1/announcements
 * @summary Migrate from RobloxWebsite project, return news notification for Private Message page
 */
exports.getAnnouncements = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/announcements',
    baseUrl: 'https://privatemessages.roblox.com',
    requestFormat: 'json',
    response: Roblox_PrivateMessages_Api_Models_GetAnnouncementsResponse,
    errors: [
        {
            status: 400,
            description: `2: Message does not exist or the current user is not authorized to view it.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://privatemessages.roblox.com/v1/announcements/metadata
 */
exports.getAnnouncementsMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/announcements/metadata',
    baseUrl: 'https://privatemessages.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ numOfAnnouncements: zod_1.z.number().int() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://privatemessages.roblox.com/v1/messages
 * @summary Gets a user's messages.
 * @param pageNumber
 * @param pageSize
 * @param messageTab
 */
exports.getMessages = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/messages',
    baseUrl: 'https://privatemessages.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        pageNumber: {
            style: 'form',
            explode: true,
        },
        pageSize: {
            style: 'form',
            explode: true,
        },
        messageTab: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        pageNumber: zod_1.z.number().int().optional(),
        pageSize: zod_1.z.number().int().optional().default(20),
        messageTab: zod_1.z.enum(['Inbox', 'Sent', 'Archive']).optional().default('Inbox'),
    },
    response: Roblox_PrivateMessages_Api_Models_GetMessagesResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://privatemessages.roblox.com/v1/messages/:messageId
 * @summary Gets a message's details.
 * @param messageId
 */
exports.getMessagesMessageid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/messages/:messageId',
    baseUrl: 'https://privatemessages.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        messageId: {
            style: 'simple',
        },
    },
    parameters: {
        messageId: zod_1.z.number().int(),
    },
    response: Roblox_PrivateMessages_Api_Models_MessageDetailsResponse,
    errors: [
        {
            status: 400,
            description: `2: Message does not exist or the current user is not authorized to view it.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://privatemessages.roblox.com/v1/messages/archive
 * @summary Archives a batch of messages.
 * @param body
 */
exports.postMessagesArchive = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/messages/archive',
    baseUrl: 'https://privatemessages.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_PrivateMessages_Api_Models_BatchMessagesRequest,
    response: Roblox_PrivateMessages_Api_Models_BatchMessagesResponse,
    errors: [
        {
            status: 400,
            description: `5: Too many ids in a batch request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://privatemessages.roblox.com/v1/messages/mark-read
 * @summary Marks a batch of messages as read.
 * @param body
 */
exports.postMessagesMarkRead = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/messages/mark-read',
    baseUrl: 'https://privatemessages.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_PrivateMessages_Api_Models_BatchMessagesRequest,
    response: Roblox_PrivateMessages_Api_Models_BatchMessagesResponse,
    errors: [
        {
            status: 400,
            description: `5: Too many ids in a batch request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://privatemessages.roblox.com/v1/messages/mark-unread
 * @summary Marks a batch of messages as unread.
 * @param body
 */
exports.postMessagesMarkUnread = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/messages/mark-unread',
    baseUrl: 'https://privatemessages.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_PrivateMessages_Api_Models_BatchMessagesRequest,
    response: Roblox_PrivateMessages_Api_Models_BatchMessagesResponse,
    errors: [
        {
            status: 400,
            description: `5: Too many ids in a batch request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api POST https://privatemessages.roblox.com/v1/messages/unarchive
 * @summary Unarchives a batch of messages.
 * @param body
 */
exports.postMessagesUnarchive = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/messages/unarchive',
    baseUrl: 'https://privatemessages.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_PrivateMessages_Api_Models_BatchMessagesRequest,
    response: Roblox_PrivateMessages_Api_Models_BatchMessagesResponse,
    errors: [
        {
            status: 400,
            description: `5: Too many ids in a batch request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://privatemessages.roblox.com/v1/messages/unread/count
 * @summary Gets unread messages for the authenticated user.
 */
exports.getMessagesUnreadCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/messages/unread/count',
    baseUrl: 'https://privatemessages.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
