// rozod/lib/endpoints/publishv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postPluginsPluginidIcon = exports.postGamesGameidThumbnailImage = exports.postBadgesBadgeidIcon = exports.postAudioVerify = exports.postAudio = exports.getAssetQuotas = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Publish_Api_AssetQuota = zod_1.z.object({
    duration: zod_1.z.string(),
    usage: zod_1.z.number().int(),
    capacity: zod_1.z.number().int(),
    expirationTime: zod_1.z.string(),
});
const Roblox_Publish_Api_AssetQuotasResponse = zod_1.z.object({
    quotas: zod_1.z.array(Roblox_Publish_Api_AssetQuota),
});
const Roblox_Publish_Api_UploadAudioRequest = zod_1.z.object({
    name: zod_1.z.string(),
    file: zod_1.z.string(),
    groupId: zod_1.z.number().int(),
    paymentSource: zod_1.z.string(),
    estimatedFileSize: zod_1.z.number().int(),
    estimatedDuration: zod_1.z.number(),
    assetPrivacy: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2)]),
});
const Roblox_Publish_Api_PublishAudioResponse = zod_1.z.object({
    Id: zod_1.z.number().int(),
    Name: zod_1.z.string(),
});
const Roblox_Publish_Api_VerifyAudioRequest = zod_1.z.object({
    name: zod_1.z.string(),
    file: zod_1.z.string(),
    groupId: zod_1.z.number().int(),
    paymentSource: zod_1.z.string(),
    fileSize: zod_1.z.number().int(),
    duration: zod_1.z.number(),
});
const Roblox_Publish_Api_VerifyAudioResponse = zod_1.z.object({
    name: zod_1.z.string(),
    price: zod_1.z.number().int(),
    balance: zod_1.z.number().int(),
    canAfford: zod_1.z.boolean(),
});
const badgeId_icon_body = zod_1.z.object({ Files: zod_1.z.instanceof(File) });
const Roblox_Publish_Api_UploadResponse = zod_1.z.object({
    targetId: zod_1.z.number().int(),
});
const thumbnail_image_body = zod_1.z.object({ Files: zod_1.z.instanceof(File) });
const pluginId_icon_body = zod_1.z.object({ Files: zod_1.z.instanceof(File) });
/**
 * @api GET https://publish.roblox.com/v1/asset-quotas
 * @summary List asset quotas of the given resource type and asset type.
 * @param resourceType Resource type of the asset quota
 * @param assetType Asset type of the asset quota
 * @param useDummyData Use dummy data for testing. This is for internal use only
 */
exports.getAssetQuotas = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/asset-quotas',
    baseUrl: 'https://publish.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        resourceType: {
            style: 'form',
            explode: true,
        },
        assetType: {
            style: 'form',
            explode: true,
        },
        useDummyData: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        resourceType: zod_1.z.string(),
        assetType: zod_1.z.string(),
        useDummyData: zod_1.z.boolean().optional(),
    },
    response: Roblox_Publish_Api_AssetQuotasResponse,
    errors: [
        {
            status: 400,
            description: `7: The asset type is not appropriate for this request.
8: The resource type is not appropriate for this request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 500,
            description: `0: Reserved for base level errors. Do not use in your endpoint directly, do not document.`,
        },
    ],
});
/**
 * @api POST https://publish.roblox.com/v1/audio
 * @summary Published an audio file and returns the new asset info.
 * @param body The file upload request body. Roblox.Publish.Api.UploadAudioRequest
 */
exports.postAudio = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/audio',
    baseUrl: 'https://publish.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Publish_Api_UploadAudioRequest,
    response: Roblox_Publish_Api_PublishAudioResponse,
    errors: [
        {
            status: 400,
            description: `3: The request did not contain a file to be uploaded.
4: The file in the request is too large.
5: The duration of the audio file is too long.
8: The file type is not supported.
9: The file is corrupted
11: Missing permissions to spend group funds.
14: The user/group does not have suffiecient funds to publish.
14: The user/group does not have suffiecient funds to publish.
15: The audio file has already been reviewed and rejected.
18: Too many requests. Try again later.
20: Error while trying to purchase the product.
22: The file size estimation error was greater than the acceptable margin of error.
23: The duration estimation error was greater than the acceptable margin of error.
24: Asset privacy is invalid.
29: Invalid argument in the request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
1: The request did not include an authorization.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `7: Failed to parse the file.
19: Asset creation was unavailable. Please try again.`,
        },
    ],
});
/**
 * @api POST https://publish.roblox.com/v1/audio/verify
 * @summary Verifies an audio file and returns a product that you can purchase to publish the audio file.
 * @param body The verify audio request body. Roblox.Publish.Api.VerifyAudioRequest
 */
exports.postAudioVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/audio/verify',
    baseUrl: 'https://publish.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Publish_Api_VerifyAudioRequest,
    response: Roblox_Publish_Api_VerifyAudioResponse,
    errors: [
        {
            status: 400,
            description: `3: The request did not contain a file to be uploaded.
4: The file in the request is too large.
5: The duration of the audio file is too long.
8: The file type is not supported.
9: The file is corrupted
18: Too many requests. Try again later.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
1: The request did not include an authorization.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 500,
            description: `7: Failed to parse the file.`,
        },
    ],
});
/**
 * @api POST https://publish.roblox.com/v1/badges/:badgeId/icon
 * @summary Overwrites a badge icon with a new one.
 * @param body
 * @param badgeId The badge Id.
 */
exports.postBadgesBadgeidIcon = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/badges/:badgeId/icon',
    baseUrl: 'https://publish.roblox.com',
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        badgeId: {
            style: 'simple',
        },
    },
    parameters: {
        badgeId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ Files: zod_1.z.instanceof(File) }),
    response: zod_1.z.object({ targetId: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `2: File not present in request.
12: Name or description is moderated.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
5: You do not have permission to manage this item.`,
        },
        {
            status: 404,
            description: `4: Target item is invalid or does not exist.`,
        },
        {
            status: 429,
            description: `3: You&#x27;re uploading too much, please wait and try again later.`,
        },
    ],
});
/**
 * @api POST https://publish.roblox.com/v1/games/:gameId/thumbnail/image
 * @summary Uploads a game thumbnail.
 * @param body
 * @param gameId The universe Id.
 */
exports.postGamesGameidThumbnailImage = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/games/:gameId/thumbnail/image',
    baseUrl: 'https://publish.roblox.com',
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ Files: zod_1.z.instanceof(File) }),
    response: zod_1.z.object({ targetId: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `1: File uploaded does not match known image format. Try converting to png.
2: File not present in request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
5: You do not have permission to manage this item.`,
        },
        {
            status: 404,
            description: `4: Target item is invalid or does not exist.`,
        },
        {
            status: 429,
            description: `3: You&#x27;re uploading too much, please wait and try again later.`,
        },
    ],
});
/**
 * @api POST https://publish.roblox.com/v1/plugins/:pluginId/icon
 * @summary Overwrites a plugin icon with a new one.
 * @param body
 * @param pluginId The plugin Id.
 */
exports.postPluginsPluginidIcon = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/plugins/:pluginId/icon',
    baseUrl: 'https://publish.roblox.com',
    requestFormat: 'form-data',
    serializationMethod: {
        body: {},
        pluginId: {
            style: 'simple',
        },
    },
    parameters: {
        pluginId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ Files: zod_1.z.instanceof(File) }),
    response: zod_1.z.object({ targetId: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `2: File not present in request.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
5: You do not have permission to manage this item.`,
        },
        {
            status: 404,
            description: `4: Target item is invalid or does not exist.`,
        },
        {
            status: 429,
            description: `3: You&#x27;re uploading too much, please wait and try again later.`,
        },
    ],
});
