// rozod/lib/endpoints/thumbnailsresizerv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSecureresizeThumbprintHashWidthHeightTypeFormatFiltertype = exports.getResizeHashWidthHeightTypeFormatFiltertype = void 0;
const zod_1 = require("zod");
const __1 = require("..");
/**
 * @api GET https://thumbnailsresizer.roblox.com/v1/resize/:hash/:width/:height/:type/:format/:filterType
 * @summary Resizes larger thumbnails to specified size and format
 * @param hash Hash of larger thumbnail
 * @param width Desired width of thumbnail
 * @param height Desired height of thumbnail
 * @param type Thumbnail Type
 * @param format Desired image format of the thumbnail
 * @param filterType E.g. is output circular
 * @param shouldModify
 */
exports.getResizeHashWidthHeightTypeFormatFiltertype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/resize/:hash/:width/:height/:type/:format/:filterType',
    baseUrl: 'https://thumbnailsresizer.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        hash: {
            style: 'simple',
        },
        width: {
            style: 'simple',
        },
        height: {
            style: 'simple',
        },
        type: {
            style: 'simple',
        },
        format: {
            style: 'simple',
        },
        filterType: {
            style: 'form',
            explode: true,
        },
        shouldModify: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        hash: zod_1.z.string(),
        width: zod_1.z.number().int(),
        height: zod_1.z.number().int(),
        type: zod_1.z.string(),
        format: zod_1.z.string(),
        filterType: zod_1.z.string(),
        shouldModify: zod_1.z.boolean().optional(),
    },
    response: zod_1.z.void(),
    errors: [],
});
/**
 * @api GET https://thumbnailsresizer.roblox.com/v1/secureresize/:thumbPrint/:hash/:width/:height/:type/:format/:filterType
 * @summary Decrypts and Resizes larger thumbnails to specified size and format
 * @param thumbPrint The thumbPrint that represents the key version
 * @param hash Hash of larger thumbnail
 * @param width Desired width of thumbnail
 * @param height Desired height of thumbnail
 * @param type Thumbnail Type
 * @param format Desired image format of the thumbnail
 * @param filterType E.g. is output circular
 */
exports.getSecureresizeThumbprintHashWidthHeightTypeFormatFiltertype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/secureresize/:thumbPrint/:hash/:width/:height/:type/:format/:filterType',
    baseUrl: 'https://thumbnailsresizer.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        thumbPrint: {
            style: 'simple',
        },
        hash: {
            style: 'simple',
        },
        width: {
            style: 'simple',
        },
        height: {
            style: 'simple',
        },
        type: {
            style: 'simple',
        },
        format: {
            style: 'simple',
        },
        filterType: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        thumbPrint: zod_1.z.string(),
        hash: zod_1.z.string(),
        width: zod_1.z.number().int(),
        height: zod_1.z.number().int(),
        type: zod_1.z.string(),
        format: zod_1.z.string(),
        filterType: zod_1.z.string().optional(),
    },
    response: zod_1.z.void(),
    errors: [],
});
