// rozod/lib/endpoints/tradesv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersUseridCanTradeWith = exports.postTradesSend = exports.getTradesMetadata = exports.postTradesExpireOutdated = exports.getTradesTradestatustypeCount = exports.getTradesTradestatustype = exports.postTradesTradeidDecline = exports.postTradesTradeidCounter = exports.postTradesTradeidAccept = exports.getTradesTradeid = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_Responses_Users_SkinnyUserResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Trades_Api_UserAssetResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    serialNumber: zod_1.z.number().int(),
    assetId: zod_1.z.number().int(),
    name: zod_1.z.string(),
    recentAveragePrice: zod_1.z.number().int(),
    originalPrice: zod_1.z.number().int(),
    assetStock: zod_1.z.number().int(),
    membershipType: zod_1.z.enum(['None', 'BC', 'TBC', 'OBC', 'RobloxPremium']),
});
const Roblox_Trades_Api_TradeOfferResponse = zod_1.z.object({
    user: Roblox_Web_Responses_Users_SkinnyUserResponse,
    userAssets: zod_1.z.array(Roblox_Trades_Api_UserAssetResponse),
    robux: zod_1.z.number().int(),
});
const Roblox_Trades_Api_TradeDetailResponse = zod_1.z.object({
    offers: zod_1.z.array(Roblox_Trades_Api_TradeOfferResponse),
    id: zod_1.z.number().int(),
    user: Roblox_Web_Responses_Users_SkinnyUserResponse,
    created: zod_1.z.string().datetime({ offset: true }),
    expiration: zod_1.z.string().datetime({ offset: true }),
    isActive: zod_1.z.boolean(),
    status: zod_1.z.enum([
        'Unknown',
        'Open',
        'Pending',
        'Completed',
        'Expired',
        'Declined',
        'RejectedDueToError',
        'Countered',
        'Processing',
        'InterventionRequired',
        'TwoStepVerificationRequired',
    ]),
});
const Roblox_Trades_Api_TradeResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    user: Roblox_Web_Responses_Users_SkinnyUserResponse,
    created: zod_1.z.string().datetime({ offset: true }),
    expiration: zod_1.z.string().datetime({ offset: true }),
    isActive: zod_1.z.boolean(),
    status: zod_1.z.enum([
        'Unknown',
        'Open',
        'Pending',
        'Completed',
        'Expired',
        'Declined',
        'RejectedDueToError',
        'Countered',
        'Processing',
        'InterventionRequired',
        'TwoStepVerificationRequired',
    ]),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Trades_Api_TradeResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Trades_Api_TradeResponse),
});
const Roblox_Trades_Api_TradeCountResponse = zod_1.z.object({
    count: zod_1.z.number().int(),
});
const Roblox_Trades_Api_TradeMetadata = zod_1.z.object({
    maxItemsPerSide: zod_1.z.number().int(),
    minValueRatio: zod_1.z.number(),
    tradeSystemMaxRobuxPercent: zod_1.z.number(),
    tradeSystemRobuxFee: zod_1.z.number(),
});
const Roblox_Trades_Api_CanTradeResponse = zod_1.z.object({
    canTrade: zod_1.z.boolean(),
    status: zod_1.z.enum([
        'Unknown',
        'CanTrade',
        'CannotTradeWithSelf',
        'SenderCannotTrade',
        'ReceiverCannotTrade',
        'SenderPrivacyTooStrict',
        'UsersCannotTrade',
        'TradeAccepterNeedsFriction',
    ]),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Trades_Api_TradeOfferRequest = zod_1.z.object({
    userId: zod_1.z.number().int(),
    userAssetIds: zod_1.z.array(zod_1.z.number()),
    robux: zod_1.z.number().int(),
});
const Roblox_Trades_Api_TradeRequest = zod_1.z.object({
    offers: zod_1.z.array(Roblox_Trades_Api_TradeOfferRequest),
});
const Roblox_Trades_Api_NewTradeResponse = zod_1.z.object({ id: zod_1.z.number().int() });
/**
 * @api GET https://trades.roblox.com/v1/trades/:tradeId
 * @summary Gets detailed information about a trade.
 * @param tradeId The trade id.
 */
exports.getTradesTradeid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/trades/:tradeId',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        tradeId: {
            style: 'simple',
        },
    },
    parameters: {
        tradeId: zod_1.z.number().int(),
    },
    response: Roblox_Trades_Api_TradeDetailResponse,
    errors: [
        {
            status: 400,
            description: `2: The trade cannot be found or you are not authorized to view it.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
4: You are not authorized to modify this trade.`,
        },
        {
            status: 404,
            description: `2: The trade cannot be found or you are not authorized to view it.`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api POST https://trades.roblox.com/v1/trades/:tradeId/accept
 * @summary Accepts a trade.
 * @param tradeId The trade id.
 */
exports.postTradesTradeidAccept = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/trades/:tradeId/accept',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        tradeId: {
            style: 'simple',
        },
    },
    parameters: {
        tradeId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: The trade cannot be found or you are not authorized to view it.
3: The trade is inactive.
4: You are not authorized to modify this trade.
6: Trade needs to be confirmed by the other party.
6: Trade needs to be confirmed by the other party.
7: The user cannot trade. See field for whether the user who cannot trade is the sender or receiver.
23: The trade reaches Two Step Verification thresholds and the user has not verified in the past time threshold.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `5: Trading system is unavailable`,
        },
    ],
});
/**
 * @api POST https://trades.roblox.com/v1/trades/:tradeId/counter
 * @summary Counters a trade.
 * @param body The trade request.
 * @param tradeId The trade id.
 */
exports.postTradesTradeidCounter = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/trades/:tradeId/counter',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        tradeId: {
            style: 'simple',
        },
    },
    parameters: {
        tradeId: zod_1.z.number().int(),
    },
    body: Roblox_Trades_Api_TradeRequest,
    response: zod_1.z.object({ id: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `2: The trade cannot be found or you are not authorized to view it.
4: You are not authorized to modify this trade.
7: The user cannot trade. See field for whether the user who cannot trade is the sender or receiver.
8: The trade request should include offers.
9: Invalid number of trade offers.
10: Invalid trade partner. See field for whether the invalid partner is the sender or receiver.
11: Cannot add negative Robux amounts to a trade.
12: One or more userAssets are invalid. See fieldData for details.
13: Invalid number of userAssets in one side of the trade.
15: The trade is unbalanced.
16: Trade value ratio is not sufficient.
17: You have insufficient Robux to make this offer.
18: Too many Robux in one side of the offer. See field for whether the side is the sender or receiver.
19: Unknown error while processing the trade.
21: Cannot trade with yourself.
22: User&#x27;s privacy settings are too strict to allow trading. See field for whether the user is the sender or receiver.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `14: You are sending too many trade requests. Please slow down and try again later.`,
        },
        {
            status: 502,
            description: `0: An unknown error occured.`,
        },
        {
            status: 503,
            description: `5: Trading system is unavailable`,
        },
    ],
});
/**
 * @api POST https://trades.roblox.com/v1/trades/:tradeId/decline
 * @summary Declines a trade.
 * @param tradeId The trade id.
 */
exports.postTradesTradeidDecline = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/trades/:tradeId/decline',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        tradeId: {
            style: 'simple',
        },
    },
    parameters: {
        tradeId: zod_1.z.number().int(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: The trade cannot be found or you are not authorized to view it.
3: The trade is inactive.
4: You are not authorized to modify this trade.
7: The user cannot trade. See field for whether the user who cannot trade is the sender or receiver.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `5: Trading system is unavailable`,
        },
    ],
});
/**
 * @api GET https://trades.roblox.com/v1/trades/:tradeStatusType
 * @summary Fetches a list of the authenticated user's trades.
 * @param tradeStatusType The trade status type.
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder Sorted by trade creation date
 */
exports.getTradesTradestatustype = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/trades/:tradeStatusType',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        tradeStatusType: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        tradeStatusType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Trades_Api_TradeResponse_,
    errors: [
        {
            status: 400,
            description: `1: Invalid trade status type.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://trades.roblox.com/v1/trades/:tradeStatusType/count
 * @summary Gets the total number of pending trades for the authenticated user.
Inbound is the only accepted tradeStatusType.
 * @param tradeStatusType The trade status type to fetch a total count for.
 */
exports.getTradesTradestatustypeCount = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/trades/:tradeStatusType/count',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        tradeStatusType: {
            style: 'simple',
        },
    },
    parameters: {
        tradeStatusType: zod_1.z.union([zod_1.z.literal(1), zod_1.z.literal(2), zod_1.z.literal(3), zod_1.z.literal(4)]),
    },
    response: zod_1.z.object({ count: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `1: Invalid trade status type.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://trades.roblox.com/v1/trades/expire-outdated
 * @summary Deprecated. TradeSession are automatically set to expire while the inbound/outbound trades are fetched.
Expires Outdated Inbound Trades for User
 */
exports.postTradesExpireOutdated = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/trades/expire-outdated',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({}),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
    ],
});
/**
 * @api GET https://trades.roblox.com/v1/trades/metadata
 * @summary Gets metadata about the trade system.
 */
exports.getTradesMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/trades/metadata',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    response: Roblox_Trades_Api_TradeMetadata,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://trades.roblox.com/v1/trades/send
 * @summary Sends a trade.
 * @param body The trade request.
 */
exports.postTradesSend = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/trades/send',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Trades_Api_TradeRequest,
    response: zod_1.z.object({ id: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `7: The user cannot trade. See field for whether the user who cannot trade is the sender or receiver.
8: The trade request should include offers.
9: Invalid number of trade offers.
10: Invalid trade partner. See field for whether the invalid partner is the sender or receiver.
11: Cannot add negative Robux amounts to a trade.
12: One or more userAssets are invalid. See fieldData for details.
13: Invalid number of userAssets in one side of the trade.
15: The trade is unbalanced.
16: Trade value ratio is not sufficient.
17: You have insufficient Robux to make this offer.
18: Too many Robux in one side of the offer. See field for whether the side is the sender or receiver.
19: Unknown error while processing the trade.
21: Cannot trade with yourself.
22: User&#x27;s privacy settings are too strict to allow trading. See field for whether the user is the sender or receiver.
23: The trade reaches Two Step Verification thresholds and the user has not verified in the past time threshold.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `14: You are sending too many trade requests. Please slow down and try again later.`,
        },
        {
            status: 502,
            description: `0: An unknown error occured.`,
        },
        {
            status: 503,
            description: `5: Trading system is unavailable`,
        },
    ],
});
/**
 * @api GET https://trades.roblox.com/v1/users/:userId/can-trade-with
 * @summary Returns whether you can trade with another user.
 * @param userId The other user's id.
 */
exports.getUsersUseridCanTradeWith = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/can-trade-with',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Trades_Api_CanTradeResponse,
    errors: [
        {
            status: 400,
            description: `10: Invalid trade partner. See field for whether the invalid partner is the sender or receiver.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
