// rozod/lib/endpoints/tradesv2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersMeCanTrade = exports.getUsersUseridTradableitems = exports.getUsersUseridCanTradeWith = exports.postTradesSend = exports.postTradesTradeidCounter = exports.getTradesTradeid = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_Responses_Users_SkinnyUserResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Trades_Api_Models_V2_ItemTarget = zod_1.z.object({
    itemType: zod_1.z.enum(['Unknown', 'Asset', 'Bundle']),
    targetId: zod_1.z.string(),
});
const Roblox_Trades_Api_Models_V2_TradableItemInstance = zod_1.z.object({
    collectibleItemInstanceId: zod_1.z.string(),
    itemTarget: Roblox_Trades_Api_Models_V2_ItemTarget,
    itemName: zod_1.z.string(),
    serialNumber: zod_1.z.number().int(),
    originalPrice: zod_1.z.number().int(),
    recentAveragePrice: zod_1.z.number().int(),
    assetStock: zod_1.z.number().int(),
    isOnHold: zod_1.z.boolean(),
});
const Roblox_Trades_Api_Models_V2_TradeOffer = zod_1.z.object({
    user: Roblox_Web_Responses_Users_SkinnyUserResponse,
    robux: zod_1.z.number().int(),
    items: zod_1.z.array(Roblox_Trades_Api_Models_V2_TradableItemInstance),
});
const Roblox_Trades_Api_Models_V2_TradeDetailsResponse = zod_1.z.object({
    tradeId: zod_1.z.number().int(),
    status: zod_1.z.enum([
        'Unknown',
        'Open',
        'Pending',
        'Completed',
        'Expired',
        'Declined',
        'RejectedDueToError',
        'Countered',
        'Processing',
        'InterventionRequired',
        'TwoStepVerificationRequired',
    ]),
    participantAOffer: Roblox_Trades_Api_Models_V2_TradeOffer,
    participantBOffer: Roblox_Trades_Api_Models_V2_TradeOffer,
});
const Roblox_Trades_Api_Models_V2_CanTradeWithResponse = zod_1.z.object({
    userId: zod_1.z.number().int(),
    targetUserId: zod_1.z.number().int(),
    canTrade: zod_1.z.boolean(),
    mutualTradeEligibility: zod_1.z.enum([
        'Unknown',
        'Eligible',
        'CallingUserIneligible',
        'TargetUserIneligible',
        'CannotTradeWithSelf',
        'CallingUserPrivacySettingsRestricted',
    ]),
});
const Roblox_Trades_Api_Models_V2_TradableItem = zod_1.z.object({
    collectibleItemId: zod_1.z.string(),
    itemTarget: Roblox_Trades_Api_Models_V2_ItemTarget,
    itemName: zod_1.z.string(),
    originalPrice: zod_1.z.number().int(),
    recentAveragePrice: zod_1.z.number().int(),
    assetStock: zod_1.z.number().int(),
    instances: zod_1.z.array(Roblox_Trades_Api_Models_V2_TradableItemInstance),
});
const Roblox_Trades_Api_Models_V2_GetUserTradableItemsResponse = zod_1.z.object({
    userId: zod_1.z.number().int(),
    items: zod_1.z.array(Roblox_Trades_Api_Models_V2_TradableItem),
    nextPageCursor: zod_1.z.string(),
});
const Roblox_Trades_Api_Models_V2_CanTradeResponse = zod_1.z.object({
    userId: zod_1.z.number().int(),
    canTrade: zod_1.z.boolean(),
    tradeEligibility: zod_1.z.enum([
        'Unknown',
        'Eligible',
        'IneligibleTradeSystemDisabled',
        'IneligibleCannotTradeWithRoblox',
        'IneligibleUserNotFound',
        'IneligibleMissingPremiumMembership',
        'IneligibleLegalOrRegulatoryRestrictions',
    ]),
});
const Roblox_Trades_Api_Models_V2_TradeOfferRequest = zod_1.z.object({
    userId: zod_1.z.number().int(),
    robux: zod_1.z.number().int(),
    collectibleItemInstanceIds: zod_1.z.array(zod_1.z.string()),
});
const Roblox_Trades_Api_Models_V2_TradeRequest = zod_1.z.object({
    senderOffer: Roblox_Trades_Api_Models_V2_TradeOfferRequest,
    recipientOffer: Roblox_Trades_Api_Models_V2_TradeOfferRequest,
});
const Roblox_Trades_Api_Models_V2_NewTradeResponse = zod_1.z.object({
    tradeId: zod_1.z.number().int(),
});
/**
 * @api GET https://trades.roblox.com/v2/trades/:tradeId
 * @summary Gets the details of a trade.
 * @param tradeId The id of the trade.
 */
exports.getTradesTradeid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/trades/:tradeId',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        tradeId: {
            style: 'simple',
        },
    },
    parameters: {
        tradeId: zod_1.z.number().int(),
    },
    response: Roblox_Trades_Api_Models_V2_TradeDetailsResponse,
    errors: [
        {
            status: 400,
            description: `0: An unknown error occured.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
4: You are not authorized to modify this trade.`,
        },
        {
            status: 403,
            description: `4: You are not authorized to modify this trade.`,
        },
        {
            status: 404,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api POST https://trades.roblox.com/v2/trades/:tradeId/counter
 * @summary Counters an existing trade.
 * @param body
 * @param tradeId
 */
exports.postTradesTradeidCounter = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/trades/:tradeId/counter',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        tradeId: {
            style: 'simple',
        },
    },
    parameters: {
        tradeId: zod_1.z.number().int(),
    },
    body: Roblox_Trades_Api_Models_V2_TradeRequest,
    response: zod_1.z.object({ tradeId: zod_1.z.number().int() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
4: You are not authorized to modify this trade.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 404,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api POST https://trades.roblox.com/v2/trades/send
 * @summary Sends a new trade.
 * @param body
 */
exports.postTradesSend = (0, __1.endpoint)({
    method: 'POST',
    path: '/v2/trades/send',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Trades_Api_Models_V2_TradeRequest,
    response: zod_1.z.object({ tradeId: zod_1.z.number().int() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
4: You are not authorized to modify this trade.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 404,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api GET https://trades.roblox.com/v2/users/:userId/can-trade-with
 * @summary Checks if the user can trade with a specific user.
 * @param userId
 */
exports.getUsersUseridCanTradeWith = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/users/:userId/can-trade-with',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Trades_Api_Models_V2_CanTradeWithResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
4: You are not authorized to modify this trade.`,
        },
    ],
});
/**
 * @api GET https://trades.roblox.com/v2/users/:userId/tradableItems
 * @summary Gets tradable items for a user.
 * @param userId The id of the user.
 * @param search Optional search query to filter items by.
 * @param itemTargetTypes Optional list of item target types to filter by.
 * @param sortBy The key to sort tradable items by.
 * @param sortOrder The sort order for the tradable items.
 * @param limit The maximum number of items to return.
 * @param cursor The pagination cursor.
 */
exports.getUsersUseridTradableitems = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/users/:userId/tradableItems',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        search: {
            style: 'form',
            explode: true,
        },
        itemTargetTypes: {
            style: 'form',
            explode: true,
        },
        sortBy: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        search: zod_1.z.string().optional(),
        itemTargetTypes: zod_1.z
            .array(zod_1.z.enum([
            'Unknown',
            'HatAccessory',
            'HairAccessory',
            'FaceAccessory',
            'NeckAccessory',
            'ShoulderAccessory',
            'FrontAccessory',
            'BackAccessory',
            'WaistAccessory',
            'Gear',
            'Face',
            'JacketAccessory',
            'SweaterAccessory',
            'DressSkirtAccessory',
            'Character',
            'DynamicHead',
            'Shoes',
            'Animation',
        ]))
            .optional(),
        sortBy: zod_1.z.enum(['Unknown', 'CreationTime', 'AcquisitionTime']).optional().default('CreationTime'),
        sortOrder: zod_1.z
            .union([zod_1.z.literal(0), zod_1.z.literal(1), zod_1.z.literal(2)])
            .optional()
            .default(2),
        limit: zod_1.z.number().int().optional().default(25),
        cursor: zod_1.z.string().optional(),
    },
    response: Roblox_Trades_Api_Models_V2_GetUserTradableItemsResponse,
    errors: [
        {
            status: 400,
            description: `25: The cursor provided is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
4: You are not authorized to modify this trade.`,
        },
        {
            status: 403,
            description: `4: You are not authorized to modify this trade.`,
        },
        {
            status: 404,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api GET https://trades.roblox.com/v2/users/me/can-trade
 * @summary Checks if the calling user can trade with others.
 */
exports.getUsersMeCanTrade = (0, __1.endpoint)({
    method: 'GET',
    path: '/v2/users/me/can-trade',
    baseUrl: 'https://trades.roblox.com',
    requestFormat: 'json',
    response: Roblox_Trades_Api_Models_V2_CanTradeResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.
4: You are not authorized to modify this trade.`,
        },
    ],
});
