// rozod/lib/endpoints/translationrolesv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getGameLocalizationRolesRolesRoleCurrentUser = exports.getGameLocalizationRolesGamesGameidRolesRoleAssignees = exports.getGameLocalizationRolesGamesGameidCurrentUserRoles = exports.patchGameLocalizationRolesGamesGameid = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Web_WebAPI_Models_ApiArrayResponse_System_String_ = zod_1.z.object({
    data: zod_1.z.array(zod_1.z.string()),
});
const Roblox_TranslationRoles_Api_Assignee = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    type: zod_1.z.enum(['user', 'group', 'groupRole']),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_TranslationRoles_Api_Assignee_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_TranslationRoles_Api_Assignee),
});
const Roblox_GameLocalization_Client_GameLocalizationRoles_Assignee = zod_1.z.object({
    assigneeType: zod_1.z.enum(['user', 'group', 'groupRole']),
    id: zod_1.z.number().int(),
});
const Roblox_GameLocalization_Client_GameLocalizationRoles_GameLocalizationRoleAssignment = zod_1.z.object({
    gameId: zod_1.z.number().int(),
    assignee: Roblox_GameLocalization_Client_GameLocalizationRoles_Assignee,
});
const Roblox_TranslationRoles_Api_GetGameLocalizationRoleAssignmentsForUserResponse = zod_1.z.object({
    games: zod_1.z.array(Roblox_GameLocalization_Client_GameLocalizationRoles_GameLocalizationRoleAssignment),
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
});
const Roblox_TranslationRoles_Api_UpdateRoleRequest = zod_1.z.object({
    assigneeId: zod_1.z.number().int(),
    assigneeType: zod_1.z.enum(['user', 'group', 'groupRole']),
    role: zod_1.z.literal('translator'),
    revoke: zod_1.z.boolean(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
/**
 * @api PATCH https://translationroles.roblox.com/v1/game-localization-roles/games/:gameId
 * @summary Assigns or revokes a role
 * @param body The request body
 * @param gameId The id of the game
 */
exports.patchGameLocalizationRolesGamesGameid = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/game-localization-roles/games/:gameId',
    baseUrl: 'https://translationroles.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    body: Roblox_TranslationRoles_Api_UpdateRoleRequest.optional(),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `3: Invalid game id
4: Invalid assignee id
6: Request body can&#x27;t be null
7: The role you are assigning has reached max limit`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
1: You must be authorized to use this endpoint`,
        },
        {
            status: 429,
            description: `5: Too many attempts. Please try again later.`,
        },
        {
            status: 503,
            description: `2: Feature is disabled`,
        },
    ],
});
/**
 * @api GET https://translationroles.roblox.com/v1/game-localization-roles/games/:gameId/current-user/roles
 * @summary Retrieves the list of roles granted to current logged-in user
 * @param gameId The id of the game
 */
exports.getGameLocalizationRolesGamesGameidCurrentUserRoles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/game-localization-roles/games/:gameId/current-user/roles',
    baseUrl: 'https://translationroles.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        gameId: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_System_String_,
    errors: [
        {
            status: 400,
            description: `3: Invalid game id`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 503,
            description: `2: Feature is disabled`,
        },
    ],
});
/**
 * @api GET https://translationroles.roblox.com/v1/game-localization-roles/games/:gameId/roles/:role/assignees
 * @summary Gets list of users assigned a specific role in a game.
 * @param gameId The id of the game
 * @param role The Roblox.GameLocalization.Client.GameLocalizationRoles.GameLocalizationRoleType
 */
exports.getGameLocalizationRolesGamesGameidRolesRoleAssignees = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/game-localization-roles/games/:gameId/roles/:role/assignees',
    baseUrl: 'https://translationroles.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        gameId: {
            style: 'simple',
        },
        role: {
            style: 'simple',
        },
    },
    parameters: {
        gameId: zod_1.z.number().int(),
        role: zod_1.z.literal('translator'),
    },
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_TranslationRoles_Api_Assignee_,
    errors: [
        {
            status: 400,
            description: `3: Invalid game id`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `1: You must be authorized to use this endpoint`,
        },
        {
            status: 503,
            description: `2: Feature is disabled`,
        },
    ],
});
/**
 * @api GET https://translationroles.roblox.com/v1/game-localization-roles/roles/:role/current-user
 * @summary Gets the list of games and associated role assignment info for the requested user and role.
 * @param role The Roblox.GameLocalization.Client.GameLocalizationRoles.GameLocalizationRoleType
 * @param exclusiveStartKey Part of pagination. Last primary key of returned items in previous operation.
 * @param pageSize Part of pagination. Maximum number of items that might be returned in the page.
 * @param groupId Optional seleted groupId of resources requested for the user and role.
 */
exports.getGameLocalizationRolesRolesRoleCurrentUser = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/game-localization-roles/roles/:role/current-user',
    baseUrl: 'https://translationroles.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        role: {
            style: 'simple',
        },
        exclusiveStartKey: {
            style: 'form',
            explode: true,
        },
        pageSize: {
            style: 'form',
            explode: true,
        },
        groupId: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        role: zod_1.z.literal('translator'),
        exclusiveStartKey: zod_1.z.string().optional(),
        pageSize: zod_1.z.number().int().optional(),
        groupId: zod_1.z.number().int().optional(),
    },
    response: Roblox_TranslationRoles_Api_GetGameLocalizationRoleAssignmentsForUserResponse,
    errors: [
        {
            status: 400,
            description: `10: Invalid page size
11: Maximum page size exceeded
12: Invalid exclusive start key`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 500,
            description: `0: An unknown error occurred`,
        },
        {
            status: 503,
            description: `2: Feature is disabled`,
        },
    ],
});
