// rozod/lib/endpoints/twostepverificationv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postUsersUseridRecoveryCodesRegenerate = exports.postUsersUseridRecoveryCodesClear = exports.getUsersUseridRecoveryCodes = exports.postUsersUseridConfigurationSmsEnable = exports.postUsersUseridConfigurationSmsDisable = exports.postUsersUseridConfigurationSecurityKeyList = exports.postUsersUseridConfigurationSecurityKeyEnableVerify = exports.postUsersUseridConfigurationSecurityKeyEnable = exports.postUsersUseridConfigurationSecurityKeyDisable = exports.postUsersUseridConfigurationEmailEnable = exports.postUsersUseridConfigurationEmailDisable = exports.postUsersUseridConfigurationAuthenticatorEnableVerify = exports.postUsersUseridConfigurationAuthenticatorEnable = exports.postUsersUseridConfigurationAuthenticatorDisable = exports.getUsersUseridConfiguration = exports.postUsersUseridChallengesSmsVerify = exports.postUsersUseridChallengesSmsSendCode = exports.postUsersUseridChallengesSecurityKeyVerifyStart = exports.postUsersUseridChallengesSecurityKeyVerifyFinish = exports.postUsersUseridChallengesRecoveryCodesVerify = exports.postUsersUseridChallengesPasswordVerify = exports.postUsersUseridChallengesPasskeyVerifyStart = exports.postUsersUseridChallengesPasskeyVerifyFinish = exports.postUsersUseridChallengesEmailVerify = exports.postUsersUseridChallengesEmailSendCode = exports.postUsersUseridChallengesCrossDeviceVerify = exports.postUsersUseridChallengesCrossDeviceRetry = exports.postUsersUseridChallengesCrossDeviceRetract = exports.postUsersUseridChallengesAuthenticatorVerify = exports.getMetadata = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_TwoStepVerification_Api_MetadataResponse = zod_1.z.object({
    twoStepVerificationEnabled: zod_1.z.boolean(),
    authenticatorQrCodeSize: zod_1.z.string(),
    emailCodeLength: zod_1.z.number().int(),
    authenticatorCodeLength: zod_1.z.number().int(),
    authenticatorHelpSiteAddress: zod_1.z.string(),
    isPasswordRequiredForEnablingAuthenticator: zod_1.z.boolean(),
    isPasswordRequiredForEnablingEmailTwoStepVerification: zod_1.z.boolean(),
    isUsingTwoStepWebviewComponent: zod_1.z.boolean(),
    isTwoStepEnabledRequiredForEmailPasswordRequirement: zod_1.z.boolean(),
    isTwoStepEnabledRequiredForAuthenticatorPasswordRequirement: zod_1.z.boolean(),
    isSingleMethodEnforcementEnabled: zod_1.z.boolean(),
    isSmsTwoStepVerificationAvailable: zod_1.z.boolean(),
    isSecurityKeyTwoStepVerificationAvailable: zod_1.z.boolean(),
    isAuthenticatorWithVerifiedPhoneEnabled: zod_1.z.boolean(),
    isPasswordRequiredForEnablingSecurityKey: zod_1.z.boolean(),
    isPasswordRequiredForEnablingSms2SV: zod_1.z.boolean(),
    isPasswordRequiredForChangingRecoveryCodes: zod_1.z.boolean(),
    isPasswordRequiredForDisablingAuthenticator: zod_1.z.boolean(),
    isPasswordRequiredForDisablingEmailTwoStepVerification: zod_1.z.boolean(),
    isPasswordRequiredForDisablingSms2SV: zod_1.z.boolean(),
    isRecoveryCodeGenerationForAuthenticatorSetupEnabled: zod_1.z.boolean(),
    isSecurityKeyOnAllPlatformsEnabled: zod_1.z.boolean(),
    receiveWarningsOnDisableTwoStep: zod_1.z.boolean(),
    isAndroidSecurityKeyEnabled: zod_1.z.boolean(),
    isSettingsTabRedesignEnabled: zod_1.z.boolean(),
    twoStepCopyTextEnrollmentStatus: zod_1.z.number().int(),
    isEppUIEnabled: zod_1.z.boolean(),
    isEppRecoveryCodesEnabled: zod_1.z.boolean(),
    maskedUserEmail: zod_1.z.string(),
    isUserU13: zod_1.z.boolean(),
    isDelayedUiEnabled: zod_1.z.boolean(),
    is2svRecoveryEnabled: zod_1.z.boolean(),
});
const Roblox_TwoStepVerification_Api_UserConfigurationMethod = zod_1.z.object({
    mediaType: zod_1.z.enum([
        'Email',
        'SMS',
        'Authenticator',
        'RecoveryCode',
        'SecurityKey',
        'CrossDevice',
        'Password',
        'Passkey',
    ]),
    enabled: zod_1.z.boolean(),
    updated: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_TwoStepVerification_Api_UserConfiguration = zod_1.z.object({
    primaryMediaType: zod_1.z.union([
        zod_1.z.literal(0),
        zod_1.z.literal(1),
        zod_1.z.literal(2),
        zod_1.z.literal(3),
        zod_1.z.literal(4),
        zod_1.z.literal(5),
        zod_1.z.literal(6),
        zod_1.z.literal(7),
    ]),
    methods: zod_1.z.array(Roblox_TwoStepVerification_Api_UserConfigurationMethod),
});
const Roblox_TwoStepVerification_Api_RecoveryCodesStatusResponse = zod_1.z.object({
    activeCount: zod_1.z.number().int(),
    created: zod_1.z.string().datetime({ offset: true }),
});
const Roblox_TwoStepVerification_Api_VerifyCodeRequest = zod_1.z.object({
    challengeId: zod_1.z.string(),
    actionType: zod_1.z.enum([
        'Unknown',
        'Login',
        'RobuxSpend',
        'ItemTrade',
        'Resale',
        'PasswordReset',
        'RevertAccount',
        'Generic',
        'GenericWithRecoveryCodes',
    ]),
    code: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_VerifyCodeResponse = zod_1.z.object({
    verificationToken: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_RetractDialogRequest = zod_1.z.object({
    challengeId: zod_1.z.string(),
    actionType: zod_1.z.enum([
        'Unknown',
        'Login',
        'RobuxSpend',
        'ItemTrade',
        'Resale',
        'PasswordReset',
        'RevertAccount',
        'Generic',
        'GenericWithRecoveryCodes',
    ]),
});
const Roblox_TwoStepVerification_Api_RetractDialogResponse = zod_1.z.object({});
const Roblox_TwoStepVerification_Api_RetryApprovalRequest = zod_1.z.object({
    challengeId: zod_1.z.string(),
    actionType: zod_1.z.enum([
        'Unknown',
        'Login',
        'RobuxSpend',
        'ItemTrade',
        'Resale',
        'PasswordReset',
        'RevertAccount',
        'Generic',
        'GenericWithRecoveryCodes',
    ]),
});
const Roblox_TwoStepVerification_Api_RetryApprovalResponse = zod_1.z.object({});
const Roblox_TwoStepVerification_Api_VerifyApprovalRequest = zod_1.z.object({
    challengeId: zod_1.z.string(),
    actionType: zod_1.z.enum([
        'Unknown',
        'Login',
        'RobuxSpend',
        'ItemTrade',
        'Resale',
        'PasswordReset',
        'RevertAccount',
        'Generic',
        'GenericWithRecoveryCodes',
    ]),
});
const Roblox_TwoStepVerification_Api_VerifyApprovalResponse = zod_1.z.object({
    verificationToken: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_SendCodeRequest = zod_1.z.object({
    challengeId: zod_1.z.string(),
    actionType: zod_1.z.enum([
        'Unknown',
        'Login',
        'RobuxSpend',
        'ItemTrade',
        'Resale',
        'PasswordReset',
        'RevertAccount',
        'Generic',
        'GenericWithRecoveryCodes',
    ]),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_TwoStepVerification_Api_VerifyStartPasskeyResponse = zod_1.z.object({
    authenticationOptions: zod_1.z.string(),
    sessionId: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_VerifyStartSecurityKeyResponse = zod_1.z.object({
    authenticationOptions: zod_1.z.string(),
    sessionId: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_DisableTwoStepVerificationRequest = zod_1.z.object({
    password: zod_1.z.string(),
    reauthenticationToken: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_Models_Request_SecureAuthenticationIntentModel = zod_1.z.object({
    clientPublicKey: zod_1.z.string(),
    clientEpochTimestamp: zod_1.z.number().int(),
    saiSignature: zod_1.z.string(),
    serverNonce: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_EnableTwoStepVerificationRequest = zod_1.z.object({
    password: zod_1.z.string(),
    secureAuthenticationIntent: Roblox_TwoStepVerification_Api_Models_Request_SecureAuthenticationIntentModel,
});
const Roblox_TwoStepVerification_Api_EnableAuthenticatorResponse = zod_1.z.object({
    setupToken: zod_1.z.string(),
    qrCodeImageUrl: zod_1.z.string(),
    manualEntryKey: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_EnableVerifyAuthenticatorRequest = zod_1.z.object({
    setupToken: zod_1.z.string(),
    code: zod_1.z.string(),
    password: zod_1.z.string(),
    secureAuthenticationIntent: Roblox_TwoStepVerification_Api_Models_Request_SecureAuthenticationIntentModel,
});
const Roblox_TwoStepVerification_Api_EnableVerifyAuthenticatorResponse = zod_1.z.object({
    recoveryCodes: zod_1.z.array(zod_1.z.string()),
});
const Roblox_TwoStepVerification_Api_DisableSecurityKeyRequest = zod_1.z.object({
    credentialNicknames: zod_1.z.array(zod_1.z.string()),
});
const Roblox_TwoStepVerification_Api_EnableSecurityKeyResponse = zod_1.z.object({
    creationOptions: zod_1.z.string(),
    sessionId: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_EnableVerifySecurityKeyRequest = zod_1.z.object({
    sessionId: zod_1.z.string(),
    credentialNickname: zod_1.z.string(),
    attestationResponse: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_SecurityKeyCredential = zod_1.z.object({
    nickname: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_ListSecurityKeyResponse = zod_1.z.object({
    credentials: zod_1.z.array(Roblox_TwoStepVerification_Api_SecurityKeyCredential),
});
const Roblox_TwoStepVerification_Api_ClearRecoveryCodesRequest = zod_1.z.object({
    password: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_RegenerateRecoveryCodesRequest = zod_1.z.object({
    password: zod_1.z.string(),
});
const Roblox_TwoStepVerification_Api_RegenerateRecoveryCodesResponse = zod_1.z.object({ recoveryCodes: zod_1.z.array(zod_1.z.string()) });
/**
 * @api GET https://twostepverification.roblox.com/v1/metadata
 * @summary Gets two step verification system metadata.
 * @param userId The user ID.
 * @param challengeId The active two step verification challenge ID if there is one.
 * @param actionType The Roblox.TwoStepVerification.Client.TwoStepVerificationActionType associated with the challenge.
 * @description The metadata endpoint takes in optional request parameters to output additional context
for when the user is unauthenticated but attempting to login with two step verification.

When supplied, all three request parameters must be sent and match up.
 */
exports.getMetadata = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/metadata',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'form',
            explode: true,
        },
        challengeId: {
            style: 'form',
            explode: true,
        },
        actionType: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int().optional(),
        challengeId: zod_1.z.string().optional(),
        actionType: zod_1.z
            .union([
            zod_1.z.literal(0),
            zod_1.z.literal(1),
            zod_1.z.literal(2),
            zod_1.z.literal(3),
            zod_1.z.literal(4),
            zod_1.z.literal(5),
            zod_1.z.literal(6),
            zod_1.z.literal(7),
            zod_1.z.literal(8),
        ])
            .optional(),
    },
    response: Roblox_TwoStepVerification_Api_MetadataResponse,
    errors: [],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/authenticator/verify
 * @summary Verifies a two step verification challenge code via authenticator app.
 * @param body The Roblox.TwoStepVerification.Api.VerifyCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesAuthenticatorVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/authenticator/verify',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_VerifyCodeRequest,
    response: zod_1.z.object({ verificationToken: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
2: The user ID is invalid.
10: The two step verification challenge code is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/cross-device/retract
 * @summary Reverts a user's dialog state from ACTIVE to PENDING.
 * @param body The Roblox.TwoStepVerification.Api.RetractDialogRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesCrossDeviceRetract = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/cross-device/retract',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_RetractDialogRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
2: The user ID is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
19: Challenge denied.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/cross-device/retry
 * @summary Retry a Cross Device two step verification approval.
 * @param body The Roblox.TwoStepVerification.Api.RetryApprovalRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesCrossDeviceRetry = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/cross-device/retry',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_RetryApprovalRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
2: The user ID is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
19: Challenge denied.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/cross-device/verify
 * @summary Verifies a two step verification approval via Cross Device. Cross Device approval does not use a verification code.
 * @param body The Roblox.TwoStepVerification.Api.VerifyApprovalRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesCrossDeviceVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/cross-device/verify',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_VerifyApprovalRequest,
    response: zod_1.z.object({ verificationToken: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `0: An unknown error occurred with the request.
1: Invalid challenge ID.
2: The user ID is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
19: Challenge denied.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/email/send-code
 * @summary Sends a two step verification challenge code via email.
 * @param body The request body.Roblox.TwoStepVerification.Api.SendCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesEmailSendCode = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/email/send-code',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_SendCodeRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user ID is invalid.`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/email/verify
 * @summary Verifies a two step verification challenge with a code sent via email.
 * @param body The request body.Roblox.TwoStepVerification.Api.VerifyCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesEmailVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/email/verify',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_VerifyCodeRequest,
    response: zod_1.z.object({ verificationToken: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
10: The two step verification challenge code is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user ID is invalid.`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/passkey/verify-finish
 * @summary Validates the assertion data returned by the passkey.
 * @param body The request bodyRoblox.TwoStepVerification.Api.VerifyCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesPasskeyVerifyFinish = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/passkey/verify-finish',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_VerifyCodeRequest,
    response: zod_1.z.object({ verificationToken: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
2: The user ID is invalid.
10: The two step verification challenge code is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/passkey/verify-start
 * @summary Provides a challenge for the passkey to authenticate.
 * @param body The request bodyRoblox.TwoStepVerification.Api.SendCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesPasskeyVerifyStart = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/passkey/verify-start',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_SendCodeRequest,
    response: Roblox_TwoStepVerification_Api_VerifyStartPasskeyResponse,
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
2: The user ID is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
8: The user is not allowed to perform the requested action.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/password/verify
 * @summary Verifies a two step verification challenge with a password (code).
 * @param body The request body.Roblox.TwoStepVerification.Api.VerifyCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesPasswordVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/password/verify',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_VerifyCodeRequest,
    response: zod_1.z.object({ verificationToken: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
4: The password is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user ID is invalid.`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/recovery-codes/verify
 * @summary Verifies a two step verification challenge via a recovery code.
 * @param body The Roblox.TwoStepVerification.Api.VerifyCodeRequest.
 * @param userId The user ID.
 * @description Once a recovery code has been used to verify a challenge it cannot be used again.
 */
exports.postUsersUseridChallengesRecoveryCodesVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/recovery-codes/verify',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_VerifyCodeRequest,
    response: zod_1.z.object({ verificationToken: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
2: The user ID is invalid.
10: The two step verification challenge code is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/security-key/verify-finish
 * @summary Validates the assertion data returned by the security key.
 * @param body The request bodyRoblox.TwoStepVerification.Api.VerifyCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesSecurityKeyVerifyFinish = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/security-key/verify-finish',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_VerifyCodeRequest,
    response: zod_1.z.object({ verificationToken: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
2: The user ID is invalid.
10: The two step verification challenge code is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/security-key/verify-start
 * @summary Provides a challenge for the security key to authenticate.
 * @param body The request bodyRoblox.TwoStepVerification.Api.SendCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesSecurityKeyVerifyStart = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/security-key/verify-start',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_SendCodeRequest,
    response: Roblox_TwoStepVerification_Api_VerifyStartSecurityKeyResponse,
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
2: The user ID is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
8: The user is not allowed to perform the requested action.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/sms/send-code
 * @summary Sends a two step verification code via SMS for the specified user.
 * @param body The request body.Roblox.TwoStepVerification.Api.SendCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesSmsSendCode = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/sms/send-code',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_SendCodeRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user ID is invalid.`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/challenges/sms/verify
 * @summary Verifies a two step verification challenge with a code sent via SMS.
 * @param body The request body.Roblox.TwoStepVerification.Api.VerifyCodeRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridChallengesSmsVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/challenges/sms/verify',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_VerifyCodeRequest,
    response: zod_1.z.object({ verificationToken: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.
2: The user ID is invalid.
10: The two step verification challenge code is invalid.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api GET https://twostepverification.roblox.com/v1/users/:userId/configuration
 * @summary Gets two step verification configuration for the specified user.
 * @param userId The Id of the user to get the configuration for.
 * @param challengeId The active challenge for the user (as an alternative when the user is unauthenticated).
 * @param actionType The action type the challengeId is associated with.
 */
exports.getUsersUseridConfiguration = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/configuration',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        challengeId: {
            style: 'form',
            explode: true,
        },
        actionType: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        challengeId: zod_1.z.string().optional(),
        actionType: zod_1.z
            .union([
            zod_1.z.literal(0),
            zod_1.z.literal(1),
            zod_1.z.literal(2),
            zod_1.z.literal(3),
            zod_1.z.literal(4),
            zod_1.z.literal(5),
            zod_1.z.literal(6),
            zod_1.z.literal(7),
            zod_1.z.literal(8),
        ])
            .optional(),
    },
    response: Roblox_TwoStepVerification_Api_UserConfiguration,
    errors: [
        {
            status: 400,
            description: `1: Invalid challenge ID.`,
        },
        {
            status: 403,
            description: `2: The user ID is invalid.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/authenticator/disable
 * @summary Disables two step verification via authenticator for the specified user.
 * @param body The Roblox.TwoStepVerification.Api.DisableTwoStepVerificationRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationAuthenticatorDisable = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/authenticator/disable',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_DisableTwoStepVerificationRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: The user ID is invalid.
4: The password is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
8: The user is not allowed to perform the requested action.`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/authenticator/enable
 * @summary Initiates enabling authenticator-based two step verification for the specified user.
 * @param body The Roblox.TwoStepVerification.Api.EnableTwoStepVerificationRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationAuthenticatorEnable = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/authenticator/enable',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_EnableTwoStepVerificationRequest,
    response: Roblox_TwoStepVerification_Api_EnableAuthenticatorResponse,
    errors: [
        {
            status: 400,
            description: `2: The user ID is invalid.
3: The email is invalid.
4: The password is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
11: The two step verification configuration is already enabled.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/authenticator/enable-verify
 * @summary Finishes enabling authenticator-based two step verification for the specified user.
 * @param body The Roblox.TwoStepVerification.Api.EnableVerifyAuthenticatorRequest.
 * @param userId The user ID.
 * @description Enabling authenticator-based two step verification requires two parts to help ensure
the user has properly stored the authenticator key in their authenticator app.
 */
exports.postUsersUseridConfigurationAuthenticatorEnableVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/authenticator/enable-verify',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_EnableVerifyAuthenticatorRequest,
    response: Roblox_TwoStepVerification_Api_EnableVerifyAuthenticatorResponse,
    errors: [
        {
            status: 400,
            description: `2: The user ID is invalid.
4: The password is invalid.
10: The two step verification challenge code is invalid.
12: Invalid setup token.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
11: The two step verification configuration is already enabled.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/email/disable
 * @summary Disables two step verification via email for the specified user.
 * @param body The Roblox.TwoStepVerification.Api.DisableTwoStepVerificationRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationEmailDisable = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/email/disable',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_DisableTwoStepVerificationRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `4: The password is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user ID is invalid.
8: The user is not allowed to perform the requested action.`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/email/enable
 * @summary Enables two step verification via email for the specified user.
 * @param body The request body.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationEmailEnable = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/email/enable',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_EnableTwoStepVerificationRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `3: The email is invalid.
4: The password is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user ID is invalid.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/security-key/disable
 * @summary Disables a batch of credentials for the specified user.
 * @param body The request bodyRoblox.TwoStepVerification.Api.DisableTwoStepVerificationRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationSecurityKeyDisable = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/security-key/disable',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_DisableSecurityKeyRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `4: The password is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
8: The user is not allowed to perform the requested action.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/security-key/enable
 * @summary Initiates security key registration by providing credential creation options.
 * @param body The request body.Roblox.TwoStepVerification.Api.EnableTwoStepVerificationRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationSecurityKeyEnable = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/security-key/enable',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_EnableTwoStepVerificationRequest,
    response: Roblox_TwoStepVerification_Api_EnableSecurityKeyResponse,
    errors: [
        {
            status: 400,
            description: `2: The user ID is invalid.
4: The password is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
9: The two step verification configuration is invalid for this action.
16: Reached limit of security keys registered.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/security-key/enable-verify
 * @summary Finishes security key registration and stores credential. Enables security key as a 2sv media type if it is a user's first key.
 * @param body The request body.Roblox.TwoStepVerification.Api.EnableVerifySecurityKeyRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationSecurityKeyEnableVerify = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/security-key/enable-verify',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_EnableVerifySecurityKeyRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: The user ID is invalid.
17: Invalid security key nickname.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
17: Invalid security key nickname.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/security-key/list
 * @summary List a user's registered security keys.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationSecurityKeyList = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/security-key/list',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_TwoStepVerification_Api_ListSecurityKeyResponse,
    errors: [
        {
            status: 400,
            description: `2: The user ID is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/sms/disable
 * @summary Disables two step verification via SMS for the specified user.
 * @param body The request body.Roblox.TwoStepVerification.Api.DisableTwoStepVerificationRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationSmsDisable = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/sms/disable',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_DisableTwoStepVerificationRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `4: The password is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user ID is invalid.
8: The user is not allowed to perform the requested action.`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/configuration/sms/enable
 * @summary Enables two step verification via SMS for the specified user.
 * @param body The request body.Roblox.TwoStepVerification.Api.EnableTwoStepVerificationRequest.
 * @param userId The user ID.
 */
exports.postUsersUseridConfigurationSmsEnable = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/configuration/sms/enable',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: Roblox_TwoStepVerification_Api_EnableTwoStepVerificationRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `4: The password is invalid.
15: The phone number is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: The user ID is invalid.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.
8: The user is not allowed to perform the requested action.`,
        },
    ],
});
/**
 * @api GET https://twostepverification.roblox.com/v1/users/:userId/recovery-codes
 * @summary Gets the current status of recovery codes for a user.
 * @param userId The user ID.
 */
exports.getUsersUseridRecoveryCodes = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/recovery-codes',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_TwoStepVerification_Api_RecoveryCodesStatusResponse,
    errors: [
        {
            status: 400,
            description: `2: The user ID is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/recovery-codes/clear
 * @summary Clears any existing recovery codes for the user.
 * @param body The Roblox.TwoStepVerification.Api.ClearRecoveryCodesRequest.
 * @param userId The user ID.
 * @description Clearing recovery codes voids any recovery codes previously generated for the user.
New recovery codes will have to be generated to pass two step verification via recovery code.
 */
exports.postUsersUseridRecoveryCodesClear = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/recovery-codes/clear',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ password: zod_1.z.string() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `2: The user ID is invalid.
4: The password is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
/**
 * @api POST https://twostepverification.roblox.com/v1/users/:userId/recovery-codes/regenerate
 * @summary Clears any existing recovery codes and generates a new batch of recovery codes.
 * @param body The Roblox.TwoStepVerification.Api.RegenerateRecoveryCodesRequest.
 * @param userId The user ID to generate recovery codes for.
 * @description Two step verification recovery codes do not enforce that two step verification must be passed when logging in.
At least one two step verification media type must be enabled to trigger the two step verification flow.
Recovery codes are intended to be used to pass two step verification when the enabled media type is unavailable.

Recovery codes generated by this endpoint do not have an expiration.

Once a recovery code generated by this endpoint has been used it cannot be used again.
 */
exports.postUsersUseridRecoveryCodesRegenerate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users/:userId/recovery-codes/regenerate',
    baseUrl: 'https://twostepverification.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ password: zod_1.z.string() }),
    response: Roblox_TwoStepVerification_Api_RegenerateRecoveryCodesResponse,
    errors: [
        {
            status: 400,
            description: `2: The user ID is invalid.
4: The password is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed`,
        },
        {
            status: 429,
            description: `5: Too many requests.`,
        },
        {
            status: 503,
            description: `7: Two step verification is currently under maintenance.`,
        },
    ],
});
