// rozod/lib/endpoints/usersv1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUsersSearch = exports.getUsersAuthenticatedRoles = exports.getUsersAuthenticatedCountryCode = exports.getUsersAuthenticatedAgeBracket = exports.getUsersAuthenticated = exports.getUsersUseridUsernameHistory = exports.getUsersUseridDisplayNamesValidate = exports.patchUsersUseridDisplayNames = exports.getUsersUserid = exports.postUsers = exports.postUsernamesUsers = exports.postGender = exports.getGender = exports.getDisplayNamesValidate = exports.postDescription = exports.getDescription = exports.postBirthdate = exports.getBirthdate = void 0;
const zod_1 = require("zod");
const __1 = require("..");
const Roblox_Users_Api_BirthdateResponse = zod_1.z.object({
    birthMonth: zod_1.z.number().int(),
    birthDay: zod_1.z.number().int(),
    birthYear: zod_1.z.number().int(),
});
const Roblox_Users_Api_BirthdateRequest = zod_1.z.object({
    birthMonth: zod_1.z.number().int(),
    birthDay: zod_1.z.number().int(),
    birthYear: zod_1.z.number().int(),
    password: zod_1.z.string(),
});
const Roblox_Web_WebAPI_ApiEmptyResponseModel = zod_1.z.object({});
const Roblox_Users_Api_DescriptionResponse = zod_1.z.object({
    description: zod_1.z.string(),
});
const Roblox_Users_Api_DescriptionRequest = zod_1.z.object({
    description: zod_1.z.string(),
});
const Roblox_Users_Api_GenderResponse = zod_1.z.object({ gender: zod_1.z.number().int() });
const Roblox_Users_Api_GenderRequest = zod_1.z.object({
    gender: zod_1.z.enum(['Unknown', 'Male', 'Female']),
});
const Roblox_Users_Api_GetUserResponse = zod_1.z.object({
    description: zod_1.z.string(),
    created: zod_1.z.string().datetime({ offset: true }),
    isBanned: zod_1.z.boolean(),
    externalAppDisplayName: zod_1.z.string(),
    hasVerifiedBadge: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Users_Api_UsernameHistoryResponse = zod_1.z.object({ name: zod_1.z.string() });
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Users_Api_UsernameHistoryResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Users_Api_UsernameHistoryResponse),
});
const Roblox_Users_Api_AuthenticatedGetUserResponse = zod_1.z.object({
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Users_Api_UserAgeBracketResponse = zod_1.z.object({
    ageBracket: zod_1.z.number().int(),
});
const Roblox_Users_Api_UserCountryCodeResponse = zod_1.z.object({
    countryCode: zod_1.z.string(),
});
const Roblox_Users_Api_UserRolesResponse = zod_1.z.object({
    roles: zod_1.z.array(zod_1.z.string()),
});
const Roblox_Users_Api_SearchGetUserResponse = zod_1.z.object({
    previousUsernames: zod_1.z.array(zod_1.z.string()),
    hasVerifiedBadge: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Users_Api_SearchGetUserResponse_ = zod_1.z.object({
    previousPageCursor: zod_1.z.string(),
    nextPageCursor: zod_1.z.string(),
    data: zod_1.z.array(Roblox_Users_Api_SearchGetUserResponse),
});
const Roblox_Users_Api_MultiGetByUsernameRequest = zod_1.z.object({
    usernames: zod_1.z.array(zod_1.z.string()),
    excludeBannedUsers: zod_1.z.boolean(),
});
const Roblox_Users_Api_MultiGetUserByNameResponse = zod_1.z.object({
    requestedUsername: zod_1.z.string(),
    hasVerifiedBadge: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Users_Api_MultiGetUserByNameResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Users_Api_MultiGetUserByNameResponse),
});
const Roblox_Users_Api_MultiGetByUserIdRequest = zod_1.z.object({
    userIds: zod_1.z.array(zod_1.z.number()),
    excludeBannedUsers: zod_1.z.boolean(),
});
const Roblox_Users_Api_MultiGetUserResponse = zod_1.z.object({
    hasVerifiedBadge: zod_1.z.boolean(),
    id: zod_1.z.number().int(),
    name: zod_1.z.string(),
    displayName: zod_1.z.string(),
});
const Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Users_Api_MultiGetUserResponse_ = zod_1.z.object({
    data: zod_1.z.array(Roblox_Users_Api_MultiGetUserResponse),
});
const Roblox_Users_Api_SetDisplayNameRequest = zod_1.z.object({
    newDisplayName: zod_1.z.string(),
});
/**
 * @api GET https://users.roblox.com/v1/birthdate
 * @summary Get the user's birthdate
 */
exports.getBirthdate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/birthdate',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    response: Roblox_Users_Api_BirthdateResponse,
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://users.roblox.com/v1/birthdate
 * @summary Update the user's birthdate
 * @param body The Roblox.Users.Api.BirthdateRequest
 */
exports.postBirthdate = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/birthdate',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Users_Api_BirthdateRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: User not found.
4: The birthdate provided is invalid.
8: Password is incorrect.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: PIN is locked.
5: Invalid birthdate change.`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.
5: Invalid birthdate change.`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/description
 * @summary Get the user's description
 */
exports.getDescription = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/description',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ description: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://users.roblox.com/v1/description
 * @summary Update the user's description
 * @param body The Roblox.Users.Api.DescriptionRequest
 */
exports.postDescription = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/description',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: zod_1.z.object({ description: zod_1.z.string() }),
    response: zod_1.z.object({ description: zod_1.z.string() }),
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: PIN is locked.`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
        {
            status: 503,
            description: `3: This feature is currently disabled. Please try again later.`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/display-names/validate
 * @summary Validate a display name for a new user.
 * @param displayName The display name.
 * @param birthdate The new user's birthdate
 */
exports.getDisplayNamesValidate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/display-names/validate',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        displayName: {
            style: 'form',
            explode: true,
        },
        birthdate: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        displayName: zod_1.z.string(),
        birthdate: zod_1.z.string().datetime({ offset: true }),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Display name is too short
2: Display name is too long
3: Display name contains invalid characters
4: Display name has been moderated
6: Request must contain a birthdate
8: Display name has too many combinations of character sets`,
        },
        {
            status: 429,
            description: `5: Display name updates for this user have been throttled`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/gender
 * @summary Get the user's gender
 */
exports.getGender = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/gender',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ gender: zod_1.z.number().int() }),
    errors: [
        {
            status: 400,
            description: `1: User not found.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api POST https://users.roblox.com/v1/gender
 * @summary Update the user's gender
 * @param body The Roblox.Users.Api.GenderRequest
 */
exports.postGender = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/gender',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Users_Api_GenderRequest,
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: User not found.
6: The gender provided is invalid.`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
2: PIN is locked.`,
        },
        {
            status: 500,
            description: `0: An unknown error occured.`,
        },
    ],
});
/**
 * @api POST https://users.roblox.com/v1/usernames/users
 * @summary Get users by usernames.
 * @param body The Roblox.Users.Api.MultiGetByUsernameRequest.
 * @description This endpoint will also check previous usernames.
Does not require X-CSRF-Token protection because this is essentially a get request but as a POST to avoid URI limits.
 */
exports.postUsernamesUsers = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/usernames/users',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Users_Api_MultiGetByUsernameRequest,
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Users_Api_MultiGetUserByNameResponse_,
    errors: [
        {
            status: 400,
            description: `2: Too many usernames.`,
        },
    ],
});
/**
 * @api POST https://users.roblox.com/v1/users
 * @summary Get users by ids.
 * @param body The Roblox.Users.Api.MultiGetByUserIdRequest.
 * @description Does not require X-CSRF-Token protection because this is essentially a get request but as a POST to avoid URI limits.
 */
exports.postUsers = (0, __1.endpoint)({
    method: 'POST',
    path: '/v1/users',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
    },
    parameters: {},
    body: Roblox_Users_Api_MultiGetByUserIdRequest,
    response: Roblox_Web_WebAPI_Models_ApiArrayResponse_Roblox_Users_Api_MultiGetUserResponse_,
    errors: [
        {
            status: 400,
            description: `1: Too many ids.`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/users/:userId
 * @summary Gets detailed user information by id.
 * @param userId The user id.
 */
exports.getUsersUserid = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    response: Roblox_Users_Api_GetUserResponse,
    errors: [
        {
            status: 404,
            description: `3: The user id is invalid.`,
        },
    ],
});
/**
 * @api PATCH https://users.roblox.com/v1/users/:userId/display-names
 * @summary Set the display name for the authorized user.
 * @param body Roblox.Users.Api.SetDisplayNameRequest
 * @param userId the user id
 */
exports.patchUsersUseridDisplayNames = (0, __1.endpoint)({
    method: 'PATCH',
    path: '/v1/users/:userId/display-names',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        body: {},
        userId: {
            style: 'simple',
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
    },
    body: zod_1.z.object({ newDisplayName: zod_1.z.string() }),
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Display name is too short
2: Display name is too long
3: Display name contains invalid characters
4: Display name has been moderated
8: Display name has too many combinations of character sets`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `0: Token Validation Failed
7: The user id is invalid.`,
        },
        {
            status: 429,
            description: `5: Display name updates for this user have been throttled`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/users/:userId/display-names/validate
 * @summary Validate a display name for an existing user.
 * @param userId The user id.
 * @param displayName The display name.
 */
exports.getUsersUseridDisplayNamesValidate = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/display-names/validate',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        displayName: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        displayName: zod_1.z.string(),
    },
    response: zod_1.z.object({}),
    errors: [
        {
            status: 400,
            description: `1: Display name is too short
2: Display name is too long
3: Display name contains invalid characters
4: Display name has been moderated
8: Display name has too many combinations of character sets`,
        },
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
        {
            status: 403,
            description: `7: The user id is invalid.`,
        },
        {
            status: 429,
            description: `5: Display name updates for this user have been throttled`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/users/:userId/username-history
 * @summary Retrieves the username history for a particular user.
 * @param userId
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 * @param sortOrder The order the results are sorted in.
 */
exports.getUsersUseridUsernameHistory = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/:userId/username-history',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        userId: {
            style: 'simple',
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
        sortOrder: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        userId: zod_1.z.number().int(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
        sortOrder: zod_1.z.enum(['Asc', 'Desc']).optional().default('Asc'),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Users_Api_UsernameHistoryResponse_,
    errors: [
        {
            status: 400,
            description: `3: The user id is invalid.`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/users/authenticated
 * @summary Gets the minimal authenticated user.
 */
exports.getUsersAuthenticated = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/authenticated',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    response: Roblox_Users_Api_AuthenticatedGetUserResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/users/authenticated/age-bracket
 * @summary Gets the age bracket of the authenticated user.
 */
exports.getUsersAuthenticatedAgeBracket = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/authenticated/age-bracket',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ ageBracket: zod_1.z.number().int() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/users/authenticated/country-code
 * @summary Gets the country code of the authenticated user.
 */
exports.getUsersAuthenticatedCountryCode = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/authenticated/country-code',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    response: zod_1.z.object({ countryCode: zod_1.z.string() }),
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/users/authenticated/roles
 * @summary Gets the (public) roles of the authenticated user, such as `"Soothsayer"` and `"BetaTester"`.
 */
exports.getUsersAuthenticatedRoles = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/authenticated/roles',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    response: Roblox_Users_Api_UserRolesResponse,
    errors: [
        {
            status: 401,
            description: `0: Authorization has been denied for this request.`,
        },
    ],
});
/**
 * @api GET https://users.roblox.com/v1/users/search
 * @summary Searches for users by keyword.
 * @param keyword The search keyword.
 * @param sessionId
 * @param limit The number of results per request.
 * @param cursor The paging cursor for the previous or next page.
 */
exports.getUsersSearch = (0, __1.endpoint)({
    method: 'GET',
    path: '/v1/users/search',
    baseUrl: 'https://users.roblox.com',
    requestFormat: 'json',
    serializationMethod: {
        keyword: {
            style: 'form',
            explode: true,
        },
        sessionId: {
            style: 'form',
            explode: true,
        },
        limit: {
            style: 'form',
            explode: true,
        },
        cursor: {
            style: 'form',
            explode: true,
        },
    },
    parameters: {
        keyword: zod_1.z.string(),
        sessionId: zod_1.z.string().optional(),
        limit: zod_1.z
            .union([zod_1.z.literal(10), zod_1.z.literal(25), zod_1.z.literal(50), zod_1.z.literal(100)])
            .optional()
            .default(10),
        cursor: zod_1.z.string().optional(),
    },
    response: Roblox_Web_WebAPI_Models_ApiPageResponse_Roblox_Users_Api_SearchGetUserResponse_,
    errors: [
        {
            status: 400,
            description: `5: The keyword was filtered.
6: The keyword is too short.`,
        },
        {
            status: 429,
            description: `4: Too many requests.`,
        },
    ],
});
