// rozod/lib/opencloud/index.js
"use strict";
// OpenCloud APIs index file
// These are Roblox's OpenCloud APIs which are separate from the legacy web APIs (https://create.roblox.com/docs/en-us/cloud)
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudV2 = exports.AssetsV1 = exports.DatastoresOrderedV1 = exports.DatastoresV1 = exports.MessagingV1 = exports.UniversesV1 = exports.v2 = exports.v1 = void 0;
// Import v1 APIs
const UniversesV1 = __importStar(require("./v1/universes"));
exports.UniversesV1 = UniversesV1;
const MessagingV1 = __importStar(require("./v1/messaging"));
exports.MessagingV1 = MessagingV1;
const DatastoresV1 = __importStar(require("./v1/datastores"));
exports.DatastoresV1 = DatastoresV1;
const DatastoresOrderedV1 = __importStar(require("./v1/datastores-ordered"));
exports.DatastoresOrderedV1 = DatastoresOrderedV1;
const AssetsV1 = __importStar(require("./v1/assets"));
exports.AssetsV1 = AssetsV1;
// Import v2 APIs
const CloudV2 = __importStar(require("./v2/cloud"));
exports.CloudV2 = CloudV2;
/**
 * OpenCloud v1 APIs
 * @description Roblox OpenCloud v1 APIs for universes, messaging, datastores, and assets
 */
exports.v1 = {
    /** Universe APIs for managing Roblox experiences */
    universes: UniversesV1,
    /** Messaging APIs for cross-server communication */
    messaging: MessagingV1,
    /** DataStore APIs for persistent data storage */
    datastores: DatastoresV1,
    /** Ordered DataStore APIs for leaderboards and sorted data */
    datastoresOrdered: DatastoresOrderedV1,
    /** Asset APIs for managing experience assets */
    assets: AssetsV1,
};
/**
 * OpenCloud v2 APIs
 * @description Roblox OpenCloud v2 APIs
 */
exports.v2 = CloudV2;
